{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment analysis and GO plots \n",
    "\n",
    "It is crucial to understand the functional significance of the list of genes identified by high-throughput genomic methods. Enrichment analysis (or pathway analysis) can identify terms which are statistically over or under-represented within the list of interest, by systematically mapping genes and proteins to their associated biological annotations (such as gene ontology GO terms or pathway membership) and then comparing the distribution of the terms within a gene set of interest with the background distribution of these terms (eg all genes represented on a microarray chip). \n",
    "\n",
    "In this template the enrichment is performed using WebGestalt in R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-12-04T13:15:30.416567Z",
     "start_time": "2018-12-04T13:15:29.529Z"
    }
   },
   "outputs": [],
   "source": [
    "suppressPackageStartupMessages(library(WebGestaltR))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Results obtained in the integrative analysis from published meta analysis and GWAS summary statistics are explored here.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 x 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>rank</th><th scope=col>Name</th><th scope=col>Score</th><th scope=col>adjP.Val</th><th scope=col>adjP.Val2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CETN2</th><td>1</td><td>CETN2 </td><td>8.192692e-05</td><td>1</td><td>0.0001638538</td></tr>\n",
       "\t<tr><th scope=row>DNAJC8</th><td>2</td><td>DNAJC8</td><td>1.638538e-04</td><td>1</td><td>0.0003277077</td></tr>\n",
       "\t<tr><th scope=row>PSMC2</th><td>3</td><td>PSMC2 </td><td>2.457808e-04</td><td>1</td><td>0.0004915615</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 x 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & rank & Name & Score & adjP.Val & adjP.Val2\\\\\n",
       "  & <int> & <chr> & <dbl> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCETN2 & 1 & CETN2  & 8.192692e-05 & 1 & 0.0001638538\\\\\n",
       "\tDNAJC8 & 2 & DNAJC8 & 1.638538e-04 & 1 & 0.0003277077\\\\\n",
       "\tPSMC2 & 3 & PSMC2  & 2.457808e-04 & 1 & 0.0004915615\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 x 5\n",
       "\n",
       "| <!--/--> | rank &lt;int&gt; | Name &lt;chr&gt; | Score &lt;dbl&gt; | adjP.Val &lt;int&gt; | adjP.Val2 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| CETN2 | 1 | CETN2  | 8.192692e-05 | 1 | 0.0001638538 |\n",
       "| DNAJC8 | 2 | DNAJC8 | 1.638538e-04 | 1 | 0.0003277077 |\n",
       "| PSMC2 | 3 | PSMC2  | 2.457808e-04 | 1 | 0.0004915615 |\n",
       "\n"
      ],
      "text/plain": [
       "       rank Name   Score        adjP.Val adjP.Val2   \n",
       "CETN2  1    CETN2  8.192692e-05 1        0.0001638538\n",
       "DNAJC8 2    DNAJC8 1.638538e-04 1        0.0003277077\n",
       "PSMC2  3    PSMC2  2.457808e-04 1        0.0004915615"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get the dataset to explore\n",
    "rra.results<-read.table(\"/mnt/data/IntegrativeAnalysis//output//RRA_paper_result\")\n",
    "head(rra.results,n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 x 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>rank</th><th scope=col>logFC.case.ctl</th><th scope=col>Var</th><th scope=col>Qpvalue</th><th scope=col>REM.Pvalue</th><th scope=col>REM.FDR</th><th scope=col>Fisher.Pvalue</th><th scope=col>Fisher.FDR</th><th scope=col>n.estimators</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ZNF264</th><td>1</td><td> 0.3552904</td><td>0.299334526</td><td>1.502152e-26</td><td>0.516086802</td><td>0.67039523</td><td>1.427335e-31</td><td>1.172413e-27</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>SVOP</th><td>2</td><td>-0.7902721</td><td>0.005249884</td><td>3.104079e-01</td><td>0.000000000</td><td>0.00000000</td><td>2.022028e-27</td><td>8.304468e-24</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>NFKB1</th><td>3</td><td> 0.3817504</td><td>0.016106871</td><td>9.565457e-03</td><td>0.002629964</td><td>0.01517535</td><td>8.712374e-27</td><td>2.263827e-23</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 x 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & rank & logFC.case.ctl & Var & Qpvalue & REM.Pvalue & REM.FDR & Fisher.Pvalue & Fisher.FDR & n.estimators\\\\\n",
       "  & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\tZNF264 & 1 &  0.3552904 & 0.299334526 & 1.502152e-26 & 0.516086802 & 0.67039523 & 1.427335e-31 & 1.172413e-27 & 2\\\\\n",
       "\tSVOP & 2 & -0.7902721 & 0.005249884 & 3.104079e-01 & 0.000000000 & 0.00000000 & 2.022028e-27 & 8.304468e-24 & 2\\\\\n",
       "\tNFKB1 & 3 &  0.3817504 & 0.016106871 & 9.565457e-03 & 0.002629964 & 0.01517535 & 8.712374e-27 & 2.263827e-23 & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 x 9\n",
       "\n",
       "| <!--/--> | rank &lt;int&gt; | logFC.case.ctl &lt;dbl&gt; | Var &lt;dbl&gt; | Qpvalue &lt;dbl&gt; | REM.Pvalue &lt;dbl&gt; | REM.FDR &lt;dbl&gt; | Fisher.Pvalue &lt;dbl&gt; | Fisher.FDR &lt;dbl&gt; | n.estimators &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| ZNF264 | 1 |  0.3552904 | 0.299334526 | 1.502152e-26 | 0.516086802 | 0.67039523 | 1.427335e-31 | 1.172413e-27 | 2 |\n",
       "| SVOP | 2 | -0.7902721 | 0.005249884 | 3.104079e-01 | 0.000000000 | 0.00000000 | 2.022028e-27 | 8.304468e-24 | 2 |\n",
       "| NFKB1 | 3 |  0.3817504 | 0.016106871 | 9.565457e-03 | 0.002629964 | 0.01517535 | 8.712374e-27 | 2.263827e-23 | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "       rank logFC.case.ctl Var         Qpvalue      REM.Pvalue  REM.FDR   \n",
       "ZNF264 1     0.3552904     0.299334526 1.502152e-26 0.516086802 0.67039523\n",
       "SVOP   2    -0.7902721     0.005249884 3.104079e-01 0.000000000 0.00000000\n",
       "NFKB1  3     0.3817504     0.016106871 9.565457e-03 0.002629964 0.01517535\n",
       "       Fisher.Pvalue Fisher.FDR   n.estimators\n",
       "ZNF264 1.427335e-31  1.172413e-27 2           \n",
       "SVOP   2.022028e-27  8.304468e-24 2           \n",
       "NFKB1  8.712374e-27  2.263827e-23 2           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get the dataset to explore\n",
    "rra.results<-read.table(\"/mnt/data/IntegrativeAnalysis//output//RRA_paper_result\")\n",
    "head(rra.results,n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-12-04T14:31:16.948030Z",
     "start_time": "2018-12-04T14:31:16.200Z"
    }
   },
   "outputs": [],
   "source": [
    "# WebGestalt categories to be explored\n",
    "categories.database=c(\"geneontology_Biological_Process\",\"geneontology_Cellular_Component\",\"geneontology_Molecular_Function\",\"pathway_KEGG\",\"pathway_Wikipathway\",\"pathway_Reactome\",\"pathway_Panther\")\n",
    "categories.short=c(\"BP\",\"CC\",\"MF\",\"KEGG\",\"Wikipathway\",\"Reactome\",\"Panther\")\n",
    "categories.toplot=c(\"BP\",\"CC\",\"MF\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-12-04T15:06:19.062640Z",
     "start_time": "2018-12-04T15:06:18.265Z"
    }
   },
   "outputs": [],
   "source": [
    "outpath=\"/mnt/data/Enrichment/output\"\n",
    "#dir.create(outpath, recursive = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-29T14:34:03.076142Z",
     "start_time": "2018-11-29T14:33:00.791Z"
    },
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for WebGestaltR {WebGestaltR}\"><tr><td>WebGestaltR {WebGestaltR}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>WebGestaltR: The R interface for enrichment analysis with WebGestalt.</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Main function for enrichment analysis\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "WebGestaltR(\n",
       "  enrichMethod = \"ORA\",\n",
       "  organism = \"hsapiens\",\n",
       "  enrichDatabase = NULL,\n",
       "  enrichDatabaseFile = NULL,\n",
       "  enrichDatabaseType = NULL,\n",
       "  enrichDatabaseDescriptionFile = NULL,\n",
       "  interestGeneFile = NULL,\n",
       "  interestGene = NULL,\n",
       "  interestGeneType = NULL,\n",
       "  collapseMethod = \"mean\",\n",
       "  referenceGeneFile = NULL,\n",
       "  referenceGene = NULL,\n",
       "  referenceGeneType = NULL,\n",
       "  referenceSet = NULL,\n",
       "  minNum = 10,\n",
       "  maxNum = 500,\n",
       "  sigMethod = \"fdr\",\n",
       "  fdrMethod = \"BH\",\n",
       "  fdrThr = 0.05,\n",
       "  topThr = 10,\n",
       "  reportNum = 20,\n",
       "  perNum = 1000,\n",
       "  gseaP = 1,\n",
       "  isOutput = TRUE,\n",
       "  outputDirectory = getwd(),\n",
       "  projectName = NULL,\n",
       "  dagColor = \"continuous\",\n",
       "  saveRawGseaResult = FALSE,\n",
       "  gseaPlotFormat = c(\"png\", \"svg\"),\n",
       "  setCoverNum = 10,\n",
       "  networkConstructionMethod = NULL,\n",
       "  neighborNum = 10,\n",
       "  highlightType = \"Seeds\",\n",
       "  highlightSeedNum = 10,\n",
       "  nThreads = 1,\n",
       "  cache = NULL,\n",
       "  hostName = \"http://www.webgestalt.org/\",\n",
       "  ...\n",
       ")\n",
       "\n",
       "WebGestaltRBatch(\n",
       "  interestGeneFolder = NULL,\n",
       "  enrichMethod = \"ORA\",\n",
       "  isParallel = FALSE,\n",
       "  nThreads = 3,\n",
       "  ...\n",
       ")\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>enrichMethod</code></td>\n",
       "<td>\n",
       "<p>Enrichment methods: <code>ORA</code>, <code>GSEA</code> or <code>NTA</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>organism</code></td>\n",
       "<td>\n",
       "<p>Currently, WebGestaltR supports 12 organisms. Users can use the function\n",
       "<code>listOrganism</code> to check available organisms. Users can also input <code>others</code> to\n",
       "perform the enrichment analysis for other organisms not supported by WebGestaltR. For\n",
       "other organisms, users need to provide the functional categories, interesting list and\n",
       "reference list (for ORA method). Because WebGestaltR does not perform the ID mapping for\n",
       "the other organisms, the above data should have the same ID type.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>enrichDatabase</code></td>\n",
       "<td>\n",
       "<p>The functional categories for the enrichment analysis. Users can use\n",
       "the function <code>listGeneSet</code> to check the available functional databases for the\n",
       "selected organism. Multiple databases in a vector are supported for ORA and GSEA.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>enrichDatabaseFile</code></td>\n",
       "<td>\n",
       "<p>Users can provide one or more GMT files as the functional\n",
       "category for enrichment analysis. The extension of the file should be <code>gmt</code> and the\n",
       "first column of the file is the category ID, the second one is the external link for the\n",
       "category. Genes annotated to the category are from the third column. All columns are\n",
       "separated by tabs. The GMT files will be combined with <code>enrichDatabase</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>enrichDatabaseType</code></td>\n",
       "<td>\n",
       "<p>The ID type of the genes in the <code>enrichDatabaseFile</code>.\n",
       "If users set <code>organism</code> as <code>others</code>, users do not need to set this ID type because\n",
       "WebGestaltR will not perform ID mapping for other organisms. The supported ID types of\n",
       "WebGestaltR for the selected organism can be found by the function <code>listIdType</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>enrichDatabaseDescriptionFile</code></td>\n",
       "<td>\n",
       "<p>Users can also provide description files for the custom\n",
       "<code>enrichDatabaseFile</code>. The extension of the description file should be <code>des</code>. The\n",
       "description file contains two columns: the first column is the category ID that should be\n",
       "exactly the same as the category ID in the custom <code>enrichDatabaseFile</code> and the second\n",
       "column is the description of the category. All columns are separated by tabs.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>interestGeneFile</code></td>\n",
       "<td>\n",
       "<p>If <code>enrichMethod</code> is <code>ORA</code> or <code>NTA</code>, the extension of\n",
       "the <code>interestGeneFile</code> should be <code>txt</code> and the file can only contain one column:\n",
       "the interesting gene list. If <code>enrichMethod</code> is <code>GSEA</code>, the extension of the\n",
       "<code>interestGeneFile</code> should be <code>rnk</code> and the file should contain two columns\n",
       "separated by tab: the gene list and the corresponding scores.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>interestGene</code></td>\n",
       "<td>\n",
       "<p>Users can also use an R object as the input. If <code>enrichMethod</code> is\n",
       "<code>ORA</code> or <code>NTA</code>, <code>interestGene</code> should be an R <code>vector</code> object\n",
       "containing the interesting gene list. If <code>enrichMethod</code> is <code>GSEA</code>,\n",
       "<code>interestGene</code> should be an R <code>data.frame</code> object containing two columns: the\n",
       "gene list and the corresponding scores.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>interestGeneType</code></td>\n",
       "<td>\n",
       "<p>The ID type of the interesting gene list. The supported ID types of\n",
       "WebGestaltR for the selected organism can be found by the function <code>listIdType</code>. If\n",
       "the <code>organism</code> is <code>others</code>, users do not need to set this parameter.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>collapseMethod</code></td>\n",
       "<td>\n",
       "<p>The method to collapse duplicate IDs with scores. <code>mean</code>,\n",
       "<code>median</code>, <code>min</code> and <code>max</code> represent the mean, median, minimum and maximum\n",
       "of scores for the duplicate IDs.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>referenceGeneFile</code></td>\n",
       "<td>\n",
       "<p>For the ORA method, the users need to upload the reference gene\n",
       "list. The extension of the <code>referenceGeneFile</code> should be <code>txt</code> and the file can\n",
       "only contain one column: the reference gene list.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>referenceGene</code></td>\n",
       "<td>\n",
       "<p>For the ORA method, users can also use an R object as the reference\n",
       "gene list. <code>referenceGene</code> should be an R <code>vector</code> object containing the\n",
       "reference gene list.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>referenceGeneType</code></td>\n",
       "<td>\n",
       "<p>The ID type of the reference gene list. The supported ID types\n",
       "of WebGestaltR for the selected organism can be found by the function <code>listIdType</code>.\n",
       "If the <code>organism</code> is <code>others</code>, users do not need to set this parameter.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>referenceSet</code></td>\n",
       "<td>\n",
       "<p>Users can directly select the reference set from existing platforms in\n",
       "WebGestaltR and do not need to provide the reference set through <code>referenceGeneFile</code>.\n",
       "All existing platforms supported in WebGestaltR can be found by the function\n",
       "<code>listReferenceSet</code>. If <code>referenceGeneFile</code> and <code>refereneceGene</code> are\n",
       "<code>NULL</code>, WebGestaltR will use the <code>referenceSet</code> as the reference gene set.\n",
       "Otherwise, WebGestaltR will use the user supplied reference set for enrichment analysis.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>minNum</code></td>\n",
       "<td>\n",
       "<p>WebGestaltR will exclude the categories with the number of annotated genes\n",
       "less than <code>minNum</code> for enrichment analysis. The default is <code>10</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>maxNum</code></td>\n",
       "<td>\n",
       "<p>WebGestaltR will exclude the categories with the number of annotated genes\n",
       "larger than <code>maxNum</code> for enrichment analysis. The default is <code>500</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>sigMethod</code></td>\n",
       "<td>\n",
       "<p>Two methods of significance are available in WebGestaltR: <code>fdr</code> and\n",
       "<code>top</code>. <code>fdr</code> means the enriched categories are identified based on the FDR and\n",
       "<code>top</code> means all categories are ranked based on FDR and then select top categories\n",
       "as the enriched categories. The default is <code>fdr</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fdrMethod</code></td>\n",
       "<td>\n",
       "<p>For the ORA method, WebGestaltR supports five FDR methods: <code>holm</code>,\n",
       "<code>hochberg</code>, <code>hommel</code>, <code>bonferroni</code>, <code>BH</code> and <code>BY</code>. The default\n",
       "is <code>BH</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fdrThr</code></td>\n",
       "<td>\n",
       "<p>The significant threshold for the <code>fdr</code> method. The default is <code>0.05</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>topThr</code></td>\n",
       "<td>\n",
       "<p>The threshold for the <code>top</code> method. The default is <code>10</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>reportNum</code></td>\n",
       "<td>\n",
       "<p>The number of enriched categories visualized in the final report. The default\n",
       "is <code>20</code>. A larger <code>reportNum</code> may be slow to render in the report.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>perNum</code></td>\n",
       "<td>\n",
       "<p>The number of permutations for the GSEA method. The default is <code>1000</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>gseaP</code></td>\n",
       "<td>\n",
       "<p>The exponential scaling factor of the phenotype score. The default is <code>1</code>.\n",
       "When p=0, ES reduces to standard K-S statistics (See original paper for more details).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>isOutput</code></td>\n",
       "<td>\n",
       "<p>If <code>isOutput</code> is TRUE, WebGestaltR will create a folder named by\n",
       "the <code>projectName</code> and save the results in the folder. Otherwise, WebGestaltR will\n",
       "only return an R <code>data.frame</code> object containing the enrichment results. If\n",
       "hundreds of gene list need to be analyzed simultaneously, it is better to set\n",
       "<code>isOutput</code> to <code>FALSE</code>. The default is <code>TRUE</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>outputDirectory</code></td>\n",
       "<td>\n",
       "<p>The output directory for the results.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>projectName</code></td>\n",
       "<td>\n",
       "<p>The name of the project. If <code>projectName</code> is <code>NULL</code>,\n",
       "WebGestaltR will use time stamp as the project name.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>dagColor</code></td>\n",
       "<td>\n",
       "<p>If <code>dagColor</code> is <code>binary</code>, the significant terms in the DAG\n",
       "structure will be colored by steel blue for ORA method or steel blue (positive related)\n",
       "and dark orange (negative related) for GSEA method. If <code>dagColor</code> is <code>continous</code>,\n",
       "the significant terms in the DAG structure will be colored by the color gradient based on\n",
       "corresponding FDRs.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>saveRawGseaResult</code></td>\n",
       "<td>\n",
       "<p>Whether the raw result from GSEA is saved as a RDS file, which can be\n",
       "used for plotting. Defaults to <code>FALSE</code>. The list includes\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt>Enrichment_Results</dt><dd><p>A data frame of GSEA results with statistics</p>\n",
       "</dd>\n",
       "<dt>Running_Sums</dt><dd><p>A matrix of running sum of scores for each gene set</p>\n",
       "</dd>\n",
       "<dt>Items_in_Set</dt><dd><p>A list with ranks of genes for each gene set</p>\n",
       "</dd>\n",
       "</dl>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>gseaPlotFormat</code></td>\n",
       "<td>\n",
       "<p>The graphic format of GSEA enrichment plots. Either <code>svg</code>,\n",
       "<code>png</code>, or <code>c(\"png\", \"svg\")</code> (default).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>setCoverNum</code></td>\n",
       "<td>\n",
       "<p>The number of expected gene sets after set cover to reduce redundancy.\n",
       "It could get fewer sets if the coverage reaches 100%. The default is <code>10</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>networkConstructionMethod</code></td>\n",
       "<td>\n",
       "<p>Netowrk construction method for NTA. Either\n",
       "<code>Network_Retrieval_Prioritization</code> or <code>Network_Expansion</code>. Network Retrieval &amp;\n",
       "Prioritization first uses random walk analysis to calculate random walk probabilities\n",
       "for the input seeds, then identifies the relationships among the seeds in the selected\n",
       "network and returns a retrieval sub-network. The seeds with the top random walk\n",
       "probabilities are highlighted in the sub-network. Network Expansion first uses random\n",
       "walk analysis to rank all genes in the selected network based on their network\n",
       "proximity to the input seeds and then return an expanded sub-network in which nodes\n",
       "are the input seeds and their top ranking neighbors and edges represent their\n",
       "relationships.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>neighborNum</code></td>\n",
       "<td>\n",
       "<p>The number of neighbors to include in NTA Network Expansion method.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>highlightType</code></td>\n",
       "<td>\n",
       "<p>The type of nodes to highlight in the NTA Network Expansion method,\n",
       "either <code>Seeds</code> or <code>Neighbors</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>highlightSeedNum</code></td>\n",
       "<td>\n",
       "<p>The number of top input seeds to highlight in NTA Network Retrieval\n",
       "&amp; Prioritizaiton method.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>nThreads</code></td>\n",
       "<td>\n",
       "<p>The number of cores to use for GSEA and set cover, and in batch function.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>cache</code></td>\n",
       "<td>\n",
       "<p>A directory to save data cache for reuse. Defaults to <code>NULL</code> and disabled.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>hostName</code></td>\n",
       "<td>\n",
       "<p>The server URL for accessing data. Mostly for development purposes.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>...</code></td>\n",
       "<td>\n",
       "<p>In batch function, passes parameters to WebGestaltR function.\n",
       "Also handles backward compatibility for some parameters in old versions.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>interestGeneFolder</code></td>\n",
       "<td>\n",
       "<p>Run WebGestaltR for gene list files in the folder.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>isParallel</code></td>\n",
       "<td>\n",
       "<p>If jobs are run parallelly in the batch.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>WebGestaltR function can perform three enrichment analyses:\n",
       "ORA (Over-Representation Analysis) and GSEA (Gene Set Enrichment Analysis).and\n",
       "NTA (Network Topology Analysis). Based on the user-uploaded gene list or gene list\n",
       "with scores, WebGestaltR function will first map the gene list to the entrez gene\n",
       "ids and then summarize the gene list based on the GO (Gene Ontology) Slim. After\n",
       "performing the enrichment analysis, WebGestaltR function also returns a user-friendly\n",
       "HTML report containing GO Slim summary and the enrichment analysis result. If functional\n",
       "categories have DAG (directed acyclic graph) structure or genes in the functional\n",
       "categories have network structure, those relationship can also be visualized in the\n",
       "report.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>The WebGestaltR function returns a data frame containing the enrichment analysis\n",
       "result and also outputs an user-friendly HTML report if <code>isOutput</code> is <code>TRUE</code>.\n",
       "The columns in the data frame depend on the <code>enrichMethod</code> and they are the following:\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt>geneSet</dt><dd><p>ID of the gene set.</p>\n",
       "</dd>\n",
       "<dt>description</dt><dd><p>Description of the gene set if available.</p>\n",
       "</dd>\n",
       "<dt>link</dt><dd><p>Link to the data source.</p>\n",
       "</dd>\n",
       "<dt>size</dt><dd><p>The number of genes in the set after filtering by <code>minNum</code> and <code>maxNum</code>.</p>\n",
       "</dd>\n",
       "<dt>overlap</dt><dd><p>The number of mapped input genes that are annotated in the gene set.</p>\n",
       "</dd>\n",
       "<dt>expect</dt><dd><p>Expected number of input genes that are annotated in the gene set.</p>\n",
       "</dd>\n",
       "<dt>enrichmentRatio</dt><dd><p>Enrichment ratio, overlap / expect.</p>\n",
       "</dd>\n",
       "<dt>enrichmentScore</dt><dd><p>Enrichment score, the maximum running sum of scores for the ranked list.</p>\n",
       "</dd>\n",
       "<dt>normalizedEnrichmentScore</dt><dd><p>Normalized enrichment score, normalized against the average\n",
       "enrichment score of all permutations.</p>\n",
       "</dd>\n",
       "<dt>leadingEdgeNum</dt><dd><p>Number of genes/phosphosites in the leading edge.</p>\n",
       "</dd>\n",
       "<dt>pValue</dt><dd><p>P-value from hypergeometric test for ORA. For GSEA, please refer to its original\n",
       "publication or online at <a href=\"https://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm\">https://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm</a>.</p>\n",
       "</dd>\n",
       "<dt>FDR</dt><dd><p>Corrected P-value for mulilple testing with <code>fdrMethod</code> for ORA.</p>\n",
       "</dd>\n",
       "<dt>overlapId</dt><dd><p>The gene/phosphosite IDs of <code>overlap</code> for ORA (entrez gene IDs or\n",
       "phosphosite sequence).</p>\n",
       "</dd>\n",
       "<dt>leadingEdgeId</dt><dd><p>Genes/phosphosites in the leading edge in entrez gene ID or\n",
       "phosphosite sequence.</p>\n",
       "</dd>\n",
       "<dt>userId</dt><dd><p>The gene/phosphosite IDs of <code>overlap</code> for ORA or <code>leadingEdgeId</code>\n",
       "for GSEA in User input IDs.</p>\n",
       "</dd>\n",
       "<dt>plotPath</dt><dd><p>Path of the GSEA enrichment plot.</p>\n",
       "</dd>\n",
       "<dt>database</dt><dd><p>Name of the source database if multiple enrichment databases are given.</p>\n",
       "</dd>\n",
       "<dt>goId</dt><dd><p>In NTA, like <code>geneSet</code>, the enriched GO terms of genes in the\n",
       "returned subnetwork.</p>\n",
       "</dd>\n",
       "<dt>interestGene</dt><dd><p>In NTA, the gene IDs in the subnetwork with 0/1 annotations indicating\n",
       "if it is from user input.</p>\n",
       "</dd>\n",
       "</dl>\n",
       "\n",
       "<p>The WebGestaltRBatch function returns a list of enrichment results.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "## Not run: \n",
       "####### ORA example #########\n",
       "geneFile &lt;- system.file(\"extdata\", \"interestingGenes.txt\", package=\"WebGestaltR\")\n",
       "refFile &lt;- system.file(\"extdata\", \"referenceGenes.txt\", package=\"WebGestaltR\")\n",
       "outputDirectory &lt;- getwd()\n",
       "enrichResult &lt;- WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"pathway_KEGG\", interestGeneFile=geneFile,\n",
       "  interestGeneType=\"genesymbol\", referenceGeneFile=refFile,\n",
       "  referenceGeneType=\"genesymbol\", isOutput=TRUE,\n",
       "  outputDirectory=outputDirectory, projectName=NULL)\n",
       "\n",
       "####### GSEA example #########\n",
       "rankFile &lt;- system.file(\"extdata\", \"GeneRankList.rnk\", package=\"WebGestaltR\")\n",
       "outputDirectory &lt;- getwd()\n",
       "enrichResult &lt;- WebGestaltR(enrichMethod=\"GSEA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"pathway_KEGG\", interestGeneFile=rankFile,\n",
       "  interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10, minNum=5,\n",
       "  outputDirectory=outputDirectory)\n",
       "\n",
       "####### NTA example #########\n",
       "enrichResult &lt;- WebGestaltR(enrichMethod=\"NTA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"network_PPI_BIOGRID\", interestGeneFile=geneFile,\n",
       "  interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10,\n",
       "  outputDirectory=getwd(), highlightSeedNum=10,\n",
       "  networkConstructionMethod=\"Network_Retrieval_Prioritization\")\n",
       "\n",
       "## End(Not run)\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>WebGestaltR</em> version 0.4.4 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{WebGestaltR}{WebGestaltR: The R interface for enrichment analysis with WebGestalt.}{WebGestaltR}\n",
       "\\aliasA{WebGestaltRBatch}{WebGestaltR}{WebGestaltRBatch}\n",
       "\\aliasA{WebGestaltR\\_batch}{WebGestaltR}{WebGestaltR.Rul.batch}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Main function for enrichment analysis\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "WebGestaltR(\n",
       "  enrichMethod = \"ORA\",\n",
       "  organism = \"hsapiens\",\n",
       "  enrichDatabase = NULL,\n",
       "  enrichDatabaseFile = NULL,\n",
       "  enrichDatabaseType = NULL,\n",
       "  enrichDatabaseDescriptionFile = NULL,\n",
       "  interestGeneFile = NULL,\n",
       "  interestGene = NULL,\n",
       "  interestGeneType = NULL,\n",
       "  collapseMethod = \"mean\",\n",
       "  referenceGeneFile = NULL,\n",
       "  referenceGene = NULL,\n",
       "  referenceGeneType = NULL,\n",
       "  referenceSet = NULL,\n",
       "  minNum = 10,\n",
       "  maxNum = 500,\n",
       "  sigMethod = \"fdr\",\n",
       "  fdrMethod = \"BH\",\n",
       "  fdrThr = 0.05,\n",
       "  topThr = 10,\n",
       "  reportNum = 20,\n",
       "  perNum = 1000,\n",
       "  gseaP = 1,\n",
       "  isOutput = TRUE,\n",
       "  outputDirectory = getwd(),\n",
       "  projectName = NULL,\n",
       "  dagColor = \"continuous\",\n",
       "  saveRawGseaResult = FALSE,\n",
       "  gseaPlotFormat = c(\"png\", \"svg\"),\n",
       "  setCoverNum = 10,\n",
       "  networkConstructionMethod = NULL,\n",
       "  neighborNum = 10,\n",
       "  highlightType = \"Seeds\",\n",
       "  highlightSeedNum = 10,\n",
       "  nThreads = 1,\n",
       "  cache = NULL,\n",
       "  hostName = \"http://www.webgestalt.org/\",\n",
       "  ...\n",
       ")\n",
       "\n",
       "WebGestaltRBatch(\n",
       "  interestGeneFolder = NULL,\n",
       "  enrichMethod = \"ORA\",\n",
       "  isParallel = FALSE,\n",
       "  nThreads = 3,\n",
       "  ...\n",
       ")\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{enrichMethod}] Enrichment methods: \\code{ORA}, \\code{GSEA} or \\code{NTA}.\n",
       "\n",
       "\\item[\\code{organism}] Currently, WebGestaltR supports 12 organisms. Users can use the function\n",
       "\\code{listOrganism} to check available organisms. Users can also input \\code{others} to\n",
       "perform the enrichment analysis for other organisms not supported by WebGestaltR. For\n",
       "other organisms, users need to provide the functional categories, interesting list and\n",
       "reference list (for ORA method). Because WebGestaltR does not perform the ID mapping for\n",
       "the other organisms, the above data should have the same ID type.\n",
       "\n",
       "\\item[\\code{enrichDatabase}] The functional categories for the enrichment analysis. Users can use\n",
       "the function \\code{listGeneSet} to check the available functional databases for the\n",
       "selected organism. Multiple databases in a vector are supported for ORA and GSEA.\n",
       "\n",
       "\\item[\\code{enrichDatabaseFile}] Users can provide one or more GMT files as the functional\n",
       "category for enrichment analysis. The extension of the file should be \\code{gmt} and the\n",
       "first column of the file is the category ID, the second one is the external link for the\n",
       "category. Genes annotated to the category are from the third column. All columns are\n",
       "separated by tabs. The GMT files will be combined with \\code{enrichDatabase}.\n",
       "\n",
       "\\item[\\code{enrichDatabaseType}] The ID type of the genes in the \\code{enrichDatabaseFile}.\n",
       "If users set \\code{organism} as \\code{others}, users do not need to set this ID type because\n",
       "WebGestaltR will not perform ID mapping for other organisms. The supported ID types of\n",
       "WebGestaltR for the selected organism can be found by the function \\code{listIdType}.\n",
       "\n",
       "\\item[\\code{enrichDatabaseDescriptionFile}] Users can also provide description files for the custom\n",
       "\\code{enrichDatabaseFile}. The extension of the description file should be \\code{des}. The\n",
       "description file contains two columns: the first column is the category ID that should be\n",
       "exactly the same as the category ID in the custom \\code{enrichDatabaseFile} and the second\n",
       "column is the description of the category. All columns are separated by tabs.\n",
       "\n",
       "\\item[\\code{interestGeneFile}] If \\code{enrichMethod} is \\code{ORA} or \\code{NTA}, the extension of\n",
       "the \\code{interestGeneFile} should be \\code{txt} and the file can only contain one column:\n",
       "the interesting gene list. If \\code{enrichMethod} is \\code{GSEA}, the extension of the\n",
       "\\code{interestGeneFile} should be \\code{rnk} and the file should contain two columns\n",
       "separated by tab: the gene list and the corresponding scores.\n",
       "\n",
       "\\item[\\code{interestGene}] Users can also use an R object as the input. If \\code{enrichMethod} is\n",
       "\\code{ORA} or \\code{NTA}, \\code{interestGene} should be an R \\code{vector} object\n",
       "containing the interesting gene list. If \\code{enrichMethod} is \\code{GSEA},\n",
       "\\code{interestGene} should be an R \\code{data.frame} object containing two columns: the\n",
       "gene list and the corresponding scores.\n",
       "\n",
       "\\item[\\code{interestGeneType}] The ID type of the interesting gene list. The supported ID types of\n",
       "WebGestaltR for the selected organism can be found by the function \\code{listIdType}. If\n",
       "the \\code{organism} is \\code{others}, users do not need to set this parameter.\n",
       "\n",
       "\\item[\\code{collapseMethod}] The method to collapse duplicate IDs with scores. \\code{mean},\n",
       "\\code{median}, \\code{min} and \\code{max} represent the mean, median, minimum and maximum\n",
       "of scores for the duplicate IDs.\n",
       "\n",
       "\\item[\\code{referenceGeneFile}] For the ORA method, the users need to upload the reference gene\n",
       "list. The extension of the \\code{referenceGeneFile} should be \\code{txt} and the file can\n",
       "only contain one column: the reference gene list.\n",
       "\n",
       "\\item[\\code{referenceGene}] For the ORA method, users can also use an R object as the reference\n",
       "gene list. \\code{referenceGene} should be an R \\code{vector} object containing the\n",
       "reference gene list.\n",
       "\n",
       "\\item[\\code{referenceGeneType}] The ID type of the reference gene list. The supported ID types\n",
       "of WebGestaltR for the selected organism can be found by the function \\code{listIdType}.\n",
       "If the \\code{organism} is \\code{others}, users do not need to set this parameter.\n",
       "\n",
       "\\item[\\code{referenceSet}] Users can directly select the reference set from existing platforms in\n",
       "WebGestaltR and do not need to provide the reference set through \\code{referenceGeneFile}.\n",
       "All existing platforms supported in WebGestaltR can be found by the function\n",
       "\\code{listReferenceSet}. If \\code{referenceGeneFile} and \\code{refereneceGene} are\n",
       "\\code{NULL}, WebGestaltR will use the \\code{referenceSet} as the reference gene set.\n",
       "Otherwise, WebGestaltR will use the user supplied reference set for enrichment analysis.\n",
       "\n",
       "\\item[\\code{minNum}] WebGestaltR will exclude the categories with the number of annotated genes\n",
       "less than \\code{minNum} for enrichment analysis. The default is \\code{10}.\n",
       "\n",
       "\\item[\\code{maxNum}] WebGestaltR will exclude the categories with the number of annotated genes\n",
       "larger than \\code{maxNum} for enrichment analysis. The default is \\code{500}.\n",
       "\n",
       "\\item[\\code{sigMethod}] Two methods of significance are available in WebGestaltR: \\code{fdr} and\n",
       "\\code{top}. \\code{fdr} means the enriched categories are identified based on the FDR and\n",
       "\\code{top} means all categories are ranked based on FDR and then select top categories\n",
       "as the enriched categories. The default is \\code{fdr}.\n",
       "\n",
       "\\item[\\code{fdrMethod}] For the ORA method, WebGestaltR supports five FDR methods: \\code{holm},\n",
       "\\code{hochberg}, \\code{hommel}, \\code{bonferroni}, \\code{BH} and \\code{BY}. The default\n",
       "is \\code{BH}.\n",
       "\n",
       "\\item[\\code{fdrThr}] The significant threshold for the \\code{fdr} method. The default is \\code{0.05}.\n",
       "\n",
       "\\item[\\code{topThr}] The threshold for the \\code{top} method. The default is \\code{10}.\n",
       "\n",
       "\\item[\\code{reportNum}] The number of enriched categories visualized in the final report. The default\n",
       "is \\code{20}. A larger \\code{reportNum} may be slow to render in the report.\n",
       "\n",
       "\\item[\\code{perNum}] The number of permutations for the GSEA method. The default is \\code{1000}.\n",
       "\n",
       "\\item[\\code{gseaP}] The exponential scaling factor of the phenotype score. The default is \\code{1}.\n",
       "When p=0, ES reduces to standard K-S statistics (See original paper for more details).\n",
       "\n",
       "\\item[\\code{isOutput}] If \\code{isOutput} is TRUE, WebGestaltR will create a folder named by\n",
       "the \\code{projectName} and save the results in the folder. Otherwise, WebGestaltR will\n",
       "only return an R \\code{data.frame} object containing the enrichment results. If\n",
       "hundreds of gene list need to be analyzed simultaneously, it is better to set\n",
       "\\code{isOutput} to \\code{FALSE}. The default is \\code{TRUE}.\n",
       "\n",
       "\\item[\\code{outputDirectory}] The output directory for the results.\n",
       "\n",
       "\\item[\\code{projectName}] The name of the project. If \\code{projectName} is \\code{NULL},\n",
       "WebGestaltR will use time stamp as the project name.\n",
       "\n",
       "\\item[\\code{dagColor}] If \\code{dagColor} is \\code{binary}, the significant terms in the DAG\n",
       "structure will be colored by steel blue for ORA method or steel blue (positive related)\n",
       "and dark orange (negative related) for GSEA method. If \\code{dagColor} is \\code{continous},\n",
       "the significant terms in the DAG structure will be colored by the color gradient based on\n",
       "corresponding FDRs.\n",
       "\n",
       "\\item[\\code{saveRawGseaResult}] Whether the raw result from GSEA is saved as a RDS file, which can be\n",
       "used for plotting. Defaults to \\code{FALSE}. The list includes\n",
       "\\begin{description}\n",
       "\n",
       "\\item[Enrichment\\_Results] A data frame of GSEA results with statistics\n",
       "\\item[Running\\_Sums] A matrix of running sum of scores for each gene set\n",
       "\\item[Items\\_in\\_Set] A list with ranks of genes for each gene set\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "\\item[\\code{gseaPlotFormat}] The graphic format of GSEA enrichment plots. Either \\code{svg},\n",
       "\\code{png}, or \\code{c(\"png\", \"svg\")} (default).\n",
       "\n",
       "\\item[\\code{setCoverNum}] The number of expected gene sets after set cover to reduce redundancy.\n",
       "It could get fewer sets if the coverage reaches 100\\%. The default is \\code{10}.\n",
       "\n",
       "\\item[\\code{networkConstructionMethod}] Netowrk construction method for NTA. Either\n",
       "\\code{Network\\_Retrieval\\_Prioritization} or \\code{Network\\_Expansion}. Network Retrieval \\&\n",
       "Prioritization first uses random walk analysis to calculate random walk probabilities\n",
       "for the input seeds, then identifies the relationships among the seeds in the selected\n",
       "network and returns a retrieval sub-network. The seeds with the top random walk\n",
       "probabilities are highlighted in the sub-network. Network Expansion first uses random\n",
       "walk analysis to rank all genes in the selected network based on their network\n",
       "proximity to the input seeds and then return an expanded sub-network in which nodes\n",
       "are the input seeds and their top ranking neighbors and edges represent their\n",
       "relationships.\n",
       "\n",
       "\\item[\\code{neighborNum}] The number of neighbors to include in NTA Network Expansion method.\n",
       "\n",
       "\\item[\\code{highlightType}] The type of nodes to highlight in the NTA Network Expansion method,\n",
       "either \\code{Seeds} or \\code{Neighbors}.\n",
       "\n",
       "\\item[\\code{highlightSeedNum}] The number of top input seeds to highlight in NTA Network Retrieval\n",
       "\\& Prioritizaiton method.\n",
       "\n",
       "\\item[\\code{nThreads}] The number of cores to use for GSEA and set cover, and in batch function.\n",
       "\n",
       "\\item[\\code{cache}] A directory to save data cache for reuse. Defaults to \\code{NULL} and disabled.\n",
       "\n",
       "\\item[\\code{hostName}] The server URL for accessing data. Mostly for development purposes.\n",
       "\n",
       "\\item[\\code{...}] In batch function, passes parameters to WebGestaltR function.\n",
       "Also handles backward compatibility for some parameters in old versions.\n",
       "\n",
       "\\item[\\code{interestGeneFolder}] Run WebGestaltR for gene list files in the folder.\n",
       "\n",
       "\\item[\\code{isParallel}] If jobs are run parallelly in the batch.\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "WebGestaltR function can perform three enrichment analyses:\n",
       "ORA (Over-Representation Analysis) and GSEA (Gene Set Enrichment Analysis).and\n",
       "NTA (Network Topology Analysis). Based on the user-uploaded gene list or gene list\n",
       "with scores, WebGestaltR function will first map the gene list to the entrez gene\n",
       "ids and then summarize the gene list based on the GO (Gene Ontology) Slim. After\n",
       "performing the enrichment analysis, WebGestaltR function also returns a user-friendly\n",
       "HTML report containing GO Slim summary and the enrichment analysis result. If functional\n",
       "categories have DAG (directed acyclic graph) structure or genes in the functional\n",
       "categories have network structure, those relationship can also be visualized in the\n",
       "report.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "The WebGestaltR function returns a data frame containing the enrichment analysis\n",
       "result and also outputs an user-friendly HTML report if \\code{isOutput} is \\code{TRUE}.\n",
       "The columns in the data frame depend on the \\code{enrichMethod} and they are the following:\n",
       "\\begin{description}\n",
       "\n",
       "\\item[geneSet] ID of the gene set.\n",
       "\\item[description] Description of the gene set if available.\n",
       "\\item[link] Link to the data source.\n",
       "\\item[size] The number of genes in the set after filtering by \\code{minNum} and \\code{maxNum}.\n",
       "\\item[overlap] The number of mapped input genes that are annotated in the gene set.\n",
       "\\item[expect] Expected number of input genes that are annotated in the gene set.\n",
       "\\item[enrichmentRatio] Enrichment ratio, overlap / expect.\n",
       "\\item[enrichmentScore] Enrichment score, the maximum running sum of scores for the ranked list.\n",
       "\\item[normalizedEnrichmentScore] Normalized enrichment score, normalized against the average\n",
       "enrichment score of all permutations.\n",
       "\\item[leadingEdgeNum] Number of genes/phosphosites in the leading edge.\n",
       "\\item[pValue] P-value from hypergeometric test for ORA. For GSEA, please refer to its original\n",
       "publication or online at \\url{https://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm}.\n",
       "\\item[FDR] Corrected P-value for mulilple testing with \\code{fdrMethod} for ORA.\n",
       "\\item[overlapId] The gene/phosphosite IDs of \\code{overlap} for ORA (entrez gene IDs or\n",
       "phosphosite sequence).\n",
       "\\item[leadingEdgeId] Genes/phosphosites in the leading edge in entrez gene ID or\n",
       "phosphosite sequence.\n",
       "\\item[userId] The gene/phosphosite IDs of \\code{overlap} for ORA or \\code{leadingEdgeId}\n",
       "for GSEA in User input IDs.\n",
       "\\item[plotPath] Path of the GSEA enrichment plot.\n",
       "\\item[database] Name of the source database if multiple enrichment databases are given.\n",
       "\\item[goId] In NTA, like \\code{geneSet}, the enriched GO terms of genes in the\n",
       "returned subnetwork.\n",
       "\\item[interestGene] In NTA, the gene IDs in the subnetwork with 0/1 annotations indicating\n",
       "if it is from user input.\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "The WebGestaltRBatch function returns a list of enrichment results.\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "## Not run: \n",
       "####### ORA example #########\n",
       "geneFile <- system.file(\"extdata\", \"interestingGenes.txt\", package=\"WebGestaltR\")\n",
       "refFile <- system.file(\"extdata\", \"referenceGenes.txt\", package=\"WebGestaltR\")\n",
       "outputDirectory <- getwd()\n",
       "enrichResult <- WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"pathway_KEGG\", interestGeneFile=geneFile,\n",
       "  interestGeneType=\"genesymbol\", referenceGeneFile=refFile,\n",
       "  referenceGeneType=\"genesymbol\", isOutput=TRUE,\n",
       "  outputDirectory=outputDirectory, projectName=NULL)\n",
       "\n",
       "####### GSEA example #########\n",
       "rankFile <- system.file(\"extdata\", \"GeneRankList.rnk\", package=\"WebGestaltR\")\n",
       "outputDirectory <- getwd()\n",
       "enrichResult <- WebGestaltR(enrichMethod=\"GSEA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"pathway_KEGG\", interestGeneFile=rankFile,\n",
       "  interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10, minNum=5,\n",
       "  outputDirectory=outputDirectory)\n",
       "\n",
       "####### NTA example #########\n",
       "enrichResult <- WebGestaltR(enrichMethod=\"NTA\", organism=\"hsapiens\",\n",
       "  enrichDatabase=\"network_PPI_BIOGRID\", interestGeneFile=geneFile,\n",
       "  interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10,\n",
       "  outputDirectory=getwd(), highlightSeedNum=10,\n",
       "  networkConstructionMethod=\"Network_Retrieval_Prioritization\")\n",
       "\n",
       "## End(Not run)\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "WebGestaltR            package:WebGestaltR             R Documentation\n",
       "\n",
       "_\bW_\be_\bb_\bG_\be_\bs_\bt_\ba_\bl_\bt_\bR: _\bT_\bh_\be _\bR _\bi_\bn_\bt_\be_\br_\bf_\ba_\bc_\be _\bf_\bo_\br _\be_\bn_\br_\bi_\bc_\bh_\bm_\be_\bn_\bt _\ba_\bn_\ba_\bl_\by_\bs_\bi_\bs _\bw_\bi_\bt_\bh _\bW_\be_\bb_\bG_\be_\bs_\bt_\ba_\bl_\bt.\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Main function for enrichment analysis\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     WebGestaltR(\n",
       "       enrichMethod = \"ORA\",\n",
       "       organism = \"hsapiens\",\n",
       "       enrichDatabase = NULL,\n",
       "       enrichDatabaseFile = NULL,\n",
       "       enrichDatabaseType = NULL,\n",
       "       enrichDatabaseDescriptionFile = NULL,\n",
       "       interestGeneFile = NULL,\n",
       "       interestGene = NULL,\n",
       "       interestGeneType = NULL,\n",
       "       collapseMethod = \"mean\",\n",
       "       referenceGeneFile = NULL,\n",
       "       referenceGene = NULL,\n",
       "       referenceGeneType = NULL,\n",
       "       referenceSet = NULL,\n",
       "       minNum = 10,\n",
       "       maxNum = 500,\n",
       "       sigMethod = \"fdr\",\n",
       "       fdrMethod = \"BH\",\n",
       "       fdrThr = 0.05,\n",
       "       topThr = 10,\n",
       "       reportNum = 20,\n",
       "       perNum = 1000,\n",
       "       gseaP = 1,\n",
       "       isOutput = TRUE,\n",
       "       outputDirectory = getwd(),\n",
       "       projectName = NULL,\n",
       "       dagColor = \"continuous\",\n",
       "       saveRawGseaResult = FALSE,\n",
       "       gseaPlotFormat = c(\"png\", \"svg\"),\n",
       "       setCoverNum = 10,\n",
       "       networkConstructionMethod = NULL,\n",
       "       neighborNum = 10,\n",
       "       highlightType = \"Seeds\",\n",
       "       highlightSeedNum = 10,\n",
       "       nThreads = 1,\n",
       "       cache = NULL,\n",
       "       hostName = \"http://www.webgestalt.org/\",\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     WebGestaltRBatch(\n",
       "       interestGeneFolder = NULL,\n",
       "       enrichMethod = \"ORA\",\n",
       "       isParallel = FALSE,\n",
       "       nThreads = 3,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "enrichMethod: Enrichment methods: 'ORA', 'GSEA' or 'NTA'.\n",
       "\n",
       "organism: Currently, WebGestaltR supports 12 organisms. Users can use\n",
       "          the function 'listOrganism' to check available organisms.\n",
       "          Users can also input 'others' to perform the enrichment\n",
       "          analysis for other organisms not supported by WebGestaltR.\n",
       "          For other organisms, users need to provide the functional\n",
       "          categories, interesting list and reference list (for ORA\n",
       "          method). Because WebGestaltR does not perform the ID mapping\n",
       "          for the other organisms, the above data should have the same\n",
       "          ID type.\n",
       "\n",
       "enrichDatabase: The functional categories for the enrichment analysis.\n",
       "          Users can use the function 'listGeneSet' to check the\n",
       "          available functional databases for the selected organism.\n",
       "          Multiple databases in a vector are supported for ORA and\n",
       "          GSEA.\n",
       "\n",
       "enrichDatabaseFile: Users can provide one or more GMT files as the\n",
       "          functional category for enrichment analysis. The extension of\n",
       "          the file should be 'gmt' and the first column of the file is\n",
       "          the category ID, the second one is the external link for the\n",
       "          category. Genes annotated to the category are from the third\n",
       "          column. All columns are separated by tabs. The GMT files will\n",
       "          be combined with 'enrichDatabase'.\n",
       "\n",
       "enrichDatabaseType: The ID type of the genes in the\n",
       "          'enrichDatabaseFile'. If users set 'organism' as 'others',\n",
       "          users do not need to set this ID type because WebGestaltR\n",
       "          will not perform ID mapping for other organisms. The\n",
       "          supported ID types of WebGestaltR for the selected organism\n",
       "          can be found by the function 'listIdType'.\n",
       "\n",
       "enrichDatabaseDescriptionFile: Users can also provide description files\n",
       "          for the custom 'enrichDatabaseFile'. The extension of the\n",
       "          description file should be 'des'. The description file\n",
       "          contains two columns: the first column is the category ID\n",
       "          that should be exactly the same as the category ID in the\n",
       "          custom 'enrichDatabaseFile' and the second column is the\n",
       "          description of the category. All columns are separated by\n",
       "          tabs.\n",
       "\n",
       "interestGeneFile: If 'enrichMethod' is 'ORA' or 'NTA', the extension of\n",
       "          the 'interestGeneFile' should be 'txt' and the file can only\n",
       "          contain one column: the interesting gene list. If\n",
       "          'enrichMethod' is 'GSEA', the extension of the\n",
       "          'interestGeneFile' should be 'rnk' and the file should\n",
       "          contain two columns separated by tab: the gene list and the\n",
       "          corresponding scores.\n",
       "\n",
       "interestGene: Users can also use an R object as the input. If\n",
       "          'enrichMethod' is 'ORA' or 'NTA', 'interestGene' should be an\n",
       "          R 'vector' object containing the interesting gene list. If\n",
       "          'enrichMethod' is 'GSEA', 'interestGene' should be an R\n",
       "          'data.frame' object containing two columns: the gene list and\n",
       "          the corresponding scores.\n",
       "\n",
       "interestGeneType: The ID type of the interesting gene list. The\n",
       "          supported ID types of WebGestaltR for the selected organism\n",
       "          can be found by the function 'listIdType'. If the 'organism'\n",
       "          is 'others', users do not need to set this parameter.\n",
       "\n",
       "collapseMethod: The method to collapse duplicate IDs with scores.\n",
       "          'mean', 'median', 'min' and 'max' represent the mean, median,\n",
       "          minimum and maximum of scores for the duplicate IDs.\n",
       "\n",
       "referenceGeneFile: For the ORA method, the users need to upload the\n",
       "          reference gene list. The extension of the 'referenceGeneFile'\n",
       "          should be 'txt' and the file can only contain one column: the\n",
       "          reference gene list.\n",
       "\n",
       "referenceGene: For the ORA method, users can also use an R object as\n",
       "          the reference gene list. 'referenceGene' should be an R\n",
       "          'vector' object containing the reference gene list.\n",
       "\n",
       "referenceGeneType: The ID type of the reference gene list. The\n",
       "          supported ID types of WebGestaltR for the selected organism\n",
       "          can be found by the function 'listIdType'. If the 'organism'\n",
       "          is 'others', users do not need to set this parameter.\n",
       "\n",
       "referenceSet: Users can directly select the reference set from existing\n",
       "          platforms in WebGestaltR and do not need to provide the\n",
       "          reference set through 'referenceGeneFile'. All existing\n",
       "          platforms supported in WebGestaltR can be found by the\n",
       "          function 'listReferenceSet'. If 'referenceGeneFile' and\n",
       "          'refereneceGene' are 'NULL', WebGestaltR will use the\n",
       "          'referenceSet' as the reference gene set. Otherwise,\n",
       "          WebGestaltR will use the user supplied reference set for\n",
       "          enrichment analysis.\n",
       "\n",
       "  minNum: WebGestaltR will exclude the categories with the number of\n",
       "          annotated genes less than 'minNum' for enrichment analysis.\n",
       "          The default is '10'.\n",
       "\n",
       "  maxNum: WebGestaltR will exclude the categories with the number of\n",
       "          annotated genes larger than 'maxNum' for enrichment analysis.\n",
       "          The default is '500'.\n",
       "\n",
       "sigMethod: Two methods of significance are available in WebGestaltR:\n",
       "          'fdr' and 'top'. 'fdr' means the enriched categories are\n",
       "          identified based on the FDR and 'top' means all categories\n",
       "          are ranked based on FDR and then select top categories as the\n",
       "          enriched categories. The default is 'fdr'.\n",
       "\n",
       "fdrMethod: For the ORA method, WebGestaltR supports five FDR methods:\n",
       "          'holm', 'hochberg', 'hommel', 'bonferroni', 'BH' and 'BY'.\n",
       "          The default is 'BH'.\n",
       "\n",
       "  fdrThr: The significant threshold for the 'fdr' method. The default\n",
       "          is '0.05'.\n",
       "\n",
       "  topThr: The threshold for the 'top' method. The default is '10'.\n",
       "\n",
       "reportNum: The number of enriched categories visualized in the final\n",
       "          report. The default is '20'. A larger 'reportNum' may be slow\n",
       "          to render in the report.\n",
       "\n",
       "  perNum: The number of permutations for the GSEA method. The default\n",
       "          is '1000'.\n",
       "\n",
       "   gseaP: The exponential scaling factor of the phenotype score. The\n",
       "          default is '1'. When p=0, ES reduces to standard K-S\n",
       "          statistics (See original paper for more details).\n",
       "\n",
       "isOutput: If 'isOutput' is TRUE, WebGestaltR will create a folder named\n",
       "          by the 'projectName' and save the results in the folder.\n",
       "          Otherwise, WebGestaltR will only return an R 'data.frame'\n",
       "          object containing the enrichment results. If hundreds of gene\n",
       "          list need to be analyzed simultaneously, it is better to set\n",
       "          'isOutput' to 'FALSE'. The default is 'TRUE'.\n",
       "\n",
       "outputDirectory: The output directory for the results.\n",
       "\n",
       "projectName: The name of the project. If 'projectName' is 'NULL',\n",
       "          WebGestaltR will use time stamp as the project name.\n",
       "\n",
       "dagColor: If 'dagColor' is 'binary', the significant terms in the DAG\n",
       "          structure will be colored by steel blue for ORA method or\n",
       "          steel blue (positive related) and dark orange (negative\n",
       "          related) for GSEA method. If 'dagColor' is 'continous', the\n",
       "          significant terms in the DAG structure will be colored by the\n",
       "          color gradient based on corresponding FDRs.\n",
       "\n",
       "saveRawGseaResult: Whether the raw result from GSEA is saved as a RDS\n",
       "          file, which can be used for plotting. Defaults to 'FALSE'.\n",
       "          The list includes\n",
       "\n",
       "          Enrichment_Results A data frame of GSEA results with\n",
       "              statistics\n",
       "\n",
       "          Running_Sums A matrix of running sum of scores for each gene\n",
       "              set\n",
       "\n",
       "          Items_in_Set A list with ranks of genes for each gene set\n",
       "\n",
       "gseaPlotFormat: The graphic format of GSEA enrichment plots. Either\n",
       "          'svg', 'png', or 'c(\"png\", \"svg\")' (default).\n",
       "\n",
       "setCoverNum: The number of expected gene sets after set cover to reduce\n",
       "          redundancy. It could get fewer sets if the coverage reaches\n",
       "          100%. The default is '10'.\n",
       "\n",
       "networkConstructionMethod: Netowrk construction method for NTA. Either\n",
       "          'Network_Retrieval_Prioritization' or 'Network_Expansion'.\n",
       "          Network Retrieval & Prioritization first uses random walk\n",
       "          analysis to calculate random walk probabilities for the input\n",
       "          seeds, then identifies the relationships among the seeds in\n",
       "          the selected network and returns a retrieval sub-network. The\n",
       "          seeds with the top random walk probabilities are highlighted\n",
       "          in the sub-network. Network Expansion first uses random walk\n",
       "          analysis to rank all genes in the selected network based on\n",
       "          their network proximity to the input seeds and then return an\n",
       "          expanded sub-network in which nodes are the input seeds and\n",
       "          their top ranking neighbors and edges represent their\n",
       "          relationships.\n",
       "\n",
       "neighborNum: The number of neighbors to include in NTA Network\n",
       "          Expansion method.\n",
       "\n",
       "highlightType: The type of nodes to highlight in the NTA Network\n",
       "          Expansion method, either 'Seeds' or 'Neighbors'.\n",
       "\n",
       "highlightSeedNum: The number of top input seeds to highlight in NTA\n",
       "          Network Retrieval & Prioritizaiton method.\n",
       "\n",
       "nThreads: The number of cores to use for GSEA and set cover, and in\n",
       "          batch function.\n",
       "\n",
       "   cache: A directory to save data cache for reuse. Defaults to 'NULL'\n",
       "          and disabled.\n",
       "\n",
       "hostName: The server URL for accessing data. Mostly for development\n",
       "          purposes.\n",
       "\n",
       "     ...: In batch function, passes parameters to WebGestaltR function.\n",
       "          Also handles backward compatibility for some parameters in\n",
       "          old versions.\n",
       "\n",
       "interestGeneFolder: Run WebGestaltR for gene list files in the folder.\n",
       "\n",
       "isParallel: If jobs are run parallelly in the batch.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     WebGestaltR function can perform three enrichment analyses: ORA\n",
       "     (Over-Representation Analysis) and GSEA (Gene Set Enrichment\n",
       "     Analysis).and NTA (Network Topology Analysis). Based on the\n",
       "     user-uploaded gene list or gene list with scores, WebGestaltR\n",
       "     function will first map the gene list to the entrez gene ids and\n",
       "     then summarize the gene list based on the GO (Gene Ontology) Slim.\n",
       "     After performing the enrichment analysis, WebGestaltR function\n",
       "     also returns a user-friendly HTML report containing GO Slim\n",
       "     summary and the enrichment analysis result. If functional\n",
       "     categories have DAG (directed acyclic graph) structure or genes in\n",
       "     the functional categories have network structure, those\n",
       "     relationship can also be visualized in the report.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     The WebGestaltR function returns a data frame containing the\n",
       "     enrichment analysis result and also outputs an user-friendly HTML\n",
       "     report if 'isOutput' is 'TRUE'.  The columns in the data frame\n",
       "     depend on the 'enrichMethod' and they are the following:\n",
       "\n",
       "     geneSet ID of the gene set.\n",
       "\n",
       "     description Description of the gene set if available.\n",
       "\n",
       "     link Link to the data source.\n",
       "\n",
       "     size The number of genes in the set after filtering by 'minNum'\n",
       "          and 'maxNum'.\n",
       "\n",
       "     overlap The number of mapped input genes that are annotated in the\n",
       "          gene set.\n",
       "\n",
       "     expect Expected number of input genes that are annotated in the\n",
       "          gene set.\n",
       "\n",
       "     enrichmentRatio Enrichment ratio, overlap / expect.\n",
       "\n",
       "     enrichmentScore Enrichment score, the maximum running sum of\n",
       "          scores for the ranked list.\n",
       "\n",
       "     normalizedEnrichmentScore Normalized enrichment score, normalized\n",
       "          against the average enrichment score of all permutations.\n",
       "\n",
       "     leadingEdgeNum Number of genes/phosphosites in the leading edge.\n",
       "\n",
       "     pValue P-value from hypergeometric test for ORA. For GSEA, please\n",
       "          refer to its original publication or online at <URL:\n",
       "          https://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm>.\n",
       "\n",
       "     FDR Corrected P-value for mulilple testing with 'fdrMethod' for\n",
       "          ORA.\n",
       "\n",
       "     overlapId The gene/phosphosite IDs of 'overlap' for ORA (entrez\n",
       "          gene IDs or phosphosite sequence).\n",
       "\n",
       "     leadingEdgeId Genes/phosphosites in the leading edge in entrez\n",
       "          gene ID or phosphosite sequence.\n",
       "\n",
       "     userId The gene/phosphosite IDs of 'overlap' for ORA or\n",
       "          'leadingEdgeId' for GSEA in User input IDs.\n",
       "\n",
       "     plotPath Path of the GSEA enrichment plot.\n",
       "\n",
       "     database Name of the source database if multiple enrichment\n",
       "          databases are given.\n",
       "\n",
       "     goId In NTA, like 'geneSet', the enriched GO terms of genes in the\n",
       "          returned subnetwork.\n",
       "\n",
       "     interestGene In NTA, the gene IDs in the subnetwork with 0/1\n",
       "          annotations indicating if it is from user input.\n",
       "\n",
       "     The WebGestaltRBatch function returns a list of enrichment\n",
       "     results.\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     ## Not run:\n",
       "     \n",
       "     ####### ORA example #########\n",
       "     geneFile <- system.file(\"extdata\", \"interestingGenes.txt\", package=\"WebGestaltR\")\n",
       "     refFile <- system.file(\"extdata\", \"referenceGenes.txt\", package=\"WebGestaltR\")\n",
       "     outputDirectory <- getwd()\n",
       "     enrichResult <- WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",\n",
       "       enrichDatabase=\"pathway_KEGG\", interestGeneFile=geneFile,\n",
       "       interestGeneType=\"genesymbol\", referenceGeneFile=refFile,\n",
       "       referenceGeneType=\"genesymbol\", isOutput=TRUE,\n",
       "       outputDirectory=outputDirectory, projectName=NULL)\n",
       "     \n",
       "     ####### GSEA example #########\n",
       "     rankFile <- system.file(\"extdata\", \"GeneRankList.rnk\", package=\"WebGestaltR\")\n",
       "     outputDirectory <- getwd()\n",
       "     enrichResult <- WebGestaltR(enrichMethod=\"GSEA\", organism=\"hsapiens\",\n",
       "       enrichDatabase=\"pathway_KEGG\", interestGeneFile=rankFile,\n",
       "       interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10, minNum=5,\n",
       "       outputDirectory=outputDirectory)\n",
       "     \n",
       "     ####### NTA example #########\n",
       "     enrichResult <- WebGestaltR(enrichMethod=\"NTA\", organism=\"hsapiens\",\n",
       "       enrichDatabase=\"network_PPI_BIOGRID\", interestGeneFile=geneFile,\n",
       "       interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10,\n",
       "       outputDirectory=getwd(), highlightSeedNum=10,\n",
       "       networkConstructionMethod=\"Network_Retrieval_Prioritization\")\n",
       "     ## End(Not run)\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?WebGestaltR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Alter the parameters of WebGestaltR at your convenience**\n",
    "\n",
    "The following code loops through all the categories and calls the WebGestaltR method to perform the analysis. The object 'result.cat' contains the result of the current category being queried and the object 'result' contains the concatenation of all categories."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " BP \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "104 11\n",
      " CC \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "93 11\n",
      " MF \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "32 11\n",
      " KEGG \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "15 11\n",
      " Wikipathway \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "5 11\n",
      " Reactome \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n",
      "210 11\n",
      " Panther \n",
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Performing the enrichment analysis...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum = minNum, :\n",
      "\"No significant gene set is identified based on FDR 0.2!\"\n"
     ]
    }
   ],
   "source": [
    "DEGenes=rownames(rra.results[1:200 ,]) # top 200 genes\n",
    "#DEGenes=rownames(meta.results[DEresult$padj<0.05 ,]) # Differentially Expressed genes, with adjusted p value > 0.05\n",
    "#if (length(DEGenes) > 200) {DEGenes=DEGenes[1:200]}\n",
    "\n",
    "# genedf.list will store genes and logFC for go plots\n",
    "#genedf.list=data.frame(ID=rownames(rra.results[DEGenes ,]),logFC=meta.results[DEGenes ,]$logFC.case.ctl)\n",
    "\n",
    "result=data.frame() # dataframe that will contain all categories results from a DE analysis file\n",
    "for (i in 1:length(categories.database)){ # for each category get the enrichment results\n",
    "    cat(\"\\n\",categories.short[[i]],\"\\n\") \n",
    "    result.cat=WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",enrichDatabase=categories.database[[i]],enrichDatabaseFile=NULL, \n",
    "             enrichDatabaseType=NULL,enrichDatabaseDescriptionFile=NULL,interestGeneFile=NULL, \n",
    "             interestGene=DEGenes,interestGeneType=\"genesymbol\",collapseMethod=\"mean\",referenceGeneFile=NULL,\n",
    "             referenceGene=NULL,referenceGeneType=\"genesymbol\",referenceSet=\"genome\", minNum=10, maxNum=500,\n",
    "             fdrMethod=\"BH\",sigMethod=\"fdr\",fdrThr=0.20,topThr=10,reportNum=20,perNum=1000,\n",
    "             isOutput=FALSE,outputDirectory=outpath,projectName=NULL,dagColor=\"binary\",hostName=\"http://www.webgestalt.org/\")\n",
    "    cat(dim(result.cat))\n",
    "    if(!is.null(dim(result.cat))){ # if there exists enrichment for this category\n",
    "        result.cat=cbind(category = categories.short[[i]], result.cat) # add column with category at front\n",
    "        result=rbind(result,result.cat)\n",
    "    } \n",
    "}\n",
    "filename=\"enrich_rra_paper\"\n",
    "write.table(result,paste(outpath,filename,sep=\"/\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-12-05T10:26:50.453448Z",
     "start_time": "2018-12-05T10:26:49.044Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 113 x 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>geneSet</th><th scope=col>description</th><th scope=col>link</th><th scope=col>size</th><th scope=col>overlap</th><th scope=col>expect</th><th scope=col>enrichmentRatio</th><th scope=col>pValue</th><th scope=col>FDR</th><th scope=col>overlapId</th><th scope=col>userId</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CC         </td><td>GO:0005743</td><td>mitochondrial inner membrane                              </td><td>http://amigo.geneontology.org/amigo/term/GO:0005743                                                                                                               </td><td>455</td><td>12</td><td>3.15469425</td><td> 3.803855</td><td>7.675239e-05</td><td>0.030318595</td><td>4711;4976;7385;10131;25875;29088;55744;55750;79590;84303;117145;285521</td><td>OPA1;MRPL24;LETMD1;CHCHD6;TRAP1;AGK;COA1;THEM4;MRPL15;COX18;NDUFB5;UQCRC2</td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:1990391</td><td>DNA repair complex                                        </td><td>http://amigo.geneontology.org/amigo/term/GO:1990391                                                                                                               </td><td> 40</td><td> 4</td><td>0.27733576</td><td>14.422951</td><td>1.657331e-04</td><td>0.030318595</td><td>4437;51317;64421;126074                                               </td><td>PHF21A;MSH3;SWSAP1;DCLRE1C                                               </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0031304</td><td>intrinsic component of mitochondrial inner membrane       </td><td>http://amigo.geneontology.org/amigo/term/GO:0031304                                                                                                               </td><td> 40</td><td> 4</td><td>0.27733576</td><td>14.422951</td><td>1.657331e-04</td><td>0.030318595</td><td>55744;55750;84303;285521                                              </td><td>CHCHD6;AGK;COA1;COX18                                                    </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0031305</td><td>integral component of mitochondrial inner membrane        </td><td>http://amigo.geneontology.org/amigo/term/GO:0031305                                                                                                               </td><td> 40</td><td> 4</td><td>0.27733576</td><td>14.422951</td><td>1.657331e-04</td><td>0.030318595</td><td>55744;55750;84303;285521                                              </td><td>CHCHD6;AGK;COA1;COX18                                                    </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0044455</td><td>mitochondrial membrane part                               </td><td>http://amigo.geneontology.org/amigo/term/GO:0044455                                                                                                               </td><td>229</td><td> 8</td><td>1.58774722</td><td> 5.038585</td><td>1.963640e-04</td><td>0.030318595</td><td>4711;4976;7385;10651;55744;55750;84303;285521                         </td><td>OPA1;CHCHD6;AGK;COA1;MTX2;COX18;NDUFB5;UQCRC2                            </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0033267</td><td>axon part                                                 </td><td>http://amigo.geneontology.org/amigo/term/GO:0033267                                                                                                               </td><td>374</td><td>10</td><td>2.59308934</td><td> 3.856404</td><td>2.785301e-04</td><td>0.035837533</td><td>1176;2596;3611;3786;4897;4976;5582;29993;54536;80228                  </td><td>OPA1;GAP43;EXOC6;NRCAM;ORAI2;AP3S1;PRKCG;PACSIN1;KCNQ3;ILK               </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0032592</td><td>integral component of mitochondrial membrane              </td><td>http://amigo.geneontology.org/amigo/term/GO:0032592                                                                                                               </td><td> 69</td><td> 4</td><td>0.47840418</td><td> 8.361131</td><td>1.343200e-03</td><td>0.136745793</td><td>55744;55750;84303;285521                                              </td><td>CHCHD6;AGK;COA1;COX18                                                    </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0098573</td><td>intrinsic component of mitochondrial membrane             </td><td>http://amigo.geneontology.org/amigo/term/GO:0098573                                                                                                               </td><td> 70</td><td> 4</td><td>0.48533758</td><td> 8.241686</td><td>1.417055e-03</td><td>0.136745793</td><td>55744;55750;84303;285521                                              </td><td>CHCHD6;AGK;COA1;COX18                                                    </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0005741</td><td>mitochondrial outer membrane                              </td><td>http://amigo.geneontology.org/amigo/term/GO:0005741                                                                                                               </td><td>180</td><td> 6</td><td>1.24801091</td><td> 4.807650</td><td>1.599920e-03</td><td>0.137237570</td><td>4976;5521;10651;25875;55750;84816                                     </td><td>OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0005758</td><td>mitochondrial intermembrane space                         </td><td>http://amigo.geneontology.org/amigo/term/GO:0005758                                                                                                               </td><td> 79</td><td> 4</td><td>0.54773812</td><td> 7.302760</td><td>2.213754e-03</td><td>0.170901805</td><td>4976;10131;55750;117145                                               </td><td>OPA1;TRAP1;AGK;THEM4                                                     </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0098798</td><td>mitochondrial protein complex                             </td><td>http://amigo.geneontology.org/amigo/term/GO:0098798                                                                                                               </td><td>266</td><td> 7</td><td>1.84428279</td><td> 3.795513</td><td>2.553334e-03</td><td>0.173338049</td><td>4711;7385;10651;29088;55750;79590;84303                               </td><td>MRPL24;CHCHD6;AGK;MRPL15;MTX2;NDUFB5;UQCRC2                              </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0031968</td><td>organelle outer membrane                                  </td><td>http://amigo.geneontology.org/amigo/term/GO:0031968                                                                                                               </td><td>204</td><td> 6</td><td>1.41441237</td><td> 4.242044</td><td>2.995177e-03</td><td>0.173338049</td><td>4976;5521;10651;25875;55750;84816                                     </td><td>OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0019867</td><td>outer membrane                                            </td><td>http://amigo.geneontology.org/amigo/term/GO:0019867                                                                                                               </td><td>206</td><td> 6</td><td>1.42827915</td><td> 4.200859</td><td>3.142824e-03</td><td>0.173338049</td><td>4976;5521;10651;25875;55750;84816                                     </td><td>OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0031970</td><td>organelle envelope lumen                                  </td><td>http://amigo.geneontology.org/amigo/term/GO:0031970                                                                                                               </td><td> 87</td><td> 4</td><td>0.60320527</td><td> 6.631242</td><td>3.143436e-03</td><td>0.173338049</td><td>4976;10131;55750;117145                                               </td><td>OPA1;TRAP1;AGK;THEM4                                                     </td></tr>\n",
       "\t<tr><td>CC         </td><td>GO:0048188</td><td>Set1C/COMPASS complex                                     </td><td>http://amigo.geneontology.org/amigo/term/GO:0048188                                                                                                               </td><td> 13</td><td> 2</td><td>0.09013412</td><td>22.189155</td><td>3.537714e-03</td><td>0.182074371</td><td>5929;80335                                                            </td><td>WDR82;RBBP5                                                              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0070402</td><td>NADPH binding                                             </td><td>http://amigo.geneontology.org/amigo/term/GO:0070402                                                                                                               </td><td> 13</td><td> 3</td><td>0.10839182</td><td>27.677366</td><td>1.526016e-04</td><td>0.173532316</td><td>2194;6715;84869                                                       </td><td>CBR4;SRD5A1;FASN                                                         </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0031435</td><td>mitogen-activated protein kinase kinase kinase binding    </td><td>http://amigo.geneontology.org/amigo/term/GO:0031435                                                                                                               </td><td> 19</td><td> 3</td><td>0.15841881</td><td>18.937145</td><td>4.984199e-04</td><td>0.173532316</td><td>998;6929;142679                                                       </td><td>TCF3;DUSP19;CDC42                                                        </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0003924</td><td>GTPase activity                                           </td><td>http://amigo.geneontology.org/amigo/term/GO:0003924                                                                                                               </td><td>293</td><td> 9</td><td>2.44298482</td><td> 3.684018</td><td>7.904017e-04</td><td>0.173532316</td><td>381;998;2785;4976;7280;8934;10381;27175;64121                         </td><td>OPA1;TUBB2A;TUBG2;RAB29;TUBB3;ARF5;GNG3;CDC42;RRAGC                      </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0046961</td><td>proton-transporting ATPase activity, rotational mechanism </td><td>http://amigo.geneontology.org/amigo/term/GO:0046961                                                                                                               </td><td> 23</td><td> 3</td><td>0.19177014</td><td>15.643728</td><td>8.889877e-04</td><td>0.173532316</td><td>529;51382;51606                                                       </td><td>ATP6V1H;ATP6V1D;ATP6V1E1                                                 </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0005525</td><td>GTP binding                                               </td><td>http://amigo.geneontology.org/amigo/term/GO:0005525                                                                                                               </td><td>369</td><td>10</td><td>3.07666007</td><td> 3.250278</td><td>1.062333e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0032550</td><td>purine ribonucleoside binding                             </td><td>http://amigo.geneontology.org/amigo/term/GO:0032550                                                                                                               </td><td>373</td><td>10</td><td>3.11001140</td><td> 3.215422</td><td>1.152068e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0001883</td><td>purine nucleoside binding                                 </td><td>http://amigo.geneontology.org/amigo/term/GO:0001883                                                                                                               </td><td>376</td><td>10</td><td>3.13502489</td><td> 3.189767</td><td>1.223363e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0032549</td><td>ribonucleoside binding                                    </td><td>http://amigo.geneontology.org/amigo/term/GO:0032549                                                                                                               </td><td>377</td><td>10</td><td>3.14336273</td><td> 3.181306</td><td>1.247915e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0001882</td><td>nucleoside binding                                        </td><td>http://amigo.geneontology.org/amigo/term/GO:0001882                                                                                                               </td><td>384</td><td>10</td><td>3.20172755</td><td> 3.123314</td><td>1.431292e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0019001</td><td>guanyl nucleotide binding                                 </td><td>http://amigo.geneontology.org/amigo/term/GO:0019001                                                                                                               </td><td>390</td><td>10</td><td>3.25175454</td><td> 3.075263</td><td>1.605429e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0032561</td><td>guanyl ribonucleotide binding                             </td><td>http://amigo.geneontology.org/amigo/term/GO:0032561                                                                                                               </td><td>390</td><td>10</td><td>3.25175454</td><td> 3.075263</td><td>1.605429e-03</td><td>0.173532316</td><td>381;998;4976;7280;8934;10381;27175;54552;55915;64121                  </td><td>OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              </td></tr>\n",
       "\t<tr><td>MF         </td><td>GO:0036442</td><td>proton-exporting ATPase activity                          </td><td>http://amigo.geneontology.org/amigo/term/GO:0036442                                                                                                               </td><td> 30</td><td> 3</td><td>0.25013496</td><td>11.993525</td><td>1.952997e-03</td><td>0.193509455</td><td>529;51382;51606                                                       </td><td>ATP6V1H;ATP6V1D;ATP6V1E1                                                 </td></tr>\n",
       "\t<tr><td>KEGG       </td><td>hsa05120  </td><td>Epithelial cell signaling in Helicobacter pylori infection</td><td>http://www.kegg.jp/kegg-bin/show_pathway?hsa05120+4790+4792+51382+51606+529+5970+998                                                                              </td><td> 68</td><td> 7</td><td>0.75565671</td><td> 9.263466</td><td>9.091745e-06</td><td>0.002863900</td><td>4790;4792;51382;51606;529;5970;998                                    </td><td>NFKB1;ATP6V1H;ATP6V1D;RELA;ATP6V1E1;NFKBIA;CDC42                         </td></tr>\n",
       "\t<tr><td>KEGG       </td><td>hsa05165  </td><td>Human papillomavirus infection                            </td><td>http://www.kegg.jp/kegg-bin/show_pathway?hsa05165+1385+27175+3693+4790+51382+51606+529+5521+5970+7148+998                                                         </td><td>339</td><td>11</td><td>3.76717097</td><td> 2.919963</td><td>1.229368e-03</td><td>0.193625407</td><td>1385;27175;3693;4790;51382;51606;529;5521;5970;7148;998               </td><td>NFKB1;ATP6V1H;ATP6V1D;TUBG2;TNXB;RELA;CREB1;PPP2R2B;ATP6V1E1;ITGB5;CDC42 </td></tr>\n",
       "\t<tr><td>Wikipathway</td><td><span style=white-space:pre-wrap>WP195     </span></td><td><span style=white-space:pre-wrap>IL-1 signaling pathway                                    </span></td><td>http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP195&amp;label[]=NFKB1&amp;label[]=HSPB2&amp;label[]=UBE2N&amp;label[]=MAPKAPK2&amp;label[]=RELA&amp;label[]=NFKBIA&amp;colors=steelblue</td><td> 57</td><td> 6</td><td>0.53195701</td><td>11.279107</td><td>1.310609e-05</td><td>0.006146754</td><td><span style=white-space:pre-wrap>9261;3316;7334;4790;4792;5970                                         </span></td><td><span style=white-space:pre-wrap>NFKB1;HSPB2;UBE2N;MAPKAPK2;RELA;NFKBIA                                   </span></td></tr>\n",
       "\t<tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-975155 </td><td>MyD88 dependent cascade initiated on endosome         </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-975155 </td><td> 94</td><td> 6</td><td>0.9530036</td><td> 6.295884</td><td>0.0003729130</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-166058 </td><td>MyD88:MAL(TIRAP) cascade initiated on plasma membrane </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-166058 </td><td> 95</td><td> 6</td><td>0.9631419</td><td> 6.229611</td><td>0.0003948163</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-168188 </td><td>Toll Like Receptor TLR6:TLR2 Cascade                  </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-168188 </td><td> 95</td><td> 6</td><td>0.9631419</td><td> 6.229611</td><td>0.0003948163</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-168138 </td><td>Toll Like Receptor 9 (TLR9) Cascade                   </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-168138 </td><td> 98</td><td> 6</td><td>0.9935569</td><td> 6.038909</td><td>0.0004666130</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-168179 </td><td>Toll Like Receptor TLR1:TLR2 Cascade                  </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-168179 </td><td> 98</td><td> 6</td><td>0.9935569</td><td> 6.038909</td><td>0.0004666130</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-181438 </td><td>Toll Like Receptor 2 (TLR2) Cascade                   </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-181438 </td><td> 98</td><td> 6</td><td>0.9935569</td><td> 6.038909</td><td>0.0004666130</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-168164 </td><td>Toll Like Receptor 3 (TLR3) Cascade                   </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-168164 </td><td> 99</td><td> 6</td><td>1.0036953</td><td> 5.977910</td><td>0.0004926798</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-193639 </td><td>p75NTR signals via NF-kB                              </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-193639 </td><td> 16</td><td> 3</td><td>0.1622134</td><td>18.494159</td><td>0.0005154108</td><td>0.03982766</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-166166 </td><td>MyD88-independent TLR4 cascade                        </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-166166 </td><td>100</td><td> 6</td><td>1.0138336</td><td> 5.918131</td><td>0.0005198679</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-937061 </td><td>TRIF(TICAM1)-mediated TLR4 signaling                  </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-937061 </td><td>100</td><td> 6</td><td>1.0138336</td><td> 5.918131</td><td>0.0005198679</td><td>0.03982766</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-1810476</td><td>RIP-mediated NFkB activation via ZBP1                 </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-1810476</td><td> 17</td><td> 3</td><td>0.1723517</td><td>17.406267</td><td>0.0006212549</td><td>0.04509003</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-1606322</td><td>ZBP1(DAI) mediated induction of type I IFNs           </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-1606322</td><td> 21</td><td> 3</td><td>0.2129051</td><td>14.090788</td><td>0.0011797998</td><td>0.07797868</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-373760 </td><td>L1CAM interactions                                    </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-373760 </td><td>117</td><td> 6</td><td>1.1861853</td><td> 5.058231</td><td>0.0011874925</td><td>0.07797868</td><td>10381;11280;2596;3786;4897;7280                       </td><td>TUBB2A;SCN11A;GAP43;NRCAM;TUBB3;KCNQ3                        </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-5260271</td><td>Diseases of Immune System                             </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-5260271</td><td> 24</td><td> 3</td><td>0.2433201</td><td>12.329439</td><td>0.0017562199</td><td>0.10090947</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-5602358</td><td>Diseases associated with the TLR signaling cascade    </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-5602358</td><td> 24</td><td> 3</td><td>0.2433201</td><td>12.329439</td><td>0.0017562199</td><td>0.10090947</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-933542 </td><td>TRAF6 mediated NF-kB activation                       </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-933542 </td><td> 24</td><td> 3</td><td>0.2433201</td><td>12.329439</td><td>0.0017562199</td><td>0.10090947</td><td>4790;4792;5970                                        </td><td>NFKB1;RELA;NFKBIA                                            </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-166016 </td><td>Toll Like Receptor 4 (TLR4) Cascade                   </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-166016 </td><td>130</td><td> 6</td><td>1.3179837</td><td> 4.552408</td><td>0.0020369838</td><td>0.11236003</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-77387  </td><td>Insulin receptor recycling                            </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-77387  </td><td> 26</td><td> 3</td><td>0.2635967</td><td>11.381021</td><td>0.0022230880</td><td>0.11790917</td><td>51382;51606;529                                       </td><td>ATP6V1H;ATP6V1D;ATP6V1E1                                     </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-193704 </td><td>p75 NTR receptor-mediated signalling                  </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-193704 </td><td> 97</td><td> 5</td><td>0.9834186</td><td> 5.084305</td><td>0.0030023844</td><td>0.15075403</td><td>23229;27018;4790;4792;5970                            </td><td>NFKB1;ARHGEF9;RELA;NFKBIA;BEX3                               </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-73887  </td><td>Death Receptor Signalling                             </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-73887  </td><td>141</td><td> 6</td><td>1.4295054</td><td> 4.197256</td><td>0.0030609954</td><td>0.15075403</td><td>23229;27018;331;4790;4792;5970                        </td><td>NFKB1;ARHGEF9;RELA;NFKBIA;XIAP;BEX3                          </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-5626467</td><td>RHO GTPases activate IQGAPs                           </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-5626467</td><td> 30</td><td> 3</td><td>0.3041501</td><td> 9.863551</td><td>0.0033709817</td><td>0.16029599</td><td>10381;7280;998                                        </td><td>TUBB2A;TUBB3;CDC42                                           </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-917977 </td><td>Transferrin endocytosis and recycling                 </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-917977 </td><td> 31</td><td> 3</td><td>0.3142884</td><td> 9.545372</td><td>0.0037048910</td><td>0.16127881</td><td>51382;51606;529                                       </td><td>ATP6V1H;ATP6V1D;ATP6V1E1                                     </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-445095 </td><td>Interaction between L1 and Ankyrins                   </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-445095 </td><td> 31</td><td> 3</td><td>0.3142884</td><td> 9.545372</td><td>0.0037048910</td><td>0.16127881</td><td>11280;3786;4897                                       </td><td>SCN11A;NRCAM;KCNQ3                                           </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-1834949</td><td>Cytosolic sensors of pathogen-associated DNA          </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-1834949</td><td> 63</td><td> 4</td><td>0.6387152</td><td> 6.262572</td><td>0.0037593386</td><td>0.16127881</td><td>4790;4792;5970;9208                                   </td><td>NFKB1;RELA;NFKBIA;LRRFIP1                                    </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-450294 </td><td>MAP kinase activation                                 </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-450294 </td><td> 64</td><td> 4</td><td>0.6488535</td><td> 6.164720</td><td>0.0039791391</td><td>0.16127881</td><td>1385;4790;7334;9261                                   </td><td>NFKB1;UBE2N;MAPKAPK2;CREB1                                   </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-1222556</td><td>ROS, RNS production in phagocytes                     </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-1222556</td><td> 32</td><td> 3</td><td>0.3244268</td><td> 9.247079</td><td>0.0040583008</td><td>0.16127881</td><td>51382;51606;529                                       </td><td>ATP6V1H;ATP6V1D;ATP6V1E1                                     </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-112316 </td><td>Neuronal System                                       </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-112316 </td><td>368</td><td>10</td><td>3.7309077</td><td> 2.680313</td><td>0.0040933707</td><td>0.16127881</td><td>10313;10611;1385;23229;2785;3786;5582;64130;83988;8500</td><td>ARHGEF9;PPFIA1;RTN3;PDLIM5;CREB1;PRKCG;GNG3;NCALD;KCNQ3;LIN7B</td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-168898 </td><td>Toll-like Receptor Cascades                           </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-168898 </td><td>155</td><td> 6</td><td>1.5714421</td><td> 3.818149</td><td>0.0048699692</td><td>0.18654688</td><td>1385;4790;4792;5970;7334;9261                         </td><td>NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-193692 </td><td>Regulated proteolysis of p75NTR                       </td><td>http://reactome.org/PathwayBrowser/#/R-HSA-193692 </td><td> 11</td><td> 2</td><td>0.1115217</td><td>17.933730</td><td>0.0052762304</td><td>0.19664653</td><td>4790;5970                                             </td><td>NFKB1;RELA                                                   </td></tr>\n",
       "\t<tr><td>Reactome</td><td>R-HSA-1445148</td><td>Translocation of SLC2A4 (GLUT4) to the plasma membrane</td><td>http://reactome.org/PathwayBrowser/#/R-HSA-1445148</td><td> 70</td><td> 4</td><td>0.7096835</td><td> 5.636315</td><td>0.0054826759</td><td>0.19896342</td><td>10381;54536;5571;7280                                 </td><td>TUBB2A;EXOC6;TUBB3;PRKAG1                                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 113 x 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " category & geneSet & description & link & size & overlap & expect & enrichmentRatio & pValue & FDR & overlapId & userId\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t CC          & GO:0005743 & mitochondrial inner membrane                               & http://amigo.geneontology.org/amigo/term/GO:0005743                                                                                                                & 455 & 12 & 3.15469425 &  3.803855 & 7.675239e-05 & 0.030318595 & 4711;4976;7385;10131;25875;29088;55744;55750;79590;84303;117145;285521 & OPA1;MRPL24;LETMD1;CHCHD6;TRAP1;AGK;COA1;THEM4;MRPL15;COX18;NDUFB5;UQCRC2\\\\\n",
       "\t CC          & GO:1990391 & DNA repair complex                                         & http://amigo.geneontology.org/amigo/term/GO:1990391                                                                                                                &  40 &  4 & 0.27733576 & 14.422951 & 1.657331e-04 & 0.030318595 & 4437;51317;64421;126074                                                & PHF21A;MSH3;SWSAP1;DCLRE1C                                               \\\\\n",
       "\t CC          & GO:0031304 & intrinsic component of mitochondrial inner membrane        & http://amigo.geneontology.org/amigo/term/GO:0031304                                                                                                                &  40 &  4 & 0.27733576 & 14.422951 & 1.657331e-04 & 0.030318595 & 55744;55750;84303;285521                                               & CHCHD6;AGK;COA1;COX18                                                    \\\\\n",
       "\t CC          & GO:0031305 & integral component of mitochondrial inner membrane         & http://amigo.geneontology.org/amigo/term/GO:0031305                                                                                                                &  40 &  4 & 0.27733576 & 14.422951 & 1.657331e-04 & 0.030318595 & 55744;55750;84303;285521                                               & CHCHD6;AGK;COA1;COX18                                                    \\\\\n",
       "\t CC          & GO:0044455 & mitochondrial membrane part                                & http://amigo.geneontology.org/amigo/term/GO:0044455                                                                                                                & 229 &  8 & 1.58774722 &  5.038585 & 1.963640e-04 & 0.030318595 & 4711;4976;7385;10651;55744;55750;84303;285521                          & OPA1;CHCHD6;AGK;COA1;MTX2;COX18;NDUFB5;UQCRC2                            \\\\\n",
       "\t CC          & GO:0033267 & axon part                                                  & http://amigo.geneontology.org/amigo/term/GO:0033267                                                                                                                & 374 & 10 & 2.59308934 &  3.856404 & 2.785301e-04 & 0.035837533 & 1176;2596;3611;3786;4897;4976;5582;29993;54536;80228                   & OPA1;GAP43;EXOC6;NRCAM;ORAI2;AP3S1;PRKCG;PACSIN1;KCNQ3;ILK               \\\\\n",
       "\t CC          & GO:0032592 & integral component of mitochondrial membrane               & http://amigo.geneontology.org/amigo/term/GO:0032592                                                                                                                &  69 &  4 & 0.47840418 &  8.361131 & 1.343200e-03 & 0.136745793 & 55744;55750;84303;285521                                               & CHCHD6;AGK;COA1;COX18                                                    \\\\\n",
       "\t CC          & GO:0098573 & intrinsic component of mitochondrial membrane              & http://amigo.geneontology.org/amigo/term/GO:0098573                                                                                                                &  70 &  4 & 0.48533758 &  8.241686 & 1.417055e-03 & 0.136745793 & 55744;55750;84303;285521                                               & CHCHD6;AGK;COA1;COX18                                                    \\\\\n",
       "\t CC          & GO:0005741 & mitochondrial outer membrane                               & http://amigo.geneontology.org/amigo/term/GO:0005741                                                                                                                & 180 &  6 & 1.24801091 &  4.807650 & 1.599920e-03 & 0.137237570 & 4976;5521;10651;25875;55750;84816                                      & OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \\\\\n",
       "\t CC          & GO:0005758 & mitochondrial intermembrane space                          & http://amigo.geneontology.org/amigo/term/GO:0005758                                                                                                                &  79 &  4 & 0.54773812 &  7.302760 & 2.213754e-03 & 0.170901805 & 4976;10131;55750;117145                                                & OPA1;TRAP1;AGK;THEM4                                                     \\\\\n",
       "\t CC          & GO:0098798 & mitochondrial protein complex                              & http://amigo.geneontology.org/amigo/term/GO:0098798                                                                                                                & 266 &  7 & 1.84428279 &  3.795513 & 2.553334e-03 & 0.173338049 & 4711;7385;10651;29088;55750;79590;84303                                & MRPL24;CHCHD6;AGK;MRPL15;MTX2;NDUFB5;UQCRC2                              \\\\\n",
       "\t CC          & GO:0031968 & organelle outer membrane                                   & http://amigo.geneontology.org/amigo/term/GO:0031968                                                                                                                & 204 &  6 & 1.41441237 &  4.242044 & 2.995177e-03 & 0.173338049 & 4976;5521;10651;25875;55750;84816                                      & OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \\\\\n",
       "\t CC          & GO:0019867 & outer membrane                                             & http://amigo.geneontology.org/amigo/term/GO:0019867                                                                                                                & 206 &  6 & 1.42827915 &  4.200859 & 3.142824e-03 & 0.173338049 & 4976;5521;10651;25875;55750;84816                                      & OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \\\\\n",
       "\t CC          & GO:0031970 & organelle envelope lumen                                   & http://amigo.geneontology.org/amigo/term/GO:0031970                                                                                                                &  87 &  4 & 0.60320527 &  6.631242 & 3.143436e-03 & 0.173338049 & 4976;10131;55750;117145                                                & OPA1;TRAP1;AGK;THEM4                                                     \\\\\n",
       "\t CC          & GO:0048188 & Set1C/COMPASS complex                                      & http://amigo.geneontology.org/amigo/term/GO:0048188                                                                                                                &  13 &  2 & 0.09013412 & 22.189155 & 3.537714e-03 & 0.182074371 & 5929;80335                                                             & WDR82;RBBP5                                                              \\\\\n",
       "\t MF          & GO:0070402 & NADPH binding                                              & http://amigo.geneontology.org/amigo/term/GO:0070402                                                                                                                &  13 &  3 & 0.10839182 & 27.677366 & 1.526016e-04 & 0.173532316 & 2194;6715;84869                                                        & CBR4;SRD5A1;FASN                                                         \\\\\n",
       "\t MF          & GO:0031435 & mitogen-activated protein kinase kinase kinase binding     & http://amigo.geneontology.org/amigo/term/GO:0031435                                                                                                                &  19 &  3 & 0.15841881 & 18.937145 & 4.984199e-04 & 0.173532316 & 998;6929;142679                                                        & TCF3;DUSP19;CDC42                                                        \\\\\n",
       "\t MF          & GO:0003924 & GTPase activity                                            & http://amigo.geneontology.org/amigo/term/GO:0003924                                                                                                                & 293 &  9 & 2.44298482 &  3.684018 & 7.904017e-04 & 0.173532316 & 381;998;2785;4976;7280;8934;10381;27175;64121                          & OPA1;TUBB2A;TUBG2;RAB29;TUBB3;ARF5;GNG3;CDC42;RRAGC                      \\\\\n",
       "\t MF          & GO:0046961 & proton-transporting ATPase activity, rotational mechanism  & http://amigo.geneontology.org/amigo/term/GO:0046961                                                                                                                &  23 &  3 & 0.19177014 & 15.643728 & 8.889877e-04 & 0.173532316 & 529;51382;51606                                                        & ATP6V1H;ATP6V1D;ATP6V1E1                                                 \\\\\n",
       "\t MF          & GO:0005525 & GTP binding                                                & http://amigo.geneontology.org/amigo/term/GO:0005525                                                                                                                & 369 & 10 & 3.07666007 &  3.250278 & 1.062333e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0032550 & purine ribonucleoside binding                              & http://amigo.geneontology.org/amigo/term/GO:0032550                                                                                                                & 373 & 10 & 3.11001140 &  3.215422 & 1.152068e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0001883 & purine nucleoside binding                                  & http://amigo.geneontology.org/amigo/term/GO:0001883                                                                                                                & 376 & 10 & 3.13502489 &  3.189767 & 1.223363e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0032549 & ribonucleoside binding                                     & http://amigo.geneontology.org/amigo/term/GO:0032549                                                                                                                & 377 & 10 & 3.14336273 &  3.181306 & 1.247915e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0001882 & nucleoside binding                                         & http://amigo.geneontology.org/amigo/term/GO:0001882                                                                                                                & 384 & 10 & 3.20172755 &  3.123314 & 1.431292e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0019001 & guanyl nucleotide binding                                  & http://amigo.geneontology.org/amigo/term/GO:0019001                                                                                                                & 390 & 10 & 3.25175454 &  3.075263 & 1.605429e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0032561 & guanyl ribonucleotide binding                              & http://amigo.geneontology.org/amigo/term/GO:0032561                                                                                                                & 390 & 10 & 3.25175454 &  3.075263 & 1.605429e-03 & 0.173532316 & 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   & OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \\\\\n",
       "\t MF          & GO:0036442 & proton-exporting ATPase activity                           & http://amigo.geneontology.org/amigo/term/GO:0036442                                                                                                                &  30 &  3 & 0.25013496 & 11.993525 & 1.952997e-03 & 0.193509455 & 529;51382;51606                                                        & ATP6V1H;ATP6V1D;ATP6V1E1                                                 \\\\\n",
       "\t KEGG        & hsa05120   & Epithelial cell signaling in Helicobacter pylori infection & http://www.kegg.jp/kegg-bin/show\\_pathway?hsa05120+4790+4792+51382+51606+529+5970+998                                                                               &  68 &  7 & 0.75565671 &  9.263466 & 9.091745e-06 & 0.002863900 & 4790;4792;51382;51606;529;5970;998                                     & NFKB1;ATP6V1H;ATP6V1D;RELA;ATP6V1E1;NFKBIA;CDC42                         \\\\\n",
       "\t KEGG        & hsa05165   & Human papillomavirus infection                             & http://www.kegg.jp/kegg-bin/show\\_pathway?hsa05165+1385+27175+3693+4790+51382+51606+529+5521+5970+7148+998                                                          & 339 & 11 & 3.76717097 &  2.919963 & 1.229368e-03 & 0.193625407 & 1385;27175;3693;4790;51382;51606;529;5521;5970;7148;998                & NFKB1;ATP6V1H;ATP6V1D;TUBG2;TNXB;RELA;CREB1;PPP2R2B;ATP6V1E1;ITGB5;CDC42 \\\\\n",
       "\t Wikipathway & WP195      & IL-1 signaling pathway                                     & http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP195\\&label{[}{]}=NFKB1\\&label{[}{]}=HSPB2\\&label{[}{]}=UBE2N\\&label{[}{]}=MAPKAPK2\\&label{[}{]}=RELA\\&label{[}{]}=NFKBIA\\&colors=steelblue &  57 &  6 & 0.53195701 & 11.279107 & 1.310609e-05 & 0.006146754 & 9261;3316;7334;4790;4792;5970                                          & NFKB1;HSPB2;UBE2N;MAPKAPK2;RELA;NFKBIA                                   \\\\\n",
       "\t ... & ... & ... & ... & ... & ... & ... & ... & ... & ... & ... & ...\\\\\n",
       "\t Reactome & R-HSA-975155  & MyD88 dependent cascade initiated on endosome          & http://reactome.org/PathwayBrowser/\\#/R-HSA-975155  &  94 &  6 & 0.9530036 &  6.295884 & 0.0003729130 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-166058  & MyD88:MAL(TIRAP) cascade initiated on plasma membrane  & http://reactome.org/PathwayBrowser/\\#/R-HSA-166058  &  95 &  6 & 0.9631419 &  6.229611 & 0.0003948163 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-168188  & Toll Like Receptor TLR6:TLR2 Cascade                   & http://reactome.org/PathwayBrowser/\\#/R-HSA-168188  &  95 &  6 & 0.9631419 &  6.229611 & 0.0003948163 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-168138  & Toll Like Receptor 9 (TLR9) Cascade                    & http://reactome.org/PathwayBrowser/\\#/R-HSA-168138  &  98 &  6 & 0.9935569 &  6.038909 & 0.0004666130 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-168179  & Toll Like Receptor TLR1:TLR2 Cascade                   & http://reactome.org/PathwayBrowser/\\#/R-HSA-168179  &  98 &  6 & 0.9935569 &  6.038909 & 0.0004666130 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-181438  & Toll Like Receptor 2 (TLR2) Cascade                    & http://reactome.org/PathwayBrowser/\\#/R-HSA-181438  &  98 &  6 & 0.9935569 &  6.038909 & 0.0004666130 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-168164  & Toll Like Receptor 3 (TLR3) Cascade                    & http://reactome.org/PathwayBrowser/\\#/R-HSA-168164  &  99 &  6 & 1.0036953 &  5.977910 & 0.0004926798 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-193639  & p75NTR signals via NF-kB                               & http://reactome.org/PathwayBrowser/\\#/R-HSA-193639  &  16 &  3 & 0.1622134 & 18.494159 & 0.0005154108 & 0.03982766 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-166166  & MyD88-independent TLR4 cascade                         & http://reactome.org/PathwayBrowser/\\#/R-HSA-166166  & 100 &  6 & 1.0138336 &  5.918131 & 0.0005198679 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-937061  & TRIF(TICAM1)-mediated TLR4 signaling                   & http://reactome.org/PathwayBrowser/\\#/R-HSA-937061  & 100 &  6 & 1.0138336 &  5.918131 & 0.0005198679 & 0.03982766 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-1810476 & RIP-mediated NFkB activation via ZBP1                  & http://reactome.org/PathwayBrowser/\\#/R-HSA-1810476 &  17 &  3 & 0.1723517 & 17.406267 & 0.0006212549 & 0.04509003 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-1606322 & ZBP1(DAI) mediated induction of type I IFNs            & http://reactome.org/PathwayBrowser/\\#/R-HSA-1606322 &  21 &  3 & 0.2129051 & 14.090788 & 0.0011797998 & 0.07797868 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-373760  & L1CAM interactions                                     & http://reactome.org/PathwayBrowser/\\#/R-HSA-373760  & 117 &  6 & 1.1861853 &  5.058231 & 0.0011874925 & 0.07797868 & 10381;11280;2596;3786;4897;7280                        & TUBB2A;SCN11A;GAP43;NRCAM;TUBB3;KCNQ3                        \\\\\n",
       "\t Reactome & R-HSA-5260271 & Diseases of Immune System                              & http://reactome.org/PathwayBrowser/\\#/R-HSA-5260271 &  24 &  3 & 0.2433201 & 12.329439 & 0.0017562199 & 0.10090947 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-5602358 & Diseases associated with the TLR signaling cascade     & http://reactome.org/PathwayBrowser/\\#/R-HSA-5602358 &  24 &  3 & 0.2433201 & 12.329439 & 0.0017562199 & 0.10090947 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-933542  & TRAF6 mediated NF-kB activation                        & http://reactome.org/PathwayBrowser/\\#/R-HSA-933542  &  24 &  3 & 0.2433201 & 12.329439 & 0.0017562199 & 0.10090947 & 4790;4792;5970                                         & NFKB1;RELA;NFKBIA                                            \\\\\n",
       "\t Reactome & R-HSA-166016  & Toll Like Receptor 4 (TLR4) Cascade                    & http://reactome.org/PathwayBrowser/\\#/R-HSA-166016  & 130 &  6 & 1.3179837 &  4.552408 & 0.0020369838 & 0.11236003 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-77387   & Insulin receptor recycling                             & http://reactome.org/PathwayBrowser/\\#/R-HSA-77387   &  26 &  3 & 0.2635967 & 11.381021 & 0.0022230880 & 0.11790917 & 51382;51606;529                                        & ATP6V1H;ATP6V1D;ATP6V1E1                                     \\\\\n",
       "\t Reactome & R-HSA-193704  & p75 NTR receptor-mediated signalling                   & http://reactome.org/PathwayBrowser/\\#/R-HSA-193704  &  97 &  5 & 0.9834186 &  5.084305 & 0.0030023844 & 0.15075403 & 23229;27018;4790;4792;5970                             & NFKB1;ARHGEF9;RELA;NFKBIA;BEX3                               \\\\\n",
       "\t Reactome & R-HSA-73887   & Death Receptor Signalling                              & http://reactome.org/PathwayBrowser/\\#/R-HSA-73887   & 141 &  6 & 1.4295054 &  4.197256 & 0.0030609954 & 0.15075403 & 23229;27018;331;4790;4792;5970                         & NFKB1;ARHGEF9;RELA;NFKBIA;XIAP;BEX3                          \\\\\n",
       "\t Reactome & R-HSA-5626467 & RHO GTPases activate IQGAPs                            & http://reactome.org/PathwayBrowser/\\#/R-HSA-5626467 &  30 &  3 & 0.3041501 &  9.863551 & 0.0033709817 & 0.16029599 & 10381;7280;998                                         & TUBB2A;TUBB3;CDC42                                           \\\\\n",
       "\t Reactome & R-HSA-917977  & Transferrin endocytosis and recycling                  & http://reactome.org/PathwayBrowser/\\#/R-HSA-917977  &  31 &  3 & 0.3142884 &  9.545372 & 0.0037048910 & 0.16127881 & 51382;51606;529                                        & ATP6V1H;ATP6V1D;ATP6V1E1                                     \\\\\n",
       "\t Reactome & R-HSA-445095  & Interaction between L1 and Ankyrins                    & http://reactome.org/PathwayBrowser/\\#/R-HSA-445095  &  31 &  3 & 0.3142884 &  9.545372 & 0.0037048910 & 0.16127881 & 11280;3786;4897                                        & SCN11A;NRCAM;KCNQ3                                           \\\\\n",
       "\t Reactome & R-HSA-1834949 & Cytosolic sensors of pathogen-associated DNA           & http://reactome.org/PathwayBrowser/\\#/R-HSA-1834949 &  63 &  4 & 0.6387152 &  6.262572 & 0.0037593386 & 0.16127881 & 4790;4792;5970;9208                                    & NFKB1;RELA;NFKBIA;LRRFIP1                                    \\\\\n",
       "\t Reactome & R-HSA-450294  & MAP kinase activation                                  & http://reactome.org/PathwayBrowser/\\#/R-HSA-450294  &  64 &  4 & 0.6488535 &  6.164720 & 0.0039791391 & 0.16127881 & 1385;4790;7334;9261                                    & NFKB1;UBE2N;MAPKAPK2;CREB1                                   \\\\\n",
       "\t Reactome & R-HSA-1222556 & ROS, RNS production in phagocytes                      & http://reactome.org/PathwayBrowser/\\#/R-HSA-1222556 &  32 &  3 & 0.3244268 &  9.247079 & 0.0040583008 & 0.16127881 & 51382;51606;529                                        & ATP6V1H;ATP6V1D;ATP6V1E1                                     \\\\\n",
       "\t Reactome & R-HSA-112316  & Neuronal System                                        & http://reactome.org/PathwayBrowser/\\#/R-HSA-112316  & 368 & 10 & 3.7309077 &  2.680313 & 0.0040933707 & 0.16127881 & 10313;10611;1385;23229;2785;3786;5582;64130;83988;8500 & ARHGEF9;PPFIA1;RTN3;PDLIM5;CREB1;PRKCG;GNG3;NCALD;KCNQ3;LIN7B\\\\\n",
       "\t Reactome & R-HSA-168898  & Toll-like Receptor Cascades                            & http://reactome.org/PathwayBrowser/\\#/R-HSA-168898  & 155 &  6 & 1.5714421 &  3.818149 & 0.0048699692 & 0.18654688 & 1385;4790;4792;5970;7334;9261                          & NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                       \\\\\n",
       "\t Reactome & R-HSA-193692  & Regulated proteolysis of p75NTR                        & http://reactome.org/PathwayBrowser/\\#/R-HSA-193692  &  11 &  2 & 0.1115217 & 17.933730 & 0.0052762304 & 0.19664653 & 4790;5970                                              & NFKB1;RELA                                                   \\\\\n",
       "\t Reactome & R-HSA-1445148 & Translocation of SLC2A4 (GLUT4) to the plasma membrane & http://reactome.org/PathwayBrowser/\\#/R-HSA-1445148 &  70 &  4 & 0.7096835 &  5.636315 & 0.0054826759 & 0.19896342 & 10381;54536;5571;7280                                  & TUBB2A;EXOC6;TUBB3;PRKAG1                                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 113 x 12\n",
       "\n",
       "| category &lt;chr&gt; | geneSet &lt;chr&gt; | description &lt;chr&gt; | link &lt;chr&gt; | size &lt;dbl&gt; | overlap &lt;dbl&gt; | expect &lt;dbl&gt; | enrichmentRatio &lt;dbl&gt; | pValue &lt;dbl&gt; | FDR &lt;dbl&gt; | overlapId &lt;chr&gt; | userId &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| CC          | GO:0005743 | mitochondrial inner membrane                               | http://amigo.geneontology.org/amigo/term/GO:0005743                                                                                                                | 455 | 12 | 3.15469425 |  3.803855 | 7.675239e-05 | 0.030318595 | 4711;4976;7385;10131;25875;29088;55744;55750;79590;84303;117145;285521 | OPA1;MRPL24;LETMD1;CHCHD6;TRAP1;AGK;COA1;THEM4;MRPL15;COX18;NDUFB5;UQCRC2 |\n",
       "| CC          | GO:1990391 | DNA repair complex                                         | http://amigo.geneontology.org/amigo/term/GO:1990391                                                                                                                |  40 |  4 | 0.27733576 | 14.422951 | 1.657331e-04 | 0.030318595 | 4437;51317;64421;126074                                                | PHF21A;MSH3;SWSAP1;DCLRE1C                                                |\n",
       "| CC          | GO:0031304 | intrinsic component of mitochondrial inner membrane        | http://amigo.geneontology.org/amigo/term/GO:0031304                                                                                                                |  40 |  4 | 0.27733576 | 14.422951 | 1.657331e-04 | 0.030318595 | 55744;55750;84303;285521                                               | CHCHD6;AGK;COA1;COX18                                                     |\n",
       "| CC          | GO:0031305 | integral component of mitochondrial inner membrane         | http://amigo.geneontology.org/amigo/term/GO:0031305                                                                                                                |  40 |  4 | 0.27733576 | 14.422951 | 1.657331e-04 | 0.030318595 | 55744;55750;84303;285521                                               | CHCHD6;AGK;COA1;COX18                                                     |\n",
       "| CC          | GO:0044455 | mitochondrial membrane part                                | http://amigo.geneontology.org/amigo/term/GO:0044455                                                                                                                | 229 |  8 | 1.58774722 |  5.038585 | 1.963640e-04 | 0.030318595 | 4711;4976;7385;10651;55744;55750;84303;285521                          | OPA1;CHCHD6;AGK;COA1;MTX2;COX18;NDUFB5;UQCRC2                             |\n",
       "| CC          | GO:0033267 | axon part                                                  | http://amigo.geneontology.org/amigo/term/GO:0033267                                                                                                                | 374 | 10 | 2.59308934 |  3.856404 | 2.785301e-04 | 0.035837533 | 1176;2596;3611;3786;4897;4976;5582;29993;54536;80228                   | OPA1;GAP43;EXOC6;NRCAM;ORAI2;AP3S1;PRKCG;PACSIN1;KCNQ3;ILK                |\n",
       "| CC          | GO:0032592 | integral component of mitochondrial membrane               | http://amigo.geneontology.org/amigo/term/GO:0032592                                                                                                                |  69 |  4 | 0.47840418 |  8.361131 | 1.343200e-03 | 0.136745793 | 55744;55750;84303;285521                                               | CHCHD6;AGK;COA1;COX18                                                     |\n",
       "| CC          | GO:0098573 | intrinsic component of mitochondrial membrane              | http://amigo.geneontology.org/amigo/term/GO:0098573                                                                                                                |  70 |  4 | 0.48533758 |  8.241686 | 1.417055e-03 | 0.136745793 | 55744;55750;84303;285521                                               | CHCHD6;AGK;COA1;COX18                                                     |\n",
       "| CC          | GO:0005741 | mitochondrial outer membrane                               | http://amigo.geneontology.org/amigo/term/GO:0005741                                                                                                                | 180 |  6 | 1.24801091 |  4.807650 | 1.599920e-03 | 0.137237570 | 4976;5521;10651;25875;55750;84816                                      | OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                      |\n",
       "| CC          | GO:0005758 | mitochondrial intermembrane space                          | http://amigo.geneontology.org/amigo/term/GO:0005758                                                                                                                |  79 |  4 | 0.54773812 |  7.302760 | 2.213754e-03 | 0.170901805 | 4976;10131;55750;117145                                                | OPA1;TRAP1;AGK;THEM4                                                      |\n",
       "| CC          | GO:0098798 | mitochondrial protein complex                              | http://amigo.geneontology.org/amigo/term/GO:0098798                                                                                                                | 266 |  7 | 1.84428279 |  3.795513 | 2.553334e-03 | 0.173338049 | 4711;7385;10651;29088;55750;79590;84303                                | MRPL24;CHCHD6;AGK;MRPL15;MTX2;NDUFB5;UQCRC2                               |\n",
       "| CC          | GO:0031968 | organelle outer membrane                                   | http://amigo.geneontology.org/amigo/term/GO:0031968                                                                                                                | 204 |  6 | 1.41441237 |  4.242044 | 2.995177e-03 | 0.173338049 | 4976;5521;10651;25875;55750;84816                                      | OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                      |\n",
       "| CC          | GO:0019867 | outer membrane                                             | http://amigo.geneontology.org/amigo/term/GO:0019867                                                                                                                | 206 |  6 | 1.42827915 |  4.200859 | 3.142824e-03 | 0.173338049 | 4976;5521;10651;25875;55750;84816                                      | OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                      |\n",
       "| CC          | GO:0031970 | organelle envelope lumen                                   | http://amigo.geneontology.org/amigo/term/GO:0031970                                                                                                                |  87 |  4 | 0.60320527 |  6.631242 | 3.143436e-03 | 0.173338049 | 4976;10131;55750;117145                                                | OPA1;TRAP1;AGK;THEM4                                                      |\n",
       "| CC          | GO:0048188 | Set1C/COMPASS complex                                      | http://amigo.geneontology.org/amigo/term/GO:0048188                                                                                                                |  13 |  2 | 0.09013412 | 22.189155 | 3.537714e-03 | 0.182074371 | 5929;80335                                                             | WDR82;RBBP5                                                               |\n",
       "| MF          | GO:0070402 | NADPH binding                                              | http://amigo.geneontology.org/amigo/term/GO:0070402                                                                                                                |  13 |  3 | 0.10839182 | 27.677366 | 1.526016e-04 | 0.173532316 | 2194;6715;84869                                                        | CBR4;SRD5A1;FASN                                                          |\n",
       "| MF          | GO:0031435 | mitogen-activated protein kinase kinase kinase binding     | http://amigo.geneontology.org/amigo/term/GO:0031435                                                                                                                |  19 |  3 | 0.15841881 | 18.937145 | 4.984199e-04 | 0.173532316 | 998;6929;142679                                                        | TCF3;DUSP19;CDC42                                                         |\n",
       "| MF          | GO:0003924 | GTPase activity                                            | http://amigo.geneontology.org/amigo/term/GO:0003924                                                                                                                | 293 |  9 | 2.44298482 |  3.684018 | 7.904017e-04 | 0.173532316 | 381;998;2785;4976;7280;8934;10381;27175;64121                          | OPA1;TUBB2A;TUBG2;RAB29;TUBB3;ARF5;GNG3;CDC42;RRAGC                       |\n",
       "| MF          | GO:0046961 | proton-transporting ATPase activity, rotational mechanism  | http://amigo.geneontology.org/amigo/term/GO:0046961                                                                                                                |  23 |  3 | 0.19177014 | 15.643728 | 8.889877e-04 | 0.173532316 | 529;51382;51606                                                        | ATP6V1H;ATP6V1D;ATP6V1E1                                                  |\n",
       "| MF          | GO:0005525 | GTP binding                                                | http://amigo.geneontology.org/amigo/term/GO:0005525                                                                                                                | 369 | 10 | 3.07666007 |  3.250278 | 1.062333e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0032550 | purine ribonucleoside binding                              | http://amigo.geneontology.org/amigo/term/GO:0032550                                                                                                                | 373 | 10 | 3.11001140 |  3.215422 | 1.152068e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0001883 | purine nucleoside binding                                  | http://amigo.geneontology.org/amigo/term/GO:0001883                                                                                                                | 376 | 10 | 3.13502489 |  3.189767 | 1.223363e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0032549 | ribonucleoside binding                                     | http://amigo.geneontology.org/amigo/term/GO:0032549                                                                                                                | 377 | 10 | 3.14336273 |  3.181306 | 1.247915e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0001882 | nucleoside binding                                         | http://amigo.geneontology.org/amigo/term/GO:0001882                                                                                                                | 384 | 10 | 3.20172755 |  3.123314 | 1.431292e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0019001 | guanyl nucleotide binding                                  | http://amigo.geneontology.org/amigo/term/GO:0019001                                                                                                                | 390 | 10 | 3.25175454 |  3.075263 | 1.605429e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0032561 | guanyl ribonucleotide binding                              | http://amigo.geneontology.org/amigo/term/GO:0032561                                                                                                                | 390 | 10 | 3.25175454 |  3.075263 | 1.605429e-03 | 0.173532316 | 381;998;4976;7280;8934;10381;27175;54552;55915;64121                   | OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC               |\n",
       "| MF          | GO:0036442 | proton-exporting ATPase activity                           | http://amigo.geneontology.org/amigo/term/GO:0036442                                                                                                                |  30 |  3 | 0.25013496 | 11.993525 | 1.952997e-03 | 0.193509455 | 529;51382;51606                                                        | ATP6V1H;ATP6V1D;ATP6V1E1                                                  |\n",
       "| KEGG        | hsa05120   | Epithelial cell signaling in Helicobacter pylori infection | http://www.kegg.jp/kegg-bin/show_pathway?hsa05120+4790+4792+51382+51606+529+5970+998                                                                               |  68 |  7 | 0.75565671 |  9.263466 | 9.091745e-06 | 0.002863900 | 4790;4792;51382;51606;529;5970;998                                     | NFKB1;ATP6V1H;ATP6V1D;RELA;ATP6V1E1;NFKBIA;CDC42                          |\n",
       "| KEGG        | hsa05165   | Human papillomavirus infection                             | http://www.kegg.jp/kegg-bin/show_pathway?hsa05165+1385+27175+3693+4790+51382+51606+529+5521+5970+7148+998                                                          | 339 | 11 | 3.76717097 |  2.919963 | 1.229368e-03 | 0.193625407 | 1385;27175;3693;4790;51382;51606;529;5521;5970;7148;998                | NFKB1;ATP6V1H;ATP6V1D;TUBG2;TNXB;RELA;CREB1;PPP2R2B;ATP6V1E1;ITGB5;CDC42  |\n",
       "| Wikipathway | WP195      | IL-1 signaling pathway                                     | http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP195&amp;label[]=NFKB1&amp;label[]=HSPB2&amp;label[]=UBE2N&amp;label[]=MAPKAPK2&amp;label[]=RELA&amp;label[]=NFKBIA&amp;colors=steelblue |  57 |  6 | 0.53195701 | 11.279107 | 1.310609e-05 | 0.006146754 | 9261;3316;7334;4790;4792;5970                                          | NFKB1;HSPB2;UBE2N;MAPKAPK2;RELA;NFKBIA                                    |\n",
       "| ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... |\n",
       "| Reactome | R-HSA-975155  | MyD88 dependent cascade initiated on endosome          | http://reactome.org/PathwayBrowser/#/R-HSA-975155  |  94 |  6 | 0.9530036 |  6.295884 | 0.0003729130 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-166058  | MyD88:MAL(TIRAP) cascade initiated on plasma membrane  | http://reactome.org/PathwayBrowser/#/R-HSA-166058  |  95 |  6 | 0.9631419 |  6.229611 | 0.0003948163 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-168188  | Toll Like Receptor TLR6:TLR2 Cascade                   | http://reactome.org/PathwayBrowser/#/R-HSA-168188  |  95 |  6 | 0.9631419 |  6.229611 | 0.0003948163 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-168138  | Toll Like Receptor 9 (TLR9) Cascade                    | http://reactome.org/PathwayBrowser/#/R-HSA-168138  |  98 |  6 | 0.9935569 |  6.038909 | 0.0004666130 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-168179  | Toll Like Receptor TLR1:TLR2 Cascade                   | http://reactome.org/PathwayBrowser/#/R-HSA-168179  |  98 |  6 | 0.9935569 |  6.038909 | 0.0004666130 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-181438  | Toll Like Receptor 2 (TLR2) Cascade                    | http://reactome.org/PathwayBrowser/#/R-HSA-181438  |  98 |  6 | 0.9935569 |  6.038909 | 0.0004666130 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-168164  | Toll Like Receptor 3 (TLR3) Cascade                    | http://reactome.org/PathwayBrowser/#/R-HSA-168164  |  99 |  6 | 1.0036953 |  5.977910 | 0.0004926798 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-193639  | p75NTR signals via NF-kB                               | http://reactome.org/PathwayBrowser/#/R-HSA-193639  |  16 |  3 | 0.1622134 | 18.494159 | 0.0005154108 | 0.03982766 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-166166  | MyD88-independent TLR4 cascade                         | http://reactome.org/PathwayBrowser/#/R-HSA-166166  | 100 |  6 | 1.0138336 |  5.918131 | 0.0005198679 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-937061  | TRIF(TICAM1)-mediated TLR4 signaling                   | http://reactome.org/PathwayBrowser/#/R-HSA-937061  | 100 |  6 | 1.0138336 |  5.918131 | 0.0005198679 | 0.03982766 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-1810476 | RIP-mediated NFkB activation via ZBP1                  | http://reactome.org/PathwayBrowser/#/R-HSA-1810476 |  17 |  3 | 0.1723517 | 17.406267 | 0.0006212549 | 0.04509003 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-1606322 | ZBP1(DAI) mediated induction of type I IFNs            | http://reactome.org/PathwayBrowser/#/R-HSA-1606322 |  21 |  3 | 0.2129051 | 14.090788 | 0.0011797998 | 0.07797868 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-373760  | L1CAM interactions                                     | http://reactome.org/PathwayBrowser/#/R-HSA-373760  | 117 |  6 | 1.1861853 |  5.058231 | 0.0011874925 | 0.07797868 | 10381;11280;2596;3786;4897;7280                        | TUBB2A;SCN11A;GAP43;NRCAM;TUBB3;KCNQ3                         |\n",
       "| Reactome | R-HSA-5260271 | Diseases of Immune System                              | http://reactome.org/PathwayBrowser/#/R-HSA-5260271 |  24 |  3 | 0.2433201 | 12.329439 | 0.0017562199 | 0.10090947 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-5602358 | Diseases associated with the TLR signaling cascade     | http://reactome.org/PathwayBrowser/#/R-HSA-5602358 |  24 |  3 | 0.2433201 | 12.329439 | 0.0017562199 | 0.10090947 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-933542  | TRAF6 mediated NF-kB activation                        | http://reactome.org/PathwayBrowser/#/R-HSA-933542  |  24 |  3 | 0.2433201 | 12.329439 | 0.0017562199 | 0.10090947 | 4790;4792;5970                                         | NFKB1;RELA;NFKBIA                                             |\n",
       "| Reactome | R-HSA-166016  | Toll Like Receptor 4 (TLR4) Cascade                    | http://reactome.org/PathwayBrowser/#/R-HSA-166016  | 130 |  6 | 1.3179837 |  4.552408 | 0.0020369838 | 0.11236003 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-77387   | Insulin receptor recycling                             | http://reactome.org/PathwayBrowser/#/R-HSA-77387   |  26 |  3 | 0.2635967 | 11.381021 | 0.0022230880 | 0.11790917 | 51382;51606;529                                        | ATP6V1H;ATP6V1D;ATP6V1E1                                      |\n",
       "| Reactome | R-HSA-193704  | p75 NTR receptor-mediated signalling                   | http://reactome.org/PathwayBrowser/#/R-HSA-193704  |  97 |  5 | 0.9834186 |  5.084305 | 0.0030023844 | 0.15075403 | 23229;27018;4790;4792;5970                             | NFKB1;ARHGEF9;RELA;NFKBIA;BEX3                                |\n",
       "| Reactome | R-HSA-73887   | Death Receptor Signalling                              | http://reactome.org/PathwayBrowser/#/R-HSA-73887   | 141 |  6 | 1.4295054 |  4.197256 | 0.0030609954 | 0.15075403 | 23229;27018;331;4790;4792;5970                         | NFKB1;ARHGEF9;RELA;NFKBIA;XIAP;BEX3                           |\n",
       "| Reactome | R-HSA-5626467 | RHO GTPases activate IQGAPs                            | http://reactome.org/PathwayBrowser/#/R-HSA-5626467 |  30 |  3 | 0.3041501 |  9.863551 | 0.0033709817 | 0.16029599 | 10381;7280;998                                         | TUBB2A;TUBB3;CDC42                                            |\n",
       "| Reactome | R-HSA-917977  | Transferrin endocytosis and recycling                  | http://reactome.org/PathwayBrowser/#/R-HSA-917977  |  31 |  3 | 0.3142884 |  9.545372 | 0.0037048910 | 0.16127881 | 51382;51606;529                                        | ATP6V1H;ATP6V1D;ATP6V1E1                                      |\n",
       "| Reactome | R-HSA-445095  | Interaction between L1 and Ankyrins                    | http://reactome.org/PathwayBrowser/#/R-HSA-445095  |  31 |  3 | 0.3142884 |  9.545372 | 0.0037048910 | 0.16127881 | 11280;3786;4897                                        | SCN11A;NRCAM;KCNQ3                                            |\n",
       "| Reactome | R-HSA-1834949 | Cytosolic sensors of pathogen-associated DNA           | http://reactome.org/PathwayBrowser/#/R-HSA-1834949 |  63 |  4 | 0.6387152 |  6.262572 | 0.0037593386 | 0.16127881 | 4790;4792;5970;9208                                    | NFKB1;RELA;NFKBIA;LRRFIP1                                     |\n",
       "| Reactome | R-HSA-450294  | MAP kinase activation                                  | http://reactome.org/PathwayBrowser/#/R-HSA-450294  |  64 |  4 | 0.6488535 |  6.164720 | 0.0039791391 | 0.16127881 | 1385;4790;7334;9261                                    | NFKB1;UBE2N;MAPKAPK2;CREB1                                    |\n",
       "| Reactome | R-HSA-1222556 | ROS, RNS production in phagocytes                      | http://reactome.org/PathwayBrowser/#/R-HSA-1222556 |  32 |  3 | 0.3244268 |  9.247079 | 0.0040583008 | 0.16127881 | 51382;51606;529                                        | ATP6V1H;ATP6V1D;ATP6V1E1                                      |\n",
       "| Reactome | R-HSA-112316  | Neuronal System                                        | http://reactome.org/PathwayBrowser/#/R-HSA-112316  | 368 | 10 | 3.7309077 |  2.680313 | 0.0040933707 | 0.16127881 | 10313;10611;1385;23229;2785;3786;5582;64130;83988;8500 | ARHGEF9;PPFIA1;RTN3;PDLIM5;CREB1;PRKCG;GNG3;NCALD;KCNQ3;LIN7B |\n",
       "| Reactome | R-HSA-168898  | Toll-like Receptor Cascades                            | http://reactome.org/PathwayBrowser/#/R-HSA-168898  | 155 |  6 | 1.5714421 |  3.818149 | 0.0048699692 | 0.18654688 | 1385;4790;4792;5970;7334;9261                          | NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                        |\n",
       "| Reactome | R-HSA-193692  | Regulated proteolysis of p75NTR                        | http://reactome.org/PathwayBrowser/#/R-HSA-193692  |  11 |  2 | 0.1115217 | 17.933730 | 0.0052762304 | 0.19664653 | 4790;5970                                              | NFKB1;RELA                                                    |\n",
       "| Reactome | R-HSA-1445148 | Translocation of SLC2A4 (GLUT4) to the plasma membrane | http://reactome.org/PathwayBrowser/#/R-HSA-1445148 |  70 |  4 | 0.7096835 |  5.636315 | 0.0054826759 | 0.19896342 | 10381;54536;5571;7280                                  | TUBB2A;EXOC6;TUBB3;PRKAG1                                     |\n",
       "\n"
      ],
      "text/plain": [
       "    category    geneSet      \n",
       "1   CC          GO:0005743   \n",
       "2   CC          GO:1990391   \n",
       "3   CC          GO:0031304   \n",
       "4   CC          GO:0031305   \n",
       "5   CC          GO:0044455   \n",
       "6   CC          GO:0033267   \n",
       "7   CC          GO:0032592   \n",
       "8   CC          GO:0098573   \n",
       "9   CC          GO:0005741   \n",
       "10  CC          GO:0005758   \n",
       "11  CC          GO:0098798   \n",
       "12  CC          GO:0031968   \n",
       "13  CC          GO:0019867   \n",
       "14  CC          GO:0031970   \n",
       "15  CC          GO:0048188   \n",
       "16  MF          GO:0070402   \n",
       "17  MF          GO:0031435   \n",
       "18  MF          GO:0003924   \n",
       "19  MF          GO:0046961   \n",
       "20  MF          GO:0005525   \n",
       "21  MF          GO:0032550   \n",
       "22  MF          GO:0001883   \n",
       "23  MF          GO:0032549   \n",
       "24  MF          GO:0001882   \n",
       "25  MF          GO:0019001   \n",
       "26  MF          GO:0032561   \n",
       "27  MF          GO:0036442   \n",
       "28  KEGG        hsa05120     \n",
       "29  KEGG        hsa05165     \n",
       "30  Wikipathway WP195        \n",
       "... ...         ...          \n",
       "84  Reactome    R-HSA-975155 \n",
       "85  Reactome    R-HSA-166058 \n",
       "86  Reactome    R-HSA-168188 \n",
       "87  Reactome    R-HSA-168138 \n",
       "88  Reactome    R-HSA-168179 \n",
       "89  Reactome    R-HSA-181438 \n",
       "90  Reactome    R-HSA-168164 \n",
       "91  Reactome    R-HSA-193639 \n",
       "92  Reactome    R-HSA-166166 \n",
       "93  Reactome    R-HSA-937061 \n",
       "94  Reactome    R-HSA-1810476\n",
       "95  Reactome    R-HSA-1606322\n",
       "96  Reactome    R-HSA-373760 \n",
       "97  Reactome    R-HSA-5260271\n",
       "98  Reactome    R-HSA-5602358\n",
       "99  Reactome    R-HSA-933542 \n",
       "100 Reactome    R-HSA-166016 \n",
       "101 Reactome    R-HSA-77387  \n",
       "102 Reactome    R-HSA-193704 \n",
       "103 Reactome    R-HSA-73887  \n",
       "104 Reactome    R-HSA-5626467\n",
       "105 Reactome    R-HSA-917977 \n",
       "106 Reactome    R-HSA-445095 \n",
       "107 Reactome    R-HSA-1834949\n",
       "108 Reactome    R-HSA-450294 \n",
       "109 Reactome    R-HSA-1222556\n",
       "110 Reactome    R-HSA-112316 \n",
       "111 Reactome    R-HSA-168898 \n",
       "112 Reactome    R-HSA-193692 \n",
       "113 Reactome    R-HSA-1445148\n",
       "    description                                               \n",
       "1   mitochondrial inner membrane                              \n",
       "2   DNA repair complex                                        \n",
       "3   intrinsic component of mitochondrial inner membrane       \n",
       "4   integral component of mitochondrial inner membrane        \n",
       "5   mitochondrial membrane part                               \n",
       "6   axon part                                                 \n",
       "7   integral component of mitochondrial membrane              \n",
       "8   intrinsic component of mitochondrial membrane             \n",
       "9   mitochondrial outer membrane                              \n",
       "10  mitochondrial intermembrane space                         \n",
       "11  mitochondrial protein complex                             \n",
       "12  organelle outer membrane                                  \n",
       "13  outer membrane                                            \n",
       "14  organelle envelope lumen                                  \n",
       "15  Set1C/COMPASS complex                                     \n",
       "16  NADPH binding                                             \n",
       "17  mitogen-activated protein kinase kinase kinase binding    \n",
       "18  GTPase activity                                           \n",
       "19  proton-transporting ATPase activity, rotational mechanism \n",
       "20  GTP binding                                               \n",
       "21  purine ribonucleoside binding                             \n",
       "22  purine nucleoside binding                                 \n",
       "23  ribonucleoside binding                                    \n",
       "24  nucleoside binding                                        \n",
       "25  guanyl nucleotide binding                                 \n",
       "26  guanyl ribonucleotide binding                             \n",
       "27  proton-exporting ATPase activity                          \n",
       "28  Epithelial cell signaling in Helicobacter pylori infection\n",
       "29  Human papillomavirus infection                            \n",
       "30  IL-1 signaling pathway                                    \n",
       "... ...                                                       \n",
       "84  MyD88 dependent cascade initiated on endosome             \n",
       "85  MyD88:MAL(TIRAP) cascade initiated on plasma membrane     \n",
       "86  Toll Like Receptor TLR6:TLR2 Cascade                      \n",
       "87  Toll Like Receptor 9 (TLR9) Cascade                       \n",
       "88  Toll Like Receptor TLR1:TLR2 Cascade                      \n",
       "89  Toll Like Receptor 2 (TLR2) Cascade                       \n",
       "90  Toll Like Receptor 3 (TLR3) Cascade                       \n",
       "91  p75NTR signals via NF-kB                                  \n",
       "92  MyD88-independent TLR4 cascade                            \n",
       "93  TRIF(TICAM1)-mediated TLR4 signaling                      \n",
       "94  RIP-mediated NFkB activation via ZBP1                     \n",
       "95  ZBP1(DAI) mediated induction of type I IFNs               \n",
       "96  L1CAM interactions                                        \n",
       "97  Diseases of Immune System                                 \n",
       "98  Diseases associated with the TLR signaling cascade        \n",
       "99  TRAF6 mediated NF-kB activation                           \n",
       "100 Toll Like Receptor 4 (TLR4) Cascade                       \n",
       "101 Insulin receptor recycling                                \n",
       "102 p75 NTR receptor-mediated signalling                      \n",
       "103 Death Receptor Signalling                                 \n",
       "104 RHO GTPases activate IQGAPs                               \n",
       "105 Transferrin endocytosis and recycling                     \n",
       "106 Interaction between L1 and Ankyrins                       \n",
       "107 Cytosolic sensors of pathogen-associated DNA              \n",
       "108 MAP kinase activation                                     \n",
       "109 ROS, RNS production in phagocytes                         \n",
       "110 Neuronal System                                           \n",
       "111 Toll-like Receptor Cascades                               \n",
       "112 Regulated proteolysis of p75NTR                           \n",
       "113 Translocation of SLC2A4 (GLUT4) to the plasma membrane    \n",
       "    link                                                                                                                                                              \n",
       "1   http://amigo.geneontology.org/amigo/term/GO:0005743                                                                                                               \n",
       "2   http://amigo.geneontology.org/amigo/term/GO:1990391                                                                                                               \n",
       "3   http://amigo.geneontology.org/amigo/term/GO:0031304                                                                                                               \n",
       "4   http://amigo.geneontology.org/amigo/term/GO:0031305                                                                                                               \n",
       "5   http://amigo.geneontology.org/amigo/term/GO:0044455                                                                                                               \n",
       "6   http://amigo.geneontology.org/amigo/term/GO:0033267                                                                                                               \n",
       "7   http://amigo.geneontology.org/amigo/term/GO:0032592                                                                                                               \n",
       "8   http://amigo.geneontology.org/amigo/term/GO:0098573                                                                                                               \n",
       "9   http://amigo.geneontology.org/amigo/term/GO:0005741                                                                                                               \n",
       "10  http://amigo.geneontology.org/amigo/term/GO:0005758                                                                                                               \n",
       "11  http://amigo.geneontology.org/amigo/term/GO:0098798                                                                                                               \n",
       "12  http://amigo.geneontology.org/amigo/term/GO:0031968                                                                                                               \n",
       "13  http://amigo.geneontology.org/amigo/term/GO:0019867                                                                                                               \n",
       "14  http://amigo.geneontology.org/amigo/term/GO:0031970                                                                                                               \n",
       "15  http://amigo.geneontology.org/amigo/term/GO:0048188                                                                                                               \n",
       "16  http://amigo.geneontology.org/amigo/term/GO:0070402                                                                                                               \n",
       "17  http://amigo.geneontology.org/amigo/term/GO:0031435                                                                                                               \n",
       "18  http://amigo.geneontology.org/amigo/term/GO:0003924                                                                                                               \n",
       "19  http://amigo.geneontology.org/amigo/term/GO:0046961                                                                                                               \n",
       "20  http://amigo.geneontology.org/amigo/term/GO:0005525                                                                                                               \n",
       "21  http://amigo.geneontology.org/amigo/term/GO:0032550                                                                                                               \n",
       "22  http://amigo.geneontology.org/amigo/term/GO:0001883                                                                                                               \n",
       "23  http://amigo.geneontology.org/amigo/term/GO:0032549                                                                                                               \n",
       "24  http://amigo.geneontology.org/amigo/term/GO:0001882                                                                                                               \n",
       "25  http://amigo.geneontology.org/amigo/term/GO:0019001                                                                                                               \n",
       "26  http://amigo.geneontology.org/amigo/term/GO:0032561                                                                                                               \n",
       "27  http://amigo.geneontology.org/amigo/term/GO:0036442                                                                                                               \n",
       "28  http://www.kegg.jp/kegg-bin/show_pathway?hsa05120+4790+4792+51382+51606+529+5970+998                                                                              \n",
       "29  http://www.kegg.jp/kegg-bin/show_pathway?hsa05165+1385+27175+3693+4790+51382+51606+529+5521+5970+7148+998                                                         \n",
       "30  http://www.wikipathways.org/wpi/PathwayWidget.php?id=WP195&label[]=NFKB1&label[]=HSPB2&label[]=UBE2N&label[]=MAPKAPK2&label[]=RELA&label[]=NFKBIA&colors=steelblue\n",
       "... ...                                                                                                                                                               \n",
       "84  http://reactome.org/PathwayBrowser/#/R-HSA-975155                                                                                                                 \n",
       "85  http://reactome.org/PathwayBrowser/#/R-HSA-166058                                                                                                                 \n",
       "86  http://reactome.org/PathwayBrowser/#/R-HSA-168188                                                                                                                 \n",
       "87  http://reactome.org/PathwayBrowser/#/R-HSA-168138                                                                                                                 \n",
       "88  http://reactome.org/PathwayBrowser/#/R-HSA-168179                                                                                                                 \n",
       "89  http://reactome.org/PathwayBrowser/#/R-HSA-181438                                                                                                                 \n",
       "90  http://reactome.org/PathwayBrowser/#/R-HSA-168164                                                                                                                 \n",
       "91  http://reactome.org/PathwayBrowser/#/R-HSA-193639                                                                                                                 \n",
       "92  http://reactome.org/PathwayBrowser/#/R-HSA-166166                                                                                                                 \n",
       "93  http://reactome.org/PathwayBrowser/#/R-HSA-937061                                                                                                                 \n",
       "94  http://reactome.org/PathwayBrowser/#/R-HSA-1810476                                                                                                                \n",
       "95  http://reactome.org/PathwayBrowser/#/R-HSA-1606322                                                                                                                \n",
       "96  http://reactome.org/PathwayBrowser/#/R-HSA-373760                                                                                                                 \n",
       "97  http://reactome.org/PathwayBrowser/#/R-HSA-5260271                                                                                                                \n",
       "98  http://reactome.org/PathwayBrowser/#/R-HSA-5602358                                                                                                                \n",
       "99  http://reactome.org/PathwayBrowser/#/R-HSA-933542                                                                                                                 \n",
       "100 http://reactome.org/PathwayBrowser/#/R-HSA-166016                                                                                                                 \n",
       "101 http://reactome.org/PathwayBrowser/#/R-HSA-77387                                                                                                                  \n",
       "102 http://reactome.org/PathwayBrowser/#/R-HSA-193704                                                                                                                 \n",
       "103 http://reactome.org/PathwayBrowser/#/R-HSA-73887                                                                                                                  \n",
       "104 http://reactome.org/PathwayBrowser/#/R-HSA-5626467                                                                                                                \n",
       "105 http://reactome.org/PathwayBrowser/#/R-HSA-917977                                                                                                                 \n",
       "106 http://reactome.org/PathwayBrowser/#/R-HSA-445095                                                                                                                 \n",
       "107 http://reactome.org/PathwayBrowser/#/R-HSA-1834949                                                                                                                \n",
       "108 http://reactome.org/PathwayBrowser/#/R-HSA-450294                                                                                                                 \n",
       "109 http://reactome.org/PathwayBrowser/#/R-HSA-1222556                                                                                                                \n",
       "110 http://reactome.org/PathwayBrowser/#/R-HSA-112316                                                                                                                 \n",
       "111 http://reactome.org/PathwayBrowser/#/R-HSA-168898                                                                                                                 \n",
       "112 http://reactome.org/PathwayBrowser/#/R-HSA-193692                                                                                                                 \n",
       "113 http://reactome.org/PathwayBrowser/#/R-HSA-1445148                                                                                                                \n",
       "    size overlap expect     enrichmentRatio pValue       FDR        \n",
       "1   455  12      3.15469425  3.803855       7.675239e-05 0.030318595\n",
       "2    40   4      0.27733576 14.422951       1.657331e-04 0.030318595\n",
       "3    40   4      0.27733576 14.422951       1.657331e-04 0.030318595\n",
       "4    40   4      0.27733576 14.422951       1.657331e-04 0.030318595\n",
       "5   229   8      1.58774722  5.038585       1.963640e-04 0.030318595\n",
       "6   374  10      2.59308934  3.856404       2.785301e-04 0.035837533\n",
       "7    69   4      0.47840418  8.361131       1.343200e-03 0.136745793\n",
       "8    70   4      0.48533758  8.241686       1.417055e-03 0.136745793\n",
       "9   180   6      1.24801091  4.807650       1.599920e-03 0.137237570\n",
       "10   79   4      0.54773812  7.302760       2.213754e-03 0.170901805\n",
       "11  266   7      1.84428279  3.795513       2.553334e-03 0.173338049\n",
       "12  204   6      1.41441237  4.242044       2.995177e-03 0.173338049\n",
       "13  206   6      1.42827915  4.200859       3.142824e-03 0.173338049\n",
       "14   87   4      0.60320527  6.631242       3.143436e-03 0.173338049\n",
       "15   13   2      0.09013412 22.189155       3.537714e-03 0.182074371\n",
       "16   13   3      0.10839182 27.677366       1.526016e-04 0.173532316\n",
       "17   19   3      0.15841881 18.937145       4.984199e-04 0.173532316\n",
       "18  293   9      2.44298482  3.684018       7.904017e-04 0.173532316\n",
       "19   23   3      0.19177014 15.643728       8.889877e-04 0.173532316\n",
       "20  369  10      3.07666007  3.250278       1.062333e-03 0.173532316\n",
       "21  373  10      3.11001140  3.215422       1.152068e-03 0.173532316\n",
       "22  376  10      3.13502489  3.189767       1.223363e-03 0.173532316\n",
       "23  377  10      3.14336273  3.181306       1.247915e-03 0.173532316\n",
       "24  384  10      3.20172755  3.123314       1.431292e-03 0.173532316\n",
       "25  390  10      3.25175454  3.075263       1.605429e-03 0.173532316\n",
       "26  390  10      3.25175454  3.075263       1.605429e-03 0.173532316\n",
       "27   30   3      0.25013496 11.993525       1.952997e-03 0.193509455\n",
       "28   68   7      0.75565671  9.263466       9.091745e-06 0.002863900\n",
       "29  339  11      3.76717097  2.919963       1.229368e-03 0.193625407\n",
       "30   57   6      0.53195701 11.279107       1.310609e-05 0.006146754\n",
       "... ...  ...     ...        ...             ...          ...        \n",
       "84   94   6      0.9530036   6.295884       0.0003729130 0.03982766 \n",
       "85   95   6      0.9631419   6.229611       0.0003948163 0.03982766 \n",
       "86   95   6      0.9631419   6.229611       0.0003948163 0.03982766 \n",
       "87   98   6      0.9935569   6.038909       0.0004666130 0.03982766 \n",
       "88   98   6      0.9935569   6.038909       0.0004666130 0.03982766 \n",
       "89   98   6      0.9935569   6.038909       0.0004666130 0.03982766 \n",
       "90   99   6      1.0036953   5.977910       0.0004926798 0.03982766 \n",
       "91   16   3      0.1622134  18.494159       0.0005154108 0.03982766 \n",
       "92  100   6      1.0138336   5.918131       0.0005198679 0.03982766 \n",
       "93  100   6      1.0138336   5.918131       0.0005198679 0.03982766 \n",
       "94   17   3      0.1723517  17.406267       0.0006212549 0.04509003 \n",
       "95   21   3      0.2129051  14.090788       0.0011797998 0.07797868 \n",
       "96  117   6      1.1861853   5.058231       0.0011874925 0.07797868 \n",
       "97   24   3      0.2433201  12.329439       0.0017562199 0.10090947 \n",
       "98   24   3      0.2433201  12.329439       0.0017562199 0.10090947 \n",
       "99   24   3      0.2433201  12.329439       0.0017562199 0.10090947 \n",
       "100 130   6      1.3179837   4.552408       0.0020369838 0.11236003 \n",
       "101  26   3      0.2635967  11.381021       0.0022230880 0.11790917 \n",
       "102  97   5      0.9834186   5.084305       0.0030023844 0.15075403 \n",
       "103 141   6      1.4295054   4.197256       0.0030609954 0.15075403 \n",
       "104  30   3      0.3041501   9.863551       0.0033709817 0.16029599 \n",
       "105  31   3      0.3142884   9.545372       0.0037048910 0.16127881 \n",
       "106  31   3      0.3142884   9.545372       0.0037048910 0.16127881 \n",
       "107  63   4      0.6387152   6.262572       0.0037593386 0.16127881 \n",
       "108  64   4      0.6488535   6.164720       0.0039791391 0.16127881 \n",
       "109  32   3      0.3244268   9.247079       0.0040583008 0.16127881 \n",
       "110 368  10      3.7309077   2.680313       0.0040933707 0.16127881 \n",
       "111 155   6      1.5714421   3.818149       0.0048699692 0.18654688 \n",
       "112  11   2      0.1115217  17.933730       0.0052762304 0.19664653 \n",
       "113  70   4      0.7096835   5.636315       0.0054826759 0.19896342 \n",
       "    overlapId                                                             \n",
       "1   4711;4976;7385;10131;25875;29088;55744;55750;79590;84303;117145;285521\n",
       "2   4437;51317;64421;126074                                               \n",
       "3   55744;55750;84303;285521                                              \n",
       "4   55744;55750;84303;285521                                              \n",
       "5   4711;4976;7385;10651;55744;55750;84303;285521                         \n",
       "6   1176;2596;3611;3786;4897;4976;5582;29993;54536;80228                  \n",
       "7   55744;55750;84303;285521                                              \n",
       "8   55744;55750;84303;285521                                              \n",
       "9   4976;5521;10651;25875;55750;84816                                     \n",
       "10  4976;10131;55750;117145                                               \n",
       "11  4711;7385;10651;29088;55750;79590;84303                               \n",
       "12  4976;5521;10651;25875;55750;84816                                     \n",
       "13  4976;5521;10651;25875;55750;84816                                     \n",
       "14  4976;10131;55750;117145                                               \n",
       "15  5929;80335                                                            \n",
       "16  2194;6715;84869                                                       \n",
       "17  998;6929;142679                                                       \n",
       "18  381;998;2785;4976;7280;8934;10381;27175;64121                         \n",
       "19  529;51382;51606                                                       \n",
       "20  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "21  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "22  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "23  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "24  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "25  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "26  381;998;4976;7280;8934;10381;27175;54552;55915;64121                  \n",
       "27  529;51382;51606                                                       \n",
       "28  4790;4792;51382;51606;529;5970;998                                    \n",
       "29  1385;27175;3693;4790;51382;51606;529;5521;5970;7148;998               \n",
       "30  9261;3316;7334;4790;4792;5970                                         \n",
       "... ...                                                                   \n",
       "84  1385;4790;4792;5970;7334;9261                                         \n",
       "85  1385;4790;4792;5970;7334;9261                                         \n",
       "86  1385;4790;4792;5970;7334;9261                                         \n",
       "87  1385;4790;4792;5970;7334;9261                                         \n",
       "88  1385;4790;4792;5970;7334;9261                                         \n",
       "89  1385;4790;4792;5970;7334;9261                                         \n",
       "90  1385;4790;4792;5970;7334;9261                                         \n",
       "91  4790;4792;5970                                                        \n",
       "92  1385;4790;4792;5970;7334;9261                                         \n",
       "93  1385;4790;4792;5970;7334;9261                                         \n",
       "94  4790;4792;5970                                                        \n",
       "95  4790;4792;5970                                                        \n",
       "96  10381;11280;2596;3786;4897;7280                                       \n",
       "97  4790;4792;5970                                                        \n",
       "98  4790;4792;5970                                                        \n",
       "99  4790;4792;5970                                                        \n",
       "100 1385;4790;4792;5970;7334;9261                                         \n",
       "101 51382;51606;529                                                       \n",
       "102 23229;27018;4790;4792;5970                                            \n",
       "103 23229;27018;331;4790;4792;5970                                        \n",
       "104 10381;7280;998                                                        \n",
       "105 51382;51606;529                                                       \n",
       "106 11280;3786;4897                                                       \n",
       "107 4790;4792;5970;9208                                                   \n",
       "108 1385;4790;7334;9261                                                   \n",
       "109 51382;51606;529                                                       \n",
       "110 10313;10611;1385;23229;2785;3786;5582;64130;83988;8500                \n",
       "111 1385;4790;4792;5970;7334;9261                                         \n",
       "112 4790;5970                                                             \n",
       "113 10381;54536;5571;7280                                                 \n",
       "    userId                                                                   \n",
       "1   OPA1;MRPL24;LETMD1;CHCHD6;TRAP1;AGK;COA1;THEM4;MRPL15;COX18;NDUFB5;UQCRC2\n",
       "2   PHF21A;MSH3;SWSAP1;DCLRE1C                                               \n",
       "3   CHCHD6;AGK;COA1;COX18                                                    \n",
       "4   CHCHD6;AGK;COA1;COX18                                                    \n",
       "5   OPA1;CHCHD6;AGK;COA1;MTX2;COX18;NDUFB5;UQCRC2                            \n",
       "6   OPA1;GAP43;EXOC6;NRCAM;ORAI2;AP3S1;PRKCG;PACSIN1;KCNQ3;ILK               \n",
       "7   CHCHD6;AGK;COA1;COX18                                                    \n",
       "8   CHCHD6;AGK;COA1;COX18                                                    \n",
       "9   OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \n",
       "10  OPA1;TRAP1;AGK;THEM4                                                     \n",
       "11  MRPL24;CHCHD6;AGK;MRPL15;MTX2;NDUFB5;UQCRC2                              \n",
       "12  OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \n",
       "13  OPA1;LETMD1;RTN4IP1;AGK;MTX2;PPP2R2B                                     \n",
       "14  OPA1;TRAP1;AGK;THEM4                                                     \n",
       "15  WDR82;RBBP5                                                              \n",
       "16  CBR4;SRD5A1;FASN                                                         \n",
       "17  TCF3;DUSP19;CDC42                                                        \n",
       "18  OPA1;TUBB2A;TUBG2;RAB29;TUBB3;ARF5;GNG3;CDC42;RRAGC                      \n",
       "19  ATP6V1H;ATP6V1D;ATP6V1E1                                                 \n",
       "20  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "21  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "22  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "23  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "24  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "25  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "26  OPA1;TUBB2A;LANCL2;TUBG2;RAB29;GNL3L;TUBB3;ARF5;CDC42;RRAGC              \n",
       "27  ATP6V1H;ATP6V1D;ATP6V1E1                                                 \n",
       "28  NFKB1;ATP6V1H;ATP6V1D;RELA;ATP6V1E1;NFKBIA;CDC42                         \n",
       "29  NFKB1;ATP6V1H;ATP6V1D;TUBG2;TNXB;RELA;CREB1;PPP2R2B;ATP6V1E1;ITGB5;CDC42 \n",
       "30  NFKB1;HSPB2;UBE2N;MAPKAPK2;RELA;NFKBIA                                   \n",
       "... ...                                                                      \n",
       "84  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "85  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "86  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "87  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "88  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "89  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "90  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "91  NFKB1;RELA;NFKBIA                                                        \n",
       "92  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "93  NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "94  NFKB1;RELA;NFKBIA                                                        \n",
       "95  NFKB1;RELA;NFKBIA                                                        \n",
       "96  TUBB2A;SCN11A;GAP43;NRCAM;TUBB3;KCNQ3                                    \n",
       "97  NFKB1;RELA;NFKBIA                                                        \n",
       "98  NFKB1;RELA;NFKBIA                                                        \n",
       "99  NFKB1;RELA;NFKBIA                                                        \n",
       "100 NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "101 ATP6V1H;ATP6V1D;ATP6V1E1                                                 \n",
       "102 NFKB1;ARHGEF9;RELA;NFKBIA;BEX3                                           \n",
       "103 NFKB1;ARHGEF9;RELA;NFKBIA;XIAP;BEX3                                      \n",
       "104 TUBB2A;TUBB3;CDC42                                                       \n",
       "105 ATP6V1H;ATP6V1D;ATP6V1E1                                                 \n",
       "106 SCN11A;NRCAM;KCNQ3                                                       \n",
       "107 NFKB1;RELA;NFKBIA;LRRFIP1                                                \n",
       "108 NFKB1;UBE2N;MAPKAPK2;CREB1                                               \n",
       "109 ATP6V1H;ATP6V1D;ATP6V1E1                                                 \n",
       "110 ARHGEF9;PPFIA1;RTN3;PDLIM5;CREB1;PRKCG;GNG3;NCALD;KCNQ3;LIN7B            \n",
       "111 NFKB1;UBE2N;MAPKAPK2;RELA;CREB1;NFKBIA                                   \n",
       "112 NFKB1;RELA                                                               \n",
       "113 TUBB2A;EXOC6;TUBB3;PRKAG1                                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "library(ggplot2)\n",
    "ggplot(data=result, aes(x=category, y=size, fill=category)) +\n",
    "       geom_bar(stat=\"identity\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alternative Enrichment Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(gprofiler2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gost<-gost(as.factor(DEGenes$V1), organism = \"hsapiens\", ordered_query = FALSE, multi_query = FALSE, significant = TRUE,\n",
    "exclude_iea = FALSE, measure_underrepresentation = FALSE, evcodes = TRUE, user_threshold = 0.05, correction_method = \"fdr\",\n",
    "domain_scope = \"annotated\", custom_bg = NULL, numeric_ns = \"\", sources = NULL, as_short_link = FALSE)\n",
    "\n",
    "gost.df<-as.data.frame(gost$result)\n",
    "gost.df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(as.matrix(gost.df),paste0(output,\"/rra_paper_gost_enrich.txt\"), sep=\"\\t\", row.names=F)\n",
    "pgost <- gostplot(gost, capped = FALSE, interactive = FALSE)\n",
    "pgost"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ppgost <- publish_gostplot(pgost, highlight_terms = head(gost.df[which(gost.df$p_value<=0.05),]$term_id, n=10), \n",
    "                       width = NA, height = NA, filename = paste0(output,\"/rra_paper_gost_enrich.jpg\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
