{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Population Stratification \n",
    "\n",
    "This is the presence of multiple subpopulations (e.g., individuals with different ethnic background) in a study. Because allele frequencies can differ between subpopulations, population stratification can lead to false positive associations and/or mask true associations. \n",
    "\n",
    "To complete this task it is necessary to have generated the bfile 'dataset.b38_IBD' and the file 'indepSNP.prune.in' from the previous tutorial (Task_2.1_QC). This bfile is going to checked for population stratification using data from the 1000 Genomes Project. Individuals with a non-European ethnic background will be removed. 1000 Genomes data need to be processed before hand.\n",
    "\n",
    "Furthermore, this guide will generate a covariate file which helps to adust for remaining population stratification within the European subjects.\n",
    "\n",
    "This protocol follows the tutorial explained in Marees AT, de Kluiver H, Stringer S, et al. A tutorial on conducting genome-wide association studies: Quality control and statistical analysis. Int J Methods Psychiatr Res. 2018;27(2):e1608.doi:10.1002/mpr.1608"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Create directories for the output files\n",
    "path=\"/mnt/data/GWAS/output/build38/task2.2_stratification\"\n",
    "if not os.path.exists(path):\n",
    "    os.makedirs(path)\n",
    "    os.makedirs(os.path.join(path,\"intermediate_datasets/\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:54:24.831494Z",
     "start_time": "2020-08-21T13:54:24.827425Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/build38/task2.2_stratification\n",
      "env: intpath=/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets\n",
      "env: ref_files=/mnt/data/GWAS/ref_files/build38/1KG_GRCh38\n"
     ]
    }
   ],
   "source": [
    "%env path= /mnt/data/GWAS/output/build38/task2.2_stratification\n",
    "%env intpath=/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets\n",
    "%env ref_files=/mnt/data/GWAS/ref_files/build38/1KG_GRCh38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# Import functions for visualisation\n",
    "source(\"scripts/QC_functions_task2.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1000 Genome data preparation\n",
    "\n",
    "Data from the 1000 Genomes Project need to be processed before used for population stratification. \n",
    "\n",
    "To complete this task it is necessary to have generated the bfile 'dataset.b38_IBD' from the previous tutorial (Task_2_QC)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T10:27:36.205521Z",
     "start_time": "2020-08-21T10:27:35.118676Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1:10177\n",
      "rs575272151\n",
      "rs540538026\n",
      "rs62635286\n",
      "rs531730856\n",
      "rs707680\n",
      "rs2691315\n",
      "1:15903\n",
      "1:54716\n",
      "1:63735\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "# Get a list of SNPs from your working dataset\n",
    "awk '{print $2}' /mnt/data/GWAS/output/build38/task2_QC/intermediate_datasets/dataset.b38.IBD.bim > $path/mySNPlist.txt\n",
    "head $path/mySNPlist.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:11:32.264601Z",
     "start_time": "2020-08-21T10:59:20.850730Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr1.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 15k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1-temporary.fam\n",
      "written.\n",
      "15569 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999887.\n",
      "15569 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr2.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 15k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2-temporary.fam\n",
      "written.\n",
      "15779 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999919.\n",
      "15779 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr3.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 13k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3-temporary.fam\n",
      "written.\n",
      "13630 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99987.\n",
      "13630 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr4.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 13k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4-temporary.fam\n",
      "written.\n",
      "13575 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999861.\n",
      "13575 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr5.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 12k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5-temporary.fam\n",
      "written.\n",
      "12082 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999893.\n",
      "12082 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr6.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 12k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6-temporary.fam\n",
      "written.\n",
      "12479 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999817.\n",
      "12479 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr7.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 11k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7-temporary.fam\n",
      "written.\n",
      "11496 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999891.\n",
      "11496 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr8.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 10k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8-temporary.fam\n",
      "written.\n",
      "10773 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999851.\n",
      "10773 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr9.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 9k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9-temporary.fam\n",
      "written.\n",
      "9106 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999823.\n",
      "9106 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr10.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf:\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10-temporary.fam\n",
      "written.\n",
      "214 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "214 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr11.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 9k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11-temporary.fam\n",
      "written.\n",
      "9684 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999867.\n",
      "9684 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr12.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 9k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12-temporary.fam\n",
      "written.\n",
      "9353 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999806.\n",
      "9353 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr13.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 7k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13-temporary.fam\n",
      "written.\n",
      "7221 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99989.\n",
      "7221 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr14.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 6k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14-temporary.fam\n",
      "written.\n",
      "6684 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999914.\n",
      "6684 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr15.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 6k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15-temporary.fam\n",
      "written.\n",
      "6284 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999839.\n",
      "6284 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr16.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 6k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16-temporary.fam\n",
      "written.\n",
      "6975 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999821.\n",
      "6975 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr17.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 6k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17-temporary.fam\n",
      "written.\n",
      "6205 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999866.\n",
      "6205 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr18.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 6k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18-temporary.fam\n",
      "written.\n",
      "6242 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999878.\n",
      "6242 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr19.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 5k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19-temporary.fam\n",
      "written.\n",
      "5438 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999846.\n",
      "5438 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr20.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 5k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20-temporary.fam\n",
      "written.\n",
      "5086 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999843.\n",
      "5086 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr21.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 3k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21-temporary.fam\n",
      "written.\n",
      "3057 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999857.\n",
      "3057 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22\n",
      "  --vcf /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr22.recode.vcf\n",
      "  --vcf-half-call haploid\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: 3k variants complete.\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22-temporary.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22-temporary.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22-temporary.fam\n",
      "written.\n",
      "3510 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999754.\n",
      "3510 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr1_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr1\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 15569 out of a possible 6436880 Sites\n",
      "Run Time = 307.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr2_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr2\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 15779 out of a possible 7074549 Sites\n",
      "Run Time = 339.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr3_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr3\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 13630 out of a possible 5827434 Sites\n",
      "Run Time = 277.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr4_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr4\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 13575 out of a possible 5727555 Sites\n",
      "Run Time = 277.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr5_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr5\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 12082 out of a possible 5261557 Sites\n",
      "Run Time = 251.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr6_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr6\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 12479 out of a possible 5017320 Sites\n",
      "Run Time = 244.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr7_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr7\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 11496 out of a possible 4708245 Sites\n",
      "Run Time = 228.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr8_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr8\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 10773 out of a possible 4592680 Sites\n",
      "Run Time = 220.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr9_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr9\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 9106 out of a possible 3546001 Sites\n",
      "Run Time = 171.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr10_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr10\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 214 out of a possible 74937 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr11_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr11\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 9684 out of a possible 4039611 Sites\n",
      "Run Time = 193.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr12_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr12\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 9353 out of a possible 3771938 Sites\n",
      "Run Time = 181.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr13_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr13\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 7221 out of a possible 2855265 Sites\n",
      "Run Time = 138.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr14_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr14\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 6684 out of a possible 2650108 Sites\n",
      "Run Time = 127.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr15_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr15\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 6284 out of a possible 2422112 Sites\n",
      "Run Time = 116.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr16_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr16\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 6975 out of a possible 2695603 Sites\n",
      "Run Time = 129.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr17_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr17\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 6205 out of a possible 2325256 Sites\n",
      "Run Time = 113.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr18_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr18\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 6242 out of a possible 2265382 Sites\n",
      "Run Time = 109.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr19_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr19\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 5438 out of a possible 1828475 Sites\n",
      "Run Time = 90.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr20_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr20\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 5086 out of a possible 1811146 Sites\n",
      "Run Time = 88.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr21_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr21\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 3057 out of a possible 1104028 Sites\n",
      "Run Time = 53.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/build38/1KG_GRCh38/ALL.chr22_GRCh38.genotypes.20170504.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.vcf.chr22\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build38/task2.2_stratification/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 3510 out of a possible 1100291 Sites\n",
      "Run Time = 54.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Creates a recode vcf per chromosome, using as input 1000 Genomes compressed vcf file per chr and previous SNPs list.\n",
    "# Then plink converts them to bfiles. \n",
    "for i in {1..22}\n",
    "do\n",
    "vcftools --gzvcf $ref_files/ALL.chr$i\\_GRCh38.genotypes.20170504.vcf.gz --snps $path/mySNPlist.txt --recode --out $path/1KG.vcf.chr$i\n",
    "plink --vcf $path/1KG.vcf.chr$i.recode.vcf --make-bed --vcf-half-call haploid --out $path/1KG.chr$i\n",
    "rm $path/1KG.vcf.chr$i.recode.vcf\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:50:12.421299Z",
     "start_time": "2020-08-21T13:50:12.360491Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Creates a file with the list of bfiles (.bim, .bed, .fam) outputed in the previous step\n",
    "# 1st-3rd parameter: path to previous plink files\n",
    "# 4th parameter: output file\n",
    "for i in {2..22}\n",
    "do\n",
    "echo $path/1KG.chr$i.bed $path/1KG.chr$i.bim $path/1KG.chr$i.fam >> $path/1KG.outlist.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:52:57.486015Z",
     "start_time": "2020-08-21T13:50:21.665190Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr1\n",
      "  --make-bed\n",
      "  --merge-list /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.outlist.txt\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "Performing single-pass merge (2504 people, 190439 variants).\n",
      "Merged fileset written to                     \n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all-merge.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all-merge.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all-merge.fam .\n",
      "190439 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999852.\n",
      "190439 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.bed\n",
      "+ /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Merge all bfiles into a single one\n",
    "plink --bfile $path/1KG.chr1 --make-bed --merge-list $path/1KG.outlist.txt --out $path/1KG.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1KG.all.bed\n",
      "1KG.all.bim\n",
      "1KG.all.fam\n",
      "1KG.all.log\n",
      "1KG.all.nosex\n",
      "1KG.outlist.txt\n",
      "1kg_ref-list.txt\n",
      "intermediate_datasets\n",
      "mySNPlist.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls $path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:14:02.348522Z",
     "start_time": "2020-08-21T14:14:02.088747Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr2.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr3.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr4.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr5.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr6.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr7.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr8.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr9.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr10.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr11.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr12.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr13.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr14.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr15.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr16.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr17.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr18.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr19.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr20.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr21.*': No such file or directory\n",
      "rm: cannot remove '/mnt/data/GWAS/output/build38/task2.2_stratification/1KG.chr22.*': No such file or directory\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'# remove individual chr plink files\\nfor i in {1..22}\\ndo\\nrm $path/1KG.chr$i.*\\ndone\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-451ac54587c8>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'# remove individual chr plink files\\nfor i in {1..22}\\ndo\\nrm $path/1KG.chr$i.*\\ndone\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2369\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2371\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2372\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'# remove individual chr plink files\\nfor i in {1..22}\\ndo\\nrm $path/1KG.chr$i.*\\ndone\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# remove individual chr plink files\n",
    "for i in {1..22}\n",
    "do\n",
    "rm $path/1KG.chr$i.*\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge the working dataset  and 1000 Genomes data sets \n",
    "\n",
    "Prior to merging 1000 Genomes data with the HapMap data we want to make sure that the files are mergeable, for this we conduct 5 steps:\n",
    "1. Extract the variants present in one dataset from the other\n",
    "2. Resolve strand issues.\n",
    "3. Remove the SNPs which after the previous two steps still differ between datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/../task2_QC/intermediate_datasets/dataset.b38.IBD\n",
      "  --extract /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all.bim\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS\n",
      "  --recode\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "316140 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "--extract: 190439 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999964.\n",
      "190439 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--recode ped to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.ped\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS.map\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Extract the variants present in 1000 Genomes dataset from the working dataset.\n",
    "# awk '{print$2}' $intpath/1kG.all.bim > $path/1kG_MDS_SNPs.txt\n",
    "plink --bfile $path/../task2_QC/intermediate_datasets/dataset.b38.IBD --extract $path/1KG.all.bim --recode --make-bed --out $intpath/dataset.b38_MDS\n",
    "# The datasets now contain the exact same variants.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set reference allele"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj.log.\n",
      "Options in effect:\n",
      "  --a1-allele /mnt/data/GWAS/output/build38/task2.2_stratification/1kg_ref-list.txt\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190439 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999964.\n",
      "--a1-allele: 190439 assignments made.\n",
      "190439 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: Impossible A1 allele assignment for variant rs587693189.\n",
      "Warning: Impossible A1 allele assignment for variant rs587656190.\n",
      "Warning: Impossible A1 allele assignment for variant rs142908143.\n",
      "Warning: Impossible A1 allele assignment for variant rs542844996.\n",
      "Warning: Impossible A1 allele assignment for variant rs532841886.\n",
      "Warning: Impossible A1 allele assignment for variant rs544320996.\n",
      "Warning: Impossible A1 allele assignment for variant rs79425337.\n",
      "Warning: Impossible A1 allele assignment for variant rs564219372.\n",
      "Warning: Impossible A1 allele assignment for variant rs370308267.\n",
      "Warning: Impossible A1 allele assignment for variant rs549442302.\n",
      "Warning: Impossible A1 allele assignment for variant rs550022116.\n",
      "Warning: Impossible A1 allele assignment for variant rs532947788.\n",
      "Warning: Impossible A1 allele assignment for variant rs143845277.\n",
      "Warning: Impossible A1 allele assignment for variant rs116229548.\n",
      "Warning: Impossible A1 allele assignment for variant rs78885634.\n",
      "Warning: Impossible A1 allele assignment for variant rs113271009.\n",
      "Warning: Impossible A1 allele assignment for variant rs572580037.\n",
      "Warning: Impossible A1 allele assignment for variant rs562270648.\n",
      "Warning: Impossible A1 allele assignment for variant rs71484804.\n",
      "Warning: Impossible A1 allele assignment for variant rs534787289.\n",
      "Warning: Impossible A1 allele assignment for variant rs532472587.\n",
      "Warning: Impossible A1 allele assignment for variant rs649129.\n",
      "Warning: Impossible A1 allele assignment for variant rs202150501.\n",
      "Warning: Impossible A1 allele assignment for variant rs558867955.\n",
      "Warning: Impossible A1 allele assignment for variant rs571706883.\n",
      "Warning: Impossible A1 allele assignment for variant rs187199126.\n",
      "Warning: Impossible A1 allele assignment for variant rs554904493.\n",
      "Warning: Impossible A1 allele assignment for variant rs143371137.\n",
      "Warning: Impossible A1 allele assignment for variant rs116335217.\n",
      "Warning: Impossible A1 allele assignment for variant rs63597060.\n",
      "Warning: Impossible A1 allele assignment for variant rs190861274.\n",
      "Warning: Impossible A1 allele assignment for variant rs11857016.\n",
      "Warning: Impossible A1 allele assignment for variant rs537370534.\n",
      "Warning: Impossible A1 allele assignment for variant rs191177490.\n",
      "Warning: Impossible A1 allele assignment for variant rs4627675.\n",
      "Warning: Impossible A1 allele assignment for variant rs374370911.\n",
      "Warning: Impossible A1 allele assignment for variant rs553808485.\n",
      "Warning: Impossible A1 allele assignment for variant rs212150.\n",
      "Warning: Impossible A1 allele assignment for variant rs529503139.\n",
      "Warning: Impossible A1 allele assignment for variant rs558672245.\n",
      "Warning: Impossible A1 allele assignment for variant rs147504593.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk '{print$2,$5}' $path/1KG.all.bim > $path/1kg_ref-list.txt\n",
    "plink --bfile $intpath/dataset.b38_MDS --reference-allele $path/1kg_ref-list.txt --make-bed --out $intpath/dataset.b38_adj\n",
    "# The 1kG_MDS7 and the dataset.b37_adj have the same reference genome for all SNPs.\n",
    "# This command will generate some warnings for impossible A1 allele assignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs575272151 G\n",
      "rs540538026 A\n",
      "rs62635286 G\n",
      "rs531730856 C\n",
      "rs707680 A\n",
      "rs2691315 T\n",
      "rs115209712 G\n",
      "rs940550 T\n",
      "rs544194668 G\n",
      "rs78348422 T\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/1kg_ref-list.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Resolve strand issues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs113271009 C T\n",
      "rs113271009 TTGATTGAT T\n",
      "rs116229548 G T\n",
      "rs116229548 GT TT\n",
      "rs116335217 A C\n",
      "rs116335217 CCCAGAGAGTTTGTATTGC C\n",
      "rs11857016 A C\n",
      "rs11857016 T C\n",
      "rs142908143 G C\n",
      "rs142908143 T C\n",
      "  94  282 1594 /mnt/data/GWAS/output/build38/task2.2_stratification/all_differences.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Check for potential strand issues.\n",
    "awk '{print$2,$5,$6}' $path/1KG.all.bim > $intpath/1kGall_tmp\n",
    "awk '{print$2,$5,$6}' $intpath/dataset.b38_adj.bim > $intpath/dataset.b38_adj_tmp\n",
    "sort $intpath/1kGall_tmp $intpath/dataset.b38_adj_tmp |uniq -u > $path/all_differences.txt\n",
    "# 94 differences between the files, some of these might be due to strand issues.\n",
    "head $path/all_differences.txt\n",
    "wc $path/all_differences.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 47  47 549 /mnt/data/GWAS/output/build38/task2.2_stratification/flip_list.txt\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected.log.\n",
      "Options in effect:\n",
      "  --a1-allele /mnt/data/GWAS/output/build38/task2.2_stratification/1kg_ref-list.txt\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_adj\n",
      "  --flip /mnt/data/GWAS/output/build38/task2.2_stratification/flip_list.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190439 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "--flip: 43 SNPs flipped.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999964.\n",
      "--a1-allele: 190439 assignments made.\n",
      "190439 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 15 variants had at least one non-A/C/G/T allele name.\n",
      "Warning: Impossible A1 allele assignment for variant rs544320996.\n",
      "Warning: Impossible A1 allele assignment for variant rs79425337.\n",
      "Warning: Impossible A1 allele assignment for variant rs370308267.\n",
      "Warning: Impossible A1 allele assignment for variant rs549442302.\n",
      "Warning: Impossible A1 allele assignment for variant rs532947788.\n",
      "Warning: Impossible A1 allele assignment for variant rs563138500.\n",
      "Warning: Impossible A1 allele assignment for variant rs143845277.\n",
      "Warning: Impossible A1 allele assignment for variant rs116229548.\n",
      "Warning: Impossible A1 allele assignment for variant rs78885634.\n",
      "Warning: Impossible A1 allele assignment for variant rs113271009.\n",
      "Warning: Impossible A1 allele assignment for variant rs564904242.\n",
      "Warning: Impossible A1 allele assignment for variant rs62566171.\n",
      "Warning: Impossible A1 allele assignment for variant rs649129.\n",
      "Warning: Impossible A1 allele assignment for variant rs202150501.\n",
      "Warning: Impossible A1 allele assignment for variant rs558867955.\n",
      "Warning: Impossible A1 allele assignment for variant rs187199126.\n",
      "Warning: Impossible A1 allele assignment for variant rs116335217.\n",
      "Warning: Impossible A1 allele assignment for variant rs63597060.\n",
      "Warning: Impossible A1 allele assignment for variant rs190861274.\n",
      "Warning: Impossible A1 allele assignment for variant rs542700889.\n",
      "Warning: Impossible A1 allele assignment for variant rs374370911.\n",
      "Warning: Impossible A1 allele assignment for variant rs553808485.\n",
      "Warning: Impossible A1 allele assignment for variant rs558672245.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Flip SNPs for resolving strand issues.\n",
    "# Print SNP-identifier and remove duplicates.\n",
    "awk '{print$1}' $path/all_differences.txt | sort -u > $path/flip_list.txt\n",
    "wc $path/flip_list.txt\n",
    "# Generates a file of 329 SNPs. These are the non-corresponding SNPs between the two files. \n",
    "# Flip the 43 non-corresponding SNPs. \n",
    "plink --bfile $intpath/dataset.b38_adj --flip $path/flip_list.txt --reference-allele $path/1kg_ref-list.txt --make-bed --out $intpath/dataset.b38.corrected\n",
    "# This command will generate some warnings for impossible A1 allele assignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Check for SNPs which are still problematic after they have been flipped.\n",
    "awk '{print$2,$5,$6}' $intpath/dataset.b38.corrected.bim > $intpath/dataset.b38.corrected_tmp\n",
    "sort $intpath/1kGall_tmp $intpath/dataset.b38.corrected_tmp |uniq -u  > $path/uncorresponding_SNPs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "74 /mnt/data/GWAS/output/build38/task2.2_stratification/uncorresponding_SNPs.txt\n",
      "rs113271009 C T\n",
      "rs113271009 TTGATTGAT A\n",
      "rs116229548 G T\n",
      "rs116229548 GT TT\n",
      "rs116335217 A C\n",
      "rs116335217 CCCAGAGAGTTTGTATTGC G\n",
      "rs11857016 T C\n",
      "rs11857016 T G\n",
      "rs142908143 G A\n",
      "rs142908143 G C\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# This file demonstrates that there are 74 differences between the files.\n",
    "wc -l $path/uncorresponding_SNPs.txt\n",
    "head $path/uncorresponding_SNPs.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove problematic SNPs from the working dataset and 1000 Genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs113271009\n",
      "rs116229548\n",
      "rs116335217\n",
      "rs11857016\n",
      "rs142908143\n",
      "rs143845277\n",
      "rs147504593\n",
      "rs187199126\n",
      "rs190861274\n",
      "rs202150501\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk '{print$1}' $path/uncorresponding_SNPs.txt | sort -u > $path/SNPs_for_exlusion.txt\n",
    "# The command above generates a list of the SNPs which caused the 74 differences between the working dataset (i.e HapMap) and the 1000 Genomes data sets after flipping and setting of the reference genome.\n",
    "head $path/SNPs_for_exlusion.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.corrected\n",
      "  --exclude /mnt/data/GWAS/output/build38/task2.2_stratification/SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190439 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "--exclude: 190402 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999964.\n",
      "190402 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/1KG.all\n",
      "  --exclude /mnt/data/GWAS/output/build38/task2.2_stratification/SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190439 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS.nosex\n",
      ".\n",
      "--exclude: 190402 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999855.\n",
      "190402 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/1kG_MDS.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove the problematic SNPs from both datasets.\n",
    "plink --bfile $intpath/dataset.b38.corrected --exclude $path/SNPs_for_exlusion.txt --make-bed --out $intpath/dataset.b38_MDS2\n",
    "plink --bfile $path/1KG.all --exclude $path/SNPs_for_exlusion.txt --make-bed --out $intpath/1kG_MDS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge the working dataset with 1000 Genomes Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2\n",
      "  --bmerge /mnt/data/GWAS/output/build38/task2.2_stratification/1kG_MDS.bed /mnt/data/GWAS/output/build38/task2.2_stratification/1kG_MDS.bim /mnt/data/GWAS/output/build38/task2.2_stratification/1kG_MDS.fam\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "495 people loaded from\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.fam.\n",
      "2504 people to be merged from\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1kG_MDS.fam.\n",
      "Of these, 2504 are new, while 0 are present in the base dataset.\n",
      "190402 markers loaded from\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38_MDS2.bim.\n",
      "190402 markers to be merged from\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/1kG_MDS.bim.\n",
      "Of these, 0 are new, while 190402 are present in the base dataset.\n",
      "Performing single-pass merge (2999 people, 190402 variants).\n",
      "Merged fileset written to                     \n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2-merge.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2-merge.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2-merge.fam .\n",
      "190402 variants loaded from .bim file.\n",
      "2999 people (237 males, 258 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.nosex .\n",
      "495 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2999 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999873.\n",
      "190402 variants and 2999 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.bed +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.bim +\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: Multiple positions seen for variant 'rs58657412'.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Merge HapMap with 1000 Genomes Data.\n",
    "plink --bfile $intpath/dataset.b38_MDS2 --bmerge $path/1kG_MDS.bed $path/1kG_MDS.bim $path/1kG_MDS.fam --allow-no-sex --make-bed --out $path/MDS_merge2\n",
    "\n",
    "# Note, we are fully aware of the sample overlap between the HapMap and 1000 Genomes datasets. However, for the purpose of this tutorial this is not important.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Perform Multidimensional Scaling (MDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2\n",
      "  --extract /mnt/data/GWAS/output/build38/task2.2_stratification/../task2_QC/indepSNP.prune.in\n",
      "  --genome\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190402 variants loaded from .bim file.\n",
      "2999 people (237 males, 258 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.nosex .\n",
      "495 phenotype values loaded from .fam.\n",
      "--extract: 128816 variants remaining.\n",
      "Using up to 4 threads (change this with --threads).\n",
      "Before main variant filters, 2999 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999861.\n",
      "128816 variants and 2999 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "IBD calculations complete.  \n",
      "Finished writing\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.genome .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2\n",
      "  --read-genome /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "190402 variants loaded from .bim file.\n",
      "2999 people (237 males, 258 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.nosex .\n",
      "495 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2999 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999873.\n",
      "190402 variants and 2999 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "Clustering... done.                        \n",
      "Cluster solution written to 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.cluster1 ,\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.cluster2 , and\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.mds .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Using a set of pruned SNPs\n",
    "plink --bfile $path/MDS_merge2 --extract $path/../task2_QC/indepSNP.prune.in --genome --out $path/MDS_merge2\n",
    "plink --bfile $path/MDS_merge2 --read-genome $path/MDS_merge2.genome --cluster --mds-plot 10 --out $path/MDS_merge2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MDS-plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2505 /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt\n",
      "FID\tIID\tSuperPop\n",
      "HG00096\tHG00096\tEUR\n",
      "HG00097\tHG00097\tEUR\n",
      "HG00099\tHG00099\tEUR\n",
      "HG00100\tHG00100\tEUR\n",
      "HG00101\tHG00101\tEUR\n",
      "HG00102\tHG00102\tEUR\n",
      "HG00103\tHG00103\tEUR\n",
      "HG00105\tHG00105\tEUR\n",
      "HG00106\tHG00106\tEUR\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# 1000 Genomes superpopulation codes\n",
    "wc -l /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt\n",
    "head /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HGX00096 HGX00096 OWN\n",
      "HGX00097 HGX00097 OWN\n",
      "HGX00099 HGX00099 OWN\n",
      "HGX00100 HGX00100 OWN\n",
      "HGX00101 HGX00101 OWN\n",
      "HGX00102 HGX00102 OWN\n",
      "HGX00103 HGX00103 OWN\n",
      "HGX00105 HGX00105 OWN\n",
      "HGX00106 HGX00106 OWN\n",
      "HGX00107 HGX00107 OWN\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Create a racefile of your own data.\n",
    "awk '{print$1,$2,\"OWN\"}' $intpath/dataset.b38_MDS.fam>$path/racefile_own.txt\n",
    "head $path/racefile_own.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID\tIID\tSuperPop\n",
      "HG00096\tHG00096\tEUR\n",
      "HG00097\tHG00097\tEUR\n",
      "HG00099\tHG00099\tEUR\n",
      "HG00100\tHG00100\tEUR\n",
      "HG00101\tHG00101\tEUR\n",
      "HG00102\tHG00102\tEUR\n",
      "HG00103\tHG00103\tEUR\n",
      "HG00105\tHG00105\tEUR\n",
      "HG00106\tHG00106\tEUR\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Concatenate racefiles.\n",
    "cat /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt $path/racefile_own.txt  > $path/racefile.txt\n",
    "head $path/racefile.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOzdd3xV5f3A8c85dyT3Zk9IQkIg7CFTgmwEF47WUeqoVq1aUau/qq2iWEHrqLO1Wq11FDcuHHUiygqgouy9R0II2ePue87vjxtCSG52cnO5+b5ffb1675nPAfnmyfc8z/dRdF1HCCFE8FE7uwFCCCH8kwAthBBBSgK0EEIEKQnQQggRpCRACyFEkJIALYQQQUoCtBBCBCkJ0EIIEaQkQAshRJCSAC2EEEFKArQQQgQpCdBCCBGkJEALIUSQkgAthBBBSgK0EEIEKQnQQggRpCRACyFEkJIALYQQQUoCtBBCBCkJ0EIIEaQkQAshRJCSAC2EEEFKArQQQgQpCdBCCBGkJEALIUSQkgAthBBBSgK0EEIEKQnQQggRpCRACyFEkJIALYQQQUoCtBBCBCkJ0EIIEaQkQAshRJCSAC2EEEFKArQQQgQpCdBCCBGkJEALIUSQkgAthBBBSgK0EEIEKQnQQggRpCRACyFEkJIALYQQQUoCtBBCBCkJ0EIIEaQkQAshRJCSAC2EEEFKArQQQgQpCdBCCBGkJEALIUSQkgAthBBBSgK0EEIEKQnQQggRpCRACyFEkJIALYQQQUoCtBBCBCkJ0EIIEaQkQAshRJAydnYDWiYsLMztdnd2K4QQ4rju3bvn5eV1xJUDHaANBoOu67quq6rq9Xpberrb7dZ1XVGUjmibECJI6Lr++OOPJyYmdnZDmrZkyZI33nijgy4eiACtKEp8fHxRUZGqqrqu+zZqmqaqqqZprbhaK84KbTk5Oenp6RkZGZ3dECHax3XXXXfhhRdmZWV1dkOatmPHjo67eIBy0L7Osq/z6+tB+z4E5u4hb8OGDYcPH+7sVggh2lmgUxyqWv0jwWQyuVyuAN89VM2aNauzmyCEaH8B6kGXl5f7QnPtFEdgbt0V5OTkHDhwoLNbIYRoZwEK0L60BrUCdCveEIqGSIpDiJAUiBSH31yz1WqVHHR7kRSHECGp0yaqVFZWVlVVddbdQ4ykOIQISZ0WoDMzM2U4c3uRFIcQIanTZhKWlpZ21q1Dj6Q4hAhJgehBK/6UlZUF4NZdhKQ4hKhvyJAhtWPOeeed59seGRm5YsWKmsOeeOKJ6dOn1zklKSnpsssuKy4u7oR21yLFkkKBpDhE1+Fw8OyzzJrFE0/QZDfvlVdesR+zcOHC5lx//vz5brf7hx9+2Lt37z333NMOLW6DAAXomgmENWJiYho/xWAwqPUEprUnnVmzZmVnZ3d2K4TocC4X552HxcJtt5GRwbnnNhGjTSZT+DEmk6k5t1BV1Wg09urV68ILL1y7dm37tLu1AlSLo/7G2NjYxrMcraum1DVJLQ7RRXz8MWefze9+BzBgALrOyy9z++3tf6NDhw599NFHffr0af9Lt0Qg+qSaptWfN7hv377Gx0G73W6tno5s5klMUhyii9izhyFDjn8dOpRduxo7ftasWYnHPPjgg825xZVXXqkoSnp6uqZpTz75ZNva21aSNAgFkuIQXcSQIeTkHP+ak8PQoY0d//DDD6875rbbbvNtNJlMtcvKu93u2tmP119/XdO01atX5+XldXq/p9MCtKqqMg66vcgoDtFFzJjB9u3MmcMXX/Doo7z/Ptde29jxcXFxPY6Jjo72bczIyNi7d2/NMbt37+7Zs2ftsxRFyc7Ovv3222+99dYOeIgWkB50KJAUh+giFIUFCxg3jg0b6NePzz8nLKyx491ut+OYmvKZl1566RNPPLFhwwabzfbVV18tWLDg17/+df1zb7jhhi1btixevLgjHqSZOm2iiiSU25FMVBFdh6IwYwYzZjTr4GuvvfbaY33sYcOGrVu3DrjzzjsdDscvf/nL/Pz8rKysf/3rX1OnTq1/bkRExE033fTAAw9Mmzat/ZrfMifZmoTCLxnFIUR9mzZt8rvdZDLNmzdv3rx5TZ7SzPeKHSdAKQ6DwVBnJqHBYAjMrbsCSXEIEZIC0YOuWYqw5q2gruutXpNQ1CcpDiFCUiB60L41vH1B2admSwDu3hXIKA4hQlKAUhxGY92uejOnXYrmkBSHECEpQAHa5XIZDAar1Tp37lyr1WowGJxOZ2Bu3RXIRBUh6muomp3PlClT4uLiagei1atXT5o0KSoqKi4ubvz48UuXLg14k+sKRIC2WCyApml2u33evHl2u92XevZtF20nKQ7RpZRR9jM/F9N0LdCGqtnt27dv+fLliqJ88sknvi0Oh+O8886bMWNGbm7u7t2777333vq/9wdeIFpgs9mA1NTUgoICXdcVRUlOTs7LywvArbuIDRs2GI1GGWYnuoJneGYhC0cwYitbRzLyIR5q5GBfNbv621977bWxY8dmZ2fPnz//V7/6FbBv376ioqL/+7//8x0/o5kDrTtY4H5ESETuODKKQ3QRP/HTKlZ9y7cKCnAnd37Kp+dzfkuv89prr91+++3Z2dljx449cuRIt27dMjMz09LSrr766muuuWbMmDFxcXEd0PwWk6neoUBSHKKLWMrSy7ncF52BK7lyMY1NxfZbzW7FihX79++fOXPmqFGjsrKy3nrrLSA8PDwnJychIeGWW25JTk6eMWPGwYMHO/pxmiQBOhTIKA7RRcQQU8rx5UxLKY2hsaU//Fazmz9//plnnpmYmAhcfvnl8+fP923v2bPnc889t3Pnzv3793u93uuuu67DnqO5Oj8LLtpOUhyiiziP8y7hkglM6EWvAgr+yl+f4ZlGjvdVs6u9xW63v/vuu16vt3v37oDT6SwtLV2/fv2wYcNqjklNTb3uuus6vZQdEqBDg9TiEF1EN7r9i3/dyZ2FFEYTPZe5AxnYyPG+ana+z6qqms3mjz76yGAwbNy40Ww2+7bPnDnztddeu+WWW+bPn3/ppZdmZmYeOnTohRdeCIahq5LiCAWS4hBdx1CGfsAHS1n6KZ+OZ3zjB1977bWWY8aMGQPMnz//mmuuycjI6H7MLbfc8uabb1qt1n379p199tkxMTETJ05MS0t78cUXA/JAjZEedCiQFIcQ9fmtZvfll1/W2TJz5syZM2cC//3vfwPQqhaRHnQokFEcQoQkCdChQFIcQoQkSXGEAklxCBGSpAcdCiTFIURIkgAdCiTFIURIkgAdCqTcqBD1NV4+VMqNigCRFIfoSvbCtTAVLoONDR3UePlQKTcqAkfKjYouowh+A8/BcNgN18Kr0Lv+cY2XDz1Zyo1KDzoUSIpDdBn/g9/BcACy4D540+9xNeVDv/rqq5KSkjp7X3vttSuuuOKKK6746quvjhw50uTxnUUCdCiQFIfoMo5Ct1pfu8FRv8c1Uj5Uyo2KgJJRHKLLGA8f1vr6AUxs6NCGyodKuVERUDJRRXQZp8EiuADGwEZIhEuaPKd2+VApNyoCTcqNiq7kL5ALW+AK6NXQQXv37vVbPlTKjYpAkxSH6GLS4IxGojMQHR3tt3zoyVVuVNF1vbPb0AKqqgKapnV2Q4QQHei6666bPXt2VlZWZzekaffcc89jjz3m8Xg64uLSgw4FMopDiJAkAToUSIpDiJAkLwlDgYziECIkSQ86FEiKQ4iQJAE6FEiKQ4iG1K9aN2TIEFVV9+/fX7NlwoQJiqKsW7fOt1dRFEVRkpKSLrvssuLi4k5o9DESoEOB1OIQXUvhKna/xJHvoIlBaPWr1vn07dv3jTfe8H3es2dPbm5uzbBoYP78+W63+4cffti7d+8999zT7s1vPgnQoUBSHKIL+f469r6OwcLhL1l2IXpj49t8VeuuvvrqmvncPr/5zW9ef/31mmN+85vfKIpSs1dVVaPR2KtXrwsvvHDt2rUd8RDNJAE6FEiKQ3QV+Yswx3Lqv8i8guF/o9vp7HmtkcPrV63zGThwYGRk5A8//AC88cYbV155Zf1zDx069NFHH/Xp06fdH6L5JECHAklxiK6ieC3dph3/2n06xT81dKzfqnU1rrrqqtdff33FihVJSUn9+vWrvevKK69UFCU9PV3TtCeffLK9n6EFJECHAklxiK4iIp3KXce/VuwgosESNA1VrfO57LLL3n///Zdffvmqq66qc+Lrr7+uadrq1avz8vI693dTGQcdCmRFFdFV9Pgl352FpQdJ4yldz5bHmfyR3wObrFqXlJQ0atSod955x28fWVGU7Ozs22+//dZbb12+fHnHPVDjpAcdCiTFIboKg4XJn1K2mTW3cGQJkz4kLMnvgb6qdVu2bFm3bt26deu2bt06ceLE1147IWH97LPPLlmyJD4+vqG73XDDDVu2bFm8eHE7P0WzSYAOBZLiEF2IKYYhc5jwLsMeIrxbQ0c1VLWudlWjzMzMxns2ERERN9100wMPPNCe7W8JSXGEAklxCFHHl19+WWfLzJkzZ86cCWzatKn+8Q6Hw/ehzt4HH3ywYxrYLBKgQ4HU4hAiJAUoxWEwGJQTGQyGwNy6K5AUhxAhKRABWlVVX4n9mugMaJrmq74v2k4mqggRkgKR4tB1XVVVr9dbe6PBYJCFUdqLpDiECEkB6sMajXV/EphMpsDcuiuQFIfoatyVJ9Nafa0WoADtcrkMBoPVap07d67VajUYDLWr/4k2khSH6Gq+u9HWzCP9lhut/T5szZo1BFmV0RqBCNAWiwXQNM1ut8+bN89ut/uSG77tou1kooroanRv08fQcLnRV155xX7MqFGjfBuDp8pojUAEaJvNput6SkqKwWBQVdVgMKSkpOi6brM192egaJykOITwq6FyoyaTKfyYmkKjwVNltEbgxkHn5eUF7F5djUxUEV1ByXZv2e7qkQWOIn3f527f58hUNXG4/2G7r7322u23356dnT127NgjR45069bgzMMawVBltEanDXTLzMysXSFbtIWkOERX4KnCVab7/qd59ZrPrgZeGDZSbnTWrFmJiYmJiYkTJ06s2Rg8VUZrdFqALi0t7axbhx5JcYiuIGmkod9lZt//rMlqzefUCf4zAY2UG3344Yd9RZTef//9mo3BU2W0RiBSHNJT7miS4hCijsbLjcbFxfXo0aP+WUFSZbSGzOULBZLiEF1OU72+5pQbbUinVxmtEaAArSiKfqKYmJjGT4mKijLWE5jWnnQkxSG6molPNzFItznlRhvS6VVGa3RaiiM2NrasrKyRs6qqqnTdT+5fEib1SYpDdDXWbk10LltabjSoqozWCESA9ltzY9++fa04S+or+SW1OIQISRLvQoGkOIQISZ0WoFVVlWRFe5FaHEKEJHntFgokxSFESOq0HrSmaX7fAYpWkBSHEH69/fbbw4cPDw8P79at2w033OCrUffII4/MmDGj5pi+ffvW+frOO+8AQ4YMiY6OLioq8m1/5513Ro8eHdjmSw46JEiKQ3QdDofj2WefnTVr1hNPPNH4SLB///vfN9100913333kyJElS5YcPHhw+vTpTqdz0qRJOTk5viVE8vPz3W73zz//XPN1165dkyZN8l3BZDL97W9/C8BDNUTWJAwFMlFFdBEul+u8886zWCy33XZbRkbGueee21CMdjgcs2fPfvLJJy+99NKYmJiBAwd+8MEHhw8fnj9//qmnnup2u9etWwcsW7Zs6tSp/fv3r/malZWVmprqu8gf//jHF154IT8/P2APWIesSRgKJMUhuoiPP/747LPP/t3vfjdgwICZM2f+4Q9/ePnll/0euWbNmpKSkl/96lc1W6xW6wUXXLBo0SKz2Zydnb1s2TJg2bJlEydOnDBhQs3Xmu4zMGDAgAsvvPDhhx/u4MdqUCBCpG9NQl3XtWNqtgTg7l2BpDhEF7Fnz54hQ4bUfB06dOiuXbv8HllYWBgREREVFVV7Y0pKSmFhITB58mRfRF6+fPnEiRMnTpxY83Xy5Mm1T7n//vtfffXVgwcPtvuzNIesSRgKJMUhuoghQ4bk5OTUfM3JyRk6dKjfIxMTE6uqqioqKmpvPHz4sK+43aRJk1asWFFSUnL06NG+ffuOGzdu5cqVJSUlmzZtqt2DBnr37n355Zd31rRvWZMwFEiKQ3QRM2bM2L59+5w5c7744otHH330/fffv/baa/0eOWrUqNjY2Pfee69mi81m++STT6ZNmwacdtppZWVlL7744vjx44Ho6OjU1NQXX3wxNTW1V69edS41Z86ct99+e/fu3R32WA2SNQlDgaQ4RBehKMqCBQvGjRu3YcOGfv36ff7552FhYX6PtFgsDz300B133LFgwYKysrJt27ZdfPHFycnJV199tW/v6NGjn3rqqZqC/RMmTHjqqafqdJ990tPTr7322qeeeqrDHqtBsiZhKJAUh+g6FEWZMWPGXXfdddFFFzU+GOymm2569tlnH3744eTk5IkTJ6alpX377bfh4eG+vZMnTy4oKJgwYYLv68SJEwsKCvwGaOCee+5xOBzt+yDNoZxcb+p8Az/81lHqynJyctLT06WanQgZ11133ezZs7Oysjq7IU275557HnvsseZUMW0FGegWCiTFIURIklocoUBqcQgRkqQHHQpkFIcQIUkCdChoY4rjCEeWsMSJDHwUIrhIiiMUtCXF8QIv/IE/ePECVqxTmfoBH5gxt1/rhBCtJD3oUNC6FIcL1yhGzWKWB4+OrqNXUXWQg0tZ2hGNFEK0lAToUNC6FMfrvP4zP9e9FBvSSW+ndgkh2kRSHKGgdSkOo7+/fR19OtMPcajNjRJCtJUE6FDQuokqpZT63Z5LrgNHOOHt0TQhWkNRlMcffzw2NrazG9K0nTt3dtzFJUCHgg0bNhiNxpYG6E/4pKFdbtwSoEUnevDBB3Nzczu7Fc0yevTojlt+RKZ6d12TmLSc5X53VVFlxRrg9ghxMvKtDyVTvUWDWjGKo4KKHHIa2vsTP7W5UUKItpIAHQpaMYpjIQsjiWxobznlbW6UEKKtJMXRRW1l63jGV1HlwlVnVwwxP/Nzb3p3SsOEOLlIikM0oRUpjoEMfImXookO44R652GEncZpmWS2Z/uEEK0iAToUtG6iykVctJKVMcSYMFmwqKiv8MpgBn/GZ6r8hyFEEJAUR1f3Du/8gT8kkFBBRS963czNl3FZZzdKiJNGh6Y4JECHglavqOLGPZShJkx55F3P9Y/yaEc0T4gQJjlo0YRWlxt14qykcgELfsfvvubrdm+YEKItZCZhKGh1udFIIi/ggglMSCDhRm5s31YJIdpIUhyhoI2Lxh7ikA1bP/q1b6uE6Ao6NMUhPehQ0LpaHBVUWLEaMPSgRwc1TAjRFtKD7qIe4IEccsopf4d3etKzs5sjxMlKXhKKJrRiospiFn/FV3/iTx/xUQe1SgjRRhKgQ0ErRnGkkjqXuf/iX+MY10GtEkK0kaQ4uig37m/4pg99+tK3s9sixEksoCkOo9GoqqrRePzl4eDBg31hUQStVqQ4TJjO4RyJzkIEsxMir6qqXq9X13Wv16soyuDBg4Hc3NyTq5fdBbV6oooQIpidkOJQFEVRFE3TzGaz2+0GUlJSbDZbWVlZkMRoSXEIIYJKQFMcJpMJcLlc8fHxwOHDh51OZ0fcWLSjVqQ4hBDBr26Arvk5UFRUlJSUBDgcjkA3SrSQpDiECEl1UxxA7S2ZmZn79++vs7ETSYpDCBFUApfiSEpKslgstbfs27dv0KBBHbeouGgXkuIQIiSdUIujoKCg/hGbN28OVGNEK7WuFocQIsjJRBUhhGg9qcUhmiApDiFCkgToUCCjOIQISX5SHKqq1skh1N/SagaDQdd1Xdd9sxZberqkOIQQQSXQKY76IbuNeWpFURISEjgW6H1X0zRNSny0F0lxCBGSThjFUTOcrva4unZ5i+jrLOu67ptKDqiqenK9nwxmMopDiJDkZ6KKn4OORdVW3kNRYmJiSktLa/8uEBYW5nK5WhqjJcUhhAgqgUtx+LLDNR9qtD0glpeX+2JrTUSWINuOJMUhREhqVg667WqHft+WVrwhFA2RURxChCQ/ozgyMzMPHDhQZ3u7R+3IyEhd16uqqlp0lqQ4WiN/MbtfJv0iMi7p7KYIEWo6NMXhJ0DXZKJrp6SDJCZKgPYrJycnPT3d/0vCsm18OZKovjgLmfI5ccMC3johQlknzCRMSUnxpZ5rtPE2BoNBOZEUYGpHjaU4dr+IamLIfTiLcBWjuVl/DzmXUrIusG0UQrSY0e9Ws9ncjveoGVFX0yX3Rf92nP/Sxc2aNavBfZY0vE5WzATY9yaVezDFMuwRvr+Wad8FrIVCiFbw04NWFGX//v2GE7XlHr55g7W75DVb2nJZUaOxURzJE9A8mKIIT6RkPZ5yFAVFRf7whQh6fnrQNTP92vM2xro3MplMsphWe2l4oorOmltQFbx2UHAcoXwn9jyKfmTU053QUCFES/gfZldfG2/jcrkMBoPVap07d67VajUYDE1GZ1VVlXqk0+3XrFmzsrOz/ezwVFG6BUMEmhtPFQYrfX6PIZxBfyZuRMCbKYRoGf8vCWuCI2AwGNpYNMO3SoumaXa7fd68eXa73dc9r7N6Sx2+TEgdDc117OIaTHHkfobmRNcwRRORTtU+CldizcBVEvA2CiFazE/k9WWHa4fCNvZbbTabruspKSm+WG8wGHyjRGw2W1suK2o0OIqjdAOxQ/BW4a7AWUzCGDY9gO0Q3aYFvI1CiBbzPw7aaDS63W5fSmHw4MFbtmwJktyCjINumUUTOJoDgIIaztSvOPQ+aRfQXQK0EO2jE8ZBR0RE1Hzeu3dvh9xYVSVf0V78pzhsuZRtQjUTlgAKqsr3vyUsge6nd0YbhRAt5j9Al5WVDR48GJg7d67dbg9sk0SL+U9xrP0TkX3Q3DiLQEMxEtmP3teA/FwU4uTgJ8VhNpvdbnedLUEyJE5SHC2w9HwG3EHOTJzlmKxEZhKRyYjHiczq7JYJEToCneJwuVz333+/LwWhqur9998fJNFZNMR/imP4o+RcimoBLx4bZVvpfoZEZyFOIgEaWWwwGOqvcyhrEraX559/fuTIkX6GQn8zGVcFVbtxV6AYSczmjOWd0UAhQlagq9m1e7lRv7U4aNVCLRKgW2bVVex7C11DNWMwY4zgrB+x9ujsZgkROgKd4ti/f39NAK3RlntILY6O1sBEFZ1DnxA/HAV0Fx4bajjm+E5onxCiVQJUbtRvLY42XlPU8D+Kw+vAHEfiBFQTUf1JOYs+12O0Hj+gaj+7X6ZiRyCbKoRoPv8Bun3LjdKqWhyi+fzX4jBYiOrHrpcwWInqjT2PzMuP7/XaybkU1czqa3EeDWRrhRDNFIhyo62rxSGaz3+Ko2AZJT8x/GG8DjQv4cmUbjy+11FAZBa9riThVCr3BbCxQojmanDRWO1EbbmH1OLoaP5THJV7CEum15VoTir3UPwzP96MI796b0QGxgi+mYw9n/iRAW6wEKI5TrICnjKKowU8VXyUgasYFEwRKEb6/xFDGIPu6uyWCRE6OqEWR+0lBGXxwODnP8Wx5W9oTsLTMFpw27B0p3I30f2PH2DPY8l5LJ7KkSUBbKwQorn8rKhSe9iyLB54UvC/okrlHgwRhMdTUYJqJOt6ovuTeu7xA3Y8x4A/kpjNsl/SbUpgmyyEaJr/Ja9qT/OrPwlQBBv/i8b2u4m8zyndiGIibhiWFFLPxVmEs5CInhx4F3cpRxbjKiYsMeBNFkI0zX+KIyEhwe/nIFFVVXXw4MEnnnhCPvg+LFy4cM6cOXV33XdNld11sCLuia8jqyJGHty19okH/lj19XkHv73nid/3rirNP5hf/sRLi6qObj2YMicYnkI+yIeT8UOHxjr/AbqwsDAzMxPIzMw8ejToBsmazebIyMisrCz54PuwZ8+eqKioE3YVfZYVddhsiY5MyspKqDIXLYkcfF1W7FHz8L9ETn4pK67cPOimyIG/zcpMMZ8yOzIuLRieQj7Ih5PxQ4fGOj+jOIxGY50yRh33jrKlZBRHs1Tu4cdZ2POx5eKppNskxr+LOZbiNfz0R6L64CrFkY9qYsx/TnhtWMNVSvkW4kehhgW89UKcTAJdLAkYPHjwtm3bfCsTDhgwYPPmzR1x71aQAO1XTk5Oenr68ZeEpRvY+iQZM1l5Ob2v4pSHUIzVk7ydR3EWEj2wscu5illyLsmTKFrDtG+kwL8QjeiEYXabN2/2er2apnm93uCJzqIhdSeqxJ5CxXaWnofXwe6XWZjKwu5sfgQgLKmJ6AyUrCf1HIb/jfAkHEc6sN1CiEb56UEnJCSUlJTUbFcUJTExsaCgIOBt80N60E1zV7B4CiXr0XXMUbgrMIQR1Z/KvYx4jMzfnFAvqaErLJlBVBauEiZ9JD1oIRoR6B50cXGxL7nhW1RF1/UgfE8oajthosrRZSRkY4rEaMZVjq5hjKRsM7oXXeOn25q+nCmKad8y5C9MXNhYdF53N99MZuPcdnkEIUR9/lMcPXv29OU3NE1r98p2ot2dkOKIPYWS9YR3x+vG0h1LdxLGoRjo/Rt6XUnVPoCKnVTtb+yKqonI3ij+//MAKNuM4wjTl1K+jco97fYkQoha/KQ4FEXp2bPnvn37fF/DwsJcLleQlOyQFEez2A5yZAk7niOiB3lfobsxRBDdH8XA0Pup2EH+t3jt9P4tGTNbeQv7YVZezikPsv4eJn0k6wCILivQozh8U719hTh0Xa/57NvbioUE25EEaL/qjuIA9vyXA+9jiqZwFdEDqNxDeDKmKNzlOI4wYwPOQtbfy2mvHT9F1zj4Ae4yMq/A0IxKsEdXkPclaeeROLb9H0mIk0Sgc9C+kO2rwlH7c7ssrSI6gp9yo7n/o2In5dtxHCEiHWsariK6TeOMFShGll3C6mtIv+iEU3a9QOEqvM5m5amBpAkM+6tEZyE6jv9aHIFvh2gLP7U4vHYmfUDBUg59ytEVGCJJO5/Dn2OOI7wb2S+gmgjvfsIp5TvIvIKYwSz7RcBaLoRohJ8ALU46flIcqHw5Bmsa0zN4by4AACAASURBVL7Fmg6w6UFcpWx9nMmfVG+po99N/PB7vC5OmRuQVgshmuAnQEdGRvrWQKm9UbrVwaxuudHcTyldy6RPyPkVNX9v+Ys4+yd2v8LRFUT19XOVqH5M+y4wDRZCNIefHHRVVVXNi8EagW+ZaL66i8a6y0GlcDleO0WrqjcaItjxLPvexJ5H4fdoLiiHidANpoG7U1ouhGiE/4GuERER7bgmoehoJ0xUKVnHjufwVLH5EaIHsvbPeCoBJryLOR69Cs+nfHc6X45m791wFJbAXni185ovhPDPf4C22+0Bbodoi9qjOIqWrFr52RvbjixFDaPPDSgGNj0IYIoi83L07WiZDBxG34HsWQx74BzIh79DSWc+gxCiHj/joE0mk29MX+3MRpB0omUcdBM01+e/KBh/6X3blv+qR+qzaUPWExaP4whD76ffzXCEbX3Y5cblpmc0up3Rv4APIBlmQzhc387tKd+G5iF2SDtfVoigEehx0DV30mvpiHuL9lKd4ihYxqKp2HIjhowJCz/sSb8Vg4HK3URksuVv6Breuyj0EuYkUyOllFEu+BjMYIZnoKkqdy2193XWzWbzQ+z4ZztfWYiuocFVvfUTBbhZokWqUxz73uC0V4b/5ZRvZw/xuiMzxh0i/WIUA4PvRnNSuYe8HBJcnAHF0B0UHVygwEE4ADHt3KzDXzP6GbJfIu/Ldr6yEF2D/1ocZrPZ6XR2SoMaJymOxuyZz+EvMVqx9uDQxzgK8DgJTyTlDIrXMPkzys9izU/0gX1wRs1pCu4ojGaUTPixPduT+ynbnkIx0vtqMq9ozysLETQ6sxZHzcbOLcFRQwK0X8cmqqSz+HSAyj04C0mawNHlRGbhriR+FBPe4dsoYhzkwWmQVOt83YLmxjAGctq5Za5idC9hSU0fKcTJqTNrccgwu5NCdYrDlosljX5/QDFgCCP/G7xuqg4y/m3iTuHQu6gufEsMrofaf6WKA6cZHgbw2Nj6BFsew1PVDi0zx0t0FqLV/Afo+gLfMtF81RNVLN2xHWDvf9F1dB0Uel2F14HjMOU7ce6mEDaBCyzwLeysuYCOfhZMBtg4F1MUYQlsmNNpzyOEABp5SVgzh9BgMAS4TaKlqkdxKEamfE7v60gYQ1gSUf04/CV9biDvcxTY8T5xsYRBIuTCQCiGmoXMIqZVf3DkkzyJ5MnYD/u5k+bk4EJK1gbmuYTo4vzU4vDloAFfPWhN01RVlSxHMDtei8MYSY8L6HEBgC0XoxWvnc9PIXoAZVuI60N4BUVurNANjsDxNwsfwc0Ag++tLjc66mk/d1p1NXHD2fMKA+6g25QAPJoQXZn/cqOqqta8FTQYDBKdg9zxcqOFq9j8CFF9GP4o1jSAvJWYYjhjBf/rj1cjLoxpHsp0NoMJUmquUXHs/3cQmUX6Rf4X/7bnMf4tcgdR/KMEaCE6mv8UR0JCgt/PIjgdr8Wx7i7GvoIljb2vV+9LHI/XxUc9cBbTryejwgnTSYZhMLT2NUqgkqp97HiOfjex6UFc/mZ+p1/EN5PZ/Ejr18oCr6vVpwrRtfgP0IWFhZmZmUBmZqYs6R38atXiUHEV4y5HPbbUrzmWX+xh3BtEZbL5GzaUN3CNLHgETxVhSUQPxBSFx+bnqP63cfoizlxJRM/WNfX7++2Lr6n6+jdVHpu8eRaiCX7GQRuNxjqjnjtulF9LyTjoJpRuZNuTRPVl0N0otd7ufjOFio2MLeV7jdHQo/6ZN4MC/2TzI+R/Tfol9Lu53Vvnsevf3Wg7Y37EphedMVlq+jRTu99CiADrhFocgwYNUlVVURRVVQcNGhQk0Vk05HiKI3YoY//L4HtRDKCT9zn7F6C58VSiu1gPRoirfwEV1sM9AINnM+27jojOgNGieJ1sfdV5cJEnrn/LRgcdXunZ+6lb0iOiSzkhQPtG182dO3fz5s1er1fTtPvuu2/Lli0y0i7InbBorO0QOZex4tds/hu5n1G1l59vZ8y/8XqxafQFP6mFS2B57TeGHef0/1it3dUJT1oie/hPr/m191P37g/ctnxt9RwphCu6kBNSHL6Bz3USCL5Rd0EyV0VSHE37/jr6XI9qJucyxr9NdH+WnMe0b9l5PVkvo/r9e3yx/QuNtqufH3ekTTEmjzB+fWXVWW9HdHZzhDiuQ1McdYfZmc3m+lvaXjip/li92iP5RBudsGisIRz7YVQTsUP5/vdUbCOyL1+eymnbTojOOnjABCjHKycVr2HTg1jTGfE4BkvgH6QhfS4x5/zZpnsZeE1YZ7dFiMCp+2tm/c5p28NozTyX2isc+ua/tPHKwueEFMcpD1CwjPxvOPV5LN2xpDHwDqp2EVNZ97TVcBQYB8ei3to/M+ZFYoey6z8Ba3xzRPdSz3kvcsaHkb3Ol/eKogup24Ou31Fve4CuM/PFR+a/tKPjE1UAczwjn8JTxbfTqdyD0cr6exlSr/KRAmNgs0JS2vHss2LAWYirBLOfN4lCiAA7oQ/rm9utqqrVagWsVmtN6dE23sZorPuTwGSSrlC7OWHRWJ/DX5KQTffpmKIxmHy5jLqqFLqfB+8c3zL6n2x9Eq+D3td2bIuFEM1wQoDWNM0Xo+12u6IodrvdF53b3tV1uVwGg8Fqtc6dO9dqtRoMhuBcEOAkdUKKwyeqP+VbMUXhOMrQy0jy9zcY/impn0Ctn77RAxj7CkPnosqPTyE6n5+JKjExMZWVlb7QHBkZWVZW1sZ7WK1Wv8uEWywWm83fdLWGySiOFij6kaLv0XKxPEWiizpjH+wKq8cx7B/Ej+qc5gkREgI9UaWsrMw3CNrr9bY9OgM2m03X9ZSUFIPBoKqqwWBISUnRdb2l0Vk0xE+KA0g4lfxF7P8AxYTZcsLwZ5eK6yJOu4ENfwlgM4UQLRO4cRR5eXkej8fr9Xo8nry8vIDdtyvwk+LwcZWRcRVH3RjsKKCBDk7YGsWhNew/ROl6ll5A5d6AN1kI0TQ/KY6O0F7joCXF0RiPjVVXYT9M1rVk/Q5g7Z0YPqD/AcL06hmEOthhZyb7dNQwJiwA2PEs2S91ZsuFOGkFOsXR/veQcdAdrDrFcehDuk3hzBXsfqU6HI94gv6RqBY0tTrFoUARDLudEY9h6Ub5Vop+JCyxU5svhPDPT8H+difjoDta9YoqvTM59AzWDFTT8bEZYRegP4quVW/QFQ4mULSco4cZ/jC7XyU8iSHNy0RrTgA1AHP5dPK+AIXUs08YZCJEF3NCgPYVRfJFUpPJVNNpN5lMLlebyoj5HQctI+3ay/GJKu4KStcz7o3j+/RdoFX/pqQrcAqnZVB4K/0y+fkOorI4uoLKPcQOrX/ZE+R9xuZH0HWG3kfK2R3yGDU2P4KzEF2jdAOD7mr3y+sedrzjcpbq/a8wh8XJDwARvOqOg65JSfuisy8d4Xa723ibVoyD9g35qKONzQhVx0dxpJ7DoLux1tR73gbv4hsso4OioxxFuYmkCVh74DjCsIfpexNHc5q+x+5XmbSQSR+y8wUOvE/V/g56FoCStQy6i0F3UfxTR1x+25suR5Ee21dd/Zema+O5K3V7QVBUChNdkP9iSeHh4RzrONev399SFovFbrdrmma32+fNm1d7eyNn+abMtOW+XcfxRWN9DrzP9r9jCGewm+RjB1VBJGh2dnxPaibRA4gfxfKLsB9m/IKm75E0nnWz0b2UbsRxmJzLmfg+lo6pUNr3Jpb+AkVh+KMdcXn7ES1lvDFhqGHb6038Xnj0Z++aRxzmaLqPNQ6+Xuo0iUCrW27UN3mkdonRmJiY8vLytsfK1NTUgoIC3/yX5OTk1o20k1EczbJoIkPmsGImQ5Ppv+tY9hkUOJpEVCbLjEz7trruXVjC8fWxGnc0B085u/7DxA/ZOJfkSXQ7vYlTClehmls1F6bmnWb7sxfoK2fbPHZG3B6ePLqxWuc/PeroebYpcZjhq8ulzKnwr0NHcfipBx0dHe2bn+Lb5XubFySdWQnQfp1QbhT4pC+OPDQ3F2qYvZRDFLggHLbPpH8uyxIZ8wLh3f1cq+h7Nj1EZC9GPOb/feD311G1H8XApIUYLJRvpWwrKWdhrBe/tjxGxU68NpKn0Ceo6003JG+FZ/O/nZE91LB4ZeSfwju7OSIYBa4etG88nC8617zW66BoGFTrAJzs6qY4TBHYdSxhmCpRIAaAcAV0sj5mew9ip/mPzsDau5j4HvvfYfer9L3RzwHZL+F1YAgHKFnP2jtIncGKmUz5rO6RR5cz7i3cpaybfZIG6NQJxuieqrNUTxgqiwqJTnBCgPZ6vbGxsZWVlZGRkaWlpb6NBoNBlrwKcieUGwXcZXQ/nSFfVveaq+mgYJxH/0bHRVRXHC2t9aaxHsOxi5aspeflZF3LoY+O5VBq6X01S89D83DK3JY9TzCJTFcj0zu7EaKr8vMizmAw+Pq2voxHUK17IikOv3JyctJTkzNibcQMQA1j/T2UrOeUb/C6SAINFCiDqO4Y/M0Ir61sC1sfJ7I3g2ejNDVM3nmUZRehmkgazykP+jnAUwkqRmsrH0yIoBe4HHRmZub+/X6GTw0aNGjz5s0dcfuWkgDt1/PPPTNSfzX7tEmUrOf0r3CV8eUoRmmE52EBHaJgPaSOIuk/MKJlV9eceGxU7MAch6uUsCQie51wQE3Gw8+5Lop/ImYQpphWPpsQwS1wAdqXF64968/3hrDtJaGlFkfHKt3A7pcZ9Q9+vJm+N/LTHynfSj+NI/mcprBDxwB5MDmFsPvh9y24cslafrwJ+2EiMrAdJGYIXgeDZzc9fgNA59sziBvJ0eVM/h9hCa19PCGCV+BqcfjSGrXjptfrbft4ZKnF0dFyNhYd2P4Dq65m/1usugqPjbTzyDvKaWDW6Q0ZMEklLOn48oN1lG/lq7F82J3V16PX+k/twHtEZuEuo3QTxhiiBzB0LvmLm9UsZxHmOEY8Rtp5lKxt4JijbH+G/e+w5FwWTSR/EVX70do6MaqLK97iLdwQRJlJ0Wp1k4z13we2/YeD1OLoaBs2bTMOfzwjbCFpv6ByJ0VrMO9kgIZDYQMMVzDrKLEQAVP8X2LtXeguxr7Elsc48D49L63enjSOnS8SN4KS9dhzKVzN0ZWMfqZZzQpLRHOz+moqdtPvD/6PybmCrN+xcR59rqPPjXzSi5SzqNzD5E9lXcT6Di5y717ojuqpRvdUDizyuio1a6I6/I/hMX3UdX93HFntLdunOQo0XcEYrmScYZz4tFVqmZzU6gbotLS0+lv8JqZbdhupxdGRqkdx5Bbz/fUMnYezFM9uticyqYjhGgp8byQ5jN4fQ9IJZ3ps1W/wjFa8bop+xGPHFHX8gNTziM4ibgSai9HPYknBGIExsrktm/ghVXuxpjc4F8ZTRc+Z5H3B4S8xx6KaOO01tj5O4SpSZ7T4DyKkHfzKvewOu9eh4z02jUclOkv/5uoqY6RSttOLdmxVBh2PTd/7iTttirv3L2X1spNY3STDgQMH6pS/8LNUR8vJmoQdqroWR9r5hCVRuhG8eLuBi6+tfKPwuQlLPLsSIYmq/eR9jteOrrHi1yy/mCXnobkY9Xdih7L3DRLGknruCVef8iXxIxjzInHDQW96aEdtikpkVmMzFbOuYdFEHEdJvxhnMZH9WDebgx/KQlx16aycY/NU6br3+No4ukbZTq0yVyvd7tW1E9bMQUfXObq2QxKjImDqziRs6Li2pKFlTcKO9vzzz48cOTI7O5uyrSyZgTGayl1YorAVMng2Retwl6IYyH6J1deQcCp5XzD8MQ59wNj/svZPZFxCQjY/3EhET8q3knIWmVf4uc262VRsx36YcW8QmdUhT6I5KfqRmEGY4zvk+ieb8n3az485DGFK+Z5mpJXrjUQfcKU5+4HGKt6ItgvcKI4OJbU4OkTpBuyHsaax+1XiRxAzlD2vYjtE4WrcpahhuMpQFMwxTM+hdB2OfPa/g6cKUwxeO5m/4cC7TPkf5ngWn07PX7PjXziPMn0pUX1PuFHFTr47h/FvYcvFdoD+t3XSA4conTWPOIo2eruNMRT85HWW6OGJKrruddH3EtPK2XatVf/8T3vY0u+y5hVaEa0VuKnejcwYbPt0FVmHsP0d/oodzxLVL+d//0zv2ScjfQOD7qR0I7YDOAsxRWOKQvcyeSE/3optH/Y8tv8Dr4OMX6Ga6TYZzc3YV6u7q31+zw83EDOYAbex702Gzj1+I83FqquwZrBiJpZunPpCJz3wSc9dpRdv9sb2M4TFntDX3fG2q3Cdd+xDli9nVkb3MhhM5K9yp04yFvzkLd7i0Vv7jy9+kAyUOrmdEKClZ3qSKVzNgD9SmLPhAMZRv8+o+DvOUk7/imWX0P1scj/BfhhFYd0cqvaw9W8YY+j1Ww59SMk6KndT9D0Vu4kfQexQhv+Nnr8m73OiB7B/AdYMStYRN7z6Ru4KwhKZ9BGrfkuPC4hr4VQXAYAtX/vq0iqPS/c6GXWXpSrXG9vPkHmuqTJX2/pfZ1W+/vmFlYqCs1irzNdNkcrhFV6vS1egdb/lqmFKwuBALJkkOo7/v7+O67GL9tRzJquuwVM2a7pC5Wx0L92moWvEDcMYQURPXMXoHopWE5nFkeXEnoKrFFMsWVezfwFRA/BUMuFdlpwHoLnI/g8H3sN+mL438dOtTHiP8G4AYQnEjWTxVCJ6ET+KZRcCjHqaiMzOe/gA0jV2/ZvK3fS7pS2PXPCz1xCmnP9F5Ge/rPx+js2SrNoKnCvvwhSl2o9o1VFYRfOgOfWal+gtCs6KAd0LCtZkddor1ha90BVBqG4OunZ5fkVRzGazw+HojIb5JznounQv62fnfPx0ereIjHg31kyM4Vh6YEnhwAI0NzGDKN9OZG8q96AYMceReg6Fq/E60NzYcxl0JyXrUEy4ijHFMuJvrL+P8W+x5hZ6X0v8yLp3XPVbBtyOorD1CU57rTOeuTWO/Oix5esZZ5kMrUjJ7nuLkrWkXcDmvzL1q1a3oeKQ9sXFlapBsRdqYTEqBtzlmseOoqBrzat9raBAdJZqz8dl18JjVLddP/Xe8D0fu4wWZcT/hXk9JA41GMJl8HPgBC4HzbGVrjg2kcTpdBoMhqCqlyROoBg4smxDbpgxMjIjOo/YQVTu5egyNDfWNFwlFP+MGk7ZJgwWzDG4irEdwlHA9O/YOI/YocQOoceF7HiOCe/y4814bFi6881kovsTP4L971D8E72vIWZQ9R1NUVRsBwVTdKc+eQscXOTe84k78RTDqtn2CU82NapBc6OeOHbYVUTMIOJH4m1siSzNjatcD09QAI9d3/Cc01miD70pLDJNzV3q2fwfZ9F6r8ehK4qOrmge3VWs6xr4MhhNRVTVpOg6Rgtnvx1pSVJKtnnX/d3pdWlZF4X1v8Lc/wp5Exia/I/iqD3NL6jSHdKD9mPjg+z4O0D0QCp34rbTbQqOAmy5xA7gyAqMFqw9mPAuu1+i52XEj+J/AwhPxmAlegCpZ5M0gW/PoNeV7H2dqV8dj7xHV7DzBfrdxM9/4sxj6xa6Stn6GMDAP2OO7YwHbrH1/3AmjzZ0H2v86vKqsxc0ujDK+nurV2ic9OHxoX6eSn6YhT2PwffQfZrf82z52pKbbeHxalRP9dQ54Rufd5qsiqtSP/St56y3Ij67sMIcoxb86DGYFI9DR1FMFty2ev/0lGMZDRW06i2RqaqObk1WFQPZcy3xg2u9yW9GZBcdLXDD7IxGY83iKYqiBOFkPwnQ/ug5C+5Ot3+SkaSiuXAWoajEDefoChLHUfwz5/xMxS6KfyRxHJseILwbhgiq9lK2jYnvsemvnPocBgsFS6nYRf7XmGIY/zYo5FxK2SZ6XsqRpUz5DGcREems/ROHF5FyZoNLrgSfiv3aijttukb/y81ZFzfc2dQ1Fp/O9CXsehGjlczf+D1Kc7Hsj7by3dqQG8N6nW/a8bbLdkQzxyjhCWrWhaYvL63qebZp5wKnGqZEpSsHF3lSJpjylrlNkaqzRPPNNNCPreelHOs+myMVt0NXdNQwvE6MFtVTpcUOMKRPNVpT1B5TjRGpMh4jSAUuxeFLZdRMV3G73TX1jCQmBjFlQ3GmceR/MwxvEj+KPa9S9ANlWzFYiO5P8RrW3IyuM3g2yZOJ7kv+ErY8wqSPWXo+lbuJ6ImziIRT6flrFk3kjBXseJZvJqFaceTS/Qx2v0zW71hyLpG9cOTjKGTwPWx8EEsag/7c2c/eLFE91XPei2y6v6moGMLY/gyHPmbUPxo6auXdtqIN3rA4ZeXdttylpog0tdto46YXnbqH8j1eNMp2e/teGvbjA/aiDUT1UHO/dYfFKYmnGHKXaoZwRfPouhvFyPEZ26B5sSYrikGxJCrD/hB2dJ3W+5fG6ExZKKOr8/OWV5ahOulU1+LIK2TlFdVTsaP6YIpi14uY48j/jgnvkDQRdznfzcBZiOMoX5+G0cqWR4k/FWMkrmLM8ahm9r/NgfeJ6seov/NJf/a/h+bCWcygu0meyOenYIwkZiBGC+7Szn3qFmtONmDC++R9zuh/Hs+516J7ObrWU5mrYWD8Y9YvZlaWbNMyzjAlDTegk5xt2PyyS4XSXZrH7lZNiubSbQWa0YqjWM9d5jGEEdFd8TiVqiNaeJzqKNEyphlPuTnckqRYu6uA7qn+20ub0q4PLk5agZtJ2C4kxeHX8UVjP+lL6gzKN1O8lrgRHF3BlM9YchaRWRhjGXo/W/5KzCkc/gyPA68dgwVLCraDRPZl4B10m8qBBUT2Zf29dJvKjn/S4xdU7sJdib0A3UVUPxxHUCB2OONeJyyxsx+9o3hs+s733OZopfcFJuVYR3b5/9ms3dXcJe7S3ZruQTWROcOUu9SjeRl2S9jml51Tno9Y9gebx667K/W0ScbDKz0eJwaTohiI6aVaU9UBV5njBhhKt3s3v+RMGWccfP3JkSMSjQiRqd7tQgK0X88///zI3obsIalU7mXnv4jsg7uUHhew/j5UI6hE90HzYIigfDMoeCqIPxV3Gc4i3GWknIElg4PvEjOEUX9n31tU7qRoPaoKCo7DKCoxQyjfxpC/UPQ9o/5evebske9wFtHjF3WHPTSTowDVFJxlRXP+bE8eZag4qFkSlIHXVIfRLy+tOvudiC8vrXKX60NmmQ9+7c7/3nv+/6J0ncXXVFXlaYkjjUe+d8cPNpRu17qNMZTv1Yfdal7zsAMdc7QaFsf4x61RPSWbHFIkQB8nAdq/Dfez9w0cuYQl0XcWMYNJnYFqomApG+dR9AMp55D/NShEZDDySX6+g8RxFCyhah++ecTmBPrfSs9LWX8vjnymfUfeZ6y9E4+dhLEcWkhUPyp3ouukns2kj0Bh5VUc+gDFSMpZTHi3xW3e/QoH3sXrYPDdpJzd7n8kbbToyqopL1jLdmp7PnaNud9SlaetuNNeuluLSlUqD2ndxhpd5br9iBYeq0b3UYu3es0xSsppxq2vOrudZoxMU2N6GzS3njTKGNNbtRfoexa6nGV6xpmmxOGSVg41EqCPkwDtV87Tg9PDD2T0HU7ZVpImMenDE3YXrmb5xXgdxI2g5GdihxA7lLJtFK4CCE/GkU/yZEo34SzEHEd0f2IGU7aJ7tOp2I05jr1vYjDR4wLyFzPuTRLGoLl4P55+t2GysPlvzKzAdohd/yYik6xrm5Xu/e4sJn+K7SCbHyb75Q75c2mY7kVpNFTmr/ase8qpmjntIUtUT3X9M86kEYaU04yfz6yc+nzEpucdez7xxA1UFRg6K2z7266BV5lj+hpW3G6b/t9Gh/GJkBPQiSriZLQhP84YtSMjYT1eG2nnH9+xfwEV2ynfTlRfCldiMKEopJ3P1idRVMyxOIvxVJAwFtshYocQ3R/bQSr3UbWXlDNIv5jll+Aux2DGeZS9b6AY2PEc8as4+AGobHsCowVrGsD31zPwdvI+Z/+C4wuyNKL7may+FncZva5s6/PrXva+jrOQrOuaHJrtsetLbrJ5HcT2U7PnNThppftY49nvHv/XEdNH3fupu/KgFtVDtXZTel9kNoQpo2aHr7zbHpmujrkvfPUch8ehj7qrgcVzhWgVCdAnuSPfsfP5Wdk78YRjjsHQnd6/BSjbTPFPHF6EYuTQx/S5EdsBStfjdbLzX5jicBXgLEUBr5v40Rz5hso9lG0msh/2XCa8x8rLyf2CzCvJ/RRPJVF9icgg7lS2PUbMUGy5zNjIN+MJ78b4dwC8VSRNwHEEe36zWj7wDko3YLDUrWvaCrtepGo/ccNYczPj3mz82IIfvcmjjKfcErb4d1WuCt1drm960dVtjCHz3MbS6JkzTCarUpWvjXvUAsT2UX/82fvDXHvlQS0yTVWMnP6Sta1PIUQ9EqBPZlX72Xg/zuKcrc70lISMRJWh96GorLsLRwGFOaDS+zr2vcaOfwDoXkwJ2PIwRaNrKBCWgMfB3tdAxWvDW4WnCjWcva9jtOJ1suF+FJ2wBCp3oRgo3ULSBPrdyJaH2foYkb2Z8gXGCIDB9/Dd2VhSGPNic9sfe0o7/TkcIOUsEk5ld9Opkugsdf0/nWFxiqtcN0UoC6dUJI0yrp7jQiFzRmMxOm3K8X8sRqty5uvW8n1aTJZBChKJjiMvlE9mzkIUE47DG/ZUHC7RMcWRdgFA0Q90m4oaRsUutjxGr6sIS8C3JpJmQ/fgKsXrAtBcGK0YLeguIlJRDJhj8NjY+jhuG8mTiB9GWDfOWEX3czjlAYbOwVnId2dz2nz6zmLq19XRGUg5m+lLGf9OJ9To6HczWx9nyQwG39vksZFpaq8LTHsWurMuNvt+fzj939bUycbDK1tWcMYQrsQNMLRu9IoQzSQvCU9yC9NxHQUF1cKMn6uLYW6cy45niR5AWDL5X6O5URS8LhQFFHSd6H7VA+xMERii6H8bG+fitYFOWALGiKNUdAAAFMlJREFUKE79F6t+i8HC8Ef54fckTcAcx7jXO/lh24O9UF/+f7bxj1tWz7Fnz7Usv91evtere5mxMEJm7olWkJeEXd3mzZuffvrpCy64oFevXl6vd/jwY3X0nYVoTkxxOQVD0o3bMyyp1duHziX3U5wFFK5C11CM6F7Ck3AWoRgxRVRPMHGV4K7C6yF2MKP/yY7ncJfgtdH3Rir30PsaLCnkL2L6EiKzTljq+2Sz5yP3oe/cmTNMGWeZdI+umgiLUQxhiubhnPciKg5o1mRFSnSKICQ96GDncjFlyr9//PEuRfHExCSMHj3+f/97q3ptsk0PYIxg5wvPf3xgZN+o7BF9Gfsq0QMANj3E/jdwl2M/ciw0qxitDLiTw19ijqVoDR4bikq3qeg6YQk4j+IsYtxb7HsNxUT/P2AIxvVGvQ69bI8W08fQzMrOFfu1Hx90jH0ofMUd9kn/sIYnKLved+37zJ0+3SRVOkXbdWgPWnLQQW3p0qW///17a9a86fGUud22wsLDixe7fvGLY7vDkijfSfTAWWeasm9ajdfOt2dycCFA8gTCkvF6QcORDwqWNE59kar9RA+kdCuOAnQX3ipKfqZiK7YDTPqIjEso/pHB9zLoz8EZnT02/esrq3a97/7m6qpmrqOqa6hmTJFK9Woj0OcS8/RXIyQ6i+AnATqoPf3002+99Xu3ezkAOrjd7g+/+CJ61KhTt27diiGcfW/itefsDj+w8JeoJsb+l21PsPpqdv2Hfrcw4DaiBhA7koheaE56nM+Yf1Oxg26T8OVDNA+nPIw5AV1n01/Z/y7xozr1iZtQtltLGmEc85fwqAy18lCzfpGK7qWmTjAuu9WWdaHZkix5DHEykQAd1MaOHet2119yzHPjjTcMTFM4+AFxw9BcG8qHHnb1ILI3FTuw5xPVj76zOPQxg2cTlkD5Npz5OArIXwyQchblO3EVEzcSUwzbn6Tnr5n8MQmjmfAuET0D/pQtENNHLVznXX2fvfKQFtWjuf/19rvcPO3liN6/lCEX4iQjLwmD1yuvvDJnzhxdrzv8KyFhmKZpeJ2Y40meyua/zvrlJYx5gbwvKNuIx07ZNg4tJHoQQOZVFH1P/DgUlbzPiB/NoLvpcRFbHqFyN4PuYtBd1dcNvoIY9RktyhmvR1Ts90b3lgHIIvTJS8LgtW7dutGjJ3m9FXW2K4r6yCMPT5w4cVz8avK+oPdvc3J7VZcbPfAeVXupOsihhZy5Cms6wNenUbEDHSLSsWYQO5RhD3XC8wgRiqRY0nFdJ0AvXLjw+ut/X1RUeGyVutqU2NiEqVMvXbDgnyYT+MqNjhyZnZ2Nq5ilv8AUTWRvBtyO4wiJ2aDgKGDfG0Rkkn4R30xh+pJAP48QIUoC9HFdJ0ADDzzwwNy5f9V1d/1dimLunXnqfb+++Le3nUL3aRx4nzW3oJoYvwBrGuXbUM1sfpiYQeg6o58BKP6JtXdW10UKiSknQgQDmajSRUVERIDf+ccKaPmHV897reKNb/JfffnF/Z/dmD79qYziZ/luOooJUzReJxMW0G0q30ypPslZiLuM3E8Z+9+APYIQoi1kFEfwioiIGDr0DEWpO/ZAURKio399cXbyRZetW/TMWT3yH9mwz334p5fwODAnoBj55QE0D2v/9P/t3VtsHNd9x/H/nJm9cElzZVq3lSpy6wg1RAp5kVPXBhK4QCvESmAgsGMnKfziIgH8oAKFDdaCDUsPhmoJcGHEKaBAj23lXoDmxQ9FHmoogBwkcCxDllQrkAFakatYFiVSpZbkXub04YSj1XLJvXB3Lme/n6fl7Fz+Oxr99+xvZ2fk9JOyY+Wrv0/ekr/4hXz9P+TyT8N+JQC6QsQRX59++umjj37jyy+vi9R/elLF4ptvvfXnP//3v/3H75yT2v9J4Vuy+L9SvimV27LnRfnokIztk9JVOXBOdFWy2/6w3NmXJJWX5Rty3275k4NRvCDAQmTQdw1UgxaRxcXFQ4fe/vGPj2s9KyIiynVT4+P7cjnvr578yguPXR1yF6/ecn//+y93PfnP46lP5OZZ2fyncvuSTPxARv74nnXpqvzuP8XNyc5vtXeDawCtkUEPrkqlsm1bzfMqlYrjOMp1x2q10q1b3i9/+V//cOzF13/6iz/76kO7Rz55f/7Zb8yWxx/5vkx8f811OZ6MPxNi7QA2ihF0AnzwwZX9+5+Yn//dli1PDQ+fTaXc7353+vXXf5BJ/eVw9juFsZ+PbvuX/z6dzWbl1KlT6XT66aefNgueOHHimWeeGRsbi7Z+wGJEHHcNZoP+0Y/+5uTJt0VERCm1xfdviPgiWsQRyYo8mMvdnpycOHjw0Jtv/tb30wcOLB479uL16/LEE3/91FOvvPzyg4ovg4H+IOIYaO+8c+vkycWVv3zf/6LuSS2yKHJnaWnHvn17X375jc2b5ebNhbffzufzhffff2J0dPny5fmf/OTWD3+YHRqK49XpAKwj7BG067paa621UqpW6+wmQzJ4I+hHHnnk17/+TOSmSJOfq6xwRPaJfOw4O5W6T+RaKrVZKS+b/aP77vufzZv3LC0Nf+97X3311VfDqxsYGIkfQTuOMzY2Njs7q5QK3g9831dKDU6r7cK777770UcfiZRbzahFfiMiIjO1WkqkqtRiKvXE176WO3t2eGzsjVOnHty8ue/VAui5kCIOM1jWWjuOY5pyfbNGU9evX69W2/+QYXanp5TSenTbNv3tb+/bufPsK68I3RlIqLAzaLXydVUqlSqXW44NB5fW+sSJE0qlfX+x9dwi5tRmx6m57vCuXV//+ON3slk5f/78rl27+longP4J6dv927dvm9ZcH3GEs+mEchzn+eefHx9/VKTlraYdEe04Y9ns3oMH//Wb33zsZz87lM2KiOzduzeV4ir1QFKF8SWh4zj1j+sjjk63PmhfEmotjz126Fe/OrnyS0LDqbsGacpxhlOp5WeffXr37t2vvfbapUuXHnrooQhqBQZS4r8kbNqFc7kcGXRLjiMvvLBndnbs8uVbWgdvS8+K/Jvp0UqNFIubT536J8/zzP6kOwPWiOwHDAsLC3fu3Ilq6wny3HPPffjhh2+88ffpdMZxHBEnnT4tosfGpgqFwtGjf3f27AfVanXLli0PP/xw1MUC6KWQzoN2Xbchl+A86E5dvXr12LFj58+fP3DgwNGjR2/cuHHnzp1sNptOp+/eUQVAuBL/U+/gjLogjA7+7LTVDnKDDlQqlffee2///v1RFwIg+Q3acZzV42UzpuZLwp44c+bMH24aCyBcfW3QIWXQntf4bSSnf/XQuXPnrl27FnUVAHosvNPslFKZTGZ6evr48ePLy8tmFMwIGkCiJT7iyOVyi4tNfg43NDRUKpU6WhUNuikiDiAqiY84SqWS1rpQKLiuq5RyXbdQKGit1+/OxWIxu0oI1SYREQdgpfhesH+tqyl1ce4HAPRJ4kfQzTesVP1PwFcz53g0WH+RgXXmzJkrV65EXQWAHuNWSDYg4gCsFN+Ioym+JAQQK3ZGHOghIg7ASiE1aNd1nXu5bsvLHKNdRByAlbgWBwB0L/ERh7mHt9baXxFMCWHrg4CIA7AS1+KwAREHYCWuxQEA3eNaHHfRoJviWhxAVBKfQXd3LQ60j4gDsFIEP1RxnO43yggaQKwkfgSNfuMsDsBKNGgbEHEAVoog4lBKdZ1REHEAiBXbIg7aa88RcQBWIuKwAREHYCUuNwoA3bMt4kDPEXEAVqJB24CIA7ASEQcAdI+IAy0QcQBWokHbgIgDsBIRBwB0j4gDLRBxAFaiQduAiAOwEhEHAHSPiAMtEHEAVqJB24CIA7ASEQcAdI+IAy0QcQBWokHbgIgDsBIRBwB0j4gDLRBxAFaiQduAiAOwEhEHAHSPiAMtEHEAVqJB24CIA7ASEQcAdI+IAy0QcQBWokHbgIgDsBIRBwB0j4gDLRBxAFaiQduAiAOwEhEHAHSPiAMtEHEAVqJB24CIA7ASEQcAdI+IAy0QcQBWCqlBu67r3Mt13XA2PQiIOAArhRFxKKXMVhzHMVOCPzsNK4g4AMRK4iMOrbXp0f6KYEoIWx8ERByAlUKKODzPa5iSSqXC2fQgIOIArBRGxGGSDaVUJpOZnp4+fvz48vKyiSk63ToRB4BY6WvEEUaDzuVyi4uLq6cPDQ2VSqWOVkWDburMmTO7du0aHx+PuhBg4CQ+gy6VSlrrQqHguq5SynXdQqGgte60O2MtRByAlfihCgB0L/Ej6KaKxWJw1h02iLM4ACtF1qDn5uai2rR9iDgAK4V3FkdTnMUBINHsjDjQQ0QcgJVCatCO4+h75fP5cDY9CIg4ACuFdC0OWZVLFIvFzz77jIgDQKIl/ocq3VnrYh1dXGLJevxQBYhKXxt04yUy4mP79u03b95smFgulyMpJubOnTvneR4NGrBMZCNoM0Am4gCQaJzFgRY4iwOwUmQN2lwVOqqtW4azOAArxfdLwqaIOADEig0RB/ck7CsiDsBKYTRopZQZ8wbdWUR83zfDYWwcEQdgpZCuxaGUqtVq9RNd1+0ihibiABArNkQc3JOwr4g4ACuF1KDL5bLrurlc7siRI7lcznXd5eXlcDY9CIg4ACtxT0IA6F7iIw7uSdhvRByAlSI4D9pcerS7Zde6ghIARMXzvEql0pc192Ol/dNRuLGRd4KoND3jJeYSmjsl9PDIZDJLS0tRF9KBJB4e6XS6UqnE4fDgTGQAiKkIGjQ38waAdkTQoJP1YQcAokLEAQAxRYMGgJiiQQPAPeLzPVnCTrPryNDQUNQldCyXy7300ktRV9GZPXv2JPFnMkk8PLr48W3k7r///sRdW3h5eTmXy0VdhUjiLtgPAIODiAMAYooGDQAxRYMGgJiiQQNATNGgASCmaNAAEFM0aACIqWQ36Fwu56wwl51dTSkVzNPpsv3QznbXmse5V8yrZQ93xNaDea2ao9rVLWveunVrfWHRHtLJ/qGK+XdNp9PVatX3/aaXujf70bzM+hfbzrJR1bzWPObIGB0dFRHXdWdnZ2NebctlY1hzyHu4o7KTeDCvU3Mku7plzeYequbmfNVqVWsd3HIw/P2c4J9679ixQ0SUUuYG4Y7jNL2QqZnY8Bbd5rKR1Nxynrm5uRBK3WC17OGO2Howr1VzIORd3U7NDT+mj/aQTnDEcf36dRHJZDLmz44+JW1k2Y1oZ7vrz6O1Du0T1kaqZQ93xNaDeX3h7+pOa06n07LyISCS/ZzgBm0+Lk1PT4e87Ea0s9115jGHsuu6juOYT159qrNlJS3nYQ93xNaDeR2R7OqOan7ggQfMrWBNvhHJfk5YBl1/V2/XdWu1WnB9L/PUWi/HvN0Fz3qe1/6yIdfcZm1m6NHXD1ntVLLWPGHu4V7V3LCqEPZwpyUFhUlEB3Oval79bDi7uv2ad+zYce3aNRGZnJy8cOFCR8v2UMJG0L7v6xVbt24VEZMHydr/9k1tZNlOdVpz+7X1++DYSLVh7uF26ul0npZP9VZSDub+bTecstusuVgsmu5cKBRMd25/2d5K2Ai6gXlbzmQylUql/kvV+ju9F4vFubm5+fl5Ecnn85s2bZqZmVln2TjU3HSekZGRUqnkuq5SytwTPoSau652nen9lqw93FHZSTyYm9Yc4a5uWfPU1NTFixdFJJvNmsR5586dpk2Hv5+T3aDNCTHmcf1HpPrPU6uzfDN9rWXjUHPTeYaHh+u/Xw6n5q6rXWd6bGuOZA93VHYSD+amNUe4q1vWbKKM+kWC2cLfz8lu0ABgsYRl0AAwOGjQABBTNGgAiCkaNADEFA0aAGKKBg0AMUWDBoCYokEDQEzRoAEgpmjQABBTNGgAiCkaNADEFA0aAGKKBg0AMUWDBoCYokEDQEzRoAEgpmjQABBTNGgAiCkaNHrPcRzHcVKpVMMU13WDxwGl1NTUlJnNdd36p8z8qyml2pktttp8aSFXhRiiQaNfglsje57X8JTjOBMTE57nOY6jtTZ3ud+0aZPv+47jDA0NZTKZtTqUUkpr7TiO53lmtjDvvR0CWjMCNGj0hem85nHDTeyNmZmZSqXi+75SSkRc111YWBCR0dHRUqm0tLTk+/7qBfP5vOnOvu9XKhUz2+HDh0XEtPtgVB6UUT8mTaVSweORkZGm82zdutVMb2eFxWLRTE+n023ObPqveSsK5gz4vm/Z+w02ggaNPgo+yK+TQpgurLU2HXN+fl4plc1mm85smnjDs0eOHJmamjLr2bJlixliB1vUWqfTafNntVpVSuVyOREplUrBGrTWExMTExMTInLjxg0RWX+FhUKhUCiIyJUrV0wBlUpFRPL5vOd5Wuug866e2bxvmXXSi7G+u8McoFeCCNW0Kt/3zbBXKVWr1cyz9b0pmDmVSlWr1WC653mm8QVMX1t90Lqu6/v+xMTEzMxM/QrrtxVMbFhP/TzB9HVWaF5F/UQz8+Tk5IULF+pX0nRmWRlZN/1g0bBDutj5sAkNGr0X5Anlctn8GXyiX6tBN0wJOnXD8Wla4dDQUP34V1Y17rVa5OpGLPe2S7N+877ScoXBRPOgYSesNbPQoNE2Ig70y/LysnmQTqfXmS04taN+YqVSafpdmYlBlpaW6iceOXLEzBwkwp22tmD+4EFHKzQzHz58WNfpqACgKUbQ6L1grGoC2WAAWz+CHh8f//zzz2u1mjkCtdae59VqNdd1R0ZGFhYWgmy6YeX1Z3EopcrlstZ6cnLy4sWLjuNs3779iy++MN89mm21M4KWlXeRcrls5pmammq5wmAljz/++OnTp0Ukl8sVi8VLly6ZV73OCNpZ4+STYrE4Nzc3Pz8vIvl8ftOmTSZjwWCiQaP3mjag+gbdMH18fHxmZiaTyZhIJBCEJA0aIgWzWtPfg3UGuXM7DdqpO+dkbGxsdnZWRFqusH4l6XS6Pi5vSHUaZg7qb7qXGl4s/0MHGQ0ag65lIgxEhQwaAGKKBg0AMUXEAQAxxQgaAGKKBg0AMUWDBoCYokEDQEzRoAEgpmjQABBTNGgAiCkaNADEFA0aAGKKBg0AMUWDBoCYokEDQEzRoAEgpmjQABBTNGgAiKn/B1ieDvBbV7cBAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "MDS_function(\"/mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.mds\",\"/mnt/data/GWAS/output/task2.2_stratification/racefile.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The plot shows that our \"own\" data falls within the European group of the 1000 genomes data. Therefore, we do not have to remove subjects.**\n",
    "\n",
    "Anyway we are running the filtering step for educational purposes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter out population stratification outliers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "head $path/MDS_merge2.mds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# A commonly used thershold is taking 6 standad deviation from population mean as theshold for excluding individuals.\n",
    "Rscript scripts/getFailures.R $path/MDS_merge2.mds $path/thresholds.txt $path/fail-pca-QC.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3000 /mnt/data/GWAS/output/build38/task2.2_stratification/MDS_merge2.mds\n",
      "1062 /mnt/data/GWAS/output/build38/task2.2_stratification/EUR_MDS_merge2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# The cut-off levels are not fixed thresholds, so you can set them manually in the plot around your population of interest (uncomment next line for doing so).\n",
    "awk '{ if ($4 <-0.02 && $5 < -0.02)  print $1,$2 }' $path/MDS_merge2.mds > $path/EUR_MDS_merge2 \n",
    "wc -l $path/MDS_merge2.mds\n",
    "wc -l $path/EUR_MDS_merge2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/../task2_QC/intermediate_datasets/dataset.b38.IBD\n",
      "  --keep /mnt/data/GWAS/output/build38/task2.2_stratification/EUR_MDS_merge2\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "316140 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "--keep: 495 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999876.\n",
      "316140 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 84090 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.hh\n",
      "); many commands treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Extract these individuals in the working dataset.\n",
    "plink --bfile $path/../task2_QC/intermediate_datasets/dataset.b38.IBD --keep $path/EUR_MDS_merge2 --make-bed --out $intpath/dataset.b38.QCed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HGX00096 HGX00096 0 0 1 1\n",
      "HGX00097 HGX00097 0 0 2 1\n",
      "HGX00099 HGX00099 0 0 2 1\n",
      "HGX00100 HGX00100 0 0 2 1\n",
      "HGX00101 HGX00101 0 0 1 1\n",
      "HGX00102 HGX00102 0 0 2 1\n",
      "HGX00103 HGX00103 0 0 1 1\n",
      "HGX00105 HGX00105 0 0 1 1\n",
      "HGX00106 HGX00106 0 0 2 1\n",
      "HGX00107 HGX00107 0 0 1 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Create covariates based on MDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.QCed.genome.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "  --extract /mnt/data/GWAS/output/build38/task2.2_stratification/../task2_QC/indepSNP.prune.in\n",
      "  --genome\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.QCed.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "316140 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "--extract: 232665 variants remaining.\n",
      "Using up to 4 threads (change this with --threads).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99987.\n",
      "232665 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "Excluding 10723 variants on non-autosomes from IBD calculation.\n",
      "IBD calculations complete.  \n",
      "Finished writing\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.QCed.genome.genome\n",
      ".\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out /mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds\n",
      "  --read-genome /mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.QCed.genome.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "316140 variants loaded from .bim file.\n",
      "495 people (237 males, 258 females) loaded from .fam.\n",
      "495 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 495 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999876.\n",
      "316140 variants and 495 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 292 are controls.\n",
      "Clustering... done.                        \n",
      "Cluster solution written to 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.cluster1 ,\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.cluster2 ,\n",
      "and\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.mds .\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 80611 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.QCed.genome.hh\n",
      "); many commands treat these as missing.\n",
      "Warning: 84090 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build38/task2.2_stratification/dataset.b38.mds.hh ); many\n",
      "commands treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Perform an MDS ONLY on HapMap data without ethnic outliers. The values of the 10 MDS dimensions are subsequently used as covariates in the association analysis in the third tutorial.\n",
    "plink --bfile $intpath/dataset.b38.QCed --extract $path/../task2_QC/indepSNP.prune.in --genome --out $path/dataset.b38.QCed.genome\n",
    "plink --bfile $intpath/dataset.b38.QCed --read-genome $path/dataset.b38.QCed.genome.genome --cluster --mds-plot 10 --out $path/dataset.b38.mds\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       FID        IID    SOL           C1           C2           C3           C4           C5           C6           C7           C8           C9          C10 \n",
      "  HGX00096   HGX00096      0 -0.000495704    0.0140154   -0.0111572   0.00363723  0.000969209  -0.00632788    0.0041895   -0.0226794    0.0104471    -0.014345 \n",
      "  HGX00097   HGX00097      0 -2.64536e-05   0.00845106   -0.0159685   0.00652097   0.00261635 -0.000613264   -0.0104634   -0.0105515    0.0052152  -0.00294776 \n",
      "  HGX00099   HGX00099      0 -0.000668454   0.00801473   -0.0200389   0.00786452 -0.000514026  -0.00662578  0.000594134    -0.017528  -0.00383219  -0.00608985 \n",
      "  HGX00100   HGX00100      0   0.00473264    0.0161124    0.0111039     0.025022   0.00180064  -0.00509499    0.0305339   -0.0268026  -0.00699904    -0.019562 \n",
      "  HGX00101   HGX00101      0 -0.000491384    0.0148019   -0.0151813    0.0112514   0.00721504  -0.00254442   0.00832269   -0.0173986   -0.0245312   -0.0155404 \n",
      "  HGX00102   HGX00102      0  -0.00195889    0.0138588   -0.0175688   0.00464131   0.00533101  -0.00363318   0.00562445   -0.0192282  -0.00802678   -0.0185233 \n",
      "  HGX00103   HGX00103      0  0.000537389    0.0116175   -0.0118092   0.00174229  -0.00154448  -0.00206074   0.00584095  -0.00607365    0.0134285  -0.00544972 \n",
      "  HGX00105   HGX00105      0  0.000600472   0.00772032   -0.0192951   0.00658362  0.000265773   0.00688948   0.00217108    0.0010934  -0.00806283  -0.00385476 \n",
      "  HGX00106   HGX00106      0   0.00237363   0.00771666   -0.0116744    0.0134582   0.00486296  -0.00495273   0.00841738  -0.00433485 -0.000955299 -0.000449247 \n",
      "FID IID C1 C2 C3 C4 C5 C6 C7 C8 C9 C10\n",
      "HGX00096 HGX00096 -0.000495704 0.0140154 -0.0111572 0.00363723 0.000969209 -0.00632788 0.0041895 -0.0226794 0.0104471 -0.014345\n",
      "HGX00097 HGX00097 -2.64536e-05 0.00845106 -0.0159685 0.00652097 0.00261635 -0.000613264 -0.0104634 -0.0105515 0.0052152 -0.00294776\n",
      "HGX00099 HGX00099 -0.000668454 0.00801473 -0.0200389 0.00786452 -0.000514026 -0.00662578 0.000594134 -0.017528 -0.00383219 -0.00608985\n",
      "HGX00100 HGX00100 0.00473264 0.0161124 0.0111039 0.025022 0.00180064 -0.00509499 0.0305339 -0.0268026 -0.00699904 -0.019562\n",
      "HGX00101 HGX00101 -0.000491384 0.0148019 -0.0151813 0.0112514 0.00721504 -0.00254442 0.00832269 -0.0173986 -0.0245312 -0.0155404\n",
      "HGX00102 HGX00102 -0.00195889 0.0138588 -0.0175688 0.00464131 0.00533101 -0.00363318 0.00562445 -0.0192282 -0.00802678 -0.0185233\n",
      "HGX00103 HGX00103 0.000537389 0.0116175 -0.0118092 0.00174229 -0.00154448 -0.00206074 0.00584095 -0.00607365 0.0134285 -0.00544972\n",
      "HGX00105 HGX00105 0.000600472 0.00772032 -0.0192951 0.00658362 0.000265773 0.00688948 0.00217108 0.0010934 -0.00806283 -0.00385476\n",
      "HGX00106 HGX00106 0.00237363 0.00771666 -0.0116744 0.0134582 0.00486296 -0.00495273 0.00841738 -0.00433485 -0.000955299 -0.000449247\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b38.mds.mds\n",
    "awk '{print$1, $2, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13}' $path/dataset.b38.mds.mds > $path/covar_mds.txt\n",
    "head $path/covar_mds.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID IID C1 C2 C3 C4 C5 C6 C7 C8 C9 C10 SEX PHENO\n",
      "HGX00096 HGX00096 -0.000495704 0.0140154 -0.0111572 0.00363723 0.000969209 -0.00632788 0.0041895 -0.0226794 0.0104471 -0.014345 1 1\n",
      "HGX00097 HGX00097 -2.64536e-05 0.00845106 -0.0159685 0.00652097 0.00261635 -0.000613264 -0.0104634 -0.0105515 0.0052152 -0.00294776 2 1\n",
      "HGX00099 HGX00099 -0.000668454 0.00801473 -0.0200389 0.00786452 -0.000514026 -0.00662578 0.000594134 -0.017528 -0.00383219 -0.00608985 2 1\n",
      "HGX00100 HGX00100 0.00473264 0.0161124 0.0111039 0.025022 0.00180064 -0.00509499 0.0305339 -0.0268026 -0.00699904 -0.019562 2 1\n",
      "HGX00101 HGX00101 -0.000491384 0.0148019 -0.0151813 0.0112514 0.00721504 -0.00254442 0.00832269 -0.0173986 -0.0245312 -0.0155404 1 1\n",
      "HGX00102 HGX00102 -0.00195889 0.0138588 -0.0175688 0.00464131 0.00533101 -0.00363318 0.00562445 -0.0192282 -0.00802678 -0.0185233 2 1\n",
      "HGX00103 HGX00103 0.000537389 0.0116175 -0.0118092 0.00174229 -0.00154448 -0.00206074 0.00584095 -0.00607365 0.0134285 -0.00544972 1 1\n",
      "HGX00105 HGX00105 0.000600472 0.00772032 -0.0192951 0.00658362 0.000265773 0.00688948 0.00217108 0.0010934 -0.00806283 -0.00385476 1 1\n",
      "HGX00106 HGX00106 0.00237363 0.00771666 -0.0116744 0.0134582 0.00486296 -0.00495273 0.00841738 -0.00433485 -0.000955299 -0.000449247 2 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk 'BEGIN {print \"SEX\",\"PHENO\"};{print $5, $6}' $intpath/dataset.b38.QCed.fam > $intpath/temp\n",
    "paste $path/covar_mds.txt $intpath/temp | sed 's/\\t/ /g'> $path/covar_mds_sex_pheno.txt\n",
    "rm $intpath/temp\n",
    "head $path/covar_mds_sex_pheno.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**For the next step you need the following files:**\n",
    "- dataset.b38.QCed (the bfile, i.e., dataset.b38.QCed.bed,dataset.b38.QCed.bim,and dataset.b38.QCed.fam)\n",
    "- covar_mds.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
