{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Task 6 Gene-wise statistics (MAGMA)\n",
    "\n",
    "In gene analysis, genetic marker data is aggregated to the level of whole genes, testing the joint association of all markers in the gene with the phenotype. Similarly, in gene-set analysis individual genes are aggregated to groups of genes sharing certain biological, functional or other characteristics.\n",
    "\n",
    "This is done using MAGMA[1]. The gene-set analysis is divided into two distinct and largely independent parts. In the first part a gene analysis is performed to quantify the degree of association each gene has with the phenotype. In addition the correlations between genes are estimated. These correlations reflect the LD between genes, and are needed in order to compensate for the dependencies between genes during the gene-set analysis. The gene p-values and gene correlation matrix are then used in the second part to perform the actual gene-set analysis.\n",
    "\n",
    "\n",
    "[1] de Leeuw CA, Mooij JM, Heskes T, Posthuma D. MAGMA: generalized gene-set analysis of GWAS data. PLoS computational biology. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657/. Published April 17, 2015. Accessed August 18, 2020."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-18T14:11:34.262741Z",
     "start_time": "2020-08-18T14:11:34.259937Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Create directories for the output files\n",
    "path=\"/mnt/data/GWAS/output/build38/task6_genewise\"\n",
    "if not os.path.exists(path):\n",
    "    os.makedirs(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/build38/task6_genewise\n",
      "env: task4path=/mnt/data/GWAS/output/build38/task4_assoc\n"
     ]
    }
   ],
   "source": [
    "%env path=/mnt/data/GWAS/output/build38/task6_genewise\n",
    "%env task4path= /mnt/data/GWAS/output/build38/task4_assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Filter out low frequency SNPs (MAF<0.01 in this example) for MAGMA**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-11T16:58:43.422025Z",
     "start_time": "2019-11-11T16:58:35.319572Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "awk '{OFS=\"\\t\"; if($6>0.01 && $6<0.99) print $0}' | sed 's/ /\\t/g' $task4path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot > $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n",
    "wc $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-11T17:01:19.824608Z",
     "start_time": "2019-11-11T17:01:11.944437Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   8976887  107722644 1254477752 /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n",
      "   8976887  107722644 1254477752 /mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "wc $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n",
    "wc $task4path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:34:52.485999Z",
     "start_time": "2019-11-12T08:34:52.050973Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS\tANNOT\n",
      "10\t100000235\t10:100000235:C:T\tC\tT\t0.2982\t0.7959\t0.6679\t0.3084\t0.1906\trs11596870\tABCC2(+148kb)|CHUK(-188.1kb)|CPN1(-42.07kb)|DNMBP(0)|DNMBP-AS1(+41.24kb)|ERLIN1(-149.9kb)\n",
      "10\t100000943\t10:100000943:G:A\tG\tA\t0.0929\t0.7945\t1.2681\t0.5058\t0.6387\trs11190359\tABCC2(+148.8kb)|CHUK(-187.4kb)|CPN1(-41.36kb)|DNMBP(0)|DNMBP-AS1(+41.94kb)|ERLIN1(-149.1kb)\n",
      "10\t100000979\t10:100000979:T:C\tT\tC\t0.0504\t0.7881\t1.7337\t0.7499\t0.4631\trs11190360\tABCC2(+148.8kb)|CHUK(-187.4kb)|CPN1(-41.33kb)|DNMBP(0)|DNMBP-AS1(+41.98kb)|ERLIN1(-149.1kb)\n",
      "10\t100002012\t10:100002012:T:C\tT\tC\t0.0374\t0.8215\t1.0097\t0.6786\t0.9886\trs11190362\tABCC2(+149.8kb)|CHUK(-186.4kb)|CPN1(-40.3kb)|DNMBP(0)|DNMBP-AS1(+43.01kb)|ERLIN1(-148.1kb)\n",
      "10\t100002038\t10:100002038:G:A\tG\tA\t0.0102\t0.8861\t3.051\t1.3529\t0.4097\trs192480913\tABCC2(+149.8kb)|CHUK(-186.3kb)|CPN1(-40.27kb)|DNMBP(0)|DNMBP-AS1(+43.04kb)|ERLIN1(-148.1kb)\n",
      "10\t100002300\t10:100002300:GA:G\tGA\tG\t0.0374\t0.8216\t1.0082\t0.6788\t0.9904\trs111354488\tABCC2(+150.1kb)|CHUK(-186.1kb)|CPN1(-40.01kb)|DNMBP(0)|DNMBP-AS1(+43.3kb)|ERLIN1(-147.8kb)\n",
      "10\t100002330\t10:100002330:AAAAG:A\tAAAAG\tA\t0.0647\t0.8095\t1.3697\t0.6481\t0.6274\tNA\tABCC2(+150.1kb)|CHUK(-186kb)|CPN1(-39.98kb)|DNMBP(0)|DNMBP-AS1(+43.33kb)|ERLIN1(-147.8kb)\n",
      "10\t100002628\t10:100002628:A:C\tA\tC\t0.4467\t0.8605\t0.7322\t0.2868\t0.2771\trs11190363\tABCC2(+150.4kb)|CHUK(-185.7kb)|CPN1(-39.68kb)|DNMBP(0)|DNMBP-AS1(+43.63kb)|ERLIN1(-147.5kb)\n",
      "10\t100002875\t10:100002875:A:G\tA\tG\t0.0648\t0.8118\t1.3222\t0.6435\t0.6642\trs7894103\tABCC2(+150.7kb)|CHUK(-185.5kb)|CPN1(-39.43kb)|DNMBP(0)|DNMBP-AS1(+43.88kb)|ERLIN1(-147.2kb)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:35:12.485850Z",
     "start_time": "2019-11-12T08:35:04.280230Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP\tCHR\tBP\n",
      "rs11596870\t10\t100000235\n",
      "rs11190359\t10\t100000943\n",
      "SNP\tP\n",
      "rs11596870\t0.1906\n",
      "rs11190359\t0.6387\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Generate  *.SNP.LOC (SNP, CHR,BP) and *.SNP.VAL files\n",
    "\n",
    "awk 'BEGIN{OFS=\"\\t\";print \"SNP\",\"CHR\",\"BP\"};{OFS=\"\\t\"; if(NR>1)  print $11,$1,$2}' $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01 > $path/dataset.b38.imputed.dosage.maf.0.01.SNP.LOC\n",
    "awk 'BEGIN{OFS=\"\\t\";print \"SNP\",\"P\"};{OFS=\"\\t\"; if(NR>1) print $11,$10}' $path/dataset.b38.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01 | sed 's/e/E/g' > $path/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL\n",
    "\n",
    "head -3 $path/dataset.b38.imputed.dosage.maf.0.01.SNP.LOC \n",
    "head -3  $path/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:35:23.242731Z",
     "start_time": "2019-11-12T08:35:12.488848Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--annotate window=50,50\n",
      "\t--snp-loc /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.LOC\n",
      "\t--gene-loc /mnt/data/GWAS/ref_files/build38/NCBI38.gene.loc\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb\n",
      "\n",
      "Start time is 14:44:20, Saturday 06 Mar 2021\n",
      "\n",
      "Starting annotation...\n",
      "Reading gene locations from file /mnt/data/GWAS/ref_files/build38/NCBI38.gene.loc... \n",
      "\tadding window: 50000bp\n",
      "\t20137 gene locations read from file\n",
      "\tchromosome  1: 2097 genes\n",
      "\tchromosome  2: 1285 genes\n",
      "\tchromosome  3: 1078 genes\n",
      "\tchromosome  4: 765 genes\n",
      "\tchromosome  5: 886 genes\n",
      "\tchromosome  6: 1053 genes\n",
      "\tchromosome  7: 942 genes\n",
      "\tchromosome  8: 691 genes\n",
      "\tchromosome  9: 803 genes\n",
      "\tchromosome 10: 748 genes\n",
      "\tchromosome 11: 1299 genes\n",
      "\tchromosome 12: 1032 genes\n",
      "\tchromosome 13: 341 genes\n",
      "\tchromosome 14: 613 genes\n",
      "\tchromosome 15: 616 genes\n",
      "\tchromosome 16: 865 genes\n",
      "\tchromosome 17: 1197 genes\n",
      "\tchromosome 18: 283 genes\n",
      "\tchromosome 19: 1423 genes\n",
      "\tchromosome 20: 540 genes\n",
      "\tchromosome 21: 246 genes\n",
      "\tchromosome 22: 461 genes\n",
      "\tchromosome  X: 825 genes\n",
      "\tchromosome  Y: 48 genes\n",
      "Reading SNP locations from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.LOC... \n",
      "\tWARNING: on line 1, chromosome code 'CHR' not recognised; skipping SNP (ID = SNP)\n",
      "\t8976886 SNP locations read from file                                                            \n",
      "\tof those, 5701055 (63.51%) mapped to at least one gene\n",
      "Writing annotation to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\tfor chromosome  1, 10 genes are empty (out of 2097)\n",
      "\tfor chromosome  2, 1 gene is empty (out of 1285)\n",
      "\tfor chromosome  4, 2 genes are empty (out of 765)\n",
      "\tfor chromosome  5, 3 genes are empty (out of 886)\n",
      "\tfor chromosome  8, 12 genes are empty (out of 691)\n",
      "\tfor chromosome  9, 26 genes are empty (out of 803)\n",
      "\tfor chromosome 14, 7 genes are empty (out of 613)\n",
      "\tfor chromosome 15, 13 genes are empty (out of 616)\n",
      "\tfor chromosome 16, 5 genes are empty (out of 865)\n",
      "\tfor chromosome 20, 3 genes are empty (out of 540)\n",
      "\tfor chromosome 21, 22 genes are empty (out of 246)\n",
      "\tfor chromosome 22, 2 genes are empty (out of 461)\n",
      "\tfor chromosome  X, 825 genes are empty (out of 825)\n",
      "\tfor chromosome  Y, 48 genes are empty (out of 48)\n",
      "\tat least one SNP mapped to each of a total of 19158 genes (out of 20137)\n",
      "\n",
      "\n",
      "End time is 14:44:34, Saturday 06 Mar 2021 (elapsed: 00:00:14)\n",
      "# window_up = 50000\n",
      "# window_down = 50000\n",
      "81399\t1:635716:736654\tNA\tNA\tNA\tNA\tNA\trs151190501\trs61769339\tNA\tNA\tNA\trs201221815\tNA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Annotate SNPS to genes in 50Kb; this parameter is not fixed, usually ranging from 0 to 500Kb. You cn modify this distance with the window parameter\n",
    "/usr/lib/magma/magma --annotate window=50,50 --snp-loc $path/dataset.b38.imputed.dosage.maf.0.01.SNP.LOC --gene-loc /mnt/data/GWAS/ref_files/build38/NCBI38.gene.loc --out $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb\n",
    "head -3 $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-20T10:37:34.844137Z",
     "start_time": "2020-08-20T10:37:34.703213Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 8 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 1 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... \n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 8 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 4 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "495 individuals read\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 1 only)... Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 4 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 5 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 5 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 9 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 9 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 2 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 2 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 10 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 10 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 11 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 11 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 3 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 3 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 12 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 12 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 13 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 13 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 14 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 14 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 15 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 15 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 16 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 16 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 17 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 17 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 18 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 18 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 19 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:10, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 19 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 21 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:11, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 21 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 22 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:11, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 22 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 7 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:11, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 7 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 20 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:11, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 20 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 6 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL N=495\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:03:11, Saturday 06 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.fam... 495 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bim (chromosome 6 only)... 316140 SNPs read, of which 5188 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 9939 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 17957 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 20094 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 10322 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 24393 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 18647 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 8413 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 4451 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 7322 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 9060 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 13504 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 10829 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 24163 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 17152 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 9788 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 14904 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 10219 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 14571 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 20338 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 15458 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "316140 SNPs read, of which 15403 used\n",
      "Preparing file /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 4225 SNPs in data (0.04707% of lines, 42.51% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 9565 SNPs in data (0.1066% of lines, 47.6% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 2294 SNPs in data (0.02555% of lines, 44.22% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 4431 SNPs in data (0.04936% of lines, 48.91% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1197 gene definitions read from file\n",
      "\tfound 1162 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 17, containing 1162 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t459 gene definitions read from file\n",
      "\tfound 439 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 22, containing 439 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t763 gene definitions read from file\n",
      "\tfound 744 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 4, containing 744 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t283 gene definitions read from file\n",
      "\tfound 283 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 18, containing 283 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 2090 SNPs in data (0.02328% of lines, 46.96% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 3726 SNPs in data (0.04151% of lines, 50.89% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 3422 SNPs in data (0.03812% of lines, 40.68% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 4735 SNPs in data (0.05275% of lines, 46.34% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t224 gene definitions read from file\n",
      "\tfound 211 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 21, containing 211 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 6812 SNPs in data (0.07588% of lines, 46.75% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 5276 SNPs in data (0.05877% of lines, 48.72% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 4478 SNPs in data (0.04988% of lines, 43.38% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t537 gene definitions read from file\n",
      "\tfound 531 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 20, containing 531 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 11365 SNPs in data (0.1266% of lines, 47.03% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 7886 SNPs in data (0.08785% of lines, 45.98% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 11105 SNPs in data (0.1237% of lines, 45.53% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1423 gene definitions read from file\n",
      "\tfound 1416 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 19, containing 1416 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 8857 SNPs in data (0.09866% of lines, 47.5% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t606 gene definitions read from file\n",
      "\tfound 600 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 14, containing 600 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t341 gene definitions read from file\n",
      "\tfound 335 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 13, containing 335 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 7204 SNPs in data (0.08025% of lines, 46.77% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 7396 SNPs in data (0.08239% of lines, 47.85% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1299 gene definitions read from file\n",
      "\tfound 1244 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 11, containing 1244 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t860 gene definitions read from file\n",
      "\tfound 806 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 16, containing 806 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 4266 SNPs in data (0.04752% of lines, 43.58% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 6245 SNPs in data (0.06957% of lines, 46.25% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t942 gene definitions read from file\n",
      "\tfound 909 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 7, containing 909 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1284 gene definitions read from file\n",
      "\tfound 1235 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 2, containing 1235 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t748 gene definitions read from file\n",
      "\tfound 727 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 10, containing 727 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 6715 SNPs in data (0.0748% of lines, 45.06% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t679 gene definitions read from file\n",
      "\tfound 652 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 8, containing 652 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1053 gene definitions read from file\n",
      "\tfound 1042 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 6, containing 1042 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t2087 gene definitions read from file\n",
      "\tfound 1974 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 1, containing 1974 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t603 gene definitions read from file\n",
      "\tfound 575 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 15, containing 575 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 9876 SNPs in data (0.11% of lines, 48.56% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t777 gene definitions read from file\n",
      "\tfound 762 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 9, containing 762 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1032 gene definitions read from file\n",
      "\tfound 1015 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 12, containing 1015 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1078 gene definitions read from file\n",
      "\tfound 1061 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 3, containing 1061 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 8976887 lines from file, containing valid SNP p-values for 8678 SNPs in data (0.09667% of lines, 48.33% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 495 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t883 gene definitions read from file\n",
      "\tfound 862 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 5, containing 862 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch21_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:36, Saturday 06 Mar 2021 (elapsed: 00:00:25)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch18_chr.genes.out\n",
      "\tprocessed genes: 271 (80.9%)     \n",
      "\n",
      "End time is 15:03:45, Saturday 06 Mar 2021 (elapsed: 00:00:35)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch22_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:48, Saturday 06 Mar 2021 (elapsed: 00:00:37)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch13_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:50, Saturday 06 Mar 2021 (elapsed: 00:00:40)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch20_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:55, Saturday 06 Mar 2021 (elapsed: 00:00:44)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch15_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:55, Saturday 06 Mar 2021 (elapsed: 00:00:45)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch14_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:03:58, Saturday 06 Mar 2021 (elapsed: 00:00:48)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch16_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:01, Saturday 06 Mar 2021 (elapsed: 00:00:51)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch8_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:03, Saturday 06 Mar 2021 (elapsed: 00:00:53)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch17_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:03, Saturday 06 Mar 2021 (elapsed: 00:00:53)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch9_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:04, Saturday 06 Mar 2021 (elapsed: 00:00:54)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch4_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:04, Saturday 06 Mar 2021 (elapsed: 00:00:54)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch10_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:05, Saturday 06 Mar 2021 (elapsed: 00:00:55)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch7_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:06, Saturday 06 Mar 2021 (elapsed: 00:00:55)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch5_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:06, Saturday 06 Mar 2021 (elapsed: 00:00:56)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch12_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:07, Saturday 06 Mar 2021 (elapsed: 00:00:57)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch11_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:08, Saturday 06 Mar 2021 (elapsed: 00:00:58)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch6_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:10, Saturday 06 Mar 2021 (elapsed: 00:00:59)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch19_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:10, Saturday 06 Mar 2021 (elapsed: 00:01:00)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch2_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:10, Saturday 06 Mar 2021 (elapsed: 00:01:00)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch3_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:12, Saturday 06 Mar 2021 (elapsed: 00:01:02)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch1_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:15, Saturday 06 Mar 2021 (elapsed: 00:01:05)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#Gene level analysis is performed using MAGMA, which compute gene-wise statistics taking into account physical distance and linkage disequilibrium (LD) between markers (de Leeuw et al. 2015). \n",
    "# All SNPs with MAF above 1% are used in these analyses, setting a distance threshold of 50kb. N is the number of individuals\n",
    "for i in {1..22}\n",
    "do\n",
    "nohup /usr/lib/magma/magma  --batch $i chr --big-data --seed 1234 --genes-only --bfile /mnt/data/GWAS/output/build38/task2.2_stratification/intermediate_datasets/dataset.b38.QCed --gene-annot $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot  --pval $path/dataset.b38.imputed.dosage.maf.0.01.SNP.PVAL  N=495 --gene-model multi --out $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:19:36.853278Z",
     "start_time": "2019-11-12T09:19:36.247662Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--merge /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\t--out /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 15:04:15, Saturday 06 Mar 2021\n",
      "\n",
      "Merging gene results files with prefix '/mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma'... \n",
      "\tWARNING: found only 0 .genes.raw batch files for chromosomes 1-22\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch1_chr.genes.out... \n",
      "\t1974 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch2_chr.genes.out... \n",
      "\t1235 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch3_chr.genes.out... \n",
      "\t1061 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch4_chr.genes.out... \n",
      "\t744 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch5_chr.genes.out... \n",
      "\t862 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch6_chr.genes.out... \n",
      "\t1042 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch7_chr.genes.out... \n",
      "\t909 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch8_chr.genes.out... \n",
      "\t652 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch9_chr.genes.out... \n",
      "\t762 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch10_chr.genes.out... \n",
      "\t727 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch11_chr.genes.out... \n",
      "\t1244 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch12_chr.genes.out... \n",
      "\t1015 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch13_chr.genes.out... \n",
      "\t335 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch14_chr.genes.out... \n",
      "\t600 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch15_chr.genes.out... \n",
      "\t575 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch16_chr.genes.out... \n",
      "\t806 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch17_chr.genes.out... \n",
      "\t1162 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch18_chr.genes.out... \n",
      "\t283 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch19_chr.genes.out... \n",
      "\t1416 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch20_chr.genes.out... \n",
      "\t531 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch21_chr.genes.out... \n",
      "\t211 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch22_chr.genes.out... \n",
      "\t439 genes read from file\n",
      "Writing merged gene analysis results to file /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out\n",
      "\n",
      "\n",
      "End time is 15:04:15, Saturday 06 Mar 2021 (elapsed: 00:00:00)\n",
      "GENE       CHR      START       STOP  NSNPS  NPARAM    N        ZSTAT      P_JOINT  P_SNPWISE_MEAN  P_SNPWISE_TOP1\n",
      "105378580    1     753398     855130      5       4  495       2.3677    0.0089488       0.0097317        0.014253\n",
      "148398       1     875741     994581     10       9  495       1.0111      0.15597         0.10946         0.23968\n",
      "26155        1     894203    1009299      9       8  495      0.90113      0.18376         0.16222         0.21856\n",
      "339451       1     909952    1015720      7       6  495      0.76407      0.22241         0.28831         0.17454\n",
      "84069        1     916497    1025108      9       8  495      0.51199      0.30433         0.41519         0.21856\n",
      "84808        1     925199    1032117      9       8  495      0.51199      0.30433         0.41519         0.21856\n",
      "57801        1     948962    1051285      9       8  495      0.82705       0.2041         0.19898         0.21856\n",
      "9636         1     963467    1064540     10       9  495      0.64915      0.25812          0.2787         0.23968\n",
      "375790       1     970101    1106119     15      12  495      -0.8976       0.8153         0.75429         0.77866\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#merge batches\n",
    "#/usr/lib/magma/magma\n",
    "magma --merge $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma --out $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
    "head $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:20:13.633421Z",
     "start_time": "2019-11-12T09:20:13.054616Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENE       CHR      START       STOP  NSNPS  NPARAM    N        ZSTAT      P_JOINT  P_SNPWISE_MEAN  P_SNPWISE_TOP1\n",
      "11214       15   85330616   85799358     20      13  495        3.813   6.8633e-05      1.8683e-05       0.0013226\n",
      "134111       5    6387347    6546721     17      14  495       3.7768   7.9438e-05      5.6238e-05       0.0018005\n",
      "26074       20   20002514   20410714     25      19  495       3.6031   0.00015722      7.9538e-05       0.0047978\n",
      "57117        4  105632627  105758729      5       4  495       2.9638    0.0015192       0.0001333        0.037451\n",
      "2197        11   65070627   65172200      1       1  495       3.5637    0.0001828       0.0001828       0.0001828\n",
      "741         11   65066403   65167738      1       1  495       3.5637    0.0001828       0.0001828       0.0001828\n",
      "30811       21   31823315   32054064     18      15  495       2.9599    0.0015389       0.0002843        0.049541\n",
      "132884       4    5484397    5759548     26      21  495        3.242   0.00059339      0.00031645       0.0069702\n",
      "51250        6  106978172  107101586      8       7  495       3.0124    0.0012961      0.00042762        0.012777\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sort -gk10,10 $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out >$path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\n",
    "head $path/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:23:49.202385Z",
     "start_time": "2019-11-12T09:23:48.793407Z"
    }
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# Add HUGO gene name using the refernece file NCBI38.gene.loc linking oficial gene symbols  and GenBank Ids\n",
    "magma<-read.table(\"/mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\", header=TRUE)\n",
    "genes<-read.table(\"/mnt/data/GWAS/ref_files/build38/NCBI38.gene.loc\")\n",
    "genes<-genes[-c(2:4)]\n",
    "head(genes)\n",
    "colnames(genes) <- c(\"GENE\",\"STRAND\",\"HUGO\")\n",
    "\n",
    "magma_merged<-merge(magma,genes, by=\"GENE\")\n",
    "magma_merged <- magma_merged[order(magma_merged$P_SNPWISE_MEAN), ]\n",
    "magma_rank<-rank(magma_merged[,10],na.last = \"keep\", ties.method = \"min\")\n",
    "magma_ranked=cbind(magma_rank, magma_merged)\n",
    "\n",
    "write.table(magma_ranked, \"/mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted.annot\", quote=FALSE, sep=\"\\t\", row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "magma_rank\tGENE\tCHR\tSTART\tSTOP\tNSNPS\tNPARAM\tN\tZSTAT\tP_JOINT\tP_SNPWISE_MEAN\tP_SNPWISE_TOP1\tSTRAND\tHUGO\n",
      "1\t11214\t15\t85330616\t85799358\t20\t13\t495\t3.813\t6.8633e-05\t1.8683e-05\t0.0013226\t+\tAKAP13\n",
      "2\t134111\t5\t6387347\t6546721\t17\t14\t495\t3.7768\t7.9438e-05\t5.6238e-05\t0.0018005\t+\tUBE2QL1\n",
      "3\t26074\t20\t20002514\t20410714\t25\t19\t495\t3.6031\t0.00015722\t7.9538e-05\t0.0047978\t+\tCFAP61\n",
      "4\t57117\t4\t105632627\t105758729\t5\t4\t495\t2.9638\t0.0015192\t0.0001333\t0.037451\t-\tINTS12\n",
      "5\t741\t11\t65066403\t65167738\t1\t1\t495\t3.5637\t0.0001828\t0.0001828\t0.0001828\t-\tZNHIT2\n",
      "5\t2197\t11\t65070627\t65172200\t1\t1\t495\t3.5637\t0.0001828\t0.0001828\t0.0001828\t-\tFAU\n",
      "7\t30811\t21\t31823315\t32054064\t18\t15\t495\t2.9599\t0.0015389\t0.0002843\t0.049541\t+\tHUNK\n",
      "8\t132884\t4\t5484397\t5759548\t26\t21\t495\t3.242\t0.00059339\t0.00031645\t0.0069702\t-\tEVC2\n",
      "9\t51250\t6\t106978172\t107101586\t8\t7\t495\t3.0124\t0.0012961\t0.00042762\t0.012777\t+\tC6orf203\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /mnt/data/GWAS/output/build38/task6_genewise/dataset.b38.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted.annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
