{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1000 Genome data preparation\n",
    "\n",
    "Data from the 1000 Genomes Project need to be processed before used for population stratification. This dataset after QC is provided, there is no need to run these first steps.\n",
    "\n",
    "To complete this task it is necessary to have generated the bfile 'dataset.b37_IBD' from the previous tutorial (Task_2_QC)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:54:24.831494Z",
     "start_time": "2020-08-21T13:54:24.827425Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/task2.2_stratification\n",
      "env: intpath=/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets\n"
     ]
    }
   ],
   "source": [
    "%env path= /mnt/data/GWAS/output/task2.2_stratification\n",
    "%env intpath=/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets\n",
    "%env ref_files=/mnt/data/GWAS/ref_files/Phase3_v5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T10:27:36.205521Z",
     "start_time": "2020-08-21T10:27:35.118676Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1:10177\n",
      "1:11008\n",
      "1:11012\n",
      "1:13110\n",
      "rs201725126\n",
      "rs200579949\n",
      "1:13273\n",
      "1:14464\n",
      "1:14599\n",
      "1:14604\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "# Get a list of SNPs from your working dataset\n",
    "awk '{print $2}' /mnt/data/GWAS/output/task2_QC/intermediate_datasets/dataset.b37.IBD.bim > $path/mySNPlist.txt\n",
    "head $path/mySNPlist.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:11:32.264601Z",
     "start_time": "2020-08-21T10:59:20.850730Z"
    },
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr1.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr1\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr1.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr1-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr1-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr1-temporary.fam written.\n",
      "497923 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999861.\n",
      "497923 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr1.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr1.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr2.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr2\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr2.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr2-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr2-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr2-temporary.fam written.\n",
      "540431 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999857.\n",
      "540431 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr2.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr2.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr3.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr3\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr3.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr3-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr3-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr3-temporary.fam written.\n",
      "462062 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "462062 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr3.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr3.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr4.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr4\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr4.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr4-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr4-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr4-temporary.fam written.\n",
      "474260 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999847.\n",
      "474260 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr4.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr4.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr5.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr5\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr5.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr5-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr5-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr5-temporary.fam written.\n",
      "410091 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999849.\n",
      "410091 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr5.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr5.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr6.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr6\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr6.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr6-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr6-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr6-temporary.fam written.\n",
      "440932 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999843.\n",
      "440932 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr6.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr6.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr7\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr7.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr7-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr7-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr7-temporary.fam written.\n",
      "384634 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999854.\n",
      "384634 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr7.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr7.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr8.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr8\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr8.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr8-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr8-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr8-temporary.fam written.\n",
      "354504 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999832.\n",
      "354504 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr8.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr8.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr9.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr9\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr9.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr9-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr9-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr9-temporary.fam written.\n",
      "281556 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999837.\n",
      "281556 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr9.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr9.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr10.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr10\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr10.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr10-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr10-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr10-temporary.fam written.\n",
      "336853 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999856.\n",
      "336853 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr10.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr10.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr11.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr11\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr11.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr11-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr11-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr11-temporary.fam written.\n",
      "328068 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999851.\n",
      "328068 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr11.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr11.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr12.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr12\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr12.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr12-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr12-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr12-temporary.fam written.\n",
      "307031 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99985.\n",
      "307031 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr12.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr12.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr13.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr13\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr13.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr13-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr13-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr13-temporary.fam written.\n",
      "243551 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999859.\n",
      "243551 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr13.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr13.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr14.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr14\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr14.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr14-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr14-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr14-temporary.fam written.\n",
      "212995 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999849.\n",
      "212995 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr14.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr14.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr15.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr15\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr15.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr15-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr15-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr15-temporary.fam written.\n",
      "185833 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999847.\n",
      "185833 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr15.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr15.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr16.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr16\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr16.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr16-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr16-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr16-temporary.fam written.\n",
      "198791 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999813.\n",
      "198791 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr16.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr16.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr17.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr17\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr17.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr17-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr17-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr17-temporary.fam written.\n",
      "176044 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999858.\n",
      "176044 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr17.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr17.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr18.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr18\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr18.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr18-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr18-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr18-temporary.fam written.\n",
      "186082 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999853.\n",
      "186082 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr18.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr18.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr19.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr19\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr19.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr19-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr19-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr19-temporary.fam written.\n",
      "154398 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999834.\n",
      "154398 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr19.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr19.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr20.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr20\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr20.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr20-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr20-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr20-temporary.fam written.\n",
      "144774 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999857.\n",
      "144774 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr20.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr20.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr21.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr21\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr21.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr21-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr21-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr21-temporary.fam written.\n",
      "93739 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999823.\n",
      "93739 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr21.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr21.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr22.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.chr22\n",
      "  --vcf /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr22.recode.vcf\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/task2.2_stratification//1KG.chr22-temporary.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr22-temporary.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr22-temporary.fam written.\n",
      "91208 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99982.\n",
      "91208 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.chr22.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr22.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr1\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 497923 out of a possible 6467265 Sites\n",
      "Run Time = 610.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr2\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 540431 out of a possible 7080712 Sites\n",
      "Run Time = 723.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr3\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 462062 out of a possible 5831616 Sites\n",
      "Run Time = 538.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr4\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 474260 out of a possible 5731977 Sites\n",
      "Run Time = 541.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr5\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 410091 out of a possible 5265131 Sites\n",
      "Run Time = 481.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr6\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 440932 out of a possible 5023483 Sites\n",
      "Run Time = 494.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr7\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 384634 out of a possible 4716205 Sites\n",
      "Run Time = 474.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr8\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 354504 out of a possible 4596555 Sites\n",
      "Run Time = 422.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr9\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 281556 out of a possible 3560268 Sites\n",
      "Run Time = 337.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr10\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 336853 out of a possible 3991662 Sites\n",
      "Run Time = 382.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr11\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 328068 out of a possible 4045150 Sites\n",
      "Run Time = 383.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr12\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 307031 out of a possible 3867557 Sites\n",
      "Run Time = 359.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr13\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 243551 out of a possible 2857612 Sites\n",
      "Run Time = 279.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr14\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 212995 out of a possible 2654741 Sites\n",
      "Run Time = 253.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr15\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 185833 out of a possible 2424342 Sites\n",
      "Run Time = 224.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr16\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 198791 out of a possible 2697571 Sites\n",
      "Run Time = 246.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr17\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 176044 out of a possible 2328930 Sites\n",
      "Run Time = 214.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr18\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 186082 out of a possible 2266920 Sites\n",
      "Run Time = 219.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr19\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 154398 out of a possible 1832210 Sites\n",
      "Run Time = 181.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr20\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 144774 out of a possible 1812578 Sites\n",
      "Run Time = 172.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr21\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 93739 out of a possible 1105434 Sites\n",
      "Run Time = 112.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/input/Phase3_v5/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/task2.2_stratification//1KG.vcf.chr22\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/task2.2_stratification//mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.8\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 91208 out of a possible 1103411 Sites\n",
      "Run Time = 107.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Creates a recode vcf per chromosome, using as input 1000 Genomes compressed vcf file per chr and previous SNPs list.\n",
    "# Then plink converts them to bfiles. \n",
    "for i in {1..22}\n",
    "do\n",
    "vcftools --gzvcf $ref_files/ALL.chr$i.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz --snps $path/mySNPlist.txt --recode --out $path/1KG.vcf.chr$i\n",
    "plink --vcf $path/1KG.vcf.chr$i.recode.vcf --make-bed --out $path/1KG.chr$i\n",
    "rm $path/1KG.vcf.chr$i.recode.vcf\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:50:12.421299Z",
     "start_time": "2020-08-21T13:50:12.360491Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Creates a file with the list of bfiles (.bim, .bed, .fam) outputed in the previous step\n",
    "# 1st-3rd parameter: path to previous plink files\n",
    "# 4th parameter: output file\n",
    "for i in {2..22}\n",
    "do\n",
    "echo $path/1KG.chr$i.bed $path/1KG.chr$i.bim $path/1KG.chr$i.fam >> $path/1KG.outlist.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:52:57.486015Z",
     "start_time": "2020-08-21T13:50:21.665190Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification//1KG.all.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/task2.2_stratification//1KG.chr1\n",
      "  --make-bed\n",
      "  --merge-list /mnt/data/GWAS/output/task2.2_stratification//1KG.outlist.txt\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification//1KG.all\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "Performing single-pass merge (2504 people, 6505760 variants).\n",
      "Merged fileset written to                     \n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all-merge.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all-merge.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all-merge.fam .\n",
      "6505760 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "6505760 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/task2.2_stratification//1KG.all.bed +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all.bim +\n",
      "/mnt/data/GWAS/output/task2.2_stratification//1KG.all.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Merge all bfiles into a single one\n",
    "plink --bfile $path/1KG.chr1 --make-bed --merge-list $path/1KG.outlist.txt --out $path/1KG.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:14:02.348522Z",
     "start_time": "2020-08-21T14:14:02.088747Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# remove individual chr plink files\n",
    "for i in {2..22}\n",
    "do\n",
    "rm $path/1KG.chr$i.*\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## QC\n",
    "\n",
    "1000 Genomes Dataset after this QC is provided (1kG.QCed.forMDS). There is no need to run these steps."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:08:56.941408Z",
     "start_time": "2020-08-21T14:07:35.739478Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /mnt/data/GWAS/output/task2.2_stratification/1KG.all\n",
      "  --geno 0.05\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "6505760 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS.nosex\n",
      ".\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "5572 variants removed due to missing genotype data (--geno).\n",
      "6500188 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove variants based on missing genotype data.\n",
    "plink --bfile $path/1KG.all --geno 0.05 --allow-no-sex --make-bed --out $intpath/1kG_MDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:11:39.790617Z",
     "start_time": "2020-08-21T14:10:39.269363Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS\n",
      "  --make-bed\n",
      "  --mind 0.03\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "6500188 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2.nosex\n",
      ".\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999931.\n",
      "6500188 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove individuals based on missing genotype data.\n",
    "plink --bfile $intpath/1kG_MDS --mind 0.03 --allow-no-sex --make-bed --out $intpath/1kG_MDS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:17:58.049728Z",
     "start_time": "2020-08-21T14:16:58.642212Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS2\n",
      "  --maf 0.05\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "6500188 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3.nosex\n",
      ".\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999931.\n",
      "452774 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "6047414 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/1kG_MDS3.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove variants based on MAF.\n",
    "plink --bfile $intpath/1kG_MDS2 --maf 0.05 --allow-no-sex --make-bed --out $intpath/1kG_MDS3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**For the next step you need the following files:**\n",
    "- 1kG_MDS3 (the bfile, i.e., 1kG_MDS3.bed, 1kG_MDS3.bim, and 1kG_MDS3.fam\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
