{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Task 3: Imputation\n",
    "\n",
    "After the QC, genotype imputation can be performed with the minimac 3 algorithm at the University of Michigan server using the HRC reference panel, and the SHAPEIT tool for haplotype phasing. After imputation, SNPs with an R2 quality estimate lower than 0.3 are excluded from further analyses according to the software recommendations. \n",
    "\n",
    "To run this notebook you need the bfile dataset.b3.QCed, generated on previous task Population stratification.\n",
    "\n",
    "https://rstudio-pubs-static.s3.amazonaws.com/452627_d519d1c86bd249e6a2d9638ef1ea836c.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Preparing-the-dataset-for-imputation\" data-toc-modified-id=\"Preparing-the-dataset-for-imputation-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Preparing the dataset for imputation</a></span></li><li><span><a href=\"#Upload-vcf.gz-files-to-the-Michigan-server-(phasing:-SHAPE-IT;-reference:HRC;-mode:-QC&amp;imputation)\" data-toc-modified-id=\"Upload-vcf.gz-files-to-the-Michigan-server-(phasing:-SHAPE-IT;-reference:HRC;-mode:-QC&amp;imputation)-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Upload vcf.gz files to the Michigan server (phasing: SHAPE-IT; reference:HRC; mode: QC&amp;imputation)</a></span></li><li><span><a href=\"#QC-of-the-imputed-genotypes\" data-toc-modified-id=\"QC-of-the-imputed-genotypes-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>QC of the imputed genotypes</a></span></li></ul></div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-22T08:31:09.380193Z",
     "start_time": "2020-08-22T08:31:09.361243Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-18T10:18:12.360263Z",
     "start_time": "2020-08-18T10:18:12.357423Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/task3_imputation\n"
     ]
    }
   ],
   "source": [
    "%env path=/mnt/data/GWAS/output/task3_imputation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Preparing the dataset for imputation\n",
    "\n",
    "We will use Will Rayner's toolbox to prepare the data. (https://www.well.ox.ac.uk/~wrayner/tools/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-22T10:38:10.337578Z",
     "start_time": "2020-08-22T10:38:05.134254Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task3_imputation/dataset.b37.outliers.freq.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers\n",
      "  --freq\n",
      "  --out /mnt/data/GWAS/output/task3_imputation/dataset.b37.outliers.freq\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999931.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/mnt/data/GWAS/output/task3_imputation/dataset.b37.outliers.freq.frq .\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 1060176 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/task3_imputation/dataset.b37.outliers.freq.hh ); many\n",
      "commands treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Determine allele frequencies in the dataset\n",
    "plink --bfile /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers  --freq --out $path/dataset.b37.outliers.freq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Run the perl script from Will Rayner's toolbox to check plink .bim files against HRC/1000G for strand, id names, positions, alleles, ref/alt assignment.** \n",
    "\n",
    "A specific population to check frequency against can be changed with -p (Default ALL, options ALL, EUR, AFR, AMR, SAS, EAS) \n",
    "\n",
    "Using the terminal (New->Terminal from Jupyter panel or from the terminal inside your docker container) create a bash script containing the perl command. Provide execution permission (chmod +x) and run it (./HRC.sh). Place the script where the path environment variable points.\n",
    "\n",
    "Next, run from terminal the Run-plink.sh created by the perl one.\n",
    "\n",
    "**IMPORTANT: Do not run this script on momic.us.es.** This script needs 30Gb of RAM to run. The ouput of this command is stored in the path where the environment variable 'path' points to. If the user needs to impute his own data, install this pipeline locally following the User Manual."
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "#!bin/bash\n",
    "perl /mnt/data/GWAS/scripts/HRC-1000G-check-bim.pl -v -b dataset.b37.outliers.bim  -f dataset.b37.outliers.freq.frq -r /mnt/data/GWAS/ref_files/HRC.r1-1.GRCh37.wgs.mac5.sites.tab -h > dataset.b37.HRC-check.log "
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-22T13:27:02.111012Z",
     "start_time": "2020-08-22T13:27:02.079592Z"
    }
   },
   "source": [
    "bash Run-plink.sh > Run-plink.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-22T16:45:21.402685Z",
     "start_time": "2020-08-22T16:45:21.390962Z"
    },
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to TEMP1.log.\n",
      "Options in effect:\n",
      "  --bfile dataset.b37.outliers\n",
      "  --exclude Exclude-dataset.b37.outliers-HRC.txt\n",
      "  --make-bed\n",
      "  --out TEMP1\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--exclude: 5436495 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "5436495 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to TEMP1.bed + TEMP1.bim + TEMP1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to TEMP2.log.\n",
      "Options in effect:\n",
      "  --bfile TEMP1\n",
      "  --make-bed\n",
      "  --out TEMP2\n",
      "  --update-chr\n",
      "  --update-map Chromosome-dataset.b37.outliers-HRC.txt\n",
      "\n",
      "Note: --update-map <filename> + parameter-free --update-chr deprecated.  Use\n",
      "--update-chr <filename> instead.\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "5436495 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--update-chr: 2 values updated, 13 variant IDs not present.\n",
      "--make-bed to TEMP2.bed + TEMP2.bim + TEMP2.fam ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to TEMP3.log.\n",
      "Options in effect:\n",
      "  --bfile TEMP2\n",
      "  --make-bed\n",
      "  --out TEMP3\n",
      "  --update-map Position-dataset.b37.outliers-HRC.txt\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--update-map: 2 values updated, 13 variant IDs not present.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "5436495 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to TEMP3.bed + TEMP3.bim + TEMP3.fam ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to TEMP4.log.\n",
      "Options in effect:\n",
      "  --bfile TEMP3\n",
      "  --flip Strand-Flip-dataset.b37.outliers-HRC.txt\n",
      "  --make-bed\n",
      "  --out TEMP4\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--flip: 1 SNP flipped.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "5436495 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to TEMP4.bed + TEMP4.bim + TEMP4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile TEMP4\n",
      "  --make-bed\n",
      "  --out dataset.b37.outliers-updated\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "--a1-allele: 5436495 assignments made.\n",
      "5436495 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to dataset.b37.outliers-updated.bed +\n",
      "dataset.b37.outliers-updated.bim + dataset.b37.outliers-updated.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr1.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 1\n",
      "  --out dataset.b37.outliers-updated-chr1\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "402326 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999986.\n",
      "--a1-allele: 402326 assignments made.\n",
      "402326 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr1.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr2.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 2\n",
      "  --out dataset.b37.outliers-updated-chr2\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "441697 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999984.\n",
      "--a1-allele: 441697 assignments made.\n",
      "441697 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr2.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr3.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 3\n",
      "  --out dataset.b37.outliers-updated-chr3\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "376451 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999982.\n",
      "--a1-allele: 376451 assignments made.\n",
      "376451 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr3.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr4.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 4\n",
      "  --out dataset.b37.outliers-updated-chr4\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "383874 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999982.\n",
      "--a1-allele: 383874 assignments made.\n",
      "383874 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr4.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr5.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 5\n",
      "  --out dataset.b37.outliers-updated-chr5\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "333822 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999984.\n",
      "--a1-allele: 333822 assignments made.\n",
      "333822 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr5.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr6.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 6\n",
      "  --out dataset.b37.outliers-updated-chr6\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "352662 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99998.\n",
      "--a1-allele: 352662 assignments made.\n",
      "352662 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr6.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr7.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 7\n",
      "  --out dataset.b37.outliers-updated-chr7\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "309668 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999984.\n",
      "--a1-allele: 309668 assignments made.\n",
      "309668 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr7.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr8.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 8\n",
      "  --out dataset.b37.outliers-updated-chr8\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "292233 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999982.\n",
      "--a1-allele: 292233 assignments made.\n",
      "292233 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr8.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr9.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 9\n",
      "  --out dataset.b37.outliers-updated-chr9\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "227584 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999988.\n",
      "--a1-allele: 227584 assignments made.\n",
      "227584 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr9.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr10.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 10\n",
      "  --out dataset.b37.outliers-updated-chr10\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "273118 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999989.\n",
      "--a1-allele: 273118 assignments made.\n",
      "273118 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr10.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr11.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 11\n",
      "  --out dataset.b37.outliers-updated-chr11\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "265922 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999987.\n",
      "--a1-allele: 265922 assignments made.\n",
      "265922 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr11.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr12.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 12\n",
      "  --out dataset.b37.outliers-updated-chr12\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "254015 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999989.\n",
      "--a1-allele: 254015 assignments made.\n",
      "254015 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr12.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr13.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 13\n",
      "  --out dataset.b37.outliers-updated-chr13\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "198466 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999987.\n",
      "--a1-allele: 198466 assignments made.\n",
      "198466 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr13.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr14.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 14\n",
      "  --out dataset.b37.outliers-updated-chr14\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "173038 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "--a1-allele: 173038 assignments made.\n",
      "173038 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr14.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr15.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 15\n",
      "  --out dataset.b37.outliers-updated-chr15\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "149280 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99999.\n",
      "--a1-allele: 149280 assignments made.\n",
      "149280 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr15.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr16.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 16\n",
      "  --out dataset.b37.outliers-updated-chr16\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "162069 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999977.\n",
      "--a1-allele: 162069 assignments made.\n",
      "162069 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr16.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr17.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 17\n",
      "  --out dataset.b37.outliers-updated-chr17\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "140234 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999993.\n",
      "--a1-allele: 140234 assignments made.\n",
      "140234 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr17.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr18.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 18\n",
      "  --out dataset.b37.outliers-updated-chr18\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "152524 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999983.\n",
      "--a1-allele: 152524 assignments made.\n",
      "152524 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr18.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr19.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 19\n",
      "  --out dataset.b37.outliers-updated-chr19\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "119631 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999985.\n",
      "--a1-allele: 119631 assignments made.\n",
      "119631 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr19.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr20.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 20\n",
      "  --out dataset.b37.outliers-updated-chr20\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "119733 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999981.\n",
      "--a1-allele: 119733 assignments made.\n",
      "119733 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr20.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr21.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 21\n",
      "  --out dataset.b37.outliers-updated-chr21\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "76336 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999982.\n",
      "--a1-allele: 76336 assignments made.\n",
      "76336 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr21.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr22.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 22\n",
      "  --out dataset.b37.outliers-updated-chr22\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "72255 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999977.\n",
      "--a1-allele: 72255 assignments made.\n",
      "72255 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr22.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.17 64-bit (28 Apr 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to dataset.b37.outliers-updated-chr23.log.\n",
      "Options in effect:\n",
      "  --a1-allele Force-Allele1-dataset.b37.outliers-HRC.txt\n",
      "  --bfile dataset.b37.outliers-updated\n",
      "  --chr 23\n",
      "  --out dataset.b37.outliers-updated-chr23\n",
      "  --recode vcf\n",
      "\n",
      "257718 MB RAM detected; reserving 128859 MB for main workspace.\n",
      "159557 out of 5436495 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999997.\n",
      "--a1-allele: 159557 assignments made.\n",
      "159557 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--recode vcf to dataset.b37.outliers-updated-chr23.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat $path/run-plink.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-22T16:11:45.516710Z",
     "start_time": "2020-08-22T16:06:35.093239Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n",
      "sort -k1,1d -k2,2n\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# compress the vcf files to upload to Michigan Server\n",
    "for i in {1..22}\n",
    "do\n",
    "    vcf-sort $path/dataset.b37.outliers-updated-chr$i.vcf | bgzip -c > $path/dataset.b37-sorted-chr$i.vcf.gz\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Upload vcf.gz files to the Michigan server (phasing: SHAPE-IT; reference:HRC; mode: QC&imputation)\n",
    "\n",
    "Log into the Michigan Imputation Server site and click on Run -> Michigan Imputation Server from the menu located at the top.\n",
    "\n",
    "Select HRC r1.1 2016 reference panel, GRCh37/hg19 array built, 0.3 rsq filter, Eagle v2.4 as the phasing algoruthm, EUR population and \"Quality control& Imputation\" as the mode; you can also selecet AES 256 encryption. Upload vcf.gz files generated in the previous step and wait for a succesfull upload and initial QC.\n",
    "\n",
    "You will receive an email when the job is completed with the link and paswword to access the results. Download them."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## QC of the imputed genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-23T10:39:27.991381Z",
     "start_time": "2020-08-23T10:38:02.955506Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_1.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr1.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr1.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_2.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr2.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr2.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_3.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr3.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr3.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_4.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr4.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr4.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_5.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr5.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr5.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_6.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr6.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr6.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_7.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr7.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr7.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_8.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr8.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr8.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_9.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr9.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr9.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_10.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr10.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr10.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_11.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr11.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr11.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_12.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr12.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr12.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_13.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr13.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr13.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_14.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr14.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr14.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_15.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr15.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr15.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_16.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr16.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr16.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_17.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr17.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr17.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_18.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr18.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr18.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_19.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr19.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr19.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_20.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr20.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr20.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_21.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr21.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr21.info.gz  \n",
      "Archive:  /mnt/data/GWAS/output/task3_imputation/imputed_files/chr_22.zip\n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.vcf.gz  \n",
      "  inflating: /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.info.gz  \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Unzip the chromosomes files. Provide the password (in double quotes) supplied by the Michigan server\n",
    "for i in {1..22}\n",
    "do\n",
    "unzip -P \"HOfK3mXPunr4Xl\" $path/imputed_files/chr_$i.zip -d $path/imputed_files \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " **Extract genotype doses**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2020-08-23T15:45:29.274Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Extract genotype doses from vcf files and generate dosage files for PLINK software.\n",
    "# 1st parameter: path to extracted zip files (from previous step). Output files will be stored in this path\n",
    "# 2nd parameter: boolean indicating whereas the rsquare filter was applied during imputation\n",
    "bash scripts/extract_dose.sh $path/imputed_files true"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Generate a .fam file for PLINK from the chr22 vcf. Then update sex and pheno with the QCed fam from task 2.2 (i.e. dataset.b37.outliers)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:17:45.398598Z",
     "start_time": "2020-08-24T07:17:41.155149Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose\n",
      "  --vcf /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.vcf.gz\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "--vcf: 218k variants complete.\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose-temporary.bed +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose-temporary.bim +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose-temporary.fam\n",
      "written.\n",
      "218642 variants loaded from .bim file.\n",
      "496 people (0 males, 0 females, 496 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "218642 variants and 496 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.bed +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.bim +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "plink --vcf $path/imputed_files/chr22.dose.vcf.gz --make-bed --out $path/imputed_files/chr22.dose\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:17:45.413643Z",
     "start_time": "2020-08-24T07:17:45.403432Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HGX00096 HGX00096 0 0 0 -9\n",
      "HGX00097 HGX00097 0 0 0 -9\n",
      "HGX00099 HGX00099 0 0 0 -9\n",
      "HGX00100 HGX00100 0 0 0 -9\n",
      "HGX00101 HGX00101 0 0 0 -9\n",
      "HGX00102 HGX00102 0 0 0 -9\n",
      "HGX00103 HGX00103 0 0 0 -9\n",
      "HGX00105 HGX00105 0 0 0 -9\n",
      "HGX00106 HGX00106 0 0 0 -9\n",
      "HGX00107 HGX00107 0 0 0 -9\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/imputed_files/chr22.dose.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:17:52.050939Z",
     "start_time": "2020-08-24T07:17:51.925187Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose\n",
      "  --make-bed\n",
      "  --mpheno 4\n",
      "  --out /mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc\n",
      "  --pheno /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers.fam\n",
      "  --update-sex /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers.fam 3\n",
      "\n",
      "32127 MB RAM detected; reserving 16063 MB for main workspace.\n",
      "218642 variants loaded from .bim file.\n",
      "496 people (0 males, 0 females, 496 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc.nosex\n",
      ".\n",
      "496 phenotype values present after --pheno.\n",
      "--update-sex: 496 people updated.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "218642 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc.bed +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc.bim +\n",
      "/mnt/data/GWAS/output/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "plink --bfile $path/imputed_files/chr22.dose --make-bed --update-sex /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers.fam 3 --pheno /mnt/data/GWAS/output/task2.2_stratification/intermediate_datasets/dataset.b37.outliers.fam --mpheno 4 --out  $path/imputed_files/chr22.dose.for.assoc\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:17:54.154521Z",
     "start_time": "2020-08-24T07:17:54.135712Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HGX00096 HGX00096 0 0 1 1\n",
      "HGX00097 HGX00097 0 0 2 1\n",
      "HGX00099 HGX00099 0 0 2 1\n",
      "HGX00100 HGX00100 0 0 2 1\n",
      "HGX00101 HGX00101 0 0 1 1\n",
      "HGX00102 HGX00102 0 0 2 1\n",
      "HGX00103 HGX00103 0 0 1 1\n",
      "HGX00105 HGX00105 0 0 1 1\n",
      "HGX00106 HGX00106 0 0 2 1\n",
      "HGX00107 HGX00107 0 0 1 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/imputed_files/chr22.dose.for.assoc.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:17:56.067109Z",
     "start_time": "2020-08-24T07:17:56.053644Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "rm $path/imputed_files/chr22.dose.fam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**For the next step you need the following files:**\n",
    "- chr22.dose.for.assoc.fam\n",
    "- chri.dose.rsq.DS.vcf.gz"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python3.6",
   "language": "python",
   "name": "python3.6"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
