{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Population Stratification \n",
    "\n",
    "This is the presence of multiple subpopulations (e.g., individuals with different ethnic background) in a study. Because allele frequencies can differ between subpopulations, population stratification can lead to false positive associations and/or mask true associations. \n",
    "\n",
    "To complete this task it is necessary to have generated the bfile 'dataset.b37_IBD' and the file 'indepSNP.prune.in' from the previous tutorial (Task_2.1_QC). This bfile is going to checked for population stratification using data from the 1000 Genomes Project. Individuals with a non-European ethnic background will be removed. 1000 Genomes data need to be processed before hand.\n",
    "\n",
    "Furthermore, this guide will generate a covariate file which helps to adust for remaining population stratification within the European subjects.\n",
    "\n",
    "This protocol follows the tutorial explained in Marees AT, de Kluiver H, Stringer S, et al. A tutorial on conducting genome-wide association studies: Quality control and statistical analysis. Int J Methods Psychiatr Res. 2018;27(2):e1608.doi:10.1002/mpr.1608"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1000 Genome data preparation\n",
    "\n",
    "Data from the 1000 Genomes Project need to be processed before used for population stratification. \n",
    "\n",
    "To complete this task it is necessary to have generated the bfile 'dataset.b37_IBD' from the previous tutorial (Task_2_QC)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:54:24.831494Z",
     "start_time": "2020-08-21T13:54:24.827425Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/build37/task2_QC\n",
      "env: intpath=/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets\n",
      "env: ref_files=/mnt/data/GWAS/ref_files/Phase3_v5\n"
     ]
    }
   ],
   "source": [
    "%env path=/mnt/data/GWAS/output/build37/task2_QC\n",
    "%env intpath=/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets\n",
    "%env ref_files=/mnt/data/GWAS/ref_files/Phase3_v5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# Import functions for visualisation\n",
    "source(\"scripts/QC_functions_task2.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T10:27:36.205521Z",
     "start_time": "2020-08-21T10:27:35.118676Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1:10177\n",
      "1:11008\n",
      "1:11012\n",
      "1:13110\n",
      "rs201725126\n",
      "rs200579949\n",
      "1:13273\n",
      "1:14464\n",
      "1:14599\n",
      "1:14604\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "# Get a list of SNPs from your working dataset\n",
    "awk '{print $2}' $intpath/dataset.b37.IBD.bim > $path/mySNPlist.txt\n",
    "head $path/mySNPlist.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:11:32.264601Z",
     "start_time": "2020-08-21T10:59:20.850730Z"
    },
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr1.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr1\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr1.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr1-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr1-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr1-temporary.fam written.\n",
      "497923 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999861.\n",
      "497923 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr1.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr1.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr2.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr2\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr2.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr2-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr2-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr2-temporary.fam written.\n",
      "540431 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999857.\n",
      "540431 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr2.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr2.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr3.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr3\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr3.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr3-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr3-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr3-temporary.fam written.\n",
      "462062 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "462062 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr3.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr3.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr4.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr4\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr4.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr4-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr4-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr4-temporary.fam written.\n",
      "474260 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999847.\n",
      "474260 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr4.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr4.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr5.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr5\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr5.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr5-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr5-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr5-temporary.fam written.\n",
      "410091 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999849.\n",
      "410091 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr5.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr5.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr6.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr6\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr6.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr6-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr6-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr6-temporary.fam written.\n",
      "440932 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999843.\n",
      "440932 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr6.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr6.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr7\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr7.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr7-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr7-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr7-temporary.fam written.\n",
      "384634 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999854.\n",
      "384634 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr7.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr7.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr8.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr8\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr8.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr8-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr8-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr8-temporary.fam written.\n",
      "354504 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999832.\n",
      "354504 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr8.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr8.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr9.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr9\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr9.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr9-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr9-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr9-temporary.fam written.\n",
      "281556 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999837.\n",
      "281556 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr9.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr9.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr10.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr10\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr10.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr10-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr10-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr10-temporary.fam written.\n",
      "336853 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999856.\n",
      "336853 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr10.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr10.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr11.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr11\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr11.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr11-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr11-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr11-temporary.fam written.\n",
      "328068 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999851.\n",
      "328068 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr11.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr11.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr12.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr12\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr12.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr12-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr12-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr12-temporary.fam written.\n",
      "307031 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99985.\n",
      "307031 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr12.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr12.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr13.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr13\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr13.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr13-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr13-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr13-temporary.fam written.\n",
      "243551 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999859.\n",
      "243551 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr13.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr13.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr14.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr14\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr14.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr14-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr14-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr14-temporary.fam written.\n",
      "212995 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999849.\n",
      "212995 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr14.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr14.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr15.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr15\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr15.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr15-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr15-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr15-temporary.fam written.\n",
      "185833 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999847.\n",
      "185833 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr15.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr15.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr16.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr16\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr16.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr16-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr16-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr16-temporary.fam written.\n",
      "198791 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999813.\n",
      "198791 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr16.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr16.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr17.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr17\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr17.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr17-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr17-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr17-temporary.fam written.\n",
      "176044 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999858.\n",
      "176044 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr17.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr17.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr18.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr18\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr18.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr18-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr18-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr18-temporary.fam written.\n",
      "186082 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999853.\n",
      "186082 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr18.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr18.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr19.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr19\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr19.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr19-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr19-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr19-temporary.fam written.\n",
      "154398 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999834.\n",
      "154398 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr19.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr19.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr20.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr20\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr20.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr20-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr20-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr20-temporary.fam written.\n",
      "144774 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999857.\n",
      "144774 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr20.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr20.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr21.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr21\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr21.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr21-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr21-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr21-temporary.fam written.\n",
      "93739 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999823.\n",
      "93739 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr21.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr21.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr22.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.chr22\n",
      "  --vcf /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr22.recode.vcf\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--vcf: /mnt/data/GWAS/output/build37/task2_QC/1KG.chr22-temporary.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr22-temporary.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr22-temporary.fam written.\n",
      "91208 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99982.\n",
      "91208 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.chr22.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr22.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr1\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 497923 out of a possible 6467265 Sites\n",
      "Run Time = 643.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr2\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 540431 out of a possible 7080712 Sites\n",
      "Run Time = 696.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr3\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 462062 out of a possible 5831616 Sites\n",
      "Run Time = 585.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr4\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 474260 out of a possible 5731977 Sites\n",
      "Run Time = 591.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr5\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 410091 out of a possible 5265131 Sites\n",
      "Run Time = 527.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr6\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 440932 out of a possible 5023483 Sites\n",
      "Run Time = 535.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr7\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 384634 out of a possible 4716205 Sites\n",
      "Run Time = 485.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr8\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 354504 out of a possible 4596555 Sites\n",
      "Run Time = 458.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr9\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 281556 out of a possible 3560268 Sites\n",
      "Run Time = 362.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr10\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 336853 out of a possible 3991662 Sites\n",
      "Run Time = 420.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr11\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 328068 out of a possible 4045150 Sites\n",
      "Run Time = 416.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr12\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 307031 out of a possible 3867557 Sites\n",
      "Run Time = 393.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr13\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 243551 out of a possible 2857612 Sites\n",
      "Run Time = 304.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr14\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 212995 out of a possible 2654741 Sites\n",
      "Run Time = 273.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr15\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 185833 out of a possible 2424342 Sites\n",
      "Run Time = 245.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr16\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 198791 out of a possible 2697571 Sites\n",
      "Run Time = 266.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr17\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 176044 out of a possible 2328930 Sites\n",
      "Run Time = 234.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr18\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 186082 out of a possible 2266920 Sites\n",
      "Run Time = 237.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr19\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 154398 out of a possible 1832210 Sites\n",
      "Run Time = 198.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr20\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 144774 out of a possible 1812578 Sites\n",
      "Run Time = 189.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr21\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 93739 out of a possible 1105434 Sites\n",
      "Run Time = 122.00 seconds\n",
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /mnt/data/GWAS/ref_files/Phase3_v5/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz\n",
      "\t--out /mnt/data/GWAS/output/build37/task2_QC/1KG.vcf.chr22\n",
      "\t--recode\n",
      "\t--snps /mnt/data/GWAS/output/build37/task2_QC/mySNPlist.txt\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 91208 out of a possible 1103411 Sites\n",
      "Run Time = 120.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Creates a recode vcf per chromosome, using as input 1000 Genomes compressed vcf file per chr and previous SNPs list.\n",
    "# Then plink converts them to bfiles. \n",
    "# This step takes long to be executed; you can skip this one and go to the next step\n",
    "for i in {1..22}\n",
    "do\n",
    "vcftools --gzvcf $ref_files/ALL.chr$i.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.clean.vcf.gz --snps $path/mySNPlist.txt --recode --out $path/1KG.vcf.chr$i\n",
    "plink --vcf $path/1KG.vcf.chr$i.recode.vcf --make-bed --out $path/1KG.chr$i\n",
    "rm $path/1KG.vcf.chr$i.recode.vcf\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:50:12.421299Z",
     "start_time": "2020-08-21T13:50:12.360491Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Creates a file with the list of bfiles (.bim, .bed, .fam) outputed in the previous step\n",
    "# 1st-3rd parameter: path to previous plink files\n",
    "# 4th parameter: output file\n",
    "for i in {2..22}\n",
    "do\n",
    "echo $path/1KG.chr$i.bed $path/1KG.chr$i.bim $path/1KG.chr$i.fam >> $path/1KG.outlist.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T13:52:57.486015Z",
     "start_time": "2020-08-21T13:50:21.665190Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1KG.all.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/1KG.chr1\n",
      "  --make-bed\n",
      "  --merge-list /mnt/data/GWAS/output/build37/task2_QC/1KG.outlist.txt\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1KG.all\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "Performing single-pass merge (2504 people, 6505760 variants).\n",
      "Merged fileset written to                     \n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all-merge.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all-merge.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all-merge.fam .\n",
      "6505760 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "6505760 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1KG.all.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1KG.all.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Merge all bfiles into a single one\n",
    "plink --bfile $path/1KG.chr1 --make-bed --merge-list $path/1KG.outlist.txt --out $path/1KG.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-21T14:14:02.348522Z",
     "start_time": "2020-08-21T14:14:02.088747Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# remove individual chr plink files\n",
    "for i in {2..22}\n",
    "do\n",
    "rm $path/1KG.chr$i.*\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge the working dataset and 1000 Genomes data sets \n",
    "\n",
    "Prior to merging 1000 Genomes data with the HapMap data we want to make sure that the files are mergeable, for this we conduct 5 steps:\n",
    "1. Extract the variants present in one dataset from the other\n",
    "2. Resolve strand issues.\n",
    "3. Remove the SNPs which after the previous two steps still differ between datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.IBD\n",
      "  --extract /mnt/data/GWAS/output/build37/task2_QC/1KG.all.bim\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS\n",
      "  --recode\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--extract: 6505760 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99997.\n",
      "6505760 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--recode ped to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.ped\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS.map\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Extract the variants present in 1000 Genomes dataset from the working dataset.\n",
    "# awk '{print$2}' $intpath/1kG.all.bim > $path/1kG_MDS_SNPs.txt\n",
    "plink --bfile $intpath/dataset.b37.IBD --extract $path/1KG.all.bim --recode --make-bed --out $intpath/dataset.b37_MDS\n",
    "# The datasets now contain the exact same variants.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set reference allele"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj.log.\n",
      "Options in effect:\n",
      "  --a1-allele /mnt/data/GWAS/output/build37/task2_QC/1kg_ref-list.txt\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505760 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99997.\n",
      "--a1-allele: 6505760 assignments made.\n",
      "6505760 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: Impossible A1 allele assignment for variant rs147619543.\n",
      "Warning: Impossible A1 allele assignment for variant rs137984339.\n",
      "Warning: Impossible A1 allele assignment for variant rs55768108.\n",
      "Warning: Impossible A1 allele assignment for variant rs60360048.\n",
      "Warning: Impossible A1 allele assignment for variant rs201618615.\n",
      "Warning: Impossible A1 allele assignment for variant rs60203926.\n",
      "Warning: Impossible A1 allele assignment for variant rs111369971.\n",
      "Warning: Impossible A1 allele assignment for variant rs11211275.\n",
      "Warning: Impossible A1 allele assignment for variant rs11375147.\n",
      "Warning: Impossible A1 allele assignment for variant rs34094074.\n",
      "Warning: Impossible A1 allele assignment for variant rs59508386.\n",
      "Warning: Impossible A1 allele assignment for variant rs57226191.\n",
      "Warning: Impossible A1 allele assignment for variant rs71311970.\n",
      "Warning: Impossible A1 allele assignment for variant rs139796930.\n",
      "Warning: Impossible A1 allele assignment for variant rs5776404.\n",
      "Warning: Impossible A1 allele assignment for variant rs11399953.\n",
      "Warning: Impossible A1 allele assignment for variant rs137892494.\n",
      "Warning: Impossible A1 allele assignment for variant rs71093259.\n",
      "Warning: Impossible A1 allele assignment for variant rs71565447.\n",
      "Warning: Impossible A1 allele assignment for variant rs34612182.\n",
      "Warning: Impossible A1 allele assignment for variant rs35359058.\n",
      "Warning: Impossible A1 allele assignment for variant rs71721827.\n",
      "Warning: Impossible A1 allele assignment for variant rs55924341.\n",
      "Warning: Impossible A1 allele assignment for variant rs145912288.\n",
      "Warning: Impossible A1 allele assignment for variant rs201659170.\n",
      "Warning: Impossible A1 allele assignment for variant rs11408370.\n",
      "Warning: Impossible A1 allele assignment for variant rs369838070.\n",
      "Warning: Impossible A1 allele assignment for variant rs35428718.\n",
      "Warning: Impossible A1 allele assignment for variant rs34109939.\n",
      "Warning: Impossible A1 allele assignment for variant rs148153906.\n",
      "Warning: Impossible A1 allele assignment for variant rs71407496.\n",
      "Warning: Impossible A1 allele assignment for variant rs146528263.\n",
      "Warning: Impossible A1 allele assignment for variant rs147872509.\n",
      "Warning: Impossible A1 allele assignment for variant rs3835759.\n",
      "Warning: Impossible A1 allele assignment for variant rs35178225.\n",
      "Warning: Impossible A1 allele assignment for variant rs5831472.\n",
      "Warning: Impossible A1 allele assignment for variant rs72339534.\n",
      "Warning: Impossible A1 allele assignment for variant rs71390158.\n",
      "Warning: Impossible A1 allele assignment for variant rs199685963.\n",
      "Warning: Impossible A1 allele assignment for variant rs149286053.\n",
      "Warning: Impossible A1 allele assignment for variant rs11442369.\n",
      "Warning: Impossible A1 allele assignment for variant rs35496875.\n",
      "Warning: Impossible A1 allele assignment for variant rs113282615.\n",
      "Warning: Impossible A1 allele assignment for variant rs72077160.\n",
      "Warning: Impossible A1 allele assignment for variant rs143937474.\n",
      "Warning: Impossible A1 allele assignment for variant rs201407212.\n",
      "Warning: Impossible A1 allele assignment for variant rs35965969.\n",
      "Warning: Impossible A1 allele assignment for variant rs67029113.\n",
      "Warning: Impossible A1 allele assignment for variant rs67949386.\n",
      "Warning: Impossible A1 allele assignment for variant rs113258194.\n",
      "Warning: Impossible A1 allele assignment for variant rs150165882.\n",
      "Warning: Impossible A1 allele assignment for variant rs34537206.\n",
      "Warning: Impossible A1 allele assignment for variant rs10668200.\n",
      "Warning: Impossible A1 allele assignment for variant rs34362926.\n",
      "Warning: Impossible A1 allele assignment for variant rs141935407.\n",
      "Warning: Impossible A1 allele assignment for variant rs1966896.\n",
      "Warning: Impossible A1 allele assignment for variant rs200822642.\n",
      "Warning: Impossible A1 allele assignment for variant rs141821078.\n",
      "Warning: Impossible A1 allele assignment for variant rs58612314.\n",
      "Warning: Impossible A1 allele assignment for variant rs35159863.\n",
      "Warning: Impossible A1 allele assignment for variant rs11378635.\n",
      "Warning: Impossible A1 allele assignment for variant rs34454560.\n",
      "Warning: Impossible A1 allele assignment for variant rs9814775.\n",
      "Warning: Impossible A1 allele assignment for variant rs78490723.\n",
      "Warning: Impossible A1 allele assignment for variant rs35732987.\n",
      "Warning: Impossible A1 allele assignment for variant rs10663307.\n",
      "Warning: Impossible A1 allele assignment for variant rs370308267.\n",
      "Warning: Impossible A1 allele assignment for variant rs200697504.\n",
      "Warning: Impossible A1 allele assignment for variant rs34562094.\n",
      "Warning: Impossible A1 allele assignment for variant rs61324533.\n",
      "Warning: Impossible A1 allele assignment for variant rs61564785.\n",
      "Warning: Impossible A1 allele assignment for variant rs34762839.\n",
      "Warning: Impossible A1 allele assignment for variant rs35882455.\n",
      "Warning: Impossible A1 allele assignment for variant rs112726145.\n",
      "Warning: Impossible A1 allele assignment for variant rs7436539.\n",
      "Warning: Impossible A1 allele assignment for variant rs148548225.\n",
      "Warning: Impossible A1 allele assignment for variant rs35746929.\n",
      "Warning: Impossible A1 allele assignment for variant rs60463614.\n",
      "Warning: Impossible A1 allele assignment for variant rs71654862.\n",
      "Warning: Impossible A1 allele assignment for variant rs34812158.\n",
      "Warning: Impossible A1 allele assignment for variant rs60101939.\n",
      "Warning: Impossible A1 allele assignment for variant rs10013793.\n",
      "Warning: Impossible A1 allele assignment for variant rs113842049.\n",
      "Warning: Impossible A1 allele assignment for variant rs71964935.\n",
      "Warning: Impossible A1 allele assignment for variant rs10690013.\n",
      "Warning: Impossible A1 allele assignment for variant rs70961801.\n",
      "Warning: Impossible A1 allele assignment for variant rs58890122.\n",
      "Warning: Impossible A1 allele assignment for variant rs57536077.\n",
      "Warning: Impossible A1 allele assignment for variant rs59300393.\n",
      "Warning: Impossible A1 allele assignment for variant rs60465499.\n",
      "Warning: Impossible A1 allele assignment for variant rs202183072.\n",
      "Warning: Impossible A1 allele assignment for variant rs60484575.\n",
      "Warning: Impossible A1 allele assignment for variant rs71584391.\n",
      "Warning: Impossible A1 allele assignment for variant rs34497958.\n",
      "Warning: Impossible A1 allele assignment for variant rs57216995.\n",
      "Warning: Impossible A1 allele assignment for variant rs34278906.\n",
      "Warning: Impossible A1 allele assignment for variant rs376653011.\n",
      "Warning: Impossible A1 allele assignment for variant rs146771059.\n",
      "Warning: Impossible A1 allele assignment for variant rs60755557.\n",
      "Warning: Impossible A1 allele assignment for variant rs33999637.\n",
      "Warning: Impossible A1 allele assignment for variant rs3066161.\n",
      "Warning: Impossible A1 allele assignment for variant rs60010478.\n",
      "Warning: Impossible A1 allele assignment for variant rs70938251.\n",
      "Warning: Impossible A1 allele assignment for variant rs5866808.\n",
      "Warning: Impossible A1 allele assignment for variant rs149654848.\n",
      "Warning: Impossible A1 allele assignment for variant rs248382.\n",
      "Warning: Impossible A1 allele assignment for variant rs112912857.\n",
      "Warning: Impossible A1 allele assignment for variant rs76139278.\n",
      "Warning: Impossible A1 allele assignment for variant rs370121313.\n",
      "Warning: Impossible A1 allele assignment for variant rs58688712.\n",
      "Warning: Impossible A1 allele assignment for variant rs35540828.\n",
      "Warning: Impossible A1 allele assignment for variant rs200293348.\n",
      "Warning: Impossible A1 allele assignment for variant rs10651764.\n",
      "Warning: Impossible A1 allele assignment for variant rs10671428.\n",
      "Warning: Impossible A1 allele assignment for variant rs34768415.\n",
      "Warning: Impossible A1 allele assignment for variant rs150215882.\n",
      "Warning: Impossible A1 allele assignment for variant rs58893321.\n",
      "Warning: Impossible A1 allele assignment for variant rs199959526.\n",
      "Warning: Impossible A1 allele assignment for variant rs201191015.\n",
      "Warning: Impossible A1 allele assignment for variant rs60383041.\n",
      "Warning: Impossible A1 allele assignment for variant rs202034055.\n",
      "Warning: Impossible A1 allele assignment for variant rs199636827.\n",
      "Warning: Impossible A1 allele assignment for variant rs200589655.\n",
      "Warning: Impossible A1 allele assignment for variant rs143772814.\n",
      "Warning: Impossible A1 allele assignment for variant rs66657443.\n",
      "Warning: Impossible A1 allele assignment for variant rs11449422.\n",
      "Warning: Impossible A1 allele assignment for variant rs78046670.\n",
      "Warning: Impossible A1 allele assignment for variant rs71536404.\n",
      "Warning: Impossible A1 allele assignment for variant rs3035346.\n",
      "Warning: Impossible A1 allele assignment for variant rs12200694.\n",
      "Warning: Impossible A1 allele assignment for variant rs58774833.\n",
      "Warning: Impossible A1 allele assignment for variant rs71541270.\n",
      "Warning: Impossible A1 allele assignment for variant rs3054945.\n",
      "Warning: Impossible A1 allele assignment for variant rs34521551.\n",
      "Warning: Impossible A1 allele assignment for variant rs144366744.\n",
      "Warning: Impossible A1 allele assignment for variant rs200474568.\n",
      "Warning: Impossible A1 allele assignment for variant rs148588315.\n",
      "Warning: Impossible A1 allele assignment for variant rs57690424.\n",
      "Warning: Impossible A1 allele assignment for variant rs35144811.\n",
      "Warning: Impossible A1 allele assignment for variant rs9371847.\n",
      "Warning: Impossible A1 allele assignment for variant rs11377175.\n",
      "Warning: Impossible A1 allele assignment for variant rs201098747.\n",
      "Warning: Impossible A1 allele assignment for variant rs35003890.\n",
      "Warning: Impossible A1 allele assignment for variant rs200899071.\n",
      "Warning: Impossible A1 allele assignment for variant rs35866128.\n",
      "Warning: Impossible A1 allele assignment for variant rs5883347.\n",
      "Warning: Impossible A1 allele assignment for variant rs5884215.\n",
      "Warning: Impossible A1 allele assignment for variant rs149316964.\n",
      "Warning: Impossible A1 allele assignment for variant rs200645214.\n",
      "Warning: Impossible A1 allele assignment for variant rs145164759.\n",
      "Warning: Impossible A1 allele assignment for variant rs150201843.\n",
      "Warning: Impossible A1 allele assignment for variant rs151177697.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855925.\n",
      "Warning: Impossible A1 allele assignment for variant rs3070857.\n",
      "Warning: Impossible A1 allele assignment for variant rs148802556.\n",
      "Warning: Impossible A1 allele assignment for variant rs141623426.\n",
      "Warning: Impossible A1 allele assignment for variant rs113055548.\n",
      "Warning: Impossible A1 allele assignment for variant rs147693269.\n",
      "Warning: Impossible A1 allele assignment for variant rs5889534.\n",
      "Warning: Impossible A1 allele assignment for variant rs5890525.\n",
      "Warning: Impossible A1 allele assignment for variant rs59255004.\n",
      "Warning: Impossible A1 allele assignment for variant rs143645350.\n",
      "Warning: Impossible A1 allele assignment for variant rs34188615.\n",
      "Warning: Impossible A1 allele assignment for variant rs71264185.\n",
      "Warning: Impossible A1 allele assignment for variant rs5892663.\n",
      "Warning: Impossible A1 allele assignment for variant rs199976662.\n",
      "Warning: Impossible A1 allele assignment for variant rs200536442.\n",
      "Warning: Impossible A1 allele assignment for variant rs200998873.\n",
      "Warning: Impossible A1 allele assignment for variant rs55953934.\n",
      "Warning: Impossible A1 allele assignment for variant rs143602363.\n",
      "Warning: Impossible A1 allele assignment for variant rs3073777.\n",
      "Warning: Impossible A1 allele assignment for variant rs34378799.\n",
      "Warning: Impossible A1 allele assignment for variant rs34014791.\n",
      "Warning: Impossible A1 allele assignment for variant rs137860744.\n",
      "Warning: Impossible A1 allele assignment for variant rs59083583.\n",
      "Warning: Impossible A1 allele assignment for variant rs11389375.\n",
      "Warning: Impossible A1 allele assignment for variant rs74176994.\n",
      "Warning: Impossible A1 allele assignment for variant rs55721416.\n",
      "Warning: Impossible A1 allele assignment for variant rs11380003.\n",
      "Warning: Impossible A1 allele assignment for variant rs11139348.\n",
      "Warning: Impossible A1 allele assignment for variant rs145306304.\n",
      "Warning: Impossible A1 allele assignment for variant rs10546076.\n",
      "Warning: Impossible A1 allele assignment for variant rs34428620.\n",
      "Warning: Impossible A1 allele assignment for variant rs200278001.\n",
      "Warning: Impossible A1 allele assignment for variant rs10544831.\n",
      "Warning: Impossible A1 allele assignment for variant rs148981849.\n",
      "Warning: Impossible A1 allele assignment for variant rs67559361.\n",
      "Warning: Impossible A1 allele assignment for variant rs200516560.\n",
      "Warning: Impossible A1 allele assignment for variant rs200148222.\n",
      "Warning: Impossible A1 allele assignment for variant rs202111789.\n",
      "Warning: Impossible A1 allele assignment for variant rs146026552.\n",
      "Warning: Impossible A1 allele assignment for variant rs5783377.\n",
      "Warning: Impossible A1 allele assignment for variant rs60225229.\n",
      "Warning: Impossible A1 allele assignment for variant rs141345608.\n",
      "Warning: Impossible A1 allele assignment for variant rs35017955.\n",
      "Warning: Impossible A1 allele assignment for variant rs34261017.\n",
      "Warning: Impossible A1 allele assignment for variant rs111532670.\n",
      "Warning: Impossible A1 allele assignment for variant rs147050495.\n",
      "Warning: Impossible A1 allele assignment for variant rs11281146.\n",
      "Warning: Impossible A1 allele assignment for variant rs10598951.\n",
      "Warning: Impossible A1 allele assignment for variant rs147482924.\n",
      "Warning: Impossible A1 allele assignment for variant rs55976849.\n",
      "Warning: Impossible A1 allele assignment for variant rs10547535.\n",
      "Warning: Impossible A1 allele assignment for variant rs113540051.\n",
      "Warning: Impossible A1 allele assignment for variant rs35308664.\n",
      "Warning: Impossible A1 allele assignment for variant rs34446072.\n",
      "Warning: Impossible A1 allele assignment for variant rs35129198.\n",
      "Warning: Impossible A1 allele assignment for variant rs72240657.\n",
      "Warning: Impossible A1 allele assignment for variant rs35692523.\n",
      "Warning: Impossible A1 allele assignment for variant rs61026944.\n",
      "Warning: Impossible A1 allele assignment for variant rs34730979.\n",
      "Warning: Impossible A1 allele assignment for variant rs147359460.\n",
      "Warning: Impossible A1 allele assignment for variant rs10676423.\n",
      "Warning: Impossible A1 allele assignment for variant rs56263226.\n",
      "Warning: Impossible A1 allele assignment for variant rs145284642.\n",
      "Warning: Impossible A1 allele assignment for variant rs5792633.\n",
      "Warning: Impossible A1 allele assignment for variant rs71036061.\n",
      "Warning: Impossible A1 allele assignment for variant rs5794069.\n",
      "Warning: Impossible A1 allele assignment for variant rs35255367.\n",
      "Warning: Impossible A1 allele assignment for variant rs11389863.\n",
      "Warning: Impossible A1 allele assignment for variant rs5795305.\n",
      "Warning: Impossible A1 allele assignment for variant rs60885762.\n",
      "Warning: Impossible A1 allele assignment for variant rs7946045.\n",
      "Warning: Impossible A1 allele assignment for variant rs200765236.\n",
      "Warning: Impossible A1 allele assignment for variant rs139159553.\n",
      "Warning: Impossible A1 allele assignment for variant rs11410265.\n",
      "Warning: Impossible A1 allele assignment for variant rs200231897.\n",
      "Warning: Impossible A1 allele assignment for variant rs145053170.\n",
      "Warning: Impossible A1 allele assignment for variant rs34967117.\n",
      "Warning: Impossible A1 allele assignment for variant rs148744178.\n",
      "Warning: Impossible A1 allele assignment for variant rs140807061.\n",
      "Warning: Impossible A1 allele assignment for variant rs34990059.\n",
      "Warning: Impossible A1 allele assignment for variant rs369621491.\n",
      "Warning: Impossible A1 allele assignment for variant rs56808655.\n",
      "Warning: Impossible A1 allele assignment for variant rs142208138.\n",
      "Warning: Impossible A1 allele assignment for variant rs11411094.\n",
      "Warning: Impossible A1 allele assignment for variant rs71076684.\n",
      "Warning: Impossible A1 allele assignment for variant rs10681665.\n",
      "Warning: Impossible A1 allele assignment for variant rs60974421.\n",
      "Warning: Impossible A1 allele assignment for variant rs58817432.\n",
      "Warning: Impossible A1 allele assignment for variant rs71087101.\n",
      "Warning: Impossible A1 allele assignment for variant rs36121482.\n",
      "Warning: Impossible A1 allele assignment for variant rs112398485.\n",
      "Warning: Impossible A1 allele assignment for variant rs35136338.\n",
      "Warning: Impossible A1 allele assignment for variant rs35814981.\n",
      "Warning: Impossible A1 allele assignment for variant rs148595414.\n",
      "Warning: Impossible A1 allele assignment for variant rs3075437.\n",
      "Warning: Impossible A1 allele assignment for variant rs200381946.\n",
      "Warning: Impossible A1 allele assignment for variant rs9530195.\n",
      "Warning: Impossible A1 allele assignment for variant rs35720894.\n",
      "Warning: Impossible A1 allele assignment for variant rs60666691.\n",
      "Warning: Impossible A1 allele assignment for variant rs34703149.\n",
      "Warning: Impossible A1 allele assignment for variant rs138141946.\n",
      "Warning: Impossible A1 allele assignment for variant rs36196814.\n",
      "Warning: Impossible A1 allele assignment for variant rs17523478.\n",
      "Warning: Impossible A1 allele assignment for variant rs61400710.\n",
      "Warning: Impossible A1 allele assignment for variant rs34317236.\n",
      "Warning: Impossible A1 allele assignment for variant rs58647073.\n",
      "Warning: Impossible A1 allele assignment for variant rs148705068.\n",
      "Warning: Impossible A1 allele assignment for variant rs11430120.\n",
      "Warning: Impossible A1 allele assignment for variant rs201503456.\n",
      "Warning: Impossible A1 allele assignment for variant rs17108655.\n",
      "Warning: Impossible A1 allele assignment for variant rs140845066.\n",
      "Warning: Impossible A1 allele assignment for variant rs111864429.\n",
      "Warning: Impossible A1 allele assignment for variant rs149532947.\n",
      "Warning: Impossible A1 allele assignment for variant rs58321309.\n",
      "Warning: Impossible A1 allele assignment for variant rs11379794.\n",
      "Warning: Impossible A1 allele assignment for variant rs71464700.\n",
      "Warning: Impossible A1 allele assignment for variant rs79779753.\n",
      "Warning: Impossible A1 allele assignment for variant rs8006262.\n",
      "Warning: Impossible A1 allele assignment for variant rs113786826.\n",
      "Warning: Impossible A1 allele assignment for variant rs79697112.\n",
      "Warning: Impossible A1 allele assignment for variant rs10686797.\n",
      "Warning: Impossible A1 allele assignment for variant rs34546417.\n",
      "Warning: Impossible A1 allele assignment for variant rs35771863.\n",
      "Warning: Impossible A1 allele assignment for variant rs112038713.\n",
      "Warning: Impossible A1 allele assignment for variant rs34854915.\n",
      "Warning: Impossible A1 allele assignment for variant rs201825062.\n",
      "Warning: Impossible A1 allele assignment for variant rs58300596.\n",
      "Warning: Impossible A1 allele assignment for variant rs71280728.\n",
      "Warning: Impossible A1 allele assignment for variant rs59912939.\n",
      "Warning: Impossible A1 allele assignment for variant rs374142105.\n",
      "Warning: Impossible A1 allele assignment for variant rs35769927.\n",
      "Warning: Impossible A1 allele assignment for variant rs58851793.\n",
      "Warning: Impossible A1 allele assignment for variant rs2521331.\n",
      "Warning: Impossible A1 allele assignment for variant rs71378042.\n",
      "Warning: Impossible A1 allele assignment for variant rs5816868.\n",
      "Warning: Impossible A1 allele assignment for variant rs111488863.\n",
      "Warning: Impossible A1 allele assignment for variant rs7188256.\n",
      "Warning: Impossible A1 allele assignment for variant rs6499785.\n",
      "Warning: Impossible A1 allele assignment for variant rs56031703.\n",
      "Warning: Impossible A1 allele assignment for variant rs142646570.\n",
      "Warning: Impossible A1 allele assignment for variant rs71382034.\n",
      "Warning: Impossible A1 allele assignment for variant rs112542398.\n",
      "Warning: Impossible A1 allele assignment for variant rs60440735.\n",
      "Warning: Impossible A1 allele assignment for variant rs71134239.\n",
      "Warning: Impossible A1 allele assignment for variant rs150906505.\n",
      "Warning: Impossible A1 allele assignment for variant rs58612689.\n",
      "Warning: Impossible A1 allele assignment for variant rs149264208.\n",
      "Warning: Impossible A1 allele assignment for variant rs142032070.\n",
      "Warning: Impossible A1 allele assignment for variant rs58183353.\n",
      "Warning: Impossible A1 allele assignment for variant rs147258226.\n",
      "Warning: Impossible A1 allele assignment for variant rs201838394.\n",
      "Warning: Impossible A1 allele assignment for variant rs71172027.\n",
      "Warning: Impossible A1 allele assignment for variant rs34990450.\n",
      "Warning: Impossible A1 allele assignment for variant rs141406346.\n",
      "Warning: Impossible A1 allele assignment for variant rs11441702.\n",
      "Warning: Impossible A1 allele assignment for variant rs146092313.\n",
      "Warning: Impossible A1 allele assignment for variant rs137914150.\n",
      "Warning: Impossible A1 allele assignment for variant rs71169322.\n",
      "Warning: Impossible A1 allele assignment for variant rs11373675.\n",
      "Warning: Impossible A1 allele assignment for variant rs146296467.\n",
      "Warning: Impossible A1 allele assignment for variant rs137928326.\n",
      "Warning: Impossible A1 allele assignment for variant rs34699436.\n",
      "Warning: Impossible A1 allele assignment for variant rs35954284.\n",
      "Warning: Impossible A1 allele assignment for variant rs146055611.\n",
      "Warning: Impossible A1 allele assignment for variant rs112839858.\n",
      "Warning: Impossible A1 allele assignment for variant rs67204288.\n",
      "Warning: Impossible A1 allele assignment for variant rs35866247.\n",
      "Warning: Impossible A1 allele assignment for variant rs35299917.\n",
      "Warning: Impossible A1 allele assignment for variant rs71168622.\n",
      "Warning: Impossible A1 allele assignment for variant rs60779505.\n",
      "Warning: Impossible A1 allele assignment for variant rs56893426.\n",
      "Warning: Impossible A1 allele assignment for variant rs138312876.\n",
      "Warning: Impossible A1 allele assignment for variant rs57569946.\n",
      "Warning: Impossible A1 allele assignment for variant rs202099727.\n",
      "Warning: Impossible A1 allele assignment for variant rs35099615.\n",
      "Warning: Impossible A1 allele assignment for variant rs61542044.\n",
      "Warning: Impossible A1 allele assignment for variant rs11428307.\n",
      "Warning: Impossible A1 allele assignment for variant rs112329508.\n",
      "Warning: Impossible A1 allele assignment for variant rs113050087.\n",
      "Warning: Impossible A1 allele assignment for variant rs34668605.\n",
      "Warning: Impossible A1 allele assignment for variant rs56915281.\n",
      "Warning: Impossible A1 allele assignment for variant rs71912127.\n",
      "Warning: Impossible A1 allele assignment for variant rs59619564.\n",
      "Warning: Impossible A1 allele assignment for variant rs141872150.\n",
      "Warning: Impossible A1 allele assignment for variant rs149102978.\n",
      "Warning: Impossible A1 allele assignment for variant rs9977354.\n",
      "Warning: Impossible A1 allele assignment for variant rs34748446.\n",
      "Warning: Impossible A1 allele assignment for variant rs34394527.\n",
      "Warning: Impossible A1 allele assignment for variant rs201423572.\n",
      "Warning: Impossible A1 allele assignment for variant rs145852171.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk '{print$2,$5}' $path/1KG.all.bim > $path/1kg_ref-list.txt\n",
    "plink --bfile $intpath/dataset.b37_MDS --reference-allele $path/1kg_ref-list.txt --make-bed --out $intpath/dataset.b37_adj\n",
    "# The 1kG_MDS7 and the dataset.b37_adj have the same reference genome for all SNPs.\n",
    "# This command will generate some warnings for impossible A1 allele assignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs200579949 G\n",
      "rs199856693 A\n",
      "rs141149254 A\n",
      "rs114420996 A\n",
      "rs201888535 C\n",
      "rs114608975 C\n",
      "rs115209712 G\n",
      "rs940550 T\n",
      "rs55700207 A\n",
      "rs147538909 T\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/1kg_ref-list.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Resolve strand issues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs10013793 G T\n",
      "rs10013793 T TTG\n",
      "rs10544831 A ATT\n",
      "rs10544831 C T\n",
      "rs10546076 C T\n",
      "rs10546076 T TGTTC\n",
      "rs10547535 A ACAG\n",
      "rs10547535 G A\n",
      "rs10566332 C CAT\n",
      "rs10566332 C T\n",
      "  996  2988 17134 /mnt/data/GWAS/output/build37/task2_QC/all_differences.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Check for potential strand issues.\n",
    "awk '{print$2,$5,$6}' $path/1KG.all.bim > $intpath/1kGall_tmp\n",
    "awk '{print$2,$5,$6}' $intpath/dataset.b37_adj.bim > $intpath/dataset.b37_adj_tmp\n",
    "sort $intpath/1kGall_tmp $intpath/dataset.b37_adj_tmp |uniq -u > $path/all_differences.txt\n",
    "# 996 differences between the files, some of these might be due to strand issues.\n",
    "head $path/all_differences.txt\n",
    "wc $path/all_differences.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 498  498 5621 /mnt/data/GWAS/output/build37/task2_QC/flip_list.txt\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected.log.\n",
      "Options in effect:\n",
      "  --a1-allele /mnt/data/GWAS/output/build37/task2_QC/1kg_ref-list.txt\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_adj\n",
      "  --flip /mnt/data/GWAS/output/build37/task2_QC/flip_list.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505760 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--flip: 497 SNPs flipped.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99997.\n",
      "--a1-allele: 6505760 assignments made.\n",
      "6505760 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 204 variants had at least one non-A/C/G/T allele name.\n",
      "Warning: Impossible A1 allele assignment for variant rs147619543.\n",
      "Warning: Impossible A1 allele assignment for variant rs137984339.\n",
      "Warning: Impossible A1 allele assignment for variant rs55768108.\n",
      "Warning: Impossible A1 allele assignment for variant rs60360048.\n",
      "Warning: Impossible A1 allele assignment for variant rs201618615.\n",
      "Warning: Impossible A1 allele assignment for variant rs60203926.\n",
      "Warning: Impossible A1 allele assignment for variant rs111369971.\n",
      "Warning: Impossible A1 allele assignment for variant rs11211275.\n",
      "Warning: Impossible A1 allele assignment for variant rs34094074.\n",
      "Warning: Impossible A1 allele assignment for variant rs142949573.\n",
      "Warning: Impossible A1 allele assignment for variant rs59508386.\n",
      "Warning: Impossible A1 allele assignment for variant rs57226191.\n",
      "Warning: Impossible A1 allele assignment for variant rs71311970.\n",
      "Warning: Impossible A1 allele assignment for variant rs139796930.\n",
      "Warning: Impossible A1 allele assignment for variant rs5776404.\n",
      "Warning: Impossible A1 allele assignment for variant rs56378853.\n",
      "Warning: Impossible A1 allele assignment for variant rs137892494.\n",
      "Warning: Impossible A1 allele assignment for variant rs111279774.\n",
      "Warning: Impossible A1 allele assignment for variant rs71093259.\n",
      "Warning: Impossible A1 allele assignment for variant rs34176728.\n",
      "Warning: Impossible A1 allele assignment for variant rs7544642.\n",
      "Warning: Impossible A1 allele assignment for variant rs71565447.\n",
      "Warning: Impossible A1 allele assignment for variant rs11452685.\n",
      "Warning: Impossible A1 allele assignment for variant rs71104850.\n",
      "Warning: Impossible A1 allele assignment for variant rs202113516.\n",
      "Warning: Impossible A1 allele assignment for variant rs201981048.\n",
      "Warning: Impossible A1 allele assignment for variant rs34827379.\n",
      "Warning: Impossible A1 allele assignment for variant rs35359058.\n",
      "Warning: Impossible A1 allele assignment for variant rs5781390.\n",
      "Warning: Impossible A1 allele assignment for variant rs71721827.\n",
      "Warning: Impossible A1 allele assignment for variant rs55924341.\n",
      "Warning: Impossible A1 allele assignment for variant rs145912288.\n",
      "Warning: Impossible A1 allele assignment for variant rs34461703.\n",
      "Warning: Impossible A1 allele assignment for variant rs201659170.\n",
      "Warning: Impossible A1 allele assignment for variant rs11408370.\n",
      "Warning: Impossible A1 allele assignment for variant rs369838070.\n",
      "Warning: Impossible A1 allele assignment for variant rs35428718.\n",
      "Warning: Impossible A1 allele assignment for variant rs34109939.\n",
      "Warning: Impossible A1 allele assignment for variant rs112058391.\n",
      "Warning: Impossible A1 allele assignment for variant rs370986157.\n",
      "Warning: Impossible A1 allele assignment for variant rs148153906.\n",
      "Warning: Impossible A1 allele assignment for variant rs71407496.\n",
      "Warning: Impossible A1 allele assignment for variant rs146528263.\n",
      "Warning: Impossible A1 allele assignment for variant rs147872509.\n",
      "Warning: Impossible A1 allele assignment for variant rs70946958.\n",
      "Warning: Impossible A1 allele assignment for variant rs3835759.\n",
      "Warning: Impossible A1 allele assignment for variant rs35178225.\n",
      "Warning: Impossible A1 allele assignment for variant rs5831472.\n",
      "Warning: Impossible A1 allele assignment for variant rs36101912.\n",
      "Warning: Impossible A1 allele assignment for variant rs72339534.\n",
      "Warning: Impossible A1 allele assignment for variant rs71390158.\n",
      "Warning: Impossible A1 allele assignment for variant rs199685963.\n",
      "Warning: Impossible A1 allele assignment for variant rs149286053.\n",
      "Warning: Impossible A1 allele assignment for variant rs11442369.\n",
      "Warning: Impossible A1 allele assignment for variant rs35496875.\n",
      "Warning: Impossible A1 allele assignment for variant rs150724730.\n",
      "Warning: Impossible A1 allele assignment for variant rs113282615.\n",
      "Warning: Impossible A1 allele assignment for variant rs11418215.\n",
      "Warning: Impossible A1 allele assignment for variant rs72077160.\n",
      "Warning: Impossible A1 allele assignment for variant rs143937474.\n",
      "Warning: Impossible A1 allele assignment for variant rs111879402.\n",
      "Warning: Impossible A1 allele assignment for variant rs200688788.\n",
      "Warning: Impossible A1 allele assignment for variant rs201407212.\n",
      "Warning: Impossible A1 allele assignment for variant rs35965969.\n",
      "Warning: Impossible A1 allele assignment for variant rs67029113.\n",
      "Warning: Impossible A1 allele assignment for variant rs67949386.\n",
      "Warning: Impossible A1 allele assignment for variant rs5836500.\n",
      "Warning: Impossible A1 allele assignment for variant rs150165882.\n",
      "Warning: Impossible A1 allele assignment for variant rs34537206.\n",
      "Warning: Impossible A1 allele assignment for variant rs10668200.\n",
      "Warning: Impossible A1 allele assignment for variant rs57569510.\n",
      "Warning: Impossible A1 allele assignment for variant rs34819411.\n",
      "Warning: Impossible A1 allele assignment for variant rs11414538.\n",
      "Warning: Impossible A1 allele assignment for variant rs6764009.\n",
      "Warning: Impossible A1 allele assignment for variant rs148298277.\n",
      "Warning: Impossible A1 allele assignment for variant rs36004545.\n",
      "Warning: Impossible A1 allele assignment for variant rs34362926.\n",
      "Warning: Impossible A1 allele assignment for variant rs141935407.\n",
      "Warning: Impossible A1 allele assignment for variant rs1966896.\n",
      "Warning: Impossible A1 allele assignment for variant rs200822642.\n",
      "Warning: Impossible A1 allele assignment for variant rs34971732.\n",
      "Warning: Impossible A1 allele assignment for variant rs35159863.\n",
      "Warning: Impossible A1 allele assignment for variant rs9814616.\n",
      "Warning: Impossible A1 allele assignment for variant rs11378635.\n",
      "Warning: Impossible A1 allele assignment for variant rs7611450.\n",
      "Warning: Impossible A1 allele assignment for variant rs34454560.\n",
      "Warning: Impossible A1 allele assignment for variant rs202138300.\n",
      "Warning: Impossible A1 allele assignment for variant rs9814775.\n",
      "Warning: Impossible A1 allele assignment for variant rs150563753.\n",
      "Warning: Impossible A1 allele assignment for variant rs78490723.\n",
      "Warning: Impossible A1 allele assignment for variant rs35732987.\n",
      "Warning: Impossible A1 allele assignment for variant rs10663307.\n",
      "Warning: Impossible A1 allele assignment for variant rs370308267.\n",
      "Warning: Impossible A1 allele assignment for variant rs200697504.\n",
      "Warning: Impossible A1 allele assignment for variant rs34562094.\n",
      "Warning: Impossible A1 allele assignment for variant rs61324533.\n",
      "Warning: Impossible A1 allele assignment for variant rs61564785.\n",
      "Warning: Impossible A1 allele assignment for variant rs144147712.\n",
      "Warning: Impossible A1 allele assignment for variant rs150552425.\n",
      "Warning: Impossible A1 allele assignment for variant rs34762839.\n",
      "Warning: Impossible A1 allele assignment for variant rs35882455.\n",
      "Warning: Impossible A1 allele assignment for variant rs59749772.\n",
      "Warning: Impossible A1 allele assignment for variant rs112726145.\n",
      "Warning: Impossible A1 allele assignment for variant rs7436539.\n",
      "Warning: Impossible A1 allele assignment for variant rs148548225.\n",
      "Warning: Impossible A1 allele assignment for variant rs35347672.\n",
      "Warning: Impossible A1 allele assignment for variant rs35746929.\n",
      "Warning: Impossible A1 allele assignment for variant rs60463614.\n",
      "Warning: Impossible A1 allele assignment for variant rs71654862.\n",
      "Warning: Impossible A1 allele assignment for variant rs34812158.\n",
      "Warning: Impossible A1 allele assignment for variant rs60101939.\n",
      "Warning: Impossible A1 allele assignment for variant rs10013793.\n",
      "Warning: Impossible A1 allele assignment for variant rs113842049.\n",
      "Warning: Impossible A1 allele assignment for variant rs11414296.\n",
      "Warning: Impossible A1 allele assignment for variant rs71964935.\n",
      "Warning: Impossible A1 allele assignment for variant rs3066106.\n",
      "Warning: Impossible A1 allele assignment for variant rs70961801.\n",
      "Warning: Impossible A1 allele assignment for variant rs58890122.\n",
      "Warning: Impossible A1 allele assignment for variant rs57536077.\n",
      "Warning: Impossible A1 allele assignment for variant rs59300393.\n",
      "Warning: Impossible A1 allele assignment for variant rs201549105.\n",
      "Warning: Impossible A1 allele assignment for variant rs60465499.\n",
      "Warning: Impossible A1 allele assignment for variant rs202183072.\n",
      "Warning: Impossible A1 allele assignment for variant rs60484575.\n",
      "Warning: Impossible A1 allele assignment for variant rs71584391.\n",
      "Warning: Impossible A1 allele assignment for variant rs150452407.\n",
      "Warning: Impossible A1 allele assignment for variant rs34497958.\n",
      "Warning: Impossible A1 allele assignment for variant rs57216995.\n",
      "Warning: Impossible A1 allele assignment for variant rs35226602.\n",
      "Warning: Impossible A1 allele assignment for variant rs34278906.\n",
      "Warning: Impossible A1 allele assignment for variant rs10657259.\n",
      "Warning: Impossible A1 allele assignment for variant rs376653011.\n",
      "Warning: Impossible A1 allele assignment for variant rs146771059.\n",
      "Warning: Impossible A1 allele assignment for variant rs60755557.\n",
      "Warning: Impossible A1 allele assignment for variant rs33999637.\n",
      "Warning: Impossible A1 allele assignment for variant rs3066161.\n",
      "Warning: Impossible A1 allele assignment for variant rs60010478.\n",
      "Warning: Impossible A1 allele assignment for variant rs70938251.\n",
      "Warning: Impossible A1 allele assignment for variant rs5866808.\n",
      "Warning: Impossible A1 allele assignment for variant rs142795569.\n",
      "Warning: Impossible A1 allele assignment for variant rs149654848.\n",
      "Warning: Impossible A1 allele assignment for variant rs35290606.\n",
      "Warning: Impossible A1 allele assignment for variant rs58450772.\n",
      "Warning: Impossible A1 allele assignment for variant rs248382.\n",
      "Warning: Impossible A1 allele assignment for variant rs112912857.\n",
      "Warning: Impossible A1 allele assignment for variant rs76139278.\n",
      "Warning: Impossible A1 allele assignment for variant rs67231143.\n",
      "Warning: Impossible A1 allele assignment for variant rs370121313.\n",
      "Warning: Impossible A1 allele assignment for variant rs143078150.\n",
      "Warning: Impossible A1 allele assignment for variant rs58688712.\n",
      "Warning: Impossible A1 allele assignment for variant rs35540828.\n",
      "Warning: Impossible A1 allele assignment for variant rs200293348.\n",
      "Warning: Impossible A1 allele assignment for variant rs141641983.\n",
      "Warning: Impossible A1 allele assignment for variant rs10651764.\n",
      "Warning: Impossible A1 allele assignment for variant rs10671428.\n",
      "Warning: Impossible A1 allele assignment for variant rs34768415.\n",
      "Warning: Impossible A1 allele assignment for variant rs150215882.\n",
      "Warning: Impossible A1 allele assignment for variant rs58893321.\n",
      "Warning: Impossible A1 allele assignment for variant rs199959526.\n",
      "Warning: Impossible A1 allele assignment for variant rs55705262.\n",
      "Warning: Impossible A1 allele assignment for variant rs201191015.\n",
      "Warning: Impossible A1 allele assignment for variant rs60383041.\n",
      "Warning: Impossible A1 allele assignment for variant rs71542640.\n",
      "Warning: Impossible A1 allele assignment for variant rs202034055.\n",
      "Warning: Impossible A1 allele assignment for variant rs199636827.\n",
      "Warning: Impossible A1 allele assignment for variant rs200589655.\n",
      "Warning: Impossible A1 allele assignment for variant rs143772814.\n",
      "Warning: Impossible A1 allele assignment for variant rs35509700.\n",
      "Warning: Impossible A1 allele assignment for variant rs66657443.\n",
      "Warning: Impossible A1 allele assignment for variant rs11449422.\n",
      "Warning: Impossible A1 allele assignment for variant rs78046670.\n",
      "Warning: Impossible A1 allele assignment for variant rs34696372.\n",
      "Warning: Impossible A1 allele assignment for variant rs71536404.\n",
      "Warning: Impossible A1 allele assignment for variant rs202087713.\n",
      "Warning: Impossible A1 allele assignment for variant rs3035346.\n",
      "Warning: Impossible A1 allele assignment for variant rs12200694.\n",
      "Warning: Impossible A1 allele assignment for variant rs150511523.\n",
      "Warning: Impossible A1 allele assignment for variant rs58774833.\n",
      "Warning: Impossible A1 allele assignment for variant rs71541270.\n",
      "Warning: Impossible A1 allele assignment for variant rs3054945.\n",
      "Warning: Impossible A1 allele assignment for variant rs34521551.\n",
      "Warning: Impossible A1 allele assignment for variant rs144366744.\n",
      "Warning: Impossible A1 allele assignment for variant rs200474568.\n",
      "Warning: Impossible A1 allele assignment for variant rs148588315.\n",
      "Warning: Impossible A1 allele assignment for variant rs57690424.\n",
      "Warning: Impossible A1 allele assignment for variant rs35144811.\n",
      "Warning: Impossible A1 allele assignment for variant rs9371847.\n",
      "Warning: Impossible A1 allele assignment for variant rs11377175.\n",
      "Warning: Impossible A1 allele assignment for variant rs201098747.\n",
      "Warning: Impossible A1 allele assignment for variant rs142171671.\n",
      "Warning: Impossible A1 allele assignment for variant rs35003890.\n",
      "Warning: Impossible A1 allele assignment for variant rs202130155.\n",
      "Warning: Impossible A1 allele assignment for variant rs143156307.\n",
      "Warning: Impossible A1 allele assignment for variant rs200899071.\n",
      "Warning: Impossible A1 allele assignment for variant rs35866128.\n",
      "Warning: Impossible A1 allele assignment for variant rs140868127.\n",
      "Warning: Impossible A1 allele assignment for variant rs5883347.\n",
      "Warning: Impossible A1 allele assignment for variant rs5884215.\n",
      "Warning: Impossible A1 allele assignment for variant rs112532470.\n",
      "Warning: Impossible A1 allele assignment for variant rs149316964.\n",
      "Warning: Impossible A1 allele assignment for variant rs200645214.\n",
      "Warning: Impossible A1 allele assignment for variant rs145164759.\n",
      "Warning: Impossible A1 allele assignment for variant rs150201843.\n",
      "Warning: Impossible A1 allele assignment for variant rs140222950.\n",
      "Warning: Impossible A1 allele assignment for variant rs151177697.\n",
      "Warning: Impossible A1 allele assignment for variant rs3070857.\n",
      "Warning: Impossible A1 allele assignment for variant rs148802556.\n",
      "Warning: Impossible A1 allele assignment for variant rs141623426.\n",
      "Warning: Impossible A1 allele assignment for variant rs79404395.\n",
      "Warning: Impossible A1 allele assignment for variant rs113055548.\n",
      "Warning: Impossible A1 allele assignment for variant rs147693269.\n",
      "Warning: Impossible A1 allele assignment for variant rs5889534.\n",
      "Warning: Impossible A1 allele assignment for variant rs5890525.\n",
      "Warning: Impossible A1 allele assignment for variant rs59255004.\n",
      "Warning: Impossible A1 allele assignment for variant rs143645350.\n",
      "Warning: Impossible A1 allele assignment for variant rs34188615.\n",
      "Warning: Impossible A1 allele assignment for variant rs71264185.\n",
      "Warning: Impossible A1 allele assignment for variant rs5892663.\n",
      "Warning: Impossible A1 allele assignment for variant rs199976662.\n",
      "Warning: Impossible A1 allele assignment for variant rs200536442.\n",
      "Warning: Impossible A1 allele assignment for variant rs201150792.\n",
      "Warning: Impossible A1 allele assignment for variant rs200998873.\n",
      "Warning: Impossible A1 allele assignment for variant rs55953934.\n",
      "Warning: Impossible A1 allele assignment for variant rs143602363.\n",
      "Warning: Impossible A1 allele assignment for variant rs374892516.\n",
      "Warning: Impossible A1 allele assignment for variant rs3073777.\n",
      "Warning: Impossible A1 allele assignment for variant rs34378799.\n",
      "Warning: Impossible A1 allele assignment for variant rs34014791.\n",
      "Warning: Impossible A1 allele assignment for variant rs137860744.\n",
      "Warning: Impossible A1 allele assignment for variant rs59083583.\n",
      "Warning: Impossible A1 allele assignment for variant rs11389375.\n",
      "Warning: Impossible A1 allele assignment for variant rs74176994.\n",
      "Warning: Impossible A1 allele assignment for variant rs371675462.\n",
      "Warning: Impossible A1 allele assignment for variant rs201243915.\n",
      "Warning: Impossible A1 allele assignment for variant rs55721416.\n",
      "Warning: Impossible A1 allele assignment for variant rs11380003.\n",
      "Warning: Impossible A1 allele assignment for variant rs35759919.\n",
      "Warning: Impossible A1 allele assignment for variant rs35789695.\n",
      "Warning: Impossible A1 allele assignment for variant rs71362130.\n",
      "Warning: Impossible A1 allele assignment for variant rs145306304.\n",
      "Warning: Impossible A1 allele assignment for variant rs71366046.\n",
      "Warning: Impossible A1 allele assignment for variant rs10546076.\n",
      "Warning: Impossible A1 allele assignment for variant rs71855060.\n",
      "Warning: Impossible A1 allele assignment for variant rs34428620.\n",
      "Warning: Impossible A1 allele assignment for variant rs200278001.\n",
      "Warning: Impossible A1 allele assignment for variant rs10544831.\n",
      "Warning: Impossible A1 allele assignment for variant rs144135557.\n",
      "Warning: Impossible A1 allele assignment for variant rs148981849.\n",
      "Warning: Impossible A1 allele assignment for variant rs67559361.\n",
      "Warning: Impossible A1 allele assignment for variant rs200516560.\n",
      "Warning: Impossible A1 allele assignment for variant rs200148222.\n",
      "Warning: Impossible A1 allele assignment for variant rs202111789.\n",
      "Warning: Impossible A1 allele assignment for variant rs111563455.\n",
      "Warning: Impossible A1 allele assignment for variant rs146026552.\n",
      "Warning: Impossible A1 allele assignment for variant rs60225229.\n",
      "Warning: Impossible A1 allele assignment for variant rs141345608.\n",
      "Warning: Impossible A1 allele assignment for variant rs57872451.\n",
      "Warning: Impossible A1 allele assignment for variant rs35017955.\n",
      "Warning: Impossible A1 allele assignment for variant rs34261017.\n",
      "Warning: Impossible A1 allele assignment for variant rs111532670.\n",
      "Warning: Impossible A1 allele assignment for variant rs148435065.\n",
      "Warning: Impossible A1 allele assignment for variant rs147050495.\n",
      "Warning: Impossible A1 allele assignment for variant rs150287660.\n",
      "Warning: Impossible A1 allele assignment for variant rs150480149.\n",
      "Warning: Impossible A1 allele assignment for variant rs11281146.\n",
      "Warning: Impossible A1 allele assignment for variant rs10598951.\n",
      "Warning: Impossible A1 allele assignment for variant rs147482924.\n",
      "Warning: Impossible A1 allele assignment for variant rs55976849.\n",
      "Warning: Impossible A1 allele assignment for variant rs10547535.\n",
      "Warning: Impossible A1 allele assignment for variant rs56871830.\n",
      "Warning: Impossible A1 allele assignment for variant rs113540051.\n",
      "Warning: Impossible A1 allele assignment for variant rs202152850.\n",
      "Warning: Impossible A1 allele assignment for variant rs35308664.\n",
      "Warning: Impossible A1 allele assignment for variant rs199932595.\n",
      "Warning: Impossible A1 allele assignment for variant rs34446072.\n",
      "Warning: Impossible A1 allele assignment for variant rs35129198.\n",
      "Warning: Impossible A1 allele assignment for variant rs72240657.\n",
      "Warning: Impossible A1 allele assignment for variant rs5788827.\n",
      "Warning: Impossible A1 allele assignment for variant rs111467735.\n",
      "Warning: Impossible A1 allele assignment for variant rs35692523.\n",
      "Warning: Impossible A1 allele assignment for variant rs5789515.\n",
      "Warning: Impossible A1 allele assignment for variant rs138140225.\n",
      "Warning: Impossible A1 allele assignment for variant rs61026944.\n",
      "Warning: Impossible A1 allele assignment for variant rs10627420.\n",
      "Warning: Impossible A1 allele assignment for variant rs34730979.\n",
      "Warning: Impossible A1 allele assignment for variant rs147359460.\n",
      "Warning: Impossible A1 allele assignment for variant rs10676423.\n",
      "Warning: Impossible A1 allele assignment for variant rs34088736.\n",
      "Warning: Impossible A1 allele assignment for variant rs56263226.\n",
      "Warning: Impossible A1 allele assignment for variant rs59677985.\n",
      "Warning: Impossible A1 allele assignment for variant rs145284642.\n",
      "Warning: Impossible A1 allele assignment for variant rs5792633.\n",
      "Warning: Impossible A1 allele assignment for variant rs71036061.\n",
      "Warning: Impossible A1 allele assignment for variant rs71046975.\n",
      "Warning: Impossible A1 allele assignment for variant rs60728415.\n",
      "Warning: Impossible A1 allele assignment for variant rs199507819.\n",
      "Warning: Impossible A1 allele assignment for variant rs5794069.\n",
      "Warning: Impossible A1 allele assignment for variant rs35255367.\n",
      "Warning: Impossible A1 allele assignment for variant rs11389863.\n",
      "Warning: Impossible A1 allele assignment for variant rs5795305.\n",
      "Warning: Impossible A1 allele assignment for variant rs60885762.\n",
      "Warning: Impossible A1 allele assignment for variant rs58386771.\n",
      "Warning: Impossible A1 allele assignment for variant rs7946045.\n",
      "Warning: Impossible A1 allele assignment for variant rs6590237.\n",
      "Warning: Impossible A1 allele assignment for variant rs149989855.\n",
      "Warning: Impossible A1 allele assignment for variant rs200765236.\n",
      "Warning: Impossible A1 allele assignment for variant rs10845439.\n",
      "Warning: Impossible A1 allele assignment for variant rs139159553.\n",
      "Warning: Impossible A1 allele assignment for variant rs11410265.\n",
      "Warning: Impossible A1 allele assignment for variant rs200231897.\n",
      "Warning: Impossible A1 allele assignment for variant rs145053170.\n",
      "Warning: Impossible A1 allele assignment for variant rs34967117.\n",
      "Warning: Impossible A1 allele assignment for variant rs146159837.\n",
      "Warning: Impossible A1 allele assignment for variant rs141872617.\n",
      "Warning: Impossible A1 allele assignment for variant rs148744178.\n",
      "Warning: Impossible A1 allele assignment for variant rs144445994.\n",
      "Warning: Impossible A1 allele assignment for variant rs377288546.\n",
      "Warning: Impossible A1 allele assignment for variant rs140807061.\n",
      "Warning: Impossible A1 allele assignment for variant rs57025244.\n",
      "Warning: Impossible A1 allele assignment for variant rs34990059.\n",
      "Warning: Impossible A1 allele assignment for variant rs369621491.\n",
      "Warning: Impossible A1 allele assignment for variant rs56808655.\n",
      "Warning: Impossible A1 allele assignment for variant rs142208138.\n",
      "Warning: Impossible A1 allele assignment for variant rs11411094.\n",
      "Warning: Impossible A1 allele assignment for variant rs71076684.\n",
      "Warning: Impossible A1 allele assignment for variant rs10681665.\n",
      "Warning: Impossible A1 allele assignment for variant rs200847440.\n",
      "Warning: Impossible A1 allele assignment for variant rs60974421.\n",
      "Warning: Impossible A1 allele assignment for variant rs58817432.\n",
      "Warning: Impossible A1 allele assignment for variant rs71087101.\n",
      "Warning: Impossible A1 allele assignment for variant rs36121482.\n",
      "Warning: Impossible A1 allele assignment for variant rs112398485.\n",
      "Warning: Impossible A1 allele assignment for variant rs35136338.\n",
      "Warning: Impossible A1 allele assignment for variant rs9568372.\n",
      "Warning: Impossible A1 allele assignment for variant rs35814981.\n",
      "Warning: Impossible A1 allele assignment for variant rs9599171.\n",
      "Warning: Impossible A1 allele assignment for variant rs148595414.\n",
      "Warning: Impossible A1 allele assignment for variant rs3075437.\n",
      "Warning: Impossible A1 allele assignment for variant rs200381946.\n",
      "Warning: Impossible A1 allele assignment for variant rs9530195.\n",
      "Warning: Impossible A1 allele assignment for variant rs35720894.\n",
      "Warning: Impossible A1 allele assignment for variant rs60666691.\n",
      "Warning: Impossible A1 allele assignment for variant rs34703149.\n",
      "Warning: Impossible A1 allele assignment for variant rs138141946.\n",
      "Warning: Impossible A1 allele assignment for variant rs142694972.\n",
      "Warning: Impossible A1 allele assignment for variant rs148340685.\n",
      "Warning: Impossible A1 allele assignment for variant rs36196814.\n",
      "Warning: Impossible A1 allele assignment for variant rs17523478.\n",
      "Warning: Impossible A1 allele assignment for variant rs61400710.\n",
      "Warning: Impossible A1 allele assignment for variant rs34317236.\n",
      "Warning: Impossible A1 allele assignment for variant rs58647073.\n",
      "Warning: Impossible A1 allele assignment for variant rs148705068.\n",
      "Warning: Impossible A1 allele assignment for variant rs11430120.\n",
      "Warning: Impossible A1 allele assignment for variant rs201503456.\n",
      "Warning: Impossible A1 allele assignment for variant rs17108655.\n",
      "Warning: Impossible A1 allele assignment for variant rs140845066.\n",
      "Warning: Impossible A1 allele assignment for variant rs111864429.\n",
      "Warning: Impossible A1 allele assignment for variant rs149532947.\n",
      "Warning: Impossible A1 allele assignment for variant rs58321309.\n",
      "Warning: Impossible A1 allele assignment for variant rs11379794.\n",
      "Warning: Impossible A1 allele assignment for variant rs71464700.\n",
      "Warning: Impossible A1 allele assignment for variant rs79779753.\n",
      "Warning: Impossible A1 allele assignment for variant rs113786826.\n",
      "Warning: Impossible A1 allele assignment for variant rs79697112.\n",
      "Warning: Impossible A1 allele assignment for variant rs10686797.\n",
      "Warning: Impossible A1 allele assignment for variant rs33911018.\n",
      "Warning: Impossible A1 allele assignment for variant rs34546417.\n",
      "Warning: Impossible A1 allele assignment for variant rs112183174.\n",
      "Warning: Impossible A1 allele assignment for variant rs35771863.\n",
      "Warning: Impossible A1 allele assignment for variant rs112038713.\n",
      "Warning: Impossible A1 allele assignment for variant rs34854915.\n",
      "Warning: Impossible A1 allele assignment for variant rs201825062.\n",
      "Warning: Impossible A1 allele assignment for variant rs58300596.\n",
      "Warning: Impossible A1 allele assignment for variant rs145730606.\n",
      "Warning: Impossible A1 allele assignment for variant rs71280728.\n",
      "Warning: Impossible A1 allele assignment for variant rs59912939.\n",
      "Warning: Impossible A1 allele assignment for variant rs374142105.\n",
      "Warning: Impossible A1 allele assignment for variant rs35769927.\n",
      "Warning: Impossible A1 allele assignment for variant rs55720616.\n",
      "Warning: Impossible A1 allele assignment for variant rs58851793.\n",
      "Warning: Impossible A1 allele assignment for variant rs71378042.\n",
      "Warning: Impossible A1 allele assignment for variant rs5816868.\n",
      "Warning: Impossible A1 allele assignment for variant rs111488863.\n",
      "Warning: Impossible A1 allele assignment for variant rs56031703.\n",
      "Warning: Impossible A1 allele assignment for variant rs142646570.\n",
      "Warning: Impossible A1 allele assignment for variant rs71382034.\n",
      "Warning: Impossible A1 allele assignment for variant rs112542398.\n",
      "Warning: Impossible A1 allele assignment for variant rs5818623.\n",
      "Warning: Impossible A1 allele assignment for variant rs60440735.\n",
      "Warning: Impossible A1 allele assignment for variant rs71134239.\n",
      "Warning: Impossible A1 allele assignment for variant rs10569851.\n",
      "Warning: Impossible A1 allele assignment for variant rs139049457.\n",
      "Warning: Impossible A1 allele assignment for variant rs150906505.\n",
      "Warning: Impossible A1 allele assignment for variant rs58612689.\n",
      "Warning: Impossible A1 allele assignment for variant rs149264208.\n",
      "Warning: Impossible A1 allele assignment for variant rs142032070.\n",
      "Warning: Impossible A1 allele assignment for variant rs193038927.\n",
      "Warning: Impossible A1 allele assignment for variant rs199817361.\n",
      "Warning: Impossible A1 allele assignment for variant rs58183353.\n",
      "Warning: Impossible A1 allele assignment for variant rs147258226.\n",
      "Warning: Impossible A1 allele assignment for variant rs201838394.\n",
      "Warning: Impossible A1 allele assignment for variant rs10650743.\n",
      "Warning: Impossible A1 allele assignment for variant rs71172027.\n",
      "Warning: Impossible A1 allele assignment for variant rs35808714.\n",
      "Warning: Impossible A1 allele assignment for variant rs34990450.\n",
      "Warning: Impossible A1 allele assignment for variant rs141406346.\n",
      "Warning: Impossible A1 allele assignment for variant rs34238208.\n",
      "Warning: Impossible A1 allele assignment for variant rs11441702.\n",
      "Warning: Impossible A1 allele assignment for variant rs146092313.\n",
      "Warning: Impossible A1 allele assignment for variant rs137914150.\n",
      "Warning: Impossible A1 allele assignment for variant rs71169322.\n",
      "Warning: Impossible A1 allele assignment for variant rs72155611.\n",
      "Warning: Impossible A1 allele assignment for variant rs11373675.\n",
      "Warning: Impossible A1 allele assignment for variant rs146296467.\n",
      "Warning: Impossible A1 allele assignment for variant rs137928326.\n",
      "Warning: Impossible A1 allele assignment for variant rs34699436.\n",
      "Warning: Impossible A1 allele assignment for variant rs71876994.\n",
      "Warning: Impossible A1 allele assignment for variant rs35954284.\n",
      "Warning: Impossible A1 allele assignment for variant rs146055611.\n",
      "Warning: Impossible A1 allele assignment for variant rs112839858.\n",
      "Warning: Impossible A1 allele assignment for variant rs140822316.\n",
      "Warning: Impossible A1 allele assignment for variant rs67204288.\n",
      "Warning: Impossible A1 allele assignment for variant rs35866247.\n",
      "Warning: Impossible A1 allele assignment for variant rs35299917.\n",
      "Warning: Impossible A1 allele assignment for variant rs71168622.\n",
      "Warning: Impossible A1 allele assignment for variant rs71170536.\n",
      "Warning: Impossible A1 allele assignment for variant rs60779505.\n",
      "Warning: Impossible A1 allele assignment for variant rs139488448.\n",
      "Warning: Impossible A1 allele assignment for variant rs56893426.\n",
      "Warning: Impossible A1 allele assignment for variant rs138312876.\n",
      "Warning: Impossible A1 allele assignment for variant rs57569946.\n",
      "Warning: Impossible A1 allele assignment for variant rs55997903.\n",
      "Warning: Impossible A1 allele assignment for variant rs202099727.\n",
      "Warning: Impossible A1 allele assignment for variant rs35099615.\n",
      "Warning: Impossible A1 allele assignment for variant rs10566332.\n",
      "Warning: Impossible A1 allele assignment for variant rs138569031.\n",
      "Warning: Impossible A1 allele assignment for variant rs61542044.\n",
      "Warning: Impossible A1 allele assignment for variant rs151208523.\n",
      "Warning: Impossible A1 allele assignment for variant rs10684409.\n",
      "Warning: Impossible A1 allele assignment for variant rs11480817.\n",
      "Warning: Impossible A1 allele assignment for variant rs11428307.\n",
      "Warning: Impossible A1 allele assignment for variant rs57337201.\n",
      "Warning: Impossible A1 allele assignment for variant rs112329508.\n",
      "Warning: Impossible A1 allele assignment for variant rs113050087.\n",
      "Warning: Impossible A1 allele assignment for variant rs34668605.\n",
      "Warning: Impossible A1 allele assignment for variant rs56915281.\n",
      "Warning: Impossible A1 allele assignment for variant rs71912127.\n",
      "Warning: Impossible A1 allele assignment for variant rs59619564.\n",
      "Warning: Impossible A1 allele assignment for variant rs141872150.\n",
      "Warning: Impossible A1 allele assignment for variant rs149102978.\n",
      "Warning: Impossible A1 allele assignment for variant rs9977354.\n",
      "Warning: Impossible A1 allele assignment for variant rs146884741.\n",
      "Warning: Impossible A1 allele assignment for variant rs34748446.\n",
      "Warning: Impossible A1 allele assignment for variant rs71192922.\n",
      "Warning: Impossible A1 allele assignment for variant rs34394527.\n",
      "Warning: Impossible A1 allele assignment for variant rs201423572.\n",
      "Warning: Impossible A1 allele assignment for variant rs145852171.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Flip SNPs for resolving strand issues.\n",
    "# Print SNP-identifier and remove duplicates.\n",
    "awk '{print$1}' $path/all_differences.txt | sort -u > $path/flip_list.txt\n",
    "wc $path/flip_list.txt\n",
    "# Generates a file of 498 SNPs. These are the non-corresponding SNPs between the two files. \n",
    "# Flip the 498 non-corresponding SNPs. \n",
    "plink --bfile $intpath/dataset.b37_adj --flip $path/flip_list.txt --reference-allele $path/1kg_ref-list.txt --make-bed --out $intpath/dataset.b37.corrected\n",
    "# This command will generate some warnings for impossible A1 allele assignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Check for SNPs which are still problematic after they have been flipped.\n",
    "awk '{print$2,$5,$6}' $intpath/dataset.b37.corrected.bim > $intpath/dataset.b37.corrected_tmp\n",
    "sort $intpath/1kGall_tmp $intpath/dataset.b37.corrected_tmp |uniq -u  > $path/uncorresponding_SNPs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "984 /mnt/data/GWAS/output/build37/task2_QC/uncorresponding_SNPs.txt\n",
      "rs10013793 A TTG\n",
      "rs10013793 G T\n",
      "rs10544831 C T\n",
      "rs10544831 T ATT\n",
      "rs10546076 A TGTTC\n",
      "rs10546076 C T\n",
      "rs10547535 G A\n",
      "rs10547535 T ACAG\n",
      "rs10566332 A G\n",
      "rs10566332 C CAT\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# This file demonstrates that there are 984 differences between the files.\n",
    "wc -l $path/uncorresponding_SNPs.txt\n",
    "head $path/uncorresponding_SNPs.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove problematic SNPs from the working dataset and 1000 Genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs10013793\n",
      "rs10544831\n",
      "rs10546076\n",
      "rs10547535\n",
      "rs10566332\n",
      "rs10569851\n",
      "rs10598951\n",
      "rs10627420\n",
      "rs10646909\n",
      "rs10650743\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk '{print$1}' $path/uncorresponding_SNPs.txt | sort -u > $path/SNPs_for_exlusion.txt\n",
    "# The command above generates a list of the SNPs which caused the 984 differences between the working dataset (i.e HapMap) and the 1000 Genomes data sets after flipping and setting of the reference genome.\n",
    "head $path/SNPs_for_exlusion.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37.corrected\n",
      "  --exclude /mnt/data/GWAS/output/build37/task2_QC/SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505760 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--exclude: 6505268 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99997.\n",
      "6505268 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.bed\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.bim\n",
      "+\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/1KG.all\n",
      "  --exclude /mnt/data/GWAS/output/build37/task2_QC/SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505760 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.nosex .\n",
      "--exclude: 6505268 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999848.\n",
      "6505268 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove the problematic SNPs from both datasets.\n",
    "plink --bfile $intpath/dataset.b37.corrected --exclude $path/SNPs_for_exlusion.txt --make-bed --out $intpath/dataset.b37_MDS2\n",
    "plink --bfile $path/1KG.all --exclude $path/SNPs_for_exlusion.txt --make-bed --out $path/1kG_MDS\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge the working dataset with 1000 Genomes Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2\n",
      "  --bmerge /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.bed /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.bim /mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.fam\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people loaded from\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.fam.\n",
      "2504 people to be merged from\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.fam.\n",
      "Of these, 2504 are new, while 0 are present in the base dataset.\n",
      "6505268 markers loaded from\n",
      "/mnt/data/GWAS/output/build37/task2_QC/intermediate_datasets/dataset.b37_MDS2.bim.\n",
      "6505268 markers to be merged from\n",
      "/mnt/data/GWAS/output/build37/task2_QC/1kG_MDS.bim.\n",
      "Of these, 0 are new, while 6505268 are present in the base dataset.\n",
      "Performing single-pass merge (3000 people, 6505268 variants).\n",
      "Merged fileset written to                     \n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2-merge.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2-merge.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2-merge.fam .\n",
      "6505268 variants loaded from .bim file.\n",
      "3000 people (237 males, 259 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.nosex .\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999868.\n",
      "6505268 variants and 3000 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: Multiple positions seen for variant 'rs11460950'.\n",
      "Warning: Multiple chromosomes seen for variant 'rs77875881'.\n",
      "Warning: Multiple chromosomes seen for variant 'rs113258194'.\n",
      "Warning: Multiple positions seen for variant 'rs35212441'.\n",
      "Warning: Multiple positions seen for variant 'rs141821078'.\n",
      "22 more multiple-position warnings: see log file.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Merge HapMap with 1000 Genomes Data.\n",
    "plink --bfile $intpath/dataset.b37_MDS2 --bmerge $path/1kG_MDS.bed $path/1kG_MDS.bim $path/1kG_MDS.fam --allow-no-sex --make-bed --out $path/MDS_merge2\n",
    "\n",
    "# Note, we are fully aware of the sample overlap between the HapMap and 1000 Genomes datasets. However, for the purpose of this tutorial this is not important.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Perform Multidimensional Scaling (MDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2\n",
      "  --extract /mnt/data/GWAS/output/build37/task2_QC/../task2_QC/indepSNP.prune.in\n",
      "  --genome\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505268 variants loaded from .bim file.\n",
      "3000 people (237 males, 259 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.nosex .\n",
      "496 phenotype values loaded from .fam.\n",
      "--extract: 390066 variants remaining.\n",
      "Using up to 4 threads (change this with --threads).\n",
      "Before main variant filters, 3000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999858.\n",
      "390066 variants and 3000 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "IBD calculations complete.  \n",
      "Finished writing /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.genome .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2\n",
      "  --read-genome /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "6505268 variants loaded from .bim file.\n",
      "3000 people (237 males, 259 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.nosex .\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3000 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999868.\n",
      "6505268 variants and 3000 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.  (2504\n",
      "phenotypes are missing.)\n",
      "Clustering... done.                        \n",
      "Cluster solution written to 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.cluster1 ,\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.cluster2 , and\n",
      "/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to /mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.mds .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Using a set of pruned SNPs\n",
    "plink --bfile $path/MDS_merge2 --extract $path/../task2_QC/indepSNP.prune.in --genome --out $path/MDS_merge2\n",
    "plink --bfile $path/MDS_merge2 --read-genome $path/MDS_merge2.genome --cluster --mds-plot 10 --out $path/MDS_merge2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MDS-plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2505 /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt\n",
      "FID\tIID\tSuperPop\n",
      "HG00096\tHG00096\tEUR\n",
      "HG00097\tHG00097\tEUR\n",
      "HG00099\tHG00099\tEUR\n",
      "HG00100\tHG00100\tEUR\n",
      "HG00101\tHG00101\tEUR\n",
      "HG00102\tHG00102\tEUR\n",
      "HG00103\tHG00103\tEUR\n",
      "HG00105\tHG00105\tEUR\n",
      "HG00106\tHG00106\tEUR\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# 1000 Genomes superpopulation codes\n",
    "wc -l /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt\n",
    "head /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HGX00096 HGX00096 OWN\n",
      "HGX00097 HGX00097 OWN\n",
      "HGX00099 HGX00099 OWN\n",
      "HGX00100 HGX00100 OWN\n",
      "HGX00101 HGX00101 OWN\n",
      "HGX00102 HGX00102 OWN\n",
      "HGX00103 HGX00103 OWN\n",
      "HGX00105 HGX00105 OWN\n",
      "HGX00106 HGX00106 OWN\n",
      "HGX00107 HGX00107 OWN\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Create a racefile of your own data.\n",
    "awk '{print$1,$2,\"OWN\"}' $intpath/dataset.b37_MDS.fam>$path/racefile_own.txt\n",
    "head $path/racefile_own.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID\tIID\tSuperPop\n",
      "HG00096\tHG00096\tEUR\n",
      "HG00097\tHG00097\tEUR\n",
      "HG00099\tHG00099\tEUR\n",
      "HG00100\tHG00100\tEUR\n",
      "HG00101\tHG00101\tEUR\n",
      "HG00102\tHG00102\tEUR\n",
      "HG00103\tHG00103\tEUR\n",
      "HG00105\tHG00105\tEUR\n",
      "HG00106\tHG00106\tEUR\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Concatenate racefiles.\n",
    "cat /mnt/data/GWAS/ref_files/1000G_superpopulation_codes.txt $path/racefile_own.txt  > $path/racefile.txt\n",
    "head $path/racefile.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOzdd3wUZf7A8c/M7Gaz6SGFJLRA6EVQqnQUlMNyYkEFsSDnnZ7l9Pydig1sZ7+GnnqnHsWO5dBTEZBejIB0BIEkkgQSSE92ky0zvz82xBA2Idkkmw35vl++fO3OzszzTIBvnn3mO99HMQwDIYQQgUdt6Q4IIYTwTgK0EEIEKAnQQggRoCRACyFEgJIALYQQAUoCtBBCBCgJ0EIIEaAkQAshRICSAC2EEAFKArQQQgQoCdBCCBGgJEALIUSAkgAthBABSgK0EEIEKAnQQggRoCRACyFEgJIALYQQAUoCtBBCBCgJ0EIIEaAkQAshRICSAC2EEAFKArQQQgQoCdBCCBGgJEALIUSAkgAthBABSgK0EEIEKAnQQggRoCRACyFEgJIALYQQAUoCtBBCBCgJ0EIIEaAkQAshRICSAC2EEAFKArQQQgQoCdBCCBGgJEALIUSAkgAthBABSgK0EEIEKAnQQggRoCRACyFEgJIALYQQAUoCtBBCBCgJ0EIIEaAkQAshRICSAC2EEAFKArQQQgQoCdBCCBGgJEALIUSAkgAthBABSgK0EEIEKAnQQggRoCRACyFEgJIALYQQAUoCtBBCBCgJ0EIIEaAkQAshRICSAC2EEAHK1NIdaBiLxeJ0Olu6F0II8YuEhITs7OzmOLO/A7SmaYZhGIahqqrb7W7o4U6n0zAMRVGao29CiABhGMYLL7wQGxvb0h05s9WrVy9evLiZTu6PAK0oSrt27fLy8lRVNQzDs1HXdVVVdV334Ww+HCWEaEVmz549derUlJSUlu7ImR04cKD5Tu6nOWjPYNkz+PWMoD0v/NN6QHngAV+PfPlljh3z4bglS0hN9bVRIUTL8fdNQlWtbNFsNvu56QDhe6zcv5/iYh+Oy8ggJ8fXRoUQLcdPAbq4uNgTmqtPcfinaSGEaKX8dJOwKi5XvfDhDuHZoXNnX4+MjSUkxIfjoqOJiPC1USFEy/FHgPY61xwSEtI256AXLPD1yKef9u24WbN8bVEI0aJaLA+6tLS0pZoWQohWocWeJExOTm6b6cySxSGEqKcWC9CFhYUt1XTLkiwOIUQ9+SNAK94UFRX5oWkhRJvVv3//6jHn0ksv9WwPCwtbv3591W4vvvjixIkTaxwSFxd3/fXX5+fnt0C/q5FiSf4mWRxCNEZ5OfPnc/vtvPgiZxzmvfXWW/aTPv300/qcf8GCBU6nMzU1NS0tbc6cOU3Q40bwU4CueoCwSmRkZN2HaJqmnsY/vW1Wjcri6NjRh+NmzWLcOF8bFSKQOBxceilWK/fcQ+fOXHLJGWK02WwOPqmeD8epqmoymbp27Tp16tQffvihafrtKz/V4jh9Y1RUVN2zHL5VUxJCnMX++18mT+bWWwF698YwePNN7ruv6RvKzMz87LPPunfv3vSnbgh/jEl1XT/9ucH09PS686CdTqd+mubspp9IFocQPjt8mP79f3k7YAAHD9a1/+233x570pNPPlmfJmbOnKkoSqdOnXRdf+mllxrX38Y6GyYNWhfJ4hDCZ/37s2HDL283bGDAgLr2f+aZZ7afdM8993g2ms3m6mXlnU5n9dmPRYsW6bq+efPm7Ozso0ePNm3/G6rFArSqqm0zD1oI4bMpU9i/n0ce4auvePZZliw5w4Oy0dHRHU+KOHmvvHPnzmlpaVX7HDp0qEuXLtWPUhRl+PDh99133913390MF9EAMoL2N8niEMJnisIHHzByJDt30rMnX36JxVLX/k6ns/wkh8Ph2Xjddde9+OKLO3futNlsy5Yt++CDD6699trTj73tttv27t27cuXK5riQemqxR73PjgllH0gtDiEaQ1GYMoUpU+q186xZs2ad/AcwcODA7du3A/fff395efkVV1xx7NixlJSUV199dcKECacfGxoaescddzzxxBMXXnhh03W/YVrZmoRCCFFPu3fv9rrdbDbPmzdv3rx5ZzyknvcVm4+fpjg0TavxJKGmaf5pOtBIFocQop78EaCr1h6sis6cXJPQD60HGsniEELUk5/qQZ/+1ImmaW12GloIIerDT2NYk6nmb4I2uyahZHEIIerJTwHa4XBomhYSEjJ37tyQkBBN0yoqKvzTdKCRWhxC+Edt1ew8xo8fHx0dXT0Qbd68eezYseHh4dHR0aNGjVqzZo3fu1yTPwK01WoFdF232+3z5s2z2+2eyQ3PdiGEaJAiiraxLZ8z1wKtrZpdenr6unXrFEVZunSpZ0t5efmll146ZcqUrKysQ4cOPfzww6d/7/c/fwRom81mGEZiYqKnQJ2maYmJiYZh2Gw2P7QeaCSLQ4jG+Dt/v4IrFrN4BjMe5uG6d66tmt3ChQtHjBhx8803Lzj5lTY9PT0vL+8Pf/hDREREu3btpkyZMmrUqGa8jPrxXx5Fdna2y+Vyu90ulys7O9tv7QYayeIQwmdb2bqJTd/y7cu8/BVfVVDxOZ/7cJ6FCxfOmDFjxowZy5Yty8nJAZKTkzt06HDzzTcvW7asoKCgqTvuo7aY6CaEaKXWsGY60xUqy/jMZOZK6noU22s1u/Xr12dkZEybNm3w4MEpKSnvvvsuEBwcvGHDhpiYmDvvvDM+Pn7KlClHjhxp7ss5IwnQ/iZZHEL4LJLIQn5ZzrSQwkjqWvrDazW7BQsWXHTRRbGxscD06dOrZjm6dOnyyiuv/PTTTxkZGW63e/bs2c12HfXV8rPgbY3U4hDCZ5dy6dVcPZrRXemaS+5TPPV3/l7H/p5qdtW32O32Dz/80O12JyQkABUVFYWFhTt27Bg4cGDVPklJSbNnz27xUnZIgBZCtCLtaf8qr97P/Sc4EUHEXOb2oU8d+3uq2Xleq6oaFBT02WefaZq2a9euoKAgz/Zp06YtXLjwzjvvXLBgwXXXXZecnJyZmfnaa68NHz682a/nTGSKw98ki0OIxhjAgI/5eA1rPufzUZwh0WLWrFnWk4YNGwYsWLDglltu6dy5c8JJd9555zvvvBMSEpKenj558uTIyMgxY8Z06NDhjTfe8MsF1UVG0P7W2CyOhISGHpeRcYaauUKclbxWs/v6669rbJk2bdq0adOA//znP37oVYPICFoIIQKUBGh/kywOIUQ9yRSHv0kWhxCinmQELYQQAUoCtL9JFocQop4kQPub1OIQwj/qLh8q5UaFEKLJpcEsmADXw67adqq7fGhrKTfa8j1oaySLQ4hGyIMb4BUYBIdgFrwN3U7fr6p8aHBwMDBlypTqn3rKjQ4fPnzBggXXXHPNGfdvKTKC9jdZUUWIRvgCboVBAKTAo/CO1/3qLh8q5UaFEKLJHYf21d62h+Ne96ujfKiUGxW1kiwOIRphFHxS7e3HMKa2XWsrHyrlRkWtpBaHEI1wPiyHy2EY7IJYuPqMx1QvHyrlRoUQovk8BlmwF2ZA19p2SktL81o+VMqNirpIFocQjdYBJtURnYGIiAiv5UNbV7lRxTCMlu5DA6iqCui63tIdEUI0o9mzZz/00EMpKSkt3ZEzmzNnzvPPP+9yuZrj5DKCFkKIACUB2t8ki0MIUU8SoP1NanEIIepJArQQQgQoCdD+JlkcQvjT6VXr+vfvr6pqRkZG1ZbRo0crirJ9+3bPp4qiKIoSFxd3/fXX5+fnt0CnT5IA7W9Si0OIxjqxiUP/JmcVnCEJ7fSqdR49evRYvHix5/Xhw4ezsrKq0qKBBQsWOJ3O1NTUtLS0OXPmNHn3608CtBCiVfluNmmL0Kwc/Zq1UzHqym/zVK27+eabF5w6MrrhhhsWLVpUtc8NN9ygKErVp6qqmkymrl27Tp069YcffmiOi6gnCdD+JlkcQvju2HKCohj6KskzGPQc7S/g8MI6dj+9ap1Hnz59wsLCUlNTgcWLF8+cOfP0YzMzMz/77LPu3bs3+UXUnwRof5MsDiF8l/8D7S/85W3CRPK31rav16p1VW688cZFixatX78+Li6uZ8+e1T+aOXOmoiidOnXSdf2ll15q6mtoAAnQQojWI7QTpQd/eVtygNBab7vXVrXO4/rrr1+yZMmbb75544031jhw0aJFuq5v3rw5Ozv76NGjTdn/BpJiSf4mWRxC+K7jFay6GGtH4kZRuIO9LzDuM687nrFqXVxc3ODBg99//32vY2RFUYYPH37ffffdfffd69ata74LqpuMoP1NsjiE8J1mZdznFO1hy53krGbsJ1jivO7oqVq3d+/e7du3b9++fd++fWPGjFm48JQJ6/nz569evbpdu3a1tXbbbbft3bt35cqVTXwV9SYjaCFEq2KOpP8jZ9yrqmpd1ZY777zz7rvvfu6556q2JCcnJycn13GS0NDQO+6444knnrjwwgvr2K35SID2twceoNrfkIZ4+WWmT/ehYP+SJXTuzLBhPjUqROv09ddf19gybdq0adOmAbt37z59//Lycs+LGp8++eSTzdPBepEpDn+TLA4hRD35aQStaVqNIs6qqrrdbv+0LoQQrZE/RtCqqnqis3ISoOu6p/p+WyNZHEKIevLHCNowjNPHy6ePqduIRmVx+GTWLF9bFEK0KD+NYU2mmr8JzGazf5oWQpx9nKWtaa0+n/kpQDscDk3TQkJC5s6dGxISomla9ep/bYrU4hCi8Vb9zlbPPb2WG1Wq2bJlCwFWZbSKPwK01WoFdF232+3z5s2z2+2eyQ3P9rZGsjiEaDyjfhkGtZUbfeutt+wnDR482LMxcKqMVvFHgLbZbIZhJCYmapqmqqqmaYmJiYZh2Gz1/R0ohBA+qK3cqNlsDj6pqtBo4FQZreK/B1Wys7P91lYgkywOIXxTsN9ddKgys6A8z0j/0ul5HZakxg7SvB6ycOHC++67b/jw4SNGjMjJyWnfvv0ZWwmEKqNVWizRLTk5uXqF7LZDanEI4RtXGY4iw/Of7jaqXjtquWFYR7nR22+/PTY2NjY2dsyYMVUbA6fKaJUWe9S7sLCwpZoWQrRGcedpcedVjpTTljp7Xh9U9/6nlxu99957PR8988wzU6dO5dR0skWLFs2YMSM1NfXqq68+evRoQsPLKjQ5fwTotjlSro3U4hDCD+ouNxodHd3R2/fRAKkyWqUtPsvXsiSLQ4gmcKZRX33KjdamxauMVvFTgFYUxThVZGRk3YeEh4ebTuOf3gohAtyYv5whSbeq3GjCSXfeeec777zjctW1yKxHVZXRJuqs71psiiMqKqqoqKiOo8rKygzDy9x/a58wkSwOIRovpP0ZBpcNLTcaUFVGq/gjQHutuZGenu7DUWdBfSWpxSGEqKdWH++EEOJs1WIBWlXV1j5Z4RupxSGEqCcZQfubZHEIIeqpxQK0rute7wEKIURTee+99wYNGhQcHNy+ffvbbrvNU6Puz3/+85QpU6r26dGjR42377//PtC/f/+IiIi8vDzP9vfff3/IkCH+7b6MoP1OsjiEaIzy8vL58+fffvvtL774Yt2ZYK+//vodd9zx4IMP5uTkrF69+siRIxMnTqyoqBg7duyGDRs8S4gcO3bM6XRu27at6u3BgwfHjh3rOYPZbH7Ox+fKmoafArSmacqpNM17cZOzntTiEMJnDofj0ksvtVqt99xzT+fOnS+55JLaYnR5eflDDz300ksvXXfddZGRkX369Pn444+PHj26YMGCoUOHOp3O7du3A2vXrp0wYUKvXr2q3qakpCQlJXlOcu+997722mvHfLr30yRkTUIhRKvx3//+d/Lkybfeemvv3r2nTZt21113vfnmm1733LJlS0FBwTXXXFO1JSQk5PLLL1++fHlQUNDw4cPXrl0LrF27dsyYMaNHj656WzV8Bnr37j116tRnnnmmmS+rVv4IkZ41CQ3D0E+q2uKH1gONZHEI4bPDhw/379+/6u2AAQMOHjzodc8TJ06EhoaGh4dX35iYmHjixAlg3Lhxnoi8bt26MWPGjBkzpurtuFO/bz7++ONvv/32kSNHmvxa6kPWJPQ3yeIQwmf9+/ffsGFD1dsNGzYMGDDA656xsbFlZWUlJSXVNx49etRT3G7s2LHr168vKCg4fvx4jx49Ro4cuXHjxoKCgt27d1cfQQPdunWbPn16Sz32LWsSCiFajSlTpuzfv/+RRx756quvnn322SVLlsyq5UnZwYMHR0VFffTRR1VbbDbb0qVLL7zwQuD8888vKip64403Ro0aBURERCQlJb3xxhtJSUldu3atcapHHnnkvffeO3ToULNdVq1kTUJ/kywOIXymKMoHH3wwcuTInTt39uzZ88svv7RYLF73tFqtTz/99B//+McPPvigqKjoxx9/vOqqq+Lj42+++WbPp0OGDHn55ZerCvaPHj365ZdfrjF89ujUqdOsWbNefvnlZrusWsmahP4mWRxCNIaiKFOmTHnggQeuvPLKupPB7rjjjvnz5z/zzDPx8fFjxozp0KHDt99+Gxwc7Pl03Lhxubm5o0eP9rwdM2ZMbm6u1wANzJkzp7y8vGkvpD6U1nWnzpP44bWOkhDirDF79uyHHnooJSWlpTtyZnPmzHn++efrU8XUB5Lo5m+BlsVhYHzN19/yra/dEkI0FymB72+NzeJo+JJXGRnUMk0HcAEXHOSgGXM72r3Kq8OQpbGECBQSoNu0SUxazWrP6zTSRjFqOMN70/tf/Es545pCQohmJlMc/hY4WRxu3CtYUX2LC9cOdqxj3T72+dZHIUQTkhG0vwXOiipP4+WEpZSe4MSnfNqXvr41J4RoKjKCbqNyyPmQD71+VEBBLrl+7o8Q4nQSoP0tQLI4ggn+kR9rOyQVKd4hRMuTAO1vAVKLo4QSHe/p5AZGCSVePxJC+JPMQbdRBzhgUOszSkUU7WDHQAb6s0tCVFEU5YUXXoiKimrpjpzZTz/91HwnlwDtbwGSxfEWbykotcXobLL/x/8kQIuW8uSTT2ZlZbV0L+plyJAhzbf8iDzq3Ub9mT/PZa4Dh9dPLVh60nMnO/3cKyFaHc/6UPKot2hKf+JPk5jk9WkUFTWEkCSS/N8rIUR1EqD9LUCyODS0L/jib/zNgiWIoDDCzJiDCU4k0YJlKEPfxPtKQkIIv5EA7W8BksXhcRd3FVJ4hCMllDzKowtYkE32UIYuY1kHOvjaUSFE05AA3aY5cLzLu0tZWkHFDdzwF/4ynvFXcEVL90sIAZLF4X8BksXh8RRPhRBixjyLWbdx20Y2VlARTLCvXRRCNCXJ4mjTZjDjSZ5czOL/8J+LuTiFlPu5v6U7JURr0qxZHDKCbtP+xJ9mMWsvexexaAhDfsNvWrpHQohfyBy0vwVIFofHQAauZvU7vDOPeZdxmQRoIQKKjKD9LdBWVAEmMWkc44AggnztnBCi6UmAFiChWYiAJFMc/hYgWRx/5+/nc/5FXJRPvq8dEkI0L8niaKNGM3o96xeysJzy27itpbsjRGsltThE0wsn/E3e/JiP+9GvpfsihPBOArS/BUgWx3u8Z2A8wAOjGOVrh4QQzUtuEvpbgGRxRBE1m9m+dkUI4Q81R9Amk0lVVZPpl8Ddr18/z8yvEEIIfzol8qqq6na7DcNwu92KovTr1w/IyspqXTcSA1yAZHEIIQLfKVkciqIoiqLrelBQkNPpBBITE202W1FRUYDEaMniEEIEFL9mcZjNZsDhcLRr1w44evRoRUVFczQshBCibjUDdNXvgby8vLi4OKC8vNzfnTqrBUgWhxAi8NUM0NVnD3Jzc7t06eLf/pz9AmpFFSFEIDslQMfFxVmt1upb0tPT+/bt23yLigshhKjNKXnQubm5p++xZ88ef3WmTZAsDiFEPUktDiGE8J3U4hBCiLZIArS/SRaHEKKevATo0x/sbsJHvTVNU1XV86Wgqc7ZukgWhxCinrxE3tNnpRs5T60oSkxMDKCqqq7rnrPpui4lPoQQog6nZHFUjWqrD2+b5C6i2+32nMrzKDmgqmrruj/ZVCSLQwhRTzVrcXjf6WRU9bENRYmMjCwsLKx+u9NisTgcjobGaMniEEIEFP9lcRiG4YmYxqkaHxCLi4s9sbUqIkuQ9ZOK42Qupbx6hrtB3nfYslqsS0KI+qnXHHTjVQ/9ni2eSY82yK9ZHC4ba3695GMl9V9zcJZUbvz+DtIWs3EGed/72hUhhD94CdDJycmeRIvqGtOG18F4SEhIiE8zqq2dX7M4CrdTvD9jx46c/ChKD1duLNrHkH/Q70FyvvW1K0IIf/Cy5FVGRobnRSPjct1KS0ub7+Si0p4/Y7KiWinbTljXyo1xo1h3NfZshv+rRTsnhDgD72sSJiYmZmdnN2EzmqbVmHT2rN7ShE20Fn7N4jB0LAnR1mMRwfmsGMeErwluz8CnsR0hKBpTmK9dEUL4g5daHIqidOnSJT09vanaqMqoqxqSV71t6K1CyeJomJIDfHsR9mxMEbjtJF/L8Ldauk9CnFX8XYtDUZSMjAztVI1pwzAMT4zWT6ra0pjTijML78nwN9FduEowXORtwXCx+ynWTiVnVUt3TghxBrVmceinamQz1ZcJ9/CsrdUG+bsWx74Xl6RelXpoBIaTkjS+6E/hDoa9zo45IL8ghQho3gP06RrZjMPh0DQtJCRk7ty5ISEhmqadcanD0zNJFKWVFUf1yk9ZHIbOoX/zzUhyV2XkpeQURoNCt5kMeJz87ehOXzshhPAf79UwqoIjJ8sbNaYNzyotuq7b7fZ58+bZ7XbPkLzG6i01eGZCamjWxJKzyk+vcOBVovpjuFHNoGCJR3dzYhMRPdh2L4OeB/lhChHQvGRxeGaHqw9XGzlutdlsQFJSUm5urufM8fHxTZsl0or4KYuj+ABqEDkr0V3RwdkRwQUMfIKQLgCJkyQ0C9EqeM/iMJlMTqfTE6P79eu3d+/eAJlbkCyO+jr2DRuux1GE4cYShyOfmCFctLmluyXE2aYFVlQJDQ2tep2WltYsDauqzFc0o9BuxI0lKBqgIg/DTXiflu6TEKJhvAfooqKifv36AXPnzrXb7f7t0lnOT1kc4d3pcBluG9akJd9PSz08gpL92I742rYQogV4CdCeBLi9e/cC8+bNA4KCgpq84arK/W2N/2pxdL2BLtdTcTzjeMecko6U53DkU1/bFkK0AC8B2uFwPP74454pCFVVH3/88TOmxIlAtGMOxzcS2hnVhAFlGaS/g8vW0t0SQtSX91occ+fOnTt3bhM2I7U4qvgpi8NZwpHPcNuIGR4dUhgRpWGOwByOIw9TWywiKERr5CWLIzk5+eeff66xvTHTEVKLowXsfBzcZHxEWTqWGNwVoBPaFS2Ycf/FEtfS/RPiLOHvLI6MjIyqANpU9aClFoe/6Q7aDWH85yRMZGo2Vx6l3VB+tY3kGeSsAXAWsfMxds39pZC/ECLAeM/iSExMrB5PpRZHE/JTFkffP5G5lOXj0SuWPPZw6geLMNxsf5D0d4gdDrDjYSL7EJbCrsd87ZAQonl5D9BNnrbhQy2Os5U/sjgMFxnvYwql4+UMejYj05pz6CcKdxA9iHFLCekE4CggaiBR5+Ao8LVDQojm5eUmYVW50eobG3NDz2q1eupveGpxVN/u8zlFXQ68gqOQdoP5/ndkforjZtxuLPFkvEeX6yr3OecJtt6LonLeX1q0r0KIWnkJ0FXlRpuqDanFUZ0/sjjs2SROpvgAWhhBMdGh+RHmMlxlKNX+uMNSGPMxxfuxJvjaISFE82plBTwli6NebFmk/pb8rTjyMAA3KbeQ9SUXf1c5vwEYblZNJrwnhTuZ8JUsfyWEb1qgFoemaVX5G41cTkW0gJAOjP+C0M4Aioahc3Q547/6JToDtiysSQx9hfixFOxoqZ4KIergJUCrquoZonqy63Rdb2Q9aFGd/1ZUiR2O4cYcsWTLjNS0sUQPrNx+bAX7XgIDexZb7yF3DVHn+NonIUQz8r6iSlXasiQsNzn/1eLodguKGceJjOOdcrTJlRvzvuPAK4SnsOkmOl1J7lq0EIxm+XYmhGgk70PjmJgYr69Fa3LwDcZ9xpBXCO1M8ozKjaVptB9Px1+jO0l/l19to+sN/Pxhi3ZUCOGd9wB94sSJ5ORkIDk5+fjx437t0dnOT7U4Mj7iyCdsuJ7dT0abDkVkPle5vcPl5K5j+Ri6zsQUTvq7ZC4lvFfNw8sy+PFlTmz0ta9CiCbgJYvDZDLVyHpuvnuUDSVZHPWybipHloKBooGbCcs4+C+G/IPg9qfsVpFHxntE9qX9Baceb/DNKPr+iZ9eY/BfiJBK/0LUyt9ZHC6Xq2/fvlXlRvv27Rsg0VnUV9ZXqCqKCcMNCptuIm8LOx8j/Z1TdrPE0PPO06IzuMvRrHS8grjRlB72W6+FEDV4n+LYs2eP2+3Wdd3tdu/Zs8fPfTq7+SOLQ7Ng6BguMIgasGTdhan7e+MqI+sL8rfU43AriZNYPoaiPSRM8r7PoTdZOYEdjzSg/0KIBvISoGNiYjzDZw9VVePj4/3fs7OVP7I4OkxBMaMoqMHYsjO4JsfWE1sm1kQcRfU6Q98HmbSOUe+heqvK4iwi/V0uWIGzUOaphWg+XgJ0fn6+54FsT5g2DEPuE7YyMcMJjmPCMsJTCE4gbwvlxyj+EdVCwmkTGj5Qg3DbKdqLLQtTRBOcUAjhjfcpji5dunjmN3Rdb44FCdsyf2RxRA4gKJI1l1O0F3dpdFxoxMTXaTeEQX+Gk6W9y9I58A/yt/nSE83KeS9y8HW63kBUf1/OIISoBy9ZHIqidOnSJT093fPWYrE4HI4AeVZFsjjqa9kIInqh28ndSMUJovrT/kLOmYtmBdCdrBhL7/v46VVG/IfQLgAlBwhOwCwjYiEawN9ZHJ5yo6qqapqmqqrD4fD0wKM5OiGaXngKSZPJ+hJTKO3OwX4USzTrrqr81FmEJY7O1xA7sjJPY8ud7H6KVZMp3t+CvRZCVOf9UW/P/z2Pele9bpKlVYSfanEM/huFuwlL4fDI1Q8AACAASURBVJynlmz/beqxq+j7IG47hhvAEktkX5aPoTSNuNEARXs4fyF97ufYN772TwjRxGqtBy2aSWOzOBLqV77ZEkvKLEp/Yu9TGZkPWiJsfHkuCRNQTn4HGvTsKfuHdSf1dxTuYvi/fO2fEKKJeQnQorVy20lbTEUuWV+AgbuC8xeQv5WPMxl4IzEQM4wtd3FsBZ2uYuBTJw9bCtsY/iAFpYR0wBLbkpcghKjGyxRHWFhY9Tzoxq/qLaprxiyObfejV3DkE2KGoQRRtI8f/g/VFG09HtHtfDpcRs4qsj6n550c/Ce5awHYAO/CKLiZ6IESnYUIKF4CdFlZmScPWgJ0c1iwwNcjn36ajh3r2qH0MF1nknQJmf9FNZF8PUX7yP5m1h/PHeccz8HX6XojzhJihmCOosRzM3A+bIbXwAlyg0GIwOJ9iiM0NLS0tNTPXRGN1e8hVl+KaqL3H/jpn8SNAoXifeSlcsE3lbl0HbqwbiTBZjp1g03wJVhhLVxfW1J8o7jtqOZT1kIUQtSb93+Tdrvdz/1oO5opi8NlN9Y81ufrv7+RlvUiWhhoZHyEohI9aMmOO1LffgaAIs4/wq8fYXIsQQ/Dp9AV5oMOv/G1W7U79G9W/YrlY2RJLSF84yVAm0wmXdc9j3pX8X/PzlbNVIsj/X/OhJh/X/xq2r5FsP1BkqZgCsKRj2rOyAzKyQsDcBWzz86h5zCOw3Y4BGHwCvSAO+G30KTJ9unvcOG3nPcXMt5tytMK0WZ4LzfqeWFU499eiQYLac/xw72Ppk/SQkIxXATHo4URN5rs/+Gy0fkq0PluIAfL2FvBKgdUwKewH7oA8C10hC+bsk+R/dn5CPueJ3ZkU55WiDbD++Rg4FToP/s0UxZH0pgg50/m4x/9e/Ste+h4C0c+wVGI20b330RHuSK6nQ/pFBbQF6JgnecgBQrhY3DBI7AWxvvaOW8G/41jK+h8DdHnNuVphWgzvNfiCAoKqqioaJEO1U1qcZxZxnsc/DeKQtkRXHZCOxAUxfmLsLzD4T+wDYIgHCac3N9QIAqlDKZBCLzekp0XorVp1locXkbQiqI4HA5PKnTVxhqLYIkAVbSbLXcR2pXyYyga7mLoQHkullj4kG4QByXggIPQDVRPebtCCIfR8EFLX4AQ4hdnqMUhJTiaXPPW4tjzZ6wdcRVTkYe7FNWCally9L3UVMACEA4ZsB1+hJWeYzxfoczwMNzua+eEEE3Pe4A+nf97drZq3hVVgmKIHkBZJkExBCfRbijDXs/I65Gz4XVW5LMPgBMwGEZCGQCG6eRA+k24BsBRSGma9/M7CjjyKbYjvl6DEKIBvOfPaZpW9QyhlBhtTeLOp2AHoZ3QKzCFMuyfhHYhZxWuMgb+lQMKuyEZNsNK8DyWqLhAASfEAOR9z5pL2f4nds2reXLDzZrLsP3M+usor3ddPSGEr7wEaFVVPXManjloXdclD7oJNe+KKqWHGfQsk9YRnsJFGwnpxMYZ0aEnIkq/ZPXlVKg4gwiHy+ByGFJ1mBu6wd1gkPUFAx5nyHxyV9c8ecUJrIn0uofESRTt9fUyhBD15T2LQ1XVqruCmqZVFYZucZLFcQa2I3x7MfYsrB049zk6XMaKsQyZz9qrKP+ZBDPtygnS6VnjT1OBOOgE35P+Ht/fTlAU1g5cdNqCsJtvwX4UDMZ+Vrk4ixBtm7+zOICYmJjqr2XR2FYjpBPh3bl4M64ytt1Hh8tIupT111KWjhkybRgRjCo5eWOwOhO8CAr2LIbOJ2ogW+72cv4Rb+O2S2gWwj+8z12cOHEiOTkZSE5OlujctJp9RZXESXx/O1vvJvEigPYTlmyZnpp9GcntiNBwl1CsV94e/IUBV8O7APFjOfQ2aYsIT/F+/sZF54L97u+fLk//wtmYkwjRRngZQWua5na7MzIyqvKg5T5hE2r2FVV63kX8OMoyiD6PihP88McM93xLznaO5BPvpqfBYTDgPDjlT/UmuA8gZjgj3sKWRez5vna0dgabHrIPe8y69+2K0A5q3LkN+3ulOylO1yOSVdXc9F0TIgB5r8XRt29fz4Mqqqr27dtXHvtuHQydYyso3kfmZ2R8wOabyPqcsG5EnUPcCKKs2BQKIBl6nT7J8SCMrXwZmkzcKJSmvzNsuFFUYvprEV3V8vyG3UhwO1h+Y9mPCyqW31jmdjR514QIRKf8I/Rk182dO3fPnj1ut1vX9UcffXTv3r0ygm5CzZjF8f0dHFvBtvs58gkjFzPoeQp3Y4mLPvpcRN9dTBjBOTouCAVXjer8VrgSnqh8V36Mn5c0RyKdYqLXDZZlN5SV5xsdxzdsGFyS7o7qqY54yhrdSy0+LM+1ijbhlCwOT+JzjRwJVVUD51kVyeKoy4rxTFxF1v/YfDMoBEUz/F+gcHwTyqsYyfRaR7mBCbKhq+chb0CBzrAYRgM4S1h1EcnTSX+PC77BFNaCF1Sd28HymWXt+mn5e9yTFoZqlpbukBCAn7M4goKCTt/S+MJJnly96luqZ/KJptFpKisnUpZGr3tJvp4N1xPUju0PYo7AKGHwWnQIAwO61Tjyd5XRGbD9TPS59LyLon2UZRDZz99XUQstiEkLQ4sOuiMfCJboLNqImvOMpw9OGx9Gqz/5UrXCYZt9/qUZszh63cPYTxn2GoU7yEslKJqydOJGL9kwPvXoMKzg+eVbVQKr8tduGDz4y0kielOew6qLKc8lorevfW0WmoV2/TSJzqLtqDmCPn2g3vgAbRjG6ePl08fUbUTzZnGYI0i4CBTyf6D3PcSNIv2dDHsPi6XmFyPK4TikwZg7qb4msKIx5mNJdhYiEJwyhlUUxRNMQ0JCgJCQEM8EdONX9TaZav4mMJslVarZtBtC5mccW8WaK+jzAFYXp/8BatBJJTiRkpleziDRWYgAcEqA9ixFaBiG3W5XFMVut3uic+OHug6HQ9O0kJCQuXPnhoSEaJoWmAsC+EHz1uLwKNxFwiTOfZ7gBNZfGx1zPCLitDJ4msLP0ynsSmiyrx0SQjQvL7U4IiMjS0tLPaE5LCysqKiokW2EhIR4XSbcarXabLYGnUqyOOpiy+TIJ8QOJ6Ivay4hrDvlxyhbxyVlp6U9q1T8QF4mwfHkrCJ6IAkXtUyfhWjlmjWLw0uAbiZJSUm5ubmeuB8fH5+dne3DSSRA187gm9H0/gOH/s3gvxKWQtnPhHVjfXdGp50yxWHACgsXbsPoxjcjcdupOEHSrzh/YYv1XYhWq1kDtP/yKLKzs10ul9vtdrlcvkXns0NzZXG4yzFZ6XwN8eMpOcSRj9n5CFkLSHAvWXJtauqwyt0MsEMYFI9g07WUHkI10fUGivZgSNajEIHFTwG6+goAbXwdgOZaUUWzEj+O5WPI30pEb9IWM/ivmB8mWMnISM7Jaf/LnkegpD2aE6WULtdT/BOZX2BNRGmjfyJCBCzv5UablicVhJMrAHBywcOq/GjRNPo/Sv9HAcrSUTQ0K+3yMGmgY1QLvlHQ7ShaOGU2kvsR1Z++D9HxspbqtRCiNv4YQXtS96qvQlu1xQ+tBxp/ZHGEJtNpKlumoQah6NHt8iIiCyo/UqC9GfMM1DTGfY6iMfgfdL4K9bREaSFESzvlJqFn2sHzRInZbK6a9jabzQ6H7wXEFEU5/Xnx4ODgioqKhsZouUnYEG4IAuNkCkcI2ECDRDgAVjLeJ20RqpkOl5Iyu4U763cndrh1J/FDZGJHNIr/bhJWX9rK055nUsLpbGx5dR/yoDVNU0/TyG60FQde4ZuRbBwNoJ+Mz7oNtwKj4BBYAbI+Z/i/GLmYzP8C2LIo2lPrOQ0Xm27kmxGsvZKSA81/Dc1r/zuOHxdUHPrYsfuNuv4eZn7r+uqa0m9uKCvLljGBaAE1Q56nWFJwcDBgNpt1XW/83Tyr1Qroum632+fNm2e32z1DYM/22jT+8cXA1OwrqgBpi+hyDYPSMQwUY8mSq1M3DgNwgHMTx56t3K3jFWy+hfXX0ukqcr5l803sfZ6dj3k/Z+56QpJApfQgm27CluXrZdSD/Sjf386Wu3HkN1MLuVtc5/5f8OA5wTnf1TXw2fPviosWh53ze8v+d6UEtWgBNW8SekapngkNz/9DQ0OL60geqAfP0ygNzYP2+pXhLBhEN/OKKga7n6TgBzqfwJKDSwEy0rtYLBXoUA5RAzH/Bf0hVDOdryF2JOiEdGL7Awx8hphhrJzg/cShnTi+EcAUSvsLKDlASIf69nzfixxfR/fbSLqkXvtvf5CUWTiK2DWPwX+rbysNkXJV0Nq7bCj0v62u2ksh8UraUkfebnfMAJkJES2gZoAuLy+PioqqPjVcWlraJC215dxn/9nxCJmf0zOYrocxDMwGBuhQAfkQD/oR3CrqyT/3qiDb4TJ+uB9LfK2PFIalcO4LbL0Pw0XRXgY8RkUeu59EUej/KEHtau3S8Q2UHGD4W6y5nISJqPUoRme4MIWhO9GbaymfpNGm9kPDDN0wWev6ojbiaevhT5wJI0xdJkvpGNECTgnQnrw3z7PdVeWNmumOXECtA+BPzZjFcWIz+VsJ7ULoXlxQaiGmAhfREQURRcXEKbijUPIJXszpxZPiRjPmU5zFhHWt9fwxw7lowy9vd9xDh0sx3Ox4lKGv1HqUoqK70B3elhKvxaBn2f4QqplBz9X3kIbTLHj5OZzKHKr0min5LaLF1HzUOyoqqrS0NCwsrLCw0LPFZDI1R20j3wK0ZHHU5egycr4lrCsH/8gFdoKq/Wyd4FQJCYcx8HnTNLf5ZnreiaHz02uMeKuuPQ/8g9x1pMyuXGhciLOIv2txaJrmCZ2eR/4Cat0TCdC1KjmAFsr+v5K/lR6/I+Fh9INYQIeDkAJaEMoj8KgvJ9eduMoIijplo+0I2+5HUTn3BUI6NslFCNHq+C/NLjk52VNc1BO1PY+WKIrSr1+grHt0FmiWLI7dT7H9YZYNJy+Voa/QeRp6AfmggMaSnVenbhuG0gO+86Xdnz/iq0Gsn8be5wAMF6m3sWIsR5cx+gNGvdcE0bn0EFmf4/ZS8lCItuyUAP3zzz9TbfLB87wfsG/fvkY2I7U4qjRLLY6clbQbRNeZ2LNZPoHivRzWiAEDCsnY1iUnOwkyoEeDGz28gD1PE96T2BFkfwVwdBnWJC5cTdpC9LonvgzSFrLnz1ScqGuvoj1svpWiPayf1uDuCXFWOyVAe6Y1qs9puN1uTwn/RrUhaxI2t4SLyPiAg69jicdVzNfDKHdhhiIoAzPs18EELzbwvAY5K+n9R1ylpC0mrBtAcCKFOzm+Dt2JUmduw+GFFPxARC9Sf1vXbvlb6TqTvg/iLGnAjUQh2oCaIfL0gW3jh7pSi6O6Zsni6PcQI99FC6FoD/Fj0KwkVqCo/KiQQHSXwghzKWzAUUTmUsq9zZPkb2HTTDbfwomNlVsMF6svpewIW+/GkU/XGQz/F0C78+h2C0eXM+ItlDp/xdqO0P4Ckn5FRV5duyVO5vDbrJpM7PAzplUIHxhyy6bVOmV0rChKly5d0tPTq++RnJyckZHRmGAqtTj8ZON0KvIo+QnNTddjtAslsgi7wQ8dSCqgVx4rLqDTlRz5mNFL0MsrR8Qe34wivDslP6Ga6fE7ulxP0V4OzOe8l/lfP0xhdLqSAY83rD/luXx3K85i+s0h8eK69jR03DZMYZVvC7aTtZSEScSe38AfQVvkLDHW3GVz2vSxL4eGdvzlV2bRIX35TWXOAt2aoA15JLjjBH+UrmyD/JfFUTX/UEMjE5Y951RV1WKx/OlPf3r++ecrKio8QVYCdFMq3Ml3s3GV0XM0wQfI3kNoET1SMHVDK6L0DXY9y7DXWfNrSg8Rez6qiRFvVx77zUgUldDOWDtguDjvL7hsfDuR8O7kruGSPaydyrilzbiYbOFuMj8j4UIi+7NqMufMY8/TjHwHa1JztdiaOYqM7X8t3/+OQ3ee/NZhoKhY2imWcNUwjNKjeo0bBDcejJSK383BryuqGN40sg3fanGcrZo+i6P0MLZMgMz/Et4Dt51dn7FxDUV57NNZ+uOSf8emLutMaC/cdj5PoXAX5jC63UzZz798+x34DEW7yfofB98gdhSAKYTxXxI3mtCulSkWWrCvXT8Tl43U24gZyo5HKN5LSBIJE4noTXlOc7XYmh14z/HBsOJ9/3HoniJmJ+sVGjrlJ4yiNHdx+qnR2QD44vIy/3dVNFLNm4S1aUwbNpvNMIzExERPgTpN0xITEw3DaOiKsWeHJs7iOPAK2+5l801kfkbOt3SbRexI3HYMjfLOGCa0uIzDN+Zs7YnbxugPCU3movW4K9g1D0u7XyaR248naiApv8GayKHXKzcGRdH9Ns59AUcBoz9sxgliZxHB7Um8mKj+ABF9WD4G1UL0oOZqsXUyXGx62L75EbveoNpNCkD+3uZ6bl40n1Ompeq4H9j4x1WkFkezSF+E4cbtZN+LdLicPU9RtI/w3hT+QFkGihl3OU4bbjumUIAu09k4E3MEHS/nyEesGMt5f6XdeQARfch4j9BkLHEU7SHyZPJ7zFBKD/H97+n4a7rd3CxXYU0keiDLxxA1gJhhxAxvllZaA2eZcfgTp6WdkjTapFkVLQgM3A72/KuiotDISXXm79FrvelnnOF3qNuBJg+utyqnBGiZ2/WDJs7iMAwKdoCBYmbSejpfQ9ZSCrcTFE55LrEj6XB59H41ovcUDANFoefv6fFbUPlvJ8zR2H/i+zu5eCPAoD9TuIP+j7H3aSxxvzRRfozDbzPyHb6bTfwYwlJ8vYA6DZjLgLnNcubA5rIbebvc0b20oEgF2PyIPXGkaf9ix/a/lgeFKyiUZRua2ehxTVD+bnfeHp06/o3WWLv91GCtmSU6tz7e06SqnvZuqmloUWXBAl+PfPppOp72zJ7bjqIRfS66i+L9hHWl513ET8Qcza+2U7iLA6/M6nH1OO16Vk+pPOTAq6y6CF1HDSKyL7Z0HIUAQdEM/Sf533PeXwiOx9Bx1l5m1pbJ7qc48qmvF9Om5W51L5tetuKWstIj+opbbNnrXCtuKasoNICyY4azVM9JdRWn6fYTxokd7opCvSTT2D7fkbPVVVd0ruHU6KyojH+1fuuliUByygjaMAyTyeR2uz3Pp3jS48rLy1uqc+LMQjpRcgjFhKJw4BX63MeueSgaOd+ydirF+3AWEdKF6IHkrgco+IG87xn5LktTKD+GasEUyobpdL6SlNlED6qc9i3PZd2VmCOJ6M15L9FtFt/9ho5Tfxk+b7qJPveR8QHmMBImtdjl+6rgR3dplt5hrFltaBnR7Q9wYjPhPRn2+hnSwGu342/lE14LKTyg75xfYbLiKDKsccrOf5QfWekydFKfdAEYlB7RUdAsirvc0B3eBkmeYbJaGY3VICU4RjGFKkUH9aBwYvtr418PXX5jadfLzOn/c8YMkjS71qfmn1lVsoimabquV1RUaJoWUPWSWrsHHuA534povvwy06fXLNg/8Gk2zaRwJxikLSLtP0T246JNWBPZPx9XCZp1yaohnWMzh40aBqBoGE4MN1Hn0G4IZYexH2X0e2y6mfYTfom/R7+m2y2k3MrKCRg6Xa6ly7WntOsuJ3EytmzKMny6mJaU851r9+sV8YNNaUttY//WkHGlLZPSw0xcw9Z7yN9KzFDvex3TN82xO0sZeI8lcZSXsGiNVbJWuw7/11l00F1yRFc0DDeeBNea88sGjqJavr+qhCWpupuKQkNVjU4Tg4Y8HGw/oRf9pHe60OS0Y41RUBj/Skj2etfoF0OssfIQUOvj/ZeqJzp7Xp+tS0+1lCZeUSWsG4OeZesfQCG0K7nfogax7mrsWfS+l12P47ZnnEi2BGu0Pw/Ano09i3VX038OSZdQtJtVl7DzYfTyU74Vxwwl9XeUHMCWTeZndLqyZmd6/p4V47Em/JJJ3Xqc2OnuOT2o00Tz19c2MPMsqB22LLI+p2AHfWvNl9z/jqPvrZa4c7WVs2zB7ZSiw7qzzOj26yDNAuAoMeKHaev/z6Y7ie6pqWplUPb8X1FPjdGKwukTjEplNB/8QHDypad8BbDGae36aFB5SxgISVC7Xy1zz61W9YlmT0kjz3bP/EYdiXctQlEU+GLiRGPiRGPtWsMwjB07DM/b1rJl/Hhfz3P1ExNHlp2yZbt74nnfTzw/beI5a9bOvcB437rjxdETB66deH7GxAtdaz/73vggdMefz+mRcGBgr+yJw35cu7LQWDFhx8b0iYO3TZxgmzjRWPvPp40dj+7454yJg3+o2foE28RBGycO3bP25d8bR5ft+HLpxPMPT7ygYuJEY+3KQsNRFFA/1QZtmTBGHxTveuEC+55/lzf4PF+lG3tf2LF2Tx37vPOYY8sz9s+fdwwId57b3jUg3PnqjPKND9l27DBGDdIHxrj6WZ1zEksX9ix8unNJP6vT89+chNL/JBc+mVRzy1MdKrcMCHd++Y77m5ml8yfaBkS6xo/UA+qn2ma3wMun37RrKjWfJKSWIXOAJHicBU8S3nSTr/cJH36Y228/5T5hxQlSf8uYj9n5GDnLUS3k78BwAsSOJGYoP/6F4Ni3vrokpePxcfc8x94/k/k5Yz7kp3/Sbw5R57B+GufMozyXnz9iyPxTmis5wLIRDH2Vg2+AQdRA2l/A4bfocBk/f4jbzoC5tL/A9x9EyzJwOzyLqtTY7uaH/6NgB73upuOvz3iaooN67jZX7nfu8gJ94B+CY8/RAEPn8KeOnO/dIQlK+v9chpvY89T8He6L3gn7alqp22HYjxmenBrDM+dUlaCsoJlxO1BUMGh3jum8PwaFxGtuhxHRTTWHKp4OFh1yh3dWtWD5ahsQ/Pqod237GYGRyHEWBOgmtvEGXCU4Sxn1PptmcuI7OlzKiXWUZdH7Pg78g/jxFO8hbhRh3TFHoijseoKESZQfxXAz8GkOvIIWzHkvsu9l8lLpcCl9/o+t95K2GEceqoYpkqg+9LqPpMmsmowjD3MkWighHc+wkEprlLmU/C30m8PKC5i0vu7bgCUZ+pq77IX7Xbpb6XKRqbzQuPjd0KpPj+9wrf6traIYDENB0SyoQRg65XnV/8VhDsPtUNwVBgrWeDUiWU0YrsWdZ0oYYfLy+0MEnmYN0DWzOJqjDdGMRi7GUUBQFCgMmc+qX5HxPpoZNYiclUT0JrQDeZtxFHLkI+w5hHbGmog9i6iBFO3myCcMfJrtD7DxBrRQLlzJ2ivpOpPMT7HGExxNSRrWOEK6kPYfdjyAKQLbUToOpWA74Y1OiM5dg+4i4YIAKmJnDseeTclBMKhlvGLoFB10h3VUCw+4KwrcceeZCve7M9c4g8KV7HWuxFGmigJj0yP2/N0uLZio9mr+bjeGoTsxRyoJw7SsdW503Haj40Rz3i6X7gaVpHHm4XODQxNU7+N60VY1ttazn50FI+gmzuKo4iikPIfQznx/BxW5WDuQ8QFh3bC0W/J11859ew5r9zpuO+0vIGclrlLUENQgDDtR53Huc6y/FttRNDMGTFzO1j+SvxUFDIVed2O4SbmF7Q/R9QY23ULve8nfguHGVUryTLr/xpfL2f83ivahBmFNoN8cn34izePgv8jfSvfbKh+wPM23v7GhGGXZxgVvhHxxWamrHN2BohpB0arhMNwVhtuhgKFZlb63WLLXOU/s1GMHaiUZbmcpphDDWYwlSnGWG8FRalCEMvI5a8wAKWLUivlvBC38oImzODwKd5N6G+HdscQw+K9svoXiHxnxFp2uAjIWzrOo2xiUjSUGZzGh3TCHcXw9A/9C2mL0co4uJ6o/GBgu+j/BnmeZ8DXfzaZwF5qF9EUEJxLVn+Mb0V2oZgp3EzUI+xFKivnh/4gZ6kvRjLxUBj2LKYzNN/v6E2ke3X8Dlb9yHMXG2rttjhKj66XmPrdYAGeZkbfHZbYqpVn6+vvLL/tf2NZnyzPXuJJ/Zcnb5Yo5x5TxpcPtILKbWnbU2PtWeWR3k2IyCn9y6xVGytVBw+dZVRPOMsMcouTtdoe0V63xAfPtQQQeCdCtnOHmu9+Qs5qYwYx4i28nYY5kzCfV90CzYO2ANZH2YzEUQjqRuxp3OTseIqw7YV058A/cFViTcORx8DXM4ZQfo9ddxAxj7a8JTcaWwQ//h9tGWTqhyUT0oNMVrJpMz7twFnHwdYb+s8E9T5nNumtQFPo/1nQ/jib28zfOzpPNPa8L+uqaUk+ANocqzhKssUpYJ9WWrad/6UwaZ+5xbdCOv1eUZell2TqoaHpJlqGoTHw7JH6IuTzP2PmP8ug+Wo9pQZ65HM/tPhk4izOSAO1vTVyLI3ctwe1B58jHfBhJ1xk1dyjLiLZkRoQ4MJz0+D3b7iN/C7qLkCTcOimz0SsY9CIrL6DbrRx8lbjz+fkjvjwHUxjtxxPSlaPL0Ey4bHSeRvGPFOzk/AXYMnHZSVtQGbW3P0Sf+7HENOBy2k/govUYBg1+mK/JHFnpzFrl6jzZnDTa+z+EqO7azvkVWhCmaikTPa4J+mmJIzhacZbqGJSku23HlIvfC9VdHN3kPL5FLzrkNnR6XhcUP8QEBMcow+a2xcq6ovFkDrqVK/mJTTMpPUxoZ5xl9L2fbrf+8mnRXg4vIKIHahD7/kJQBMc3oZpw2zFbcTk59wWOLaNwD1ow5nB0J+0Gk76YsBT6PcS2+7l0L0tT6HQ1R79CDSakA11vJH4s668jPIXCXbhK6DuHsGSOfBIQD60YLlJvp2Q/3W6l20117FiapW9+2D70EevmR+3jXwmxtPslBLsdlP6shyerqoncLe6C/e7kKWZL9C87HNvsKtinH1npGPePUPsJY++bFSOflRDcRskctKhdeA963cvG6bhsGE7iRlZudxZz6C1yV2MKJeNdTKGU/YwpFMOJ7gJw6+AmbQHB7XGWENGH/O+w3pWlDAAAGNJJREFUdiB9Me0GU7yfzbMwR/LDA0T2IWoAUQMxKuh1D6qFzP/S+WpyVuIowBJLzGCCEzgcANEZyPqS0M4Me40V40ieXsfw3GXDEqWU5xv5e13Lbyo970/WpDEmwF1uLL/ZFtlNLflZn/Sf0PghWvyQmnMRCSNMCSOIOUdb9VubojF8XrMtZSDaNllX29+aZkUVRz7LR/FRNF8NomgXET1pNxhTOPk7AU5s4rNO7HyEgu2c99KSLdNTt8eiWajIw5pU+YeuOwhJxmWjxx10u5GclShmQpMJ64k5GlcZcSPoegP5qZz7Etlfkf893X+H6kkBU9n/N4r20nUmF6xg30tsvYeBzzT2R9MkrO0p3MXxjRg6al3jj6geanQfbc2dZX1vDb7onbAfF1TWwC88qMcM0M5/xhreWS3+ua7vavGDtckfhl78XmhUT5lNFs1CRtD+1jRZHEc+wWmn5+0c+jc5a3GUMOg5dj1BWRrAT2/Q8deEpbB7Hkt7ZmT83tI+BkVDUSg/RlA0zlIMcJehq/z4IgZEDyC8F3nfYe3IOY/z7ToK9nBiK1oQx1ZwwTen9GTf8/w6g4x30Z2EdGDMx436iTStmOF0uZ6jXzHirTOmV/f/rUU1UXpE3/XPinb9KgcrkSnqie3u758qLzqsh3eSEYxoSfL3L6CNHj162bJlXj6I6IMtnZzVKBolP5I0mS13ERRNyq0AnaeSuZT9f0ULxRSM4UbVcORjQFA0UYPocAkRKZQfp+IEed/jKgINexYJFzFxNWWZdL0RaxKqmSsyyF1L5eJ3J2nBFGyjaE/D7gr6TaepDHyGiN5ePzR0ste6crdUFmjsO8vSYbw5fog28J7KaQqTVbloUUi3X5snLQxtufuXQoCMoP2v/lkcn37Kpk07HnnkYFBQaN++PdpXz+KIG0WHS8jfimImKIaSA/y6WtlPWyZhyRTvRXehl0eHFUdY8gE0M+Z2VByncAfOIszhGE7cDgwIT6H3PcSOBGh3Hvv/RtLFHF7Iz59i6DVncoe9wYH5RPSm8zWN/XH4i8tmuCuwRCvbni8H7MeN4nSt+9VBKHQYX/NfgRasSA6cCASSxRGI0tPTV6xYe/fdR8vLH7NYxur6jhtumDF9+qXDhw8PCwur3KniOEt7goElBmchAx6n592VH60YT+xw0hbiKAIddwWWWMKSyd9O52vI+oyQZOyZuEorJwF63s6RpVz8HdbEyjOUZVC4C0ssxfvocHmAjpTr7cR2d+qTdnOo0nmSOWOZc+LbocVp7h8XOkY8JakXorH8Vywp8LWRAP3hhx/ee++fjh49bhg2PA9cK0Fduw6ZN+/2G264oXKnkgPseY7yYxz9muAELO0Y+Q5R5wCsnEDBD+jOytLCuoPwHpSfQAHNghrCgMf46RXcLmwZOIswhXD+AjqcuXhbIMjb7d72fLk5TBk+z2qNq9djeFueKe96mTmmv7ZselmvmUH7/r+9uw+OoszzAP7t7nnJdF5mEiBmAiYjYoAE5BQXfNsVhFML9FSEW09WXNkrKHH1SpZj4VYleGfd6p54WiUrKpS3d7DlbbbKrasTvas9FddyDxFZ5cXoIYEAEiSYaJhk3rrvjwfaYTLJ9EwmPd2T7+ev8MzT3U+37Xd6nvlN95aIlsDMZt/oabxMpqEa1oDmHLTVzFRxzJw5c+7cTYC4WNYB6Hr0ts93/t3SJ3fvPtfp5Dtob8FXH0JyY+4O9HUi1gMA8R7oCUx/Bq4KSG5UNLTsvnfnn0Yjfubss2XdKtxlKG+A4kZpHb7zSwSmOSWdAXz4T33ffVqduNiz/6WIyUVqrnTt+ee+/23uHT1NCc1z3/jrsnktZUxnsj8GtNUyVnFEIpF77lmxdesKXf8yuX0i4mr8wDXX1K5cuVLXdRzcgtuPY9wCeEfhvbvh8kGL4qNH8dUeaAmMuRbV38Ps13DhosMnRnUo8zDqCmhxzN+PCxei831cvgHTn4Hiw9HfYezNw7jD+eapkE59lDi9P5H8y5HBjbvedcXPfONv9UxfUwJAdkNiOJMT8EtC29m0adOJE6eBY+LaOZmux1XVv2jRIkmS4FLxxRuInsaUh6FFMfpK7FmLqc3YsxaNa7D37zF2Pvo6EPkSVdeicyu++QyVl8OlYsojZ1fnHY1ZryF+5tupZyeY0ezbvyXiqZAmL83ivpz+8bwWIefhWWu1jFUcDz74oCSdSSSiKe2ngDDg9XoOHDgKAFf/K3oO4qLFuOQ+TPwbKCVnb46hlKDyUsx86eyPCX3jKqfeWjFmDK7+N8z9n9SNuSuclc4ASqqky1eVTFnmVfikPSp2/JLQdlas+PcXX1wej3f3v4IGIEkV48a93NJy+4wZAID4GRzfjopJOHMIH6+HJKHvJG49jE+fQ+QUAlOxeyXKLkbj36JmLiR+YCLKM35JOFJ88803N91007Ztz3i9MSDt53dJls90dNx7442hh1Ysxhdv4A9/iXA7dq9E935cMAelIcR70HMQfScx6gpccD3KJ2DO7xHpxH9/D/91Fbo+tnqviChXDGirDVLFUV5e/vrrr99117ORCIC+lFdXAjWALF/d0LDjjy3PPn1nGAe34OQOjL0ZocXwVKH9t/jqY1z6D9i9ChN/jINb8M4dmLq+pQU7//Nd/PkOTH8Gh3J7YC0RFQA/81otYxXHxo3TP/zw2l273o3He5LbJ6LkOxND0sTQ6nv/MLHkLVwwD8e3w1WG169A5Z+hehbcFdA17H0MuoZTfzRukXH4MLzSVOz7R3z9Ccb+xTDtFxHlHa+g7Wj69AluN1Lu9ePzee+666/unq1PLN8DyY1d9+P4dgSaEJiCOb9H9z5c+xvEe+BSMfctHPrVeWucsAyBqbhkBeq/b+WOENFQMKCtZuZeHGvWrLn00iZZngpAxLQkjfv+/fffuXTpzdMisYvuQ+18lF+C+oU4vRuhxZBcmLAM7/0ASEC9CHsfxwXXG2urrESFX8G42zDmmmHaKSIaDqzisK/Zsze+9daPAZckxWXZX1YWeOyxh8px/MXnn19yfe2tVyI484FP391Scn1LXV19R0fHC5s2PvLII/jqT4A00BOpiSi/WMUxQl111VGfr1SSxui6nEh4e3p6f/7zl//6oaffO3Dmvuc+Cf2oZ8urH/62/fYlS5YeO3bs5ZfP/Kbl6MHPXa/uaP+8K1DosRNRHvAK2mo//SmeeCJzt7Y2XHfd/Pb23bruAw4B0kpgG6QTMPZdcrtvUpT2WKy0vv6RCRMS4fDmcPjRQOD5adNmzpkTnD9/vujX0oK6OpytmyaivCqqZxIqiqLruq7rsiwnEokc1vCsrmP5cgBYtgzTp6O1FRs2nH3NCS07d2boE/3hDz/2eB5bfOhnR94CwgBeAD6APhOYAb0bMFrGx95cGYsAwOe3vHy04ngwUHly04+U/5De3xXfVhJ7tcZ95XS0th5+6mOvvwP1HxV839nCluJrWQU8jeFixRW0JElVVVWdnZ2yLCdvTpKkbK+FZVm+DPjg/fcBYPJkqCqiUezdC7FaJ7TMnq+++cZgfQ6VlNxw221H/u/EFJyRoAM4AISBF4HXgCOA0eIBpkCSAMB1tHz5V5FXSt2N05RP4zFZ06Te0Jx3Pniu1O1+as2XDRf23vK97oLvO1vYUnwtE73eg46+H7QkSX6/v6urS5IkI5RFWGe79SKY4rjnHvxLph+LVFdXf/nlaOBAcuPjwC+Bo+mXkNzueRdffCIQqN+16xOXy1VXh+7uibNmyZs3b37lldKLL8Z11+VpB4goieNv2J8c0MaeeL3eaDQ6AgPajEmTJrW2tgFm7ncsA+INr2bcuMqHH35o9+727dtLAWXjxlXz5g3zQIlGvGKYg/76669FthqJXPQhOxSLFi16/PFfmHjzkgAF0ABl/PgffPbZLwDs27dv1qy97e3tTGcip7OozM6YzTACOrdvCItAxieqRCIRXderqmYC3uQfE4p7cZxvBlBbWjo9EPCLdAbQ1NQ0d+7chQsXGp1aWjL/vpyIbMiKgNaTGBfOqqqqxjOqR5KMWenxeO64444bbthQW3s1UGG0Tzz7jxIAgARUud0nrrlm3LZtj+7Zszt5DaNGjQqFQsY/Dx9GR0eeRk9EFirYzZJ6enoydxqRJEm67LLLNm/G3Xff3tpau3fvrwEXIAExQHe7fS6X5Ha7q6oq1q5dO2PGjEmTJpWUlBR61ESUfxYFtKIoKZPOOddBO52Ze3EA8PmwcOGY6uql27cHt2z5VVdXZ7hEra4Ovvra7yKRSENDg/nPH5WVqKjI3I2I7MaKKg6j/FmSzs6oGv/MoQ4aI+wLRl3XI5HI2rVrlyxZcujQoQULFhR6RET0rWIos+t/vSyuqVlmR0SOVgw3S3K5UudS3G63NZu2m4xVHAPasAEnTuSwHKs4iBzKooCORqOKoqiq2tzcrKqqoiiRiJlfYRSh3LOytRVff53DcqziIHIoK74k9Pl8vb29mqb19vauX78+ud2CrRMROZQVV9DhcFjX9WAwqCiKLMuKogSDQV3Xw+HwIEuFQqGSfiwY7XAzWcWRxujRyKlynFUcRA5l3/tBp9z6zpBD7QcR0TAphntx9JfxbnZpU1hUcRARjQTMO6uxioOITCpYQOdQBF0cWMVBRCbxCpqIyKZ4Lw6rsYqDiEzivTiIiHLn+CqOtM/w7n9NTUREyXgvDquxioOITOK9OKzGKg4iMon34iAisin73oujWLGKg4hMKsC9OCQp942yioOIbKUYbthPRETZKtjNkoZg5RNPAMCCBbjkEhw9iq1bz77giJaXXsIDD+S0nnUvbk3cifLybLe+Zg0CAVx4YeH3nS1sKb4W4C7gFQyPAgS08XOVXLWNHw9AJBVUFeKfTmnZuTPX9Rw+MP7yHgTLs926LKOsDOPHF37f2cKW4msBjmPY2Pd+0GkVwRz07Nl4882clly+HD/5CRoasl3uqafQ0IBbbslpo0Q0KM5BFxVWcRCRSbyCJiLKHa+giYhGIga01XgvDiIyiQFtNd6Lg4hMYkATEdkUA9pqrOIgIpNYxUFElDtWcRARjUQMaKuxioOITGJAW41VHERkEgOaiMimGNBWYxUHEZnEKg4iotyxioOIaCRiQFuNVRxEZBID2mqs4iAikxjQREQ2xYC2Gqs4iMgkVnEQEeWOVRxERCMRA9pqrOIgIpMY0FZjFQcRmcSAJiKyKQa01VjFQUQmsYqDiCh3rOIgIhqJGNBWYxUHEZnEgLYaqziIyCQGNBGRTTGgrcYqDiIyiVUcRES5YxUHEdFIZFFAK4oinU9RFGs2bTes4iAik6wIaFmWxaSEkc4ANE0T8xUjDas4iMgklwXb0HVdluVEIpHcqCgKp5KJiAZh0TWsy5X6TuB2u63ZtN2wioOITLKiikPMaciy7PV6V69e/eSTT0YiEXH5nO3WWcVBRLYyrFUcVgS0qqq9vb39230+XzgczmpVDGgishXHl9mFw2Fd14PBoKIosiwrihIMBnVdzzadiwOrOIjIJCu+JBSOHz9u2bbsbKhVHDU12S53+DC83lw3SkSFU7BCt1AoJOamiYgorYIFdFdXV6E2XVis4iAik6yr4kiLVRxE5GiO/5KQiIhyYFFAS5Kkn8/v91uzabthFQcRmWRFQKed4ggEAhZs2oZ4Lw4iMsmKMru0U8ZtbW2DLyXLctoZatZ+ENEIYV0ddLZqampOnz6d0hiNRgsymDxiFQcRmVSwJ6qIC2RWcRCRo7GKg4hoJCpYQGua5qzHIeYLqziIyCReQVuNVRxEZJJFXxL2f35K/2esEBFRMj6T0Gqs4iAikyy6F8dAzyRkFQcROVoxVHHwmYRERNmyKKCj0aiiKKqqNjc3q6qqKEokErFm03bDKg4iMsmKgPb5fAA0Tevt7V2/fn1vb6+YoxDtIw2rOIjIJD6TkIjIpgrwU29x69Hclh3oDkqOsgx4IYfFFgA7gFO5bPG7wCngQC6LElEmLpcrFosNx5odFtD23FAepa14sTmHltY49PTwer19fX2FHkgWnHh6eDyeWCxmh9NjJFYiExE5QgECmjd0JiIyowAB7awPO0REhcIpDiIim2JAExHZFAOaiOg89vmezL7PJBw6J/5SUVXVVatWFXoU2Zk8efKRI0cKPYqsOfH08Pl8jvt5V2VlpaIohR5FdiKRiJrTnSPzznmloEREIwSnOIiIbIoBTURkUwxoIiKbYkATEdkUA5qIyKYY0ERENsWAJiKyKWcHtKqq0jnitrP9ybJs9Ml22eFgZrsD9ZHOZ/PR8ghnpVhP5oHGXKhDnXHM1dXVyQMr7Cnt7B+qiP+uHo8nHo9rmpb2VvfiOIrdTN5ZM8sWaswD9RFnRkVFBQBFUTo7O20+2ozL2nDMFh/hrIbtxJN5kDEX5FBnHLOqqr29veLhfPF4XNd18YeZZfPOwT/1rq2tBSDLsnhAuCRJaW9kKhpT3qJNLluQMWfs09XVZcFQhzhaHuGsFOvJPNCYDRYfajNjTvkxfWFPaQdPcZw8eRKA1+sV/8zqU9JQlh0KM9sdvI+u65Z9whrKaHmEs1KsJ/PgrD/U2Y7Z4/Hg3IeAghxnBwe0+Li0evVqi5cdCjPbHaSPOJUVRZEkSXzyGqZxZhxJxj48wlkp1pN5EAU51FmNedSoUeJRsGJ+oyDH2WFz0MlP9VYUJZFIGPf3Ei8NtDvi7c541eVymV/W4jGbHJu49BjWD1lmRjJQHyuPcL7GnLIqC45wtkMyBoYCncz5GnP/V6051ObHXFtb+8UXXwBobGzct29fVsvmkcOuoDVN08+prq4GIOaDMPB/+7SGsmy2sh2z+bEN98kxlNFaeYTNjCfbPhlfyi+nnMzDt11rhm1yzKFQSKRzMBgU6Wx+2fxy2BV0CvG27PV6Y7FY8peqyU96D4VCXV1d3d3dAPx+fyAQaGtrG2RZO4w5bZ+ysrJwOKwoiizL4pnwFow559EO0j7cnHWEsxq2E0/mtGMu4KHOOOampqb9+/cDKCkpETPOY8eOFTFt/XF2dkCLghjxd/JHpOTPU/3n8kX7QMvaYcxp+5SWliZ/v2zNmHMe7SDtth1zQY5wVsN24smcdswFPNQZxyymMpIXMbpZf5ydHdBEREXMYXPQREQjBwOaiMimGNBERDbFgCYisikGNBGRTTGgiYhsigFNRGRTDGgiIptiQBMR2RQDmojIphjQREQ2xYAmIrIpBjQRkU0xoImIbIoBTURkUwxoIiKbYkATEdkUA5qIyKYY0ERENsWApvyTJEmSJLfbndKiKIrxt0GW5aamJtFNUZTkl0T//mRZNtPNtkzumsWjIhtiQNNwMR6N7HK5Ul6SJKm+vt7lckmSpOu6eMp9IBDQNE2SJJ/P5/V6B0ooWZZ1XZckyeVyiW5WPnvbAoxmMjCgaViI5BV/pzzEXmhra4vFYpqmybIMQFGUnp4eABUVFeFwuK+vT9O0/gv6/X6RzpqmxWIx0W3dunUARNwbV+XGMJKvSd1ut/F3WVlZ2j7V1dWi3cwKQ6GQaPd4PCY7i/wVb0VGT4OmaUX2fkNDwYCmYWR8kB9kFkKksK7rIjG7u7tlWS4pKUnbWYR4yqvNzc1NTU1iPWPGjBGX2MYWdV33eDzin/F4XJZlVVUBhMNhYw26rtfX19fX1wM4deoUgMFXGAwGg8EggCNHjogBxGIxAH6/3+Vy6bpuJG//zuJ9S6yTWUyD+/YyhyhfjClUEVWaponLXlmWE4mEeDU5m4zObrc7Ho8b7S6XSwSfQeRa/5NWURRN0+rr69va2pJXmLwtozFlPcl9jPZBVij2IrlRdG5sbNy3b1/yStJ2xrkr67QfLFIOSA4Hn4oJA5ryz5hPiEaj4p/GJ/qBAjqlxUjqlPNTRKHP50u+/kW/4B4oIvsHMc6PS7F+8b6ScYVGo/gj5SAM1BkMaDKNUxw0XCKRiPjD4/EM0s0o7UhujMViab8rE9MgfX19yY3Nzc2iszEjnG20Gf2NP7Jaoei8bt06PUlWAyBKi1fQlH/GtaqYkDUuYJOvoOvq6o4dO5ZIJMQZqOu6y+VKJBKKopSVlfX09Bhz0ykrT67ikGU5Go3qut7Y2Lh//35Jkmpqajo6OsR3j2JbZq6gce5dJBqNij5NTU0ZV2isZNasWW+//TYAVVVDoVBra6vY60GuoKUBik9CoVBXV1d3dzcAv98fCATEHAuNTAxoyr+0AZQc0CntdXV1bW1tXq9XTIkYjEmSFClTCmK1It+NdRrzzmYCWkqqOamqqurs7ASQcYXJK/F4PMnT5SmzOimdjfGnPUopO8v/Q0cyBjSNdBlnhIkKhXPQREQ2xYAmIrIpTnEQEdkUr6CJiGyKAU1EZFMMaCIim2JAExHZFAOaiMimGNBERDbFgCYisikGNBGRTTGgiYhsigFNRGRTDGgiIptiQBMR2RQDmojIphjQREQ2xYAmIrKp/wdRHE2/ZMyv/AAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Generate population stratification plot.\n",
    "# 1st parameter: full path to MDS dataset.\n",
    "# 2nd parameter: full path to racefile.\n",
    "# Output pdf is stored in the same directory as the input data file\n",
    "\n",
    "# Read lines represent the mean of each component, blue lines represent +/- 6 SD\n",
    "MDS_function(\"/mnt/data/GWAS/output/build37/task2_QC/MDS_merge2.mds\",\"/mnt/data/GWAS/output/build37/task2_QC/racefile.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The plot shows that our \"own\" data falls within the European group of the 1000 genomes data. Therefore, we do not have to remove subjects. Anyway we are running the filtering step for educational purposes.**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter out population stratification outliers\n",
    "\n",
    "A commonly used thershold is taking 6 standad deviation from population mean as theshold for excluding individuals.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      FID       IID    SOL           C1           C2           C3           C4           C5           C6           C7           C8           C9          C10 \n",
      "  HG00096   HG00096      0   -0.0446801   -0.0405616   0.00817707   0.00831041  -0.00148972  -0.00256521    0.0129851  -0.00525496   0.00716223  -0.00357411 \n",
      "  HG00097   HG00097      0   -0.0449411   -0.0403871   0.00771918   0.00673488  -0.00144457  -0.00301694    0.0088201   -0.0147722    0.0118504  -0.00546051 \n",
      "  HG00099   HG00099      0   -0.0464586   -0.0404293   0.00748735    0.0055747   -0.0037389  -0.00526758    0.0216535   0.00931718    0.0073412   -0.0079541 \n",
      "  HG00100   HG00100      0   -0.0424155    -0.040779   0.00726761   0.00767328  0.000635239  -0.00315927    0.0206008  -0.00180234  -0.00483294   -0.0114511 \n",
      "  HG00101   HG00101      0   -0.0437484   -0.0402676   0.00779164    0.0062208  -0.00288284  -0.00481741    0.0238685   0.00486814     0.003675  -0.00676118 \n",
      "  HG00102   HG00102      0   -0.0458155   -0.0399496   0.00700356   0.00663759  -0.00448207  -0.00127916    0.0150039   -0.0115593    0.0125762  -0.00120998 \n",
      "  HG00103   HG00103      0    -0.044493   -0.0393203   0.00820489   0.00712922  -0.00151199  0.000250689   0.00912332   -0.0106162    0.0132493  -0.00246611 \n",
      "  HG00105   HG00105      0   -0.0435366   -0.0401327   0.00768635   0.00676965  -0.00116273  -0.00433211    0.0158071  -0.00263603   0.00951726  -0.00362134 \n",
      "  HG00106   HG00106      0   -0.0460944   -0.0405581   0.00658582   0.00576979 -0.000932999  -0.00455734    0.0151276  -0.00273433   0.00506603  -0.00560208 \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/MDS_merge2.mds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# Run getFailures.R to identify those individuals\n",
    "# 1st parameter: full path to MDS dataset.\n",
    "# 2nd parameter: full path to racefile.\n",
    "# 3nd parameter: full path to threshold file\n",
    "# 4rd parameter: output file\n",
    "path<-\"/mnt/data/GWAS/output/build37/task2_QC\"\n",
    "get_failures(paste0(path,\"/MDS_merge2.mds\"), paste0(path,\"/racefile.txt\"), paste0(path,\"/thresholds.txt\"), paste0(path,\"/fail-pca-QC.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 /mnt/data/GWAS/output/build37/task2_QC/fail-pca-QC.txt\n",
      "\"x\"\n",
      "\"1\" -0.0584498391348963\n",
      "\"2\" -0.0266824483402528\n",
      "\"3\" -0.0613468589365413\n",
      "\"4\" -0.0160607207851287\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "wc -l $path/fail-pca-QC.txt\n",
    "head $path/thresholds.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/../task2_QC/intermediate_datasets/dataset.b37.IBD\n",
      "  --exclude /mnt/data/GWAS/output/build37/task2_QC/fail-pca-QC.txt\n",
      "  --make-bed\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--exclude: 7076087 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999931.\n",
      "7076087 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--make-bed to /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim +\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 1060176 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.hh ); many commands\n",
      "treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Extract these individuals from the working dataset.\n",
    "plink --bfile $path/../task2_QC/intermediate_datasets/dataset.b37.IBD --exclude $path/fail-pca-QC.txt --make-bed --out $path/dataset.b37.QCed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Create covariates based on MDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.genome.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "  --extract /mnt/data/GWAS/output/build37/task2_QC/indepSNP.prune.in\n",
      "  --genome\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "--extract: 473808 variants remaining.\n",
      "Using up to 4 threads (change this with --threads).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999881.\n",
      "473808 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "Excluding 18945 variants on non-autosomes from IBD calculation.\n",
      "IBD calculations complete.  \n",
      "Finished writing\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.genome.genome .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds\n",
      "  --read-genome /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.genome.genome\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "7076087 variants loaded from .bim file.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 496 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999931.\n",
      "7076087 variants and 496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "Clustering... done.                        \n",
      "Cluster solution written to 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.cluster1 ,\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.cluster2 , and\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.mds .\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 116599 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.genome.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: 1060176 het. haploid genotypes present (see\n",
      "/mnt/data/GWAS/output/build37/task2_QC/dataset.b37.mds.hh ); many commands\n",
      "treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Perform an MDS ONLY on HapMap data without ethnic outliers. The values of the 10 MDS dimensions are subsequently used as covariates in the association analysis in the third tutorial.\n",
    "plink --bfile $path/dataset.b37.QCed --extract $path/indepSNP.prune.in --genome --out $path/dataset.b37.QCed.genome\n",
    "plink --bfile $path/dataset.b37.QCed --read-genome $path/dataset.b37.QCed.genome.genome --cluster --mds-plot 10 --out $path/dataset.b37.mds\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       FID        IID    SOL           C1           C2           C3           C4           C5           C6           C7           C8           C9          C10 \n",
      "  HGX00096   HGX00096      0 -0.000130168    0.0148262  -0.00246446  -0.00800308   0.00270028  -0.00956846   0.00552736    0.0035352   -0.0125891    0.0173253 \n",
      "  HGX00097   HGX00097      0 -0.000255721    0.0139714  -0.00270998   -0.0107841    0.0176906  -0.00404297 -0.000167033   -0.0165653  -0.00459382    0.0126304 \n",
      "  HGX00099   HGX00099      0   0.00133495    0.0184345  -0.00508777   -0.0169761   -0.0099984  -0.00443926   0.00353687   0.00513696  -0.00120698    0.0253017 \n",
      "  HGX00100   HGX00100      0  -0.00332729    0.0183462    0.0179926  -0.00526761  -0.00754295  -0.00621013 -0.000163235  -0.00216034    -0.029367     0.026805 \n",
      "  HGX00101   HGX00101      0  0.000702511    0.0211981  -0.00266448   -0.0127535  -0.00542672  -0.00392478  -0.00215021  -0.00114056   -0.0138876    0.0281407 \n",
      "  HGX00102   HGX00102      0   0.00240392    0.0189573  -0.00523283  -0.00730245    0.0144611  -0.00904578   0.00350927   0.00110892   -0.0118359    0.0204098 \n",
      "  HGX00103   HGX00103      0  0.000293596    0.0129437  -0.00398425  -0.00442949    0.0150845  0.000305078   0.00590681   0.00564816 -0.000408353  -0.00234775 \n",
      "  HGX00105   HGX00105      0 -0.000439395    0.0169464   -0.0077837   -0.0139003   0.00431063   0.00565619  -0.00403705    0.0110402   0.00550992   0.00879913 \n",
      "  HGX00106   HGX00106      0  -0.00107784    0.0154828   0.00139142   -0.0113328 -0.000484044  -0.00598903   -0.0103982    0.0123405   -0.0104462    0.0088006 \n",
      "FID IID C1 C2 C3 C4 C5 C6 C7 C8 C9 C10\n",
      "HGX00096 HGX00096 -0.000130168 0.0148262 -0.00246446 -0.00800308 0.00270028 -0.00956846 0.00552736 0.0035352 -0.0125891 0.0173253\n",
      "HGX00097 HGX00097 -0.000255721 0.0139714 -0.00270998 -0.0107841 0.0176906 -0.00404297 -0.000167033 -0.0165653 -0.00459382 0.0126304\n",
      "HGX00099 HGX00099 0.00133495 0.0184345 -0.00508777 -0.0169761 -0.0099984 -0.00443926 0.00353687 0.00513696 -0.00120698 0.0253017\n",
      "HGX00100 HGX00100 -0.00332729 0.0183462 0.0179926 -0.00526761 -0.00754295 -0.00621013 -0.000163235 -0.00216034 -0.029367 0.026805\n",
      "HGX00101 HGX00101 0.000702511 0.0211981 -0.00266448 -0.0127535 -0.00542672 -0.00392478 -0.00215021 -0.00114056 -0.0138876 0.0281407\n",
      "HGX00102 HGX00102 0.00240392 0.0189573 -0.00523283 -0.00730245 0.0144611 -0.00904578 0.00350927 0.00110892 -0.0118359 0.0204098\n",
      "HGX00103 HGX00103 0.000293596 0.0129437 -0.00398425 -0.00442949 0.0150845 0.000305078 0.00590681 0.00564816 -0.000408353 -0.00234775\n",
      "HGX00105 HGX00105 -0.000439395 0.0169464 -0.0077837 -0.0139003 0.00431063 0.00565619 -0.00403705 0.0110402 0.00550992 0.00879913\n",
      "HGX00106 HGX00106 -0.00107784 0.0154828 0.00139142 -0.0113328 -0.000484044 -0.00598903 -0.0103982 0.0123405 -0.0104462 0.0088006\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b37.mds.mds\n",
    "awk '{print$1, $2, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13}' $path/dataset.b37.mds.mds > $path/covar_mds.txt\n",
    "head $path/covar_mds.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID IID C1 C2 C3 C4 C5 C6 C7 C8 C9 C10 SEX PHENO\n",
      "HGX00096 HGX00096 -0.000130168 0.0148262 -0.00246446 -0.00800308 0.00270028 -0.00956846 0.00552736 0.0035352 -0.0125891 0.0173253 1 1\n",
      "HGX00097 HGX00097 -0.000255721 0.0139714 -0.00270998 -0.0107841 0.0176906 -0.00404297 -0.000167033 -0.0165653 -0.00459382 0.0126304 2 1\n",
      "HGX00099 HGX00099 0.00133495 0.0184345 -0.00508777 -0.0169761 -0.0099984 -0.00443926 0.00353687 0.00513696 -0.00120698 0.0253017 2 1\n",
      "HGX00100 HGX00100 -0.00332729 0.0183462 0.0179926 -0.00526761 -0.00754295 -0.00621013 -0.000163235 -0.00216034 -0.029367 0.026805 2 1\n",
      "HGX00101 HGX00101 0.000702511 0.0211981 -0.00266448 -0.0127535 -0.00542672 -0.00392478 -0.00215021 -0.00114056 -0.0138876 0.0281407 1 1\n",
      "HGX00102 HGX00102 0.00240392 0.0189573 -0.00523283 -0.00730245 0.0144611 -0.00904578 0.00350927 0.00110892 -0.0118359 0.0204098 2 1\n",
      "HGX00103 HGX00103 0.000293596 0.0129437 -0.00398425 -0.00442949 0.0150845 0.000305078 0.00590681 0.00564816 -0.000408353 -0.00234775 1 1\n",
      "HGX00105 HGX00105 -0.000439395 0.0169464 -0.0077837 -0.0139003 0.00431063 0.00565619 -0.00403705 0.0110402 0.00550992 0.00879913 1 1\n",
      "HGX00106 HGX00106 -0.00107784 0.0154828 0.00139142 -0.0113328 -0.000484044 -0.00598903 -0.0103982 0.0123405 -0.0104462 0.0088006 2 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "awk 'BEGIN {print \"SEX\",\"PHENO\"};{print $5, $6}' $intpath/dataset.b37.QCed.fam > $intpath/temp\n",
    "paste $path/covar_mds.txt $intpath/temp | sed 's/\\t/ /g'> $path/covar_mds_sex_pheno.txt\n",
    "rm $intpath/temp\n",
    "head $path/covar_mds_sex_pheno.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**For the next step you need the following files:**\n",
    "- dataset.b37.QCed (the bfile, i.e., dataset.b37.QCed.bed,dataset.b37.QCed.bim,and dataset.b37.QCed.fam)\n",
    "- covar_mds.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
