{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Task 4 Association analysis\n",
    "\n",
    "The association analysis tries to identify genetic variants (SNPs) that can be associated with a trait.\n",
    "\n",
    "To complete this task it is necessary to have generated the genotypes dossages 'chri.dose.rsq.DS.vcf.gz', the fam file 'chr22.dose.for.assoc.fam' updated with phenotype and sex and the covariates file 'covar_mds.txt' from the Task 2.2 Stratification Analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:34:28.366878Z",
     "start_time": "2020-08-26T16:34:26.281738Z"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Create directories for the output files\n",
    "path=\"/mnt/data/GWAS/output/build37/task4_assoc\"\n",
    "if not os.path.exists(path):\n",
    "    os.makedirs(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:34:28.836773Z",
     "start_time": "2020-08-26T16:34:28.829382Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/build37/task4_assoc\n",
      "env: task3path=/mnt/data/GWAS/output/build37/task3_imputation/imputed_files\n"
     ]
    }
   ],
   "source": [
    "# Set an environment variable to hold the path to the output directory\n",
    "# It is recommended to send the output to the datavolume (so that you don't fill up the home directory). You will be able to acces it from your host machine\n",
    "%env path= /mnt/data/GWAS/output/build37/task4_assoc\n",
    "%env task3path= /mnt/data/GWAS/output/build37/task3_imputation/imputed_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Association analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T07:33:00.643036Z",
     "start_time": "2020-08-24T07:19:41.169738Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr1.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr1.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr1.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr1.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr1.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr2.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr2.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr2.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr2.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr2.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr3.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr3.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr3.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr3.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr3.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr4.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr4.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr4.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr4.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr4.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr5.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr5.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr5.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr5.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr5.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr6.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr6.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr6.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr6.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr6.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr7.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr7.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr7.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr7.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr7.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr8.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr8.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr8.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr8.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr8.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr9.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr9.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr9.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr9.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr9.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr10.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr10.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr10.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr10.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr10.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr11.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr11.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr11.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr11.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr11.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr12.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr12.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr12.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr12.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr12.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr13.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr13.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr13.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr13.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr13.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr14.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr14.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr14.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr14.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr14.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr15.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr15.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr15.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr15.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr15.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr16.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr16.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr16.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr16.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr16.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr17.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr17.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr17.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr17.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr17.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr18.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr18.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr18.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr18.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr18.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr19.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr19.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr19.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr19.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr19.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr20.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr20.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr20.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr20.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr20.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr21.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr21.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr21.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr21.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr21.imputed.dosage.assoc.dosage .\n",
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/chr22.imputed.dosage.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt\n",
      "  --dosage /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
      "  --fam /mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.for.assoc.fam\n",
      "  --hide-covar\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/chr22.imputed.dosage\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. '--linear hide-covar'.\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "496 people (237 males, 259 females) loaded from .fam.\n",
      "496 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "496 people pass filters and QC.\n",
      "Among remaining phenotypes, 203 are cases and 293 are controls.\n",
      "--dosage: Reading from\n",
      "/mnt/data/GWAS/output/build37/task3_imputation/imputed_files/chr22.dose.rsq.0.3.DS.vcf.gz.\n",
      "--dosage: Results saved to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/chr22.imputed.dosage.assoc.dosage .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Perform association analysis with PLINK (Purcell et al. 2007)\n",
    "# Association of genotype dosages with the AD case-control status is explored through regression model adjusted by sex, and the 10 MDS dimensions as covariates using PLINK.\n",
    "for i in {1..22}\n",
    "do\n",
    "plink --fam $task3path/chr22.dose.for.assoc.fam --out $path/chr$i.imputed.dosage --ci 0.95 --covar /mnt/data/GWAS/output/build37/task2_QC/covar_mds.txt --hide-covar --dosage $task3path/chr$i.dose.rsq.0.3.DS.vcf.gz format=1 noheader\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T10:34:52.669368Z",
     "start_time": "2020-08-24T10:34:49.420082Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Merge chr results in a single file\n",
    "for i in {2..22}\n",
    "do\n",
    "awk '{if (NR>1) print $0}' $path/chr$i.imputed.dosage.assoc.dosage > $path/chr$i.imputed.dosage.assoc.dosage.nh\n",
    "done\n",
    "for i in {2..22}\n",
    "do cat $path/chr$i.imputed.dosage.assoc.dosage.nh; done > $path/chr2-22.imputed.dosage.assoc.dosage.nh\n",
    "cat $path/chr1.imputed.dosage.assoc.dosage $path/chr2-22.imputed.dosage.assoc.dosage.nh > $path/dataset.b37.imputed.dosage.full.assoc.dosage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T10:35:09.069403Z",
     "start_time": "2020-08-24T10:34:52.674425Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16093211 /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage\n",
      "7856796 /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage.clean\n",
      "SNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\n",
      "1:49298:T:C\tT\tC\t0.6621\t0.4660\t0.9172\t0.4038\t0.8305\n",
      "1:54676:C:T\tC\tT\t0.3611\t0.4268\t1.2771\t0.4364\t0.5752\n",
      "1:86028:T:C\tT\tC\t0.0606\t0.9142\t1.0473\t0.5194\t0.929\n",
      "1:91536:G:T\tG\tT\t0.4853\t0.5284\t0.5885\t0.3625\t0.1436\n",
      "1:234313:C:T\tC\tT\t0.0676\t0.3908\t0.6839\t0.7846\t0.6283\n",
      "1:534192:C:T\tC\tT\t0.2262\t0.4186\t1.3682\t0.4805\t0.5141\n",
      "1:534583:C:G\tC\tG\t0.0119\t0.6564\t35.3463\t1.6454\t0.03026\n",
      "1:546697:A:G\tA\tG\t0.8784\t0.4792\t1.1050\t0.6354\t0.8752\n",
      "1:564862:T:C\tT\tC\t0.0380\t0.6638\t2.6774\t0.8295\t0.2351\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Remove missing values\n",
    "wc -l $path/dataset.b37.imputed.dosage.full.assoc.dosage\n",
    "awk '{OFS=\"\\t\"; gsub(\"^ +\",\"\",$0); gsub(\" +\",\"\\t\",$0); print $0}' $path/dataset.b37.imputed.dosage.full.assoc.dosage |awk '{if ($8!=\"NA\") print $0}' > $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean\n",
    "wc -l $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean\n",
    "head $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add rs ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T10:47:50.654359Z",
     "start_time": "2020-08-24T10:42:06.540741Z"
    }
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "## Add rs ID\n",
    "path= \"/mnt/data/GWAS/output/build37/task4_assoc\"\n",
    "assoc_results<-read.table(paste(path,\"dataset.b37.imputed.dosage.full.assoc.dosage.clean\",sep=\"/\"), sep=\"\\t\", header=T)\n",
    "annot<-read.table(\"/mnt/data/GWAS/ref_files/HRC.r1-1.GRCh37.minimac4.output.annot.clean.txt\", sep=\"\\t\", header=T)\n",
    "head(assoc_results)\n",
    "head(annot)\n",
    "merged <-merge(assoc_results,annot, by=\"SNP\", all.x=TRUE)\n",
    "write.table(merged,paste(path,\"dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs\",sep=\"/\"), sep=\"\\t\", row.names=FALSE , quote=FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-24T10:48:02.008608Z",
     "start_time": "2020-08-24T10:47:56.032761Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS\n",
      "10:100000625:A:G\tA\tG\t0.4496\t0.9979\t1.2232\t0.2799\t0.4717\trs7899632\n",
      "10:100000645:A:C\tA\tC\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs61875309\n",
      "10:100001867:C:T\tC\tT\t0.0105\t0.9292\t1.2906\t1.0324\t0.8048\trs150203744\n",
      "10:100002464:T:C\tT\tC\t0.0112\t0.9812\t0.7513\t1.0186\t0.7789\trs111551711\n",
      "10:100003242:T:G\tT\tG\t0.1401\t1.0713\t1.9542\t0.3874\t0.08371\trs12258651\n",
      "10:100003304:A:G\tA\tG\t0.0362\t0.9602\t2.6659\t0.7874\t0.213\trs72828461\n",
      "10:100003785:T:C\tT\tC\t0.3236\t0.9361\t1.1277\t0.2847\t0.6729\trs1359508\n",
      "10:100004360:G:A\tG\tA\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs1048754\n",
      "10:100004441:G:C\tG\tC\t0.6734\t0.9275\t0.906\t0.2859\t0.7298\trs1048757\n",
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS\n",
      "10\t100000625\t10:100000625:A:G\tA\tG\t0.4496\t0.9979\t1.2232\t0.2799\t0.4717\trs7899632\n",
      "10\t100000645\t10:100000645:A:C\tA\tC\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs61875309\n",
      "10\t100001867\t10:100001867:C:T\tC\tT\t0.0105\t0.9292\t1.2906\t1.0324\t0.8048\trs150203744\n",
      "10\t100002464\t10:100002464:T:C\tT\tC\t0.0112\t0.9812\t0.7513\t1.0186\t0.7789\trs111551711\n",
      "10\t100003242\t10:100003242:T:G\tT\tG\t0.1401\t1.0713\t1.9542\t0.3874\t0.08371\trs12258651\n",
      "10\t100003304\t10:100003304:A:G\tA\tG\t0.0362\t0.9602\t2.6659\t0.7874\t0.213\trs72828461\n",
      "10\t100003785\t10:100003785:T:C\tT\tC\t0.3236\t0.9361\t1.1277\t0.2847\t0.6729\trs1359508\n",
      "10\t100004360\t10:100004360:G:A\tG\tA\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs1048754\n",
      "10\t100004441\t10:100004441:G:C\tG\tC\t0.6734\t0.9275\t0.906\t0.2859\t0.7298\trs1048757\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs\n",
    "sed 's/:/\\t/g' $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs | awk 'BEGIN{OFS=\"\\t\"; print \"CHR\",\"BP\",\"SNP\",\"A1\",\"A2\",\"FRQ\",\"INFO\",\"OR\",\"SE\",\"P\",\"RS\"};{OFS=\"\\t\"; if (NR>1) print $1,$2,$1\":\"$2\":\"$3\":\"$4,$5,$6,$7,$8,$9,$10,$11,$12}' > $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.chr.bp\n",
    "head $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.chr.bp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add nearest genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:38:55.189416Z",
     "start_time": "2020-08-26T16:37:34.016604Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b3.45 64-bit (13 Jan 2017)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.log.\n",
      "Options in effect:\n",
      "  --annotate /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.chr.bp ranges=/mnt/data/GWAS/ref_files/hg19.refGene.plink.txt\n",
      "  --border 200\n",
      "  --out /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated\n",
      "\n",
      "257659 MB RAM detected; reserving 128829 MB for main workspace.\n",
      "--annotate ranges: 23967 annotations loaded from\n",
      "/mnt/data/GWAS/ref_files/hg19.refGene.plink.txt (counting multi-chromosome\n",
      "annotations once per spanned chromosome).\n",
      "--annotate: 6506795 out of 7856795 rows annotated; new report written to\n",
      "/mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot\n",
      ".\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#Annotate results \n",
    "plink --annotate $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.chr.bp ranges=/mnt/data/GWAS/ref_files/hg19.refGene.plink.txt --border 200 --out $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:52:03.201082Z",
     "start_time": "2020-08-26T16:51:47.208209Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  548  6576 84823 /mnt/data/GWAS/output/build37/task4_assoc/dataset.b37.imputed.dosage.assoc.200kb.annot.tops\n",
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tP\tSE\tRS\tANNOT\n",
      "2\t62514587\t2:62514587:C:G\tC\tG\t0.4279\t1.0372\t5.8969\t0.351\t4.288e-07\trs12614194\tB3GNT2(+62.72kb)|COMMD1(+151.4kb)|MIR5192(+81.54kb)\n",
      "10\t79119844\t10:79119844:G:C\tG\tC\t0.0307\t0.9511\t0.0072\t1.0354\t1.88e-06\trs118093470\tKCNMA1(0)\n",
      "10\t79110632\t10:79110632:C:T\tC\tT\t0.0307\t0.9515\t0.0072\t1.0369\t1.986e-06\trs144195574\tKCNMA1(0)\n",
      "15\t86321907\t15:86321907:C:T\tC\tT\t0.5786\t0.9746\t5.9959\t0.3792\t2.319e-06\trs2542594\tAKAP13(+29.32kb)|KLHL25(0)|MIR1276(+8.098kb)\n",
      "6\t29814513\t6:29814513:G:C\tG\tC\t0.0278\t0.8949\t26.9058\t0.7017\t2.706e-06\trs13216671\tHCG4(+53.66kb)|HCG4B(-77.86kb)|HCG8(-165.4kb)|HCG9(-128.4kb)|HLA-A(-95.73kb)|HLA-F(+119.4kb)|HLA-F-AS1(+97.69kb)|HLA-G(+15.61kb)|HLA-H(-40.87kb)|HLA-J(-159.2kb)|IFITM4P(+95.59kb)|LOC554223(+48.93kb)|MOG(+174.4kb)|ZFP57(+169.6kb)|ZNRD1-AS1(-154.3kb)\n",
      "19\t48578889\t19:48578889:A:C\tA\tC\t0.7349\t0.9911\t0.1667\t0.3899\t4.345e-06\trs1991722\tBSPH1(+83.46kb)|CABP5(+31.58kb)|CARD8(-132.5kb)|ELSPBP1(+50.48kb)|LIG1(-39.81kb)|LOC100505812(-180kb)|PLA2G4C(0)|SNAR-A1(+141.3kb)|SNAR-A2(+141.3kb)|SNAR-A3(+151.7kb)|SNAR-A4(+151.7kb)|SNAR-A5(+151.7kb)|SNAR-A6(+151.7kb)|SNAR-A7(+151.7kb)|SNAR-A8(+151.7kb)|SNAR-A9(+151.7kb)|SNAR-A10(+151.7kb)|SNAR-A11(+151.7kb)|SNAR-A12(+130.6kb)|SNAR-A13(+130.6kb)|SNAR-A14(+151.7kb)|SNAR-C1(+119.8kb)|SNAR-C2(+119.8kb)|SNAR-C3(+125.2kb)|SNAR-C4(+136kb)|SNAR-C5(+119.8kb)|SULT2A1(+189.2kb)|ZNF114(-195.8kb)\n",
      "19\t48584799\t19:48584799:G:T\tG\tT\t0.7349\t0.9911\t0.1667\t0.3903\t4.416e-06\trs4801743\tBSPH1(+89.37kb)|CABP5(+37.49kb)|CARD8(-126.5kb)|ELSPBP1(+56.39kb)|LIG1(-33.9kb)|LOC100505812(-174.1kb)|PLA2G4C(0)|SNAR-A1(+147.2kb)|SNAR-A2(+147.2kb)|SNAR-A3(+157.6kb)|SNAR-A4(+157.6kb)|SNAR-A5(+157.6kb)|SNAR-A6(+157.6kb)|SNAR-A7(+157.6kb)|SNAR-A8(+157.6kb)|SNAR-A9(+157.6kb)|SNAR-A10(+157.6kb)|SNAR-A11(+157.6kb)|SNAR-A12(+136.5kb)|SNAR-A13(+136.5kb)|SNAR-A14(+157.6kb)|SNAR-C1(+125.7kb)|SNAR-C2(+125.7kb)|SNAR-C3(+131.1kb)|SNAR-C4(+141.9kb)|SNAR-C5(+125.7kb)|SULT2A1(+195.1kb)|ZNF114(-189.9kb)\n",
      "19\t48584800\t19:48584800:A:C\tA\tC\t0.7349\t0.9911\t0.1667\t0.3903\t4.416e-06\trs4802429\tBSPH1(+89.37kb)|CABP5(+37.49kb)|CARD8(-126.5kb)|ELSPBP1(+56.39kb)|LIG1(-33.9kb)|LOC100505812(-174.1kb)|PLA2G4C(0)|SNAR-A1(+147.2kb)|SNAR-A2(+147.2kb)|SNAR-A3(+157.6kb)|SNAR-A4(+157.6kb)|SNAR-A5(+157.6kb)|SNAR-A6(+157.6kb)|SNAR-A7(+157.6kb)|SNAR-A8(+157.6kb)|SNAR-A9(+157.6kb)|SNAR-A10(+157.6kb)|SNAR-A11(+157.6kb)|SNAR-A12(+136.5kb)|SNAR-A13(+136.5kb)|SNAR-A14(+157.6kb)|SNAR-C1(+125.7kb)|SNAR-C2(+125.7kb)|SNAR-C3(+131.1kb)|SNAR-C4(+141.9kb)|SNAR-C5(+125.7kb)|SULT2A1(+195.1kb)|ZNF114(-189.9kb)\n",
      "10\t78977362\t10:78977362:A:G\tA\tG\t0.0268\t0.9626\t0.005\t1.1571\t4.684e-06\trs143642187\tKCNMA1(0)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sort -gk 10,10 $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot | awk '{OFS=\"\\t\"; if($10<0.0001) print $0}' | sed 's/ /\\t/g' > $path/temp\n",
    "awk 'BEGIN {OFS=\"\\t\"; print \"CHR\",\"BP\",\"SNP\",\"A1\",\"A2\",\"FRQ\",\"INFO\",\"OR\",\"P\",\"SE\",\"RS\",\"ANNOT\"};{OFS=\"\\t\"; print $0}' $path/temp > $path/dataset.b37.imputed.dosage.assoc.200kb.annot.tops\n",
    "wc $path/dataset.b37.imputed.dosage.assoc.200kb.annot.tops\n",
    "head $path/dataset.b37.imputed.dosage.assoc.200kb.annot.tops\n",
    "rm $path/temp\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:47:46.312975Z",
     "start_time": "2020-08-26T16:47:31.688027Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cp $path/dataset.b37.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot\n",
    "# comment for me; copy by now but change file names in the future"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-26T16:47:46.338721Z",
     "start_time": "2020-08-26T16:47:46.323888Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS ANNOT\n",
      "10\t100000625\t10:100000625:A:G\tA\tG\t0.4496\t0.9979\t1.2232\t0.2799\t0.4717\trs7899632 HPS1(-175.3kb)|LOXL4(-6.817kb)|MIR1287(-154.3kb)|MIR4685(-190.4kb)|PYROXD2(-142.7kb)|R3HCC1L(0)\n",
      "10\t100000645\t10:100000645:A:C\tA\tC\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs61875309 HPS1(-175.3kb)|LOXL4(-6.797kb)|MIR1287(-154.3kb)|MIR4685(-190.4kb)|PYROXD2(-142.7kb)|R3HCC1L(0)\n",
      "10\t100001867\t10:100001867:C:T\tC\tT\t0.0105\t0.9292\t1.2906\t1.0324\t0.8048\trs150203744 HPS1(-174.1kb)|LOXL4(-5.575kb)|MIR1287(-153.1kb)|MIR4685(-189.2kb)|PYROXD2(-141.5kb)|R3HCC1L(0)\n",
      "10\t100002464\t10:100002464:T:C\tT\tC\t0.0112\t0.9812\t0.7513\t1.0186\t0.7789\trs111551711 HPS1(-173.5kb)|LOXL4(-4.978kb)|MIR1287(-152.5kb)|MIR4685(-188.6kb)|PYROXD2(-140.9kb)|R3HCC1L(0)\n",
      "10\t100003242\t10:100003242:T:G\tT\tG\t0.1401\t1.0713\t1.9542\t0.3874\t0.08371\trs12258651 HPS1(-172.7kb)|LOXL4(-4.2kb)|MIR1287(-151.7kb)|MIR4685(-187.8kb)|PYROXD2(-140.1kb)|R3HCC1L(0)\n",
      "10\t100003304\t10:100003304:A:G\tA\tG\t0.0362\t0.9602\t2.6659\t0.7874\t0.213\trs72828461 HPS1(-172.6kb)|LOXL4(-4.138kb)|MIR1287(-151.7kb)|MIR4685(-187.7kb)|PYROXD2(-140kb)|R3HCC1L(0)\n",
      "10\t100003785\t10:100003785:T:C\tT\tC\t0.3236\t0.9361\t1.1277\t0.2847\t0.6729\trs1359508 HPS1(-172.2kb)|LOXL4(-3.657kb)|MIR1287(-151.2kb)|MIR4685(-187.3kb)|PYROXD2(-139.5kb)|R3HCC1L(0)\n",
      "10\t100004360\t10:100004360:G:A\tG\tA\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs1048754 HPS1(-171.6kb)|LOXL4(-3.082kb)|MIR1287(-150.6kb)|MIR4685(-186.7kb)|PYROXD2(-139kb)|R3HCC1L(0)\n",
      "10\t100004441\t10:100004441:G:C\tG\tC\t0.6734\t0.9275\t0.906\t0.2859\t0.7298\trs1048757 HPS1(-171.5kb)|LOXL4(-3.001kb)|MIR1287(-150.5kb)|MIR4685(-186.6kb)|PYROXD2(-138.9kb)|R3HCC1L(0)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**For the next step you need the following file:**\n",
    "- dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
