{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Task 6 Gene-wise statistics (MAGMA)\n",
    "\n",
    "In gene analysis, genetic marker data is aggregated to the level of whole genes, testing the joint association of all markers in the gene with the phenotype. Similarly, in gene-set analysis individual genes are aggregated to groups of genes sharing certain biological, functional or other characteristics.\n",
    "\n",
    "This is done using MAGMA[1]. The gene-set analysis is divided into two distinct and largely independent parts. In the first part a gene analysis is performed to quantify the degree of association each gene has with the phenotype. In addition the correlations between genes are estimated. These correlations reflect the LD between genes, and are needed in order to compensate for the dependencies between genes during the gene-set analysis. The gene p-values and gene correlation matrix are then used in the second part to perform the actual gene-set analysis.\n",
    "\n",
    "\n",
    "[1] de Leeuw CA, Mooij JM, Heskes T, Posthuma D. MAGMA: generalized gene-set analysis of GWAS data. PLoS computational biology. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657/. Published April 17, 2015. Accessed August 18, 2020."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-18T14:11:34.262741Z",
     "start_time": "2020-08-18T14:11:34.259937Z"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Create directories for the output files\n",
    "path=\"/mnt/data/GWAS/output/build37/task6_genewise\"\n",
    "if not os.path.exists(path):\n",
    "    os.makedirs(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: path=/mnt/data/GWAS/output/build37/task6_genewise\n",
      "env: task3path=/mnt/data/GWAS/output/build37/task3_imputation/imputed_files\n",
      "env: task4path=/mnt/data/GWAS/output/build37/task4_assoc\n"
     ]
    }
   ],
   "source": [
    "%env path=/mnt/data/GWAS/output/build37/task6_genewise\n",
    "%env task3path= /mnt/data/GWAS/output/build37/task3_imputation/imputed_files\n",
    "%env task4path= /mnt/data/GWAS/output/build37/task4_assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Filter out low frequency SNPs (MAF<0.01) for MAGMA**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-11T16:58:43.422025Z",
     "start_time": "2019-11-11T16:58:35.319572Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "awk '{OFS=\"\\t\"; if($6>0.01 && $6<0.99) print $0}' | sed 's/ /\\t/g' $task4path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot > $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n",
    "wc $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:34:52.485999Z",
     "start_time": "2019-11-12T08:34:52.050973Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS\tANNOT\n",
      "10\t100000625\t10:100000625:A:G\tA\tG\t0.4496\t0.9979\t1.2232\t0.2799\t0.4717\trs7899632\tHPS1(-175.3kb)|LOXL4(-6.817kb)|MIR1287(-154.3kb)|MIR4685(-190.4kb)|PYROXD2(-142.7kb)|R3HCC1L(0)\n",
      "10\t100000645\t10:100000645:A:C\tA\tC\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs61875309\tHPS1(-175.3kb)|LOXL4(-6.797kb)|MIR1287(-154.3kb)|MIR4685(-190.4kb)|PYROXD2(-142.7kb)|R3HCC1L(0)\n",
      "10\t100001867\t10:100001867:C:T\tC\tT\t0.0105\t0.9292\t1.2906\t1.0324\t0.8048\trs150203744\tHPS1(-174.1kb)|LOXL4(-5.575kb)|MIR1287(-153.1kb)|MIR4685(-189.2kb)|PYROXD2(-141.5kb)|R3HCC1L(0)\n",
      "10\t100002464\t10:100002464:T:C\tT\tC\t0.0112\t0.9812\t0.7513\t1.0186\t0.7789\trs111551711\tHPS1(-173.5kb)|LOXL4(-4.978kb)|MIR1287(-152.5kb)|MIR4685(-188.6kb)|PYROXD2(-140.9kb)|R3HCC1L(0)\n",
      "10\t100003242\t10:100003242:T:G\tT\tG\t0.1401\t1.0713\t1.9542\t0.3874\t0.08371\trs12258651\tHPS1(-172.7kb)|LOXL4(-4.2kb)|MIR1287(-151.7kb)|MIR4685(-187.8kb)|PYROXD2(-140.1kb)|R3HCC1L(0)\n",
      "10\t100003304\t10:100003304:A:G\tA\tG\t0.0362\t0.9602\t2.6659\t0.7874\t0.213\trs72828461\tHPS1(-172.6kb)|LOXL4(-4.138kb)|MIR1287(-151.7kb)|MIR4685(-187.7kb)|PYROXD2(-140kb)|R3HCC1L(0)\n",
      "10\t100003785\t10:100003785:T:C\tT\tC\t0.3236\t0.9361\t1.1277\t0.2847\t0.6729\trs1359508\tHPS1(-172.2kb)|LOXL4(-3.657kb)|MIR1287(-151.2kb)|MIR4685(-187.3kb)|PYROXD2(-139.5kb)|R3HCC1L(0)\n",
      "10\t100004360\t10:100004360:G:A\tG\tA\t0.2238\t0.9554\t0.6617\t0.3298\t0.2105\trs1048754\tHPS1(-171.6kb)|LOXL4(-3.082kb)|MIR1287(-150.6kb)|MIR4685(-186.7kb)|PYROXD2(-139kb)|R3HCC1L(0)\n",
      "10\t100004441\t10:100004441:G:C\tG\tC\t0.6734\t0.9275\t0.906\t0.2859\t0.7298\trs1048757\tHPS1(-171.5kb)|LOXL4(-3.001kb)|MIR1287(-150.5kb)|MIR4685(-186.6kb)|PYROXD2(-138.9kb)|R3HCC1L(0)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:35:12.485850Z",
     "start_time": "2019-11-12T08:35:04.280230Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP\tCHR\tBP\n",
      "rs7899632\t10\t100000625\n",
      "rs61875309\t10\t100000645\n",
      "SNP\tP\n",
      "rs7899632\t0.4717\n",
      "rs61875309\t0.2105\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Generate  *.SNP.LOC (SNP, CHR,BP) and *.SNP.VAL files\n",
    "\n",
    "awk 'BEGIN{OFS=\"\\t\";print \"SNP\",\"CHR\",\"BP\"};{OFS=\"\\t\"; if(NR>1)  print $11,$1,$2}' $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01 > $path/dataset.b37.imputed.dosage.maf.0.01.SNP.LOC\n",
    "awk 'BEGIN{OFS=\"\\t\";print \"SNP\",\"P\"};{OFS=\"\\t\"; if(NR>1) print $11,$10}' $path/dataset.b37.imputed.assoc.dosage.clean.rs.200kb.annot.maf.0.01 | sed 's/e/E/g' > $path/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL\n",
    "\n",
    "head -3 $path/dataset.b37.imputed.dosage.maf.0.01.SNP.LOC \n",
    "head -3  $path/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**de 200 kb pasa a 50kb - esta bien?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T08:35:23.242731Z",
     "start_time": "2019-11-12T08:35:12.488848Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--annotate window=50,50\n",
      "\t--snp-loc /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.LOC\n",
      "\t--gene-loc /mnt/data/GWAS/ref_files/NCBI37.3.gene.loc\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb\n",
      "\n",
      "Start time is 11:49:16, Sunday 14 Mar 2021\n",
      "\n",
      "Starting annotation...\n",
      "Reading gene locations from file /mnt/data/GWAS/ref_files/NCBI37.3.gene.loc... \n",
      "\tadding window: 50000bp\n",
      "\t19427 gene locations read from file\n",
      "\tchromosome  1: 2016 genes\n",
      "\tchromosome  2: 1226 genes\n",
      "\tchromosome  3: 1050 genes\n",
      "\tchromosome  4: 745 genes\n",
      "\tchromosome  5: 856 genes\n",
      "\tchromosome  6: 1016 genes\n",
      "\tchromosome  7: 906 genes\n",
      "\tchromosome  8: 669 genes\n",
      "\tchromosome  9: 775 genes\n",
      "\tchromosome 10: 723 genes\n",
      "\tchromosome 11: 1275 genes\n",
      "\tchromosome 12: 1009 genes\n",
      "\tchromosome 13: 320 genes\n",
      "\tchromosome 14: 595 genes\n",
      "\tchromosome 15: 586 genes\n",
      "\tchromosome 16: 817 genes\n",
      "\tchromosome 17: 1147 genes\n",
      "\tchromosome 18: 271 genes\n",
      "\tchromosome 19: 1389 genes\n",
      "\tchromosome 20: 527 genes\n",
      "\tchromosome 21: 215 genes\n",
      "\tchromosome 22: 442 genes\n",
      "\tchromosome  X: 805 genes\n",
      "\tchromosome  Y: 47 genes\n",
      "Reading SNP locations from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.LOC... \n",
      "\tWARNING: on line 1, chromosome code 'CHR' not recognised; skipping SNP (ID = SNP)\n",
      "\t7856795 SNP locations read from file                                                            \n",
      "\tof those, 4909274 (62.48%) mapped to at least one gene\n",
      "Writing annotation to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\tfor chromosome  1, 5 genes are empty (out of 2016)\n",
      "\tfor chromosome  2, 2 genes are empty (out of 1226)\n",
      "\tfor chromosome  4, 2 genes are empty (out of 745)\n",
      "\tfor chromosome  5, 2 genes are empty (out of 856)\n",
      "\tfor chromosome  9, 11 genes are empty (out of 775)\n",
      "\tfor chromosome 10, 1 gene is empty (out of 723)\n",
      "\tfor chromosome 15, 2 genes are empty (out of 586)\n",
      "\tfor chromosome 16, 1 gene is empty (out of 817)\n",
      "\tfor chromosome 21, 1 gene is empty (out of 215)\n",
      "\tfor chromosome  X, 805 genes are empty (out of 805)\n",
      "\tfor chromosome  Y, 47 genes are empty (out of 47)\n",
      "\tat least one SNP mapped to each of a total of 18548 genes (out of 19427)\n",
      "\n",
      "\n",
      "End time is 11:49:29, Sunday 14 Mar 2021 (elapsed: 00:00:13)\n",
      "# window_up = 50000\n",
      "# window_down = 50000\n",
      "79501\t1:19091:120008\trs200943160\trs2462492\trs114608975\trs6702460\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Annotate\n",
    "/usr/lib/magma/magma --annotate window=50,50 --snp-loc $path/dataset.b37.imputed.dosage.maf.0.01.SNP.LOC --gene-loc /mnt/data/GWAS/ref_files/NCBI37.3.gene.loc --out $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb\n",
    "head -3 $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-20T10:37:34.844137Z",
     "start_time": "2020-08-20T10:37:34.703213Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 1 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 1 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 2 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 2 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 3 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 4 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 3 only)... Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 4 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 6 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 6 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 5 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 5 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 8 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 8 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 7 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 7 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 10 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 10 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 9 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 9 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 17 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 17 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 18 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 18 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 11 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 11 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 19 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 19 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 12 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 12 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 13 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 13 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 22 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 22 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 14 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 14 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 21 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 21 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 15 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 15 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 20 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 20 only)... Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--batch 16 chr\n",
      "\t--big-data\n",
      "\t--seed 1234\n",
      "\t--genes-only\n",
      "\t--bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed\n",
      "\t--gene-annot /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot\n",
      "\t--pval /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL N=496\n",
      "\t--gene-model multi\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 11:49:29, Sunday 14 Mar 2021\n",
      "\n",
      "Loading PLINK-format data...\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.fam... 496 individuals read\n",
      "Reading file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bim (chromosome 16 only)... 7076087 SNPs read, of which 371410 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 524714 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 164227 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 224452 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 498385 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 209209 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 485255 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 98123 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 432041 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 405260 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 295895 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 195795 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 343970 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 254900 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 96709 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 151312 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 188159 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 353501 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 194819 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 465189 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 568940 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "7076087 SNPs read, of which 331587 used\n",
      "Preparing file /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed.bed... \n",
      "\n",
      "Reading SNP p-values from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL... \n",
      "\tdetected 2 variables in file\n",
      "\tusing variable: SNP (SNP id)\n",
      "\tusing variable: P (p-value)\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 73842 SNPs in data (0.9398% of lines, 76.35% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 298990 SNPs in data (3.805% of lines, 80.5% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t442 gene definitions read from file\n",
      "\tfound 439 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 22, containing 439 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 121699 SNPs in data (1.549% of lines, 74.1% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 410713 SNPs in data (5.227% of lines, 78.27% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t669 gene definitions read from file\n",
      "\tfound 660 genes containing SNPs in genotype data\n",
      "\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 122373 SNPs in data (1.558% of lines, 80.87% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 143235 SNPs in data (1.823% of lines, 76.12% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 339856 SNPs in data (4.326% of lines, 78.66% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 8, containing 660 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1389 gene definitions read from file\n",
      "\tfound 1388 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 19, containing 1388 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t854 gene definitions read from file\n",
      "\tfound 854 genes containing SNPs in genotype data\n",
      "\n",
      "\t527 gene definitions read from file\n",
      "\tfound 527 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 5, containing 854 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 20, containing 527 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 316130 SNPs in data (4.024% of lines, 78.01% of SNPs in data)\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 232834 SNPs in data (2.963% of lines, 78.69% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 390688 SNPs in data (4.973% of lines, 78.39% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 78155 SNPs in data (0.9947% of lines, 79.65% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 152746 SNPs in data (1.944% of lines, 78.01% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1147 gene definitions read from file\n",
      "\tfound 1141 genes containing SNPs in genotype data\n",
      "\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 155600 SNPs in data (1.98% of lines, 79.87% of SNPs in data)\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 166396 SNPs in data (2.118% of lines, 79.54% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 17, containing 1141 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t214 gene definitions read from file\n",
      "\tfound 214 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 21, containing 214 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t2011 gene definitions read from file\n",
      "\tfound 2002 genes containing SNPs in genotype data\n",
      "\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 176252 SNPs in data (2.243% of lines, 78.53% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t584 gene definitions read from file\n",
      "\tfound 573 genes containing SNPs in genotype data\n",
      "\n",
      "\t271 gene definitions read from file\n",
      "\tfound 271 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 15, containing 573 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 1, containing 2002 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t764 gene definitions read from file\n",
      "\tfound 759 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 18, containing 271 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t906 gene definitions read from file\n",
      "\tfound 901 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 9, containing 759 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 202513 SNPs in data (2.578% of lines, 79.45% of SNPs in data)\n",
      "\t743 gene definitions read from file\n",
      "\tfound 731 genes containing SNPs in genotype data\n",
      "\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t816 gene definitions read from file\n",
      "\tfound 810 genes containing SNPs in genotype data\n",
      "\n",
      "\t595 gene definitions read from file\n",
      "\tfound 594 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 7, containing 901 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 16, containing 810 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 14, containing 594 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 4, containing 731 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 278923 SNPs in data (3.55% of lines, 78.9% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t320 gene definitions read from file\n",
      "\tfound 320 genes containing SNPs in genotype data\n",
      "\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 271545 SNPs in data (3.456% of lines, 78.94% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 450446 SNPs in data (5.733% of lines, 79.17% of SNPs in data)\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 13, containing 320 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t722 gene definitions read from file\n",
      "\tfound 720 genes containing SNPs in genotype data\n",
      "\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 259485 SNPs in data (3.303% of lines, 78.26% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 384488 SNPs in data (4.894% of lines, 79.23% of SNPs in data)\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 10, containing 720 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1275 gene definitions read from file\n",
      "\tfound 1275 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 11, containing 1275 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1009 gene definitions read from file\n",
      "\tfound 1009 genes containing SNPs in genotype data\n",
      "\n",
      "\t1224 gene definitions read from file\n",
      "\tfound 1220 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 12, containing 1009 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 2, containing 1220 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\tread 7856796 lines from file, containing valid SNP p-values for 359532 SNPs in data (4.576% of lines, 77.29% of SNPs in data)\n",
      "Filtering phenotype/covariate missing values... 496 individuals remaining\n",
      "\n",
      "Loading gene annotation from file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot... \n",
      "\t1050 gene definitions read from file\n",
      "\tfound 1050 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 3, containing 1050 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\t1016 gene definitions read from file\n",
      "\tfound 1016 genes containing SNPs in genotype data\n",
      "\n",
      "Starting gene analysis... )     \n",
      "\tusing multi-model containing:\n",
      "\t\tSNPwise-mean\n",
      "\t\tSNPwise-top\n",
      "\tanalysing chromosome 6, containing 1016 genes\n",
      "\tWARNING: disabling 'big-data' mode, sample size is too small\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch21_chr.genes.out\n",
      "\tprocessed genes: 229 (18%)     \n",
      "\n",
      "End time is 11:58:48, Sunday 14 Mar 2021 (elapsed: 00:09:19)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch18_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:02:03, Sunday 14 Mar 2021 (elapsed: 00:12:34)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch22_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:04:21, Sunday 14 Mar 2021 (elapsed: 00:14:52)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch13_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:05:20, Sunday 14 Mar 2021 (elapsed: 00:15:51)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch20_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:07:48, Sunday 14 Mar 2021 (elapsed: 00:18:19)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch14_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:10:12, Sunday 14 Mar 2021 (elapsed: 00:20:43)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch15_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:11:08, Sunday 14 Mar 2021 (elapsed: 00:21:39)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch16_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:12:59, Sunday 14 Mar 2021 (elapsed: 00:23:30)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch9_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:15:13, Sunday 14 Mar 2021 (elapsed: 00:25:44)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch8_chr.genes.out\n",
      "\n",
      "\tprocessed genes: 744 (70.9%)     \n",
      "End time is 12:15:23, Sunday 14 Mar 2021 (elapsed: 00:25:54)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch17_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:15:44, Sunday 14 Mar 2021 (elapsed: 00:26:15)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch10_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:16:13, Sunday 14 Mar 2021 (elapsed: 00:26:44)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch5_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:16:20, Sunday 14 Mar 2021 (elapsed: 00:26:51)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch4_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:16:25, Sunday 14 Mar 2021 (elapsed: 00:26:56)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch12_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:17:33, Sunday 14 Mar 2021 (elapsed: 00:28:04)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch7_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:17:48, Sunday 14 Mar 2021 (elapsed: 00:28:19)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch6_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:20:11, Sunday 14 Mar 2021 (elapsed: 00:30:42)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch3_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:20:33, Sunday 14 Mar 2021 (elapsed: 00:31:04)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch11_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:20:52, Sunday 14 Mar 2021 (elapsed: 00:31:23)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch2_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:21:01, Sunday 14 Mar 2021 (elapsed: 00:31:32)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch19_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:21:01, Sunday 14 Mar 2021 (elapsed: 00:31:32)\n",
      "\twriting gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch1_chr.genes.out\n",
      "\n",
      "\n",
      "End time is 12:23:27, Sunday 14 Mar 2021 (elapsed: 00:33:58)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#Gene level analysis is performed using MAGMA, which compute gene-wise statistics taking into account physical distance and linkage disequilibrium (LD) between markers (de Leeuw et al. 2015). \n",
    "# All SNPs with MAF above 5% are used in these analyses, setting a distance threshold of 50kb, which usually ranges from 0 to 200kb\n",
    "# N: number of individuals\n",
    "for i in {1..22}\n",
    "do\n",
    "nohup /usr/lib/magma/magma  --batch $i chr --big-data --seed 1234 --genes-only --bfile /mnt/data/GWAS/output/build37/task2_QC/dataset.b37.QCed --gene-annot $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot  --pval $path/dataset.b37.imputed.dosage.maf.0.01.SNP.PVAL  N=496 --gene-model multi --out $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:19:36.853278Z",
     "start_time": "2019-11-12T09:19:36.247662Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to MAGMA v1.06 (linux)\n",
      "Using flags:\n",
      "\t--merge /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\t--out /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
      "\n",
      "Start time is 12:23:27, Sunday 14 Mar 2021\n",
      "\n",
      "Merging gene results files with prefix '/mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma'... \n",
      "\tWARNING: found only 0 .genes.raw batch files for chromosomes 1-22\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch1_chr.genes.out... \n",
      "\t2002 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch2_chr.genes.out... \n",
      "\t1220 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch3_chr.genes.out... \n",
      "\t1050 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch4_chr.genes.out... \n",
      "\t731 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch5_chr.genes.out... \n",
      "\t854 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch6_chr.genes.out... \n",
      "\t1016 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch7_chr.genes.out... \n",
      "\t901 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch8_chr.genes.out... \n",
      "\t660 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch9_chr.genes.out... \n",
      "\t759 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch10_chr.genes.out... \n",
      "\t720 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch11_chr.genes.out... \n",
      "\t1275 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch12_chr.genes.out... \n",
      "\t1009 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch13_chr.genes.out... \n",
      "\t320 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch14_chr.genes.out... \n",
      "\t594 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch15_chr.genes.out... \n",
      "\t573 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch16_chr.genes.out... \n",
      "\t810 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch17_chr.genes.out... \n",
      "\t1141 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch18_chr.genes.out... \n",
      "\t271 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch19_chr.genes.out... \n",
      "\t1388 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch20_chr.genes.out... \n",
      "\t527 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch21_chr.genes.out... \n",
      "\t214 genes read from file\n",
      "Reading file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.batch22_chr.genes.out... \n",
      "\t439 genes read from file\n",
      "Writing merged gene analysis results to file /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out\n",
      "\n",
      "\n",
      "End time is 12:23:27, Sunday 14 Mar 2021 (elapsed: 00:00:00)\n",
      "GENE       CHR      START       STOP  NSNPS  NPARAM    N        ZSTAT      P_JOINT  P_SNPWISE_MEAN  P_SNPWISE_TOP1\n",
      "79501        1      19091     120008      1       1  496      -1.4684        0.929           0.929           0.929\n",
      "148398       1     809993     929961    186      30  496     -0.61238      0.72986         0.73539         0.62993\n",
      "26155        1     829583     944679    213      31  496     -0.65743      0.74455         0.76571         0.63687\n",
      "339451       1     845967     951099    182      29  496     -0.73459       0.7687         0.74982         0.69474\n",
      "84069        1     851872     960488    181      29  496     -0.86454      0.80635         0.81309         0.70204\n",
      "84808        1     860579     967473    169      27  496     -0.82834      0.79626         0.79972         0.69029\n",
      "57801        1     884342     986608    183      24  496     -0.68288      0.75266         0.79127         0.63263\n",
      "9636         1     898847     999920    192      26  496     -0.83035      0.79683         0.82944         0.67171\n",
      "375790       1     905503    1041499    277      28  496      -1.5092      0.93438         0.92836         0.88454\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#merge batches\n",
    "#/usr/lib/magma/magma\n",
    "magma --merge $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma --out $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma\n",
    "head $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:20:13.633421Z",
     "start_time": "2019-11-12T09:20:13.054616Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENE       CHR      START       STOP  NSNPS  NPARAM    N        ZSTAT      P_JOINT  P_SNPWISE_MEAN  P_SNPWISE_TOP1\n",
      "11214       15   85873818   86342589   1212      32  496       4.1297   1.8165e-05      9.2671e-06      0.00077629\n",
      "64410       15   86252557   86388189    508      38  496       4.1298   1.8151e-05      2.5588e-05      0.00024736\n",
      "343413       1  159720282  159836047    235      41  496       3.7093   0.00010392      6.7899e-05       0.0034542\n",
      "134111       5    6387460    6546834    386      47  496       3.4649   0.00026519      6.8813e-05        0.012499\n",
      "1053        14   23536515   23638820    147      28  496       3.4645   0.00026563      0.00010603       0.0079492\n",
      "56833        1  159746440  159857282    237      37  496       3.7065   0.00010508      0.00010668       0.0035125\n",
      "284677       1  159754264  159875544    260      33  496       3.5985   0.00016002      0.00012396       0.0035435\n",
      "8698        19    3128250    3230335    395      50  496       2.8686    0.0020613      0.00017936        0.063752\n",
      "10090        6  149018063  149448126   1001      83  496       3.2994   0.00048447      0.00017961         0.01721\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sort -gk10,10 $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out >$path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\n",
    "head $path/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:23:49.202385Z",
     "start_time": "2019-11-12T09:23:48.793407Z"
    }
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# Annotate genes using the reference file NCBI37.3.gene.loc\n",
    "magma<-read.table(\"/mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\", header=TRUE)\n",
    "genes<-read.table(\"/mnt/data/GWAS/ref_files/NCBI37.3.gene.loc\")\n",
    "colnames(genes) <- c(\"GENE\",\"CHR\",\"START\",\"STOP\",\"STRAND\",\"HUGO\")\n",
    "\n",
    "magma_merged<-merge(magma,genes, by=\"GENE\")\n",
    "magma_merged <- magma_merged[order(magma_merged$P_SNPWISE_MEAN), ]\n",
    "magma_rank<-rank(magma_merged[,10],na.last = \"keep\", ties.method = \"min\")\n",
    "magma_ranked=cbind(magma_rank, magma_merged)\n",
    "\n",
    "write.table(magma_ranked, \"/mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted.annot\", quote=FALSE, sep=\"\\t\", row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    GENE CHR     START      STOP NSNPS NPARAM   N  ZSTAT    P_JOINT\n",
      "1  11214  15  85873818  86342589  1212     32 496 4.1297 1.8165e-05\n",
      "2  64410  15  86252557  86388189   508     38 496 4.1298 1.8151e-05\n",
      "3 343413   1 159720282 159836047   235     41 496 3.7093 1.0392e-04\n",
      "4 134111   5   6387460   6546834   386     47 496 3.4649 2.6519e-04\n",
      "5   1053  14  23536515  23638820   147     28 496 3.4645 2.6563e-04\n",
      "6  56833   1 159746440 159857282   237     37 496 3.7065 1.0508e-04\n",
      "  P_SNPWISE_MEAN P_SNPWISE_TOP1\n",
      "1     9.2671e-06     0.00077629\n",
      "2     2.5588e-05     0.00024736\n",
      "3     6.7899e-05     0.00345420\n",
      "4     6.8813e-05     0.01249900\n",
      "5     1.0603e-04     0.00794920\n",
      "6     1.0668e-04     0.00351250\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "magma<-read.table(\"/mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted\", header=TRUE)\n",
    "head(magma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       GENE CHR  START   STOP STRAND         HUGO\n",
      "1     79501   1  69091  70008      +        OR4F5\n",
      "2 100996442   1 142447 174392      - LOC100996442\n",
      "3    729759   1 367659 368597      +       OR4F29\n",
      "4     81399   1 621096 622034      -       OR4F16\n",
      "5    148398   1 859993 879961      +       SAMD11\n",
      "6     26155   1 879583 894679      -        NOC2L\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "genes<-read.table(\"/mnt/data/GWAS/ref_files/NCBI37.3.gene.loc\")\n",
    "colnames(genes) <- c(\"GENE\",\"CHR\",\"START\",\"STOP\",\"STRAND\",\"HUGO\")\n",
    "head(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-12T09:24:15.606932Z",
     "start_time": "2019-11-12T09:24:15.153136Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "magma_rank\tGENE\tCHR.x\tSTART.x\tSTOP.x\tNSNPS\tNPARAM\tN\tZSTAT\tP_JOINT\tP_SNPWISE_MEAN\tP_SNPWISE_TOP1\tCHR.y\tSTART.y\tSTOP.y\tSTRAND\tHUGO\n",
      "1\t11214\t15\t85873818\t86342589\t1212\t32\t496\t4.1297\t1.8165e-05\t9.2671e-06\t0.00077629\t15\t85923818\t86292589\t+\tAKAP13\n",
      "2\t64410\t15\t86252557\t86388189\t508\t38\t496\t4.1298\t1.8151e-05\t2.5588e-05\t0.00024736\t15\t86302557\t86338189\t-\tKLHL25\n",
      "3\t343413\t1\t159720282\t159836047\t235\t41\t496\t3.7093\t0.00010392\t6.7899e-05\t0.0034542\t1\t159770282\t159786047\t+\tFCRL6\n",
      "4\t134111\t5\t6387460\t6546834\t386\t47\t496\t3.4649\t0.00026519\t6.8813e-05\t0.012499\t5\t6437460\t6496834\t+\tUBE2QL1\n",
      "5\t1053\t14\t23536515\t23638820\t147\t28\t496\t3.4645\t0.00026563\t0.00010603\t0.0079492\t14\t23586515\t23588820\t-\tCEBPE\n",
      "6\t56833\t1\t159746440\t159857282\t237\t37\t496\t3.7065\t0.00010508\t0.00010668\t0.0035125\t1\t159796440\t159807282\t+\tSLAMF8\n",
      "7\t284677\t1\t159754264\t159875544\t260\t33\t496\t3.5985\t0.00016002\t0.00012396\t0.0035435\t1\t159804264\t159825544\t-\tC1orf204\n",
      "8\t8698\t19\t3128250\t3230335\t395\t50\t496\t2.8686\t0.0020613\t0.00017936\t0.063752\t19\t3178250\t3180335\t+\tS1PR4\n",
      "9\t10090\t6\t149018063\t149448126\t1001\t83\t496\t3.2994\t0.00048447\t0.00017961\t0.01721\t6\t149068063\t149398126\t+\tUST\n",
      "18465\t7476\t3\t13810082\t13971618\t338\t43\t496\t-1.7793\t0.96241\t0.993\t0.86559\t3\t13860082\t13921618\t-\tWNT7A\n",
      "18466\t51617\t5\t173422607\t173586182\t277\t29\t496\t-2.6862\t0.99639\t0.99304\t0.99017\t5\t173472607\t173536182\t+\tHMP19\n",
      "18467\t55773\t3\t99929661\t100094096\t178\t16\t496\t-1.7151\t0.95684\t0.9932\t0.87006\t3\t99979661\t100044096\t+\tTBC1D23\n",
      "18468\t124923\t17\t26884799\t26991377\t29\t5\t496\t-2.0588\t0.98024\t0.99368\t0.94421\t17\t26934799\t26941377\t-\tSGK494\n",
      "18469\t56954\t3\t100003483\t100125844\t118\t15\t496\t-1.1783\t0.88065\t0.99399\t0.74384\t3\t100053483\t100075844\t+\tNIT2\n",
      "18470\t26664\t19\t14859943\t14996188\t421\t40\t496\t-1.7973\t0.96386\t0.99412\t0.8927\t19\t14909943\t14946188\t-\tOR7C1\n",
      "18471\t50515\t12\t104800692\t105205792\t974\t95\t496\t-0.46072\t0.6775\t0.99434\t0.37459\t12\t104850692\t105155792\t+\tCHST11\n",
      "18472\t10252\t4\t124267946\t124374915\t194\t35\t496\t-2.7879\t0.99735\t0.99619\t0.98737\t4\t124317946\t124324915\t+\tSPRY1\n",
      "18473\t58499\t9\t109572469\t109825525\t292\t31\t496\t-1.825\t0.966\t0.99759\t0.88653\t9\t109622469\t109775525\t+\tZNF462\n",
      "18474\t2890\t5\t152820084\t153243429\t1085\t49\t496\t-1.8282\t0.96624\t0.99837\t0.87223\t5\t152870084\t153193429\t+\tGRIA1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted.annot\n",
    "tail /mnt/data/GWAS/output/build37/task6_genewise/dataset.b37.imputed.dosage.maf.0.01.LOC.50kb.genes.annot.magma.genes.out.sorted.annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
