{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Meta Analysis of GWAS data\n",
    "\n",
    "Different GWAS results can be combined by means of the standard error based weights meta-analysis method implemented in METAL, corrected by study-specific inflation factors. A Cochranâ€™s Q test for heterogeneity and I2 estimates is generated to evaluate the potential effect of study heterogeneity on the results.\n",
    "\n",
    "Note this template is not run as only one GWAS dataset is available.\n",
    "\n",
    "https://genome.sph.umich.edu/wiki/METAL_Documentation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-01T08:39:58.851761Z",
     "start_time": "2018-02-01T08:39:57.886702Z"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input File Columns**\n",
    "\n",
    "Each input file should include the following information:\n",
    "\n",
    "- A column with marker name, which should be consistent across studies\n",
    "- A column indicating the tested allele\n",
    "- A column indicating the other allele\n",
    "\n",
    "\n",
    "If you are carrying out a sample size weighted analysis (based on p-values), you will also need:\n",
    "\n",
    "- A column indicating the direction of effect for the tested allele\n",
    "- A column indicating the corresponding p-value\n",
    "- An optional column indicating the sample size (if the sample size varies by marker)\n",
    "\n",
    "\n",
    "If you are carrying out a meta-analysis based on standard errors, you will need:\n",
    "\n",
    "-A column indicating the estimated effect size for each marker\n",
    "- A column indicating the standard error of this effect size estimate\n",
    "\n",
    "The header for each of these columns must be specified so that METAL knows how to interpret the data. Additional columns including allele frequency information, strand information, and others can also be present."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-12T10:33:19.150333Z",
     "start_time": "2021-01-12T10:33:19.141981Z"
    }
   },
   "source": [
    "**Selecting an Analysis Scheme**\n",
    "\n",
    "     1) SCHEME SAMPLESIZE        - default approach, uses p-value and direction of effect, weighted according to sample size\n",
    "     \n",
    "     The weight for each MARKER can be stored in a column in the table (specified with the WEIGHTLABEL or WEIGHT commands). Most commonly, the weight will be the number of individuals contributing to that particular p-value.\n",
    "     \n",
    "     WEIGHTLABEL     N\n",
    "     \n",
    "     Alternatively, the same weight can be used for all markers for that inputfile (in which case the fixed weight can be set  with the DEFAULTWEIGHT command). The WEIGHTLABEL command takes precedence over the DEFAULTWEIGHT command, so the WEIGHT column label in use must not match any columns in the inputfile.\n",
    "     \n",
    "     WEIGHTLABEL     DONTUSECOLUMN\n",
    "     DEFAULTWEIGHT   1000\n",
    "    \n",
    "    \n",
    "    2) SCHEME STDERR            - classical approach, uses effect size estimates and standard errors\n",
    "\n",
    "     For this approach, you need to specify the label for the standard error column:\n",
    "\n",
    "     STDERR SE                "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SCHEME SAMPLESIZE "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHR\tBP\tSNP\tA1\tA2\tFRQ\tINFO\tOR\tSE\tP\tRS\tANNOT\n",
      "10\t100000235\t10:100000235:C:T\tC\tT\t0.2982\t0.7959\t0.6679\t0.3084\t0.1906\trs11596870\tABCC2(+148kb)|CHUK(-188.1kb)|CPN1(-42.07kb)|DNMBP(0)|DNMBP-AS1(+41.24kb)|ERLIN1(-149.9kb)\n",
      "10\t100000943\t10:100000943:G:A\tG\tA\t0.0929\t0.7945\t1.2681\t0.5058\t0.6387\trs11190359\tABCC2(+148.8kb)|CHUK(-187.4kb)|CPN1(-41.36kb)|DNMBP(0)|DNMBP-AS1(+41.94kb)|ERLIN1(-149.1kb)\n",
      "10\t100000979\t10:100000979:T:C\tT\tC\t0.0504\t0.7881\t1.7337\t0.7499\t0.4631\trs11190360\tABCC2(+148.8kb)|CHUK(-187.4kb)|CPN1(-41.33kb)|DNMBP(0)|DNMBP-AS1(+41.98kb)|ERLIN1(-149.1kb)\n",
      "10\t100002012\t10:100002012:T:C\tT\tC\t0.0374\t0.8215\t1.0097\t0.6786\t0.9886\trs11190362\tABCC2(+149.8kb)|CHUK(-186.4kb)|CPN1(-40.3kb)|DNMBP(0)|DNMBP-AS1(+43.01kb)|ERLIN1(-148.1kb)\n",
      "10\t100002038\t10:100002038:G:A\tG\tA\t0.0102\t0.8861\t3.051\t1.3529\t0.4097\trs192480913\tABCC2(+149.8kb)|CHUK(-186.3kb)|CPN1(-40.27kb)|DNMBP(0)|DNMBP-AS1(+43.04kb)|ERLIN1(-148.1kb)\n",
      "10\t100002300\t10:100002300:GA:G\tGA\tG\t0.0374\t0.8216\t1.0082\t0.6788\t0.9904\trs111354488\tABCC2(+150.1kb)|CHUK(-186.1kb)|CPN1(-40.01kb)|DNMBP(0)|DNMBP-AS1(+43.3kb)|ERLIN1(-147.8kb)\n",
      "10\t100002330\t10:100002330:AAAAG:A\tAAAAG\tA\t0.0647\t0.8095\t1.3697\t0.6481\t0.6274\tNA\tABCC2(+150.1kb)|CHUK(-186kb)|CPN1(-39.98kb)|DNMBP(0)|DNMBP-AS1(+43.33kb)|ERLIN1(-147.8kb)\n",
      "10\t100002628\t10:100002628:A:C\tA\tC\t0.4467\t0.8605\t0.7322\t0.2868\t0.2771\trs11190363\tABCC2(+150.4kb)|CHUK(-185.7kb)|CPN1(-39.68kb)|DNMBP(0)|DNMBP-AS1(+43.63kb)|ERLIN1(-147.5kb)\n",
      "10\t100002875\t10:100002875:A:G\tA\tG\t0.0648\t0.8118\t1.3222\t0.6435\t0.6642\trs7894103\tABCC2(+150.7kb)|CHUK(-185.5kb)|CPN1(-39.43kb)|DNMBP(0)|DNMBP-AS1(+43.88kb)|ERLIN1(-147.2kb)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sed 's/ /\\t/g' /mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot > /mnt/data/GWAS/output/build38/task4_assoc/temp\n",
    "head /mnt/data/GWAS/output/build38/task4_assoc/temp\n",
    "mv /mnt/data/GWAS/output/build38/task4_assoc/temp  /mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis\n",
      "This version released on 2011-03-25\n",
      "\n",
      "# This program faciliates meta-analysis of genome-wide association studies.\n",
      "# Commonly used commands are listed below:\n",
      "#\n",
      "# Options for describing input files ...\n",
      "#   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)\n",
      "#   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')\n",
      "#   MARKERLABEL      [LABEL]                     (default = 'MARKER')\n",
      "#   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')\n",
      "#   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')\n",
      "#   FLIP\n",
      "#\n",
      "# Options for filtering input files ...\n",
      "#   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)\n",
      "#                    (available conditions are <, >, <=, >=, =, !=, IN)\n",
      "#   REMOVEFILTERS\n",
      "#\n",
      "# Options for sample size weighted meta-analysis ...\n",
      "#   WEIGHTLABEL      [LABEL]                     (default = 'N')\n",
      "#   PVALUELABEL      [LABEL]                     (default = 'PVALUE')\n",
      "#   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)\n",
      "#   MINWEIGHT        [NUMBER]                    (default = 1.0)\n",
      "#\n",
      "# Options for inverse variance weighted meta-analysis ...\n",
      "#   STDERRLABEL      [LABEL]                     (default = 'STDERR')\n",
      "#   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)\n",
      "#\n",
      "# Options to enable tracking of allele frequencies ...\n",
      "#   AVERAGEFREQ      [ON|OFF]                    (default = OFF)\n",
      "#   MINMAXFREQ       [ON|OFF]                    (default = OFF)\n",
      "#   FREQLABEL        [LABEL]                     (default = 'FREQ')\n",
      "#\n",
      "# Options to enable tracking of user defined variables ...\n",
      "#   CUSTOMVARIABLE   [VARNAME]\n",
      "#   LABEL            [VARNAME] AS [HEADER]\n",
      "#\n",
      "# Options to enable explicit strand information ...\n",
      "#   USESTRAND        [ON|OFF]                    (default = OFF)\n",
      "#   STRANDLABEL      [LABEL]                     (default = 'STRAND')\n",
      "#\n",
      "# Automatic genomic control correction of input statistics ...\n",
      "#   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)\n",
      "#\n",
      "# Options for general analysis control ...\n",
      "#   PROCESSFILE      [FILENAME]\n",
      "#   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')\n",
      "#   MAXWARNINGS      [NUMBER]                    (default = 20)\n",
      "#   VERBOSE          [ON|OFF]                    (default = 'OFF')\n",
      "#   LOGPVALUE        [ON|OFF]                    (default = 'OFF')\n",
      "#   ANALYZE          [HETEROGENEITY]\n",
      "#   CLEAR\n",
      "\n",
      "# Options for general run control ...\n",
      "#   SOURCE           [SCRIPTFILE]\n",
      "#   RETURN\n",
      "#   QUIT\n",
      "\n",
      "## Set marker header to RS ...\n",
      "## Set allele headers to A2 and A1 ...\n",
      "## Set effect header to OR ...\n",
      "## Set p-value header to P ...\n",
      "## Set column separator to TAB ...\n",
      "## Genomic control correction of input statistics enabled\n",
      "## Meta-analysis will be based on sample sizes, p-values and direction of effect ...\n",
      "## Set default weight to 10000.00 ...\n",
      "###########################################################################\n",
      "## Processing file '/mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL'\n",
      "## WARNING: No 'N' column found -- using DEFAULTWEIGHT = 10000\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/aaaaaaaaaaaaaacc'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'c/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/t'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'c/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/t'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/t'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'c/g'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/ta'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Bad alleles for marker 'NA', expecting 'a/aaaag' found 'a/c'\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'rs138064304' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: Marker 'NA' duplicated in input, first instance used, others skipped\n",
      "## WARNING: An additional 358 rows with duplicate marker names were ignored\n",
      "## WARNING: Allele names don't match previous occurences at 2177042 additional markers\n",
      "## Genomic control parameter is 1.110, adjusting test statistics\n",
      "## Processed 6799446 markers ...\n",
      "\n",
      "###########################################################################\n",
      "## Processing file '/mnt/data/MetaAnalysis/input/IGAP_stage_1_2_combined_for_METAL.txt'\n",
      "## WARNING: No 'N' column found -- using DEFAULTWEIGHT = 10000\n",
      "## Genomic control parameter is 25.515, adjusting test statistics\n",
      "## Processed 11632 markers ...\n",
      "\n",
      "## Set output file prefix and suffix to /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_ and .tbl ...\n",
      "###########################################################################\n",
      "## Running second pass analysis to evaluate heterogeneity...\n",
      "## Processing file '/mnt/data/MetaAnalysis/input/IGAP_stage_1_2_combined_for_METAL.txt'\n",
      "## Processing file '/mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL'\n",
      "\n",
      "###########################################################################\n",
      "## Executing meta-analysis ...\n",
      "## Complete results will be stored in file '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl'\n",
      "## Column descriptions will be stored in file '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl.info'\n",
      "## Completed meta-analysis for 6802988 markers!\n",
      "## Smallest p-value is 5.502e-06 at marker 'rs12614194'\n",
      "\n",
      "## Clearing all stored statistics ...\n",
      "# Clearing user defined filters ...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "metal\n",
    "\n",
    "MARKER RS\n",
    "ALLELE A2 A1\n",
    "EFFECT OR\n",
    "PVAL P\n",
    "SEPARATOR TAB\n",
    "GENOMICCONTROL ON\n",
    "SCHEME SAMPLESIZE\n",
    "DEFAULTWEIGHT   10000\n",
    "\n",
    "PROCESS /mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL\n",
    "PROCESS /mnt/data/MetaAnalysis/input/IGAP_stage_1_2_combined_for_METAL.txt\n",
    "\n",
    "OUTFILE /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_ .tbl\n",
    "\n",
    "ANALYZE HETEROGENEITY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MarkerName\tAllele1\tAllele2\tWeight\tZscore\tP-value\tDirection\tHetISq\tHetChiSq\tHetDf\tHetPVal\n",
      "rs2326918\ta\tg\t10000.00\t-0.589\t0.5559\t-?\t0.0\t0.000\t0\t1\n",
      "rs66941928\tt\tc\t10000.00\t-1.204\t0.2287\t-?\t0.0\t0.000\t0\t1\n",
      "rs6039163\tt\tc\t10000.00\t0.266\t0.7905\t+?\t0.0\t0.000\t0\t1\n",
      "rs62234673\tt\tc\t10000.00\t0.494\t0.6214\t+?\t0.0\t0.000\t0\t1\n",
      "rs6977693\tt\tc\t10000.00\t-1.333\t0.1825\t-?\t0.0\t0.000\t0\t1\n",
      "rs12364336\ta\tg\t10000.00\t-0.639\t0.5229\t-?\t0.0\t0.000\t0\t1\n",
      "rs11250701\ta\tg\t10000.00\t-0.583\t0.5596\t-?\t0.0\t0.000\t0\t1\n",
      "rs12562373\ta\tg\t10000.00\t-0.078\t0.9382\t-?\t0.0\t0.000\t0\t1\n",
      "rs4766166\ta\tg\t10000.00\t0.616\t0.5376\t+?\t0.0\t0.000\t0\t1\n",
      "# This file contains a short description of the columns in the\n",
      "# meta-analysis summary file, named '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl'\n",
      "\n",
      "# Marker    - this is the marker name\n",
      "# Allele1   - the first allele for this marker in the first file where it occurs\n",
      "# Allele2   - the second allele for this marker in the first file where it occurs\n",
      "# Weight    - the sum of the individual study weights (typically, N) for this marker\n",
      "# Z-score   - the combined z-statistic for this marker\n",
      "# P-value   - meta-analysis p-value\n",
      "# Direction - summary of effect direction for each study, with one '+' or '-' per study\n",
      "  6802989  74832879 369785336 /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl\n",
    "head /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.SAMPLESIZE_1.tbl.info\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SCHEME STDERR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis\n",
      "This version released on 2011-03-25\n",
      "\n",
      "# This program faciliates meta-analysis of genome-wide association studies.\n",
      "# Commonly used commands are listed below:\n",
      "#\n",
      "# Options for describing input files ...\n",
      "#   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)\n",
      "#   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')\n",
      "#   MARKERLABEL      [LABEL]                     (default = 'MARKER')\n",
      "#   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')\n",
      "#   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')\n",
      "#   FLIP\n",
      "#\n",
      "# Options for filtering input files ...\n",
      "#   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)\n",
      "#                    (available conditions are <, >, <=, >=, =, !=, IN)\n",
      "#   REMOVEFILTERS\n",
      "#\n",
      "# Options for sample size weighted meta-analysis ...\n",
      "#   WEIGHTLABEL      [LABEL]                     (default = 'N')\n",
      "#   PVALUELABEL      [LABEL]                     (default = 'PVALUE')\n",
      "#   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)\n",
      "#   MINWEIGHT        [NUMBER]                    (default = 1.0)\n",
      "#\n",
      "# Options for inverse variance weighted meta-analysis ...\n",
      "#   STDERRLABEL      [LABEL]                     (default = 'STDERR')\n",
      "#   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)\n",
      "#\n",
      "# Options to enable tracking of allele frequencies ...\n",
      "#   AVERAGEFREQ      [ON|OFF]                    (default = OFF)\n",
      "#   MINMAXFREQ       [ON|OFF]                    (default = OFF)\n",
      "#   FREQLABEL        [LABEL]                     (default = 'FREQ')\n",
      "#\n",
      "# Options to enable tracking of user defined variables ...\n",
      "#   CUSTOMVARIABLE   [VARNAME]\n",
      "#   LABEL            [VARNAME] AS [HEADER]\n",
      "#\n",
      "# Options to enable explicit strand information ...\n",
      "#   USESTRAND        [ON|OFF]                    (default = OFF)\n",
      "#   STRANDLABEL      [LABEL]                     (default = 'STRAND')\n",
      "#\n",
      "# Automatic genomic control correction of input statistics ...\n",
      "#   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)\n",
      "#\n",
      "# Options for general analysis control ...\n",
      "#   PROCESSFILE      [FILENAME]\n",
      "#   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')\n",
      "#   MAXWARNINGS      [NUMBER]                    (default = 20)\n",
      "#   VERBOSE          [ON|OFF]                    (default = 'OFF')\n",
      "#   LOGPVALUE        [ON|OFF]                    (default = 'OFF')\n",
      "#   ANALYZE          [HETEROGENEITY]\n",
      "#   CLEAR\n",
      "\n",
      "# Options for general run control ...\n",
      "#   SOURCE           [SCRIPTFILE]\n",
      "#   RETURN\n",
      "#   QUIT\n",
      "\n",
      "## Set marker header to SNP ...\n",
      "## Set allele headers to A2 and A1 ...\n",
      "## Set effect header to OR ...\n",
      "## Set p-value header to P ...\n",
      "## Set column separator to TAB ...\n",
      "## Tracking of extreme allele frequencies enabled\n",
      "## Genomic control correction of input statistics enabled\n",
      "## Meta-analysis will be based on effect sizes and their standard errors ...\n",
      "## Set standard error header to SE ...\n",
      "###########################################################################\n",
      "## Processing file '/mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL'\n",
      "## ERROR: Averaging of allele frequencies requested, but no 'FREQ' column found\n",
      "\n",
      "###########################################################################\n",
      "## Processing file '/mnt/data/MetaAnalysis/input/IGAP_stage_1_2_combined_for_METAL.txt'\n",
      "## ERROR: No 'SNP' column found\n",
      "\n",
      "## Set output file prefix and suffix to /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_ and .tbl ...\n",
      "###########################################################################\n",
      "## Running second pass analysis to evaluate heterogeneity...\n",
      "\n",
      "###########################################################################\n",
      "## Executing meta-analysis ...\n",
      "## Complete results will be stored in file '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_1.tbl'\n",
      "## Column descriptions will be stored in file '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_1.tbl.info'\n",
      "## Completed meta-analysis for 0 markers!\n",
      "\n",
      "## Clearing all stored statistics ...\n",
      "# Clearing user defined filters ...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "metal\n",
    "\n",
    "MARKER SNP\n",
    "ALLELE A2 A1\n",
    "EFFECT OR\n",
    "PVAL P\n",
    "SEPARATOR TAB\n",
    "MINMAXFREQ ON\n",
    "GENOMICCONTROL ON\n",
    "SCHEME STDERR\n",
    "\n",
    "STDERR SE\n",
    "\n",
    "PROCESS /mnt/data/GWAS/output/build38/task4_assoc/dataset.b38.imputed.dosage.full.assoc.dosage.clean.rs.200kb.annotated.annot.METAL\n",
    "PROCESS /mnt/data/MetaAnalysis/input/IGAP_stage_1_2_combined_for_METAL.txt\n",
    "\n",
    "OUTFILE /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_ .tbl\n",
    "\n",
    "ANALYZE HETEROGENEITY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MarkerName\tAllele1\tAllele2\tMinFreq\tMaxFreq\tEffect\tStdErr\tP-value\tDirection\tHetISq\tHetChiSq\tHetDf\tHetPVal\n",
      "# This file contains a short description of the columns in the\n",
      "# meta-analysis summary file, named '/mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_1.tbl'\n",
      "\n",
      "# Marker    - this is the marker name\n",
      "# Allele1   - the first allele for this marker in the first file where it occurs\n",
      "# Allele2   - the second allele for this marker in the first file where it occurs\n",
      "# MinFreq     - minimum frequency for allele 1 across all studies\n",
      "# MaxFreq     - maximum frequency for allele 1 across all studies\n",
      "# Effect    - overall estimated effect size for allele1\n",
      "# StdErr    - overall standard error for effect size estimate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_1.tbl\n",
    "head /mnt/data/MetaAnalysis/output/dataset.b38.IGAP.METAL.STDERR_1.tbl.info"
   ]
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": "block",
   "toc_window_display": false
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
